
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Reference &#8212; pypath 0.8.34 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Webservice" href="webservice.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="webservice.html" title="Webservice"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Installation"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pypath 0.8.34 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="reference">
<h1>Reference<a class="headerlink" href="#reference" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-pypath.annot">
<span id="annot-module"></span><h2>annot module<a class="headerlink" href="#module-pypath.annot" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pypath.annot.Adhesome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Adhesome</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Adhesome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Adhesome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.AnnotationBase">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">AnnotationBase</code><span class="sig-paren">(</span><em>name</em>, <em>ncbi_tax_id=9606</em>, <em>input_method=None</em>, <em>input_args=None</em>, <em>entity_type='protein'</em>, <em>swissprot_only=True</em>, <em>proteins=()</em>, <em>complexes=()</em>, <em>reference_set=()</em>, <em>infer_complexes=True</em>, <em>dump=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase" title="Permalink to this definition">¶</a></dt>
<dd><dl class="method">
<dt id="pypath.annot.AnnotationBase.add_complexes_by_inference">
<code class="descname">add_complexes_by_inference</code><span class="sig-paren">(</span><em>complexes=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.add_complexes_by_inference"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.add_complexes_by_inference" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates complex annotations by in silico inference and adds them
to this annotation set.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.all_proteins">
<code class="descname">all_proteins</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.all_proteins"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.all_proteins" title="Permalink to this definition">¶</a></dt>
<dd><p>All UniProt IDs annotated in this resource.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.annotate_complex">
<code class="descname">annotate_complex</code><span class="sig-paren">(</span><em>cplex</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.annotate_complex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.annotate_complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Infers annotations for a single complex.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.complex_inference">
<code class="descname">complex_inference</code><span class="sig-paren">(</span><em>complexes=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.complex_inference"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.complex_inference" title="Permalink to this definition">¶</a></dt>
<dd><p>Annotates all complexes in <cite>complexes</cite>, by default in the default
complex database (existing in the <cite>complex</cite> module or generated
on demand according to the module’s current settings).</p>
<p>Dict with complexes as keys and sets of annotations as values.
Complexes with no valid information in this annotation resource
won’t be in the dict.</p>
<dl class="simple">
<dt>complexes<span class="classifier">iterable</span></dt><dd><p>Iterable yielding complexes.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.get_subset">
<code class="descname">get_subset</code><span class="sig-paren">(</span><em>method=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.get_subset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.get_subset" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a subset by filtering based on <code class="docutils literal notranslate"><span class="pre">kwargs</span></code>.
Each argument should be a name and a value or set of values.
Elements having the provided values in the annotation will be
returned.
Returns a set of UniProt IDs.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.load_proteins">
<code class="descname">load_proteins</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.load_proteins"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.load_proteins" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a set of all UniProt IDs to have a base set of the entire
proteome.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.annot.AnnotationBase.reload">
<code class="descname">reload</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#AnnotationBase.reload"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.AnnotationBase.reload" title="Permalink to this definition">¶</a></dt>
<dd><p>Reloads the object from the module level.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pypath.annot.CancerGeneCensus">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CancerGeneCensus</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CancerGeneCensus"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CancerGeneCensus" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Cancersea">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Cancersea</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Cancersea"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Cancersea" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.CellPhoneDB">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CellPhoneDB</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CellPhoneDB"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CellPhoneDB" title="Permalink to this definition">¶</a></dt>
<dd><dl class="attribute">
<dt id="pypath.annot.CellPhoneDB.record">
<code class="descname">record</code><a class="headerlink" href="#pypath.annot.CellPhoneDB.record" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.dataio.CellPhoneDBAnnotation</span></code></p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pypath.annot.CellPhoneDBComplex">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CellPhoneDBComplex</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CellPhoneDBComplex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CellPhoneDBComplex" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.CellSurfaceProteinAtlas">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CellSurfaceProteinAtlas</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CellSurfaceProteinAtlas"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CellSurfaceProteinAtlas" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Comppi">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Comppi</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Comppi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Comppi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Corum">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Corum</code><span class="sig-paren">(</span><em>name</em>, <em>annot_attr</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Corum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Corum" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.CorumFuncat">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CorumFuncat</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CorumFuncat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CorumFuncat" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.CorumGO">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">CorumGO</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#CorumGO"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.CorumGO" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Cpad">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Cpad</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Cpad"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Cpad" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Dgidb">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Dgidb</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Dgidb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Dgidb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Disgenet">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Disgenet</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Disgenet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Disgenet" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Exocarta">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Exocarta</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Exocarta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Exocarta" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.GOIntercell">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">GOIntercell</code><span class="sig-paren">(</span><em>categories=None</em>, <em>go_annot=None</em>, <em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#GOIntercell"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.GOIntercell" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.GuideToPharmacology">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">GuideToPharmacology</code><span class="sig-paren">(</span><em>load_sources=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#GuideToPharmacology"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.GuideToPharmacology" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Hgnc">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Hgnc</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Hgnc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Hgnc" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.HpmrComplex">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">HpmrComplex</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#HpmrComplex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.HpmrComplex" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.HumanPlasmaMembraneReceptome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">HumanPlasmaMembraneReceptome</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#HumanPlasmaMembraneReceptome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.HumanPlasmaMembraneReceptome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.HumanProteinAtlas">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">HumanProteinAtlas</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#HumanProteinAtlas"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.HumanProteinAtlas" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Integrins">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Integrins</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Integrins"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Integrins" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Intogen">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Intogen</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Intogen"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Intogen" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.KeggPathways">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">KeggPathways</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#KeggPathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.KeggPathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Kinasedotcom">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Kinasedotcom</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Kinasedotcom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Kinasedotcom" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Kirouac2010">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Kirouac2010</code><span class="sig-paren">(</span><em>load_sources=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Kirouac2010"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Kirouac2010" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.LigandReceptor">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">LigandReceptor</code><span class="sig-paren">(</span><em>name</em>, <em>ligand_col=None</em>, <em>receptor_col=None</em>, <em>ligand_id_type=None</em>, <em>receptor_id_type=None</em>, <em>record_processor_method=None</em>, <em>record_extra_fields=None</em>, <em>record_defaults=None</em>, <em>extra_fields_methods=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#LigandReceptor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.LigandReceptor" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Locate">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Locate</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>literature=True</em>, <em>external=True</em>, <em>predictions=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Locate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Locate" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Matrisome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Matrisome</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Matrisome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Matrisome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Matrixdb">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Matrixdb</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Matrixdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Matrixdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Membranome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Membranome</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Membranome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Membranome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.NetpathPathways">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">NetpathPathways</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#NetpathPathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.NetpathPathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Opm">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Opm</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Opm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Opm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Phosphatome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Phosphatome</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Phosphatome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Phosphatome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Ramilowski2015">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Ramilowski2015</code><span class="sig-paren">(</span><em>load_sources=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Ramilowski2015"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Ramilowski2015" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Ramilowski2015Location">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Ramilowski2015Location</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Ramilowski2015Location"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Ramilowski2015Location" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.SignalinkPathways">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">SignalinkPathways</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#SignalinkPathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.SignalinkPathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.SignorPathways">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">SignorPathways</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#SignorPathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.SignorPathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Surfaceome">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Surfaceome</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Surfaceome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Surfaceome" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Tfcensus">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Tfcensus</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Tfcensus"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Tfcensus" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Topdb">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Topdb</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Topdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Topdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Vesiclepedia">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Vesiclepedia</code><span class="sig-paren">(</span><em>ncbi_tax_id=9606</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Vesiclepedia"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Vesiclepedia" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.annot.Zhong2015">
<em class="property">class </em><code class="descclassname">pypath.annot.</code><code class="descname">Zhong2015</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#Zhong2015"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.Zhong2015" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pypath.annot.get_db">
<code class="descclassname">pypath.annot.</code><code class="descname">get_db</code><span class="sig-paren">(</span><em>keep_annotators=True</em>, <em>create_dataframe=False</em>, <em>use_complexes=True</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#get_db"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.get_db" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the current database instance and initializes it if does
not exist yet.</p>
</dd></dl>

<dl class="function">
<dt id="pypath.annot.init_db">
<code class="descclassname">pypath.annot.</code><code class="descname">init_db</code><span class="sig-paren">(</span><em>keep_annotators=True</em>, <em>create_dataframe=False</em>, <em>use_complexes=True</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/annot.html#init_db"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.annot.init_db" title="Permalink to this definition">¶</a></dt>
<dd><p>Initializes or reloads the annotation database.
The database will be assigned to the <code class="docutils literal notranslate"><span class="pre">db</span></code> attribute of this module.</p>
</dd></dl>

</div>
<div class="section" id="bel-module">
<h2>bel module<a class="headerlink" href="#bel-module" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-pypath.complex">
<span id="complex-module"></span><h2>complex module<a class="headerlink" href="#module-pypath.complex" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pypath.complex.AbstractComplexResource">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">AbstractComplexResource</code><span class="sig-paren">(</span><em>name</em>, <em>ncbi_tax_id=9606</em>, <em>input_method=None</em>, <em>input_args=None</em>, <em>dump=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#AbstractComplexResource"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.AbstractComplexResource" title="Permalink to this definition">¶</a></dt>
<dd><p>A resource which provides information about molecular complexes.</p>
</dd></dl>

<dl class="class">
<dt id="pypath.complex.CellPhoneDB">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">CellPhoneDB</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#CellPhoneDB"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.CellPhoneDB" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Compleat">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Compleat</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Compleat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Compleat" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.ComplexAggregator">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">ComplexAggregator</code><span class="sig-paren">(</span><em>resources=None</em>, <em>pickle_file=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#ComplexAggregator"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.ComplexAggregator" title="Permalink to this definition">¶</a></dt>
<dd><dl class="method">
<dt id="pypath.complex.ComplexAggregator.reload">
<code class="descname">reload</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#ComplexAggregator.reload"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.ComplexAggregator.reload" title="Permalink to this definition">¶</a></dt>
<dd><p>Reloads the object from the module level.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pypath.complex.ComplexPortal">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">ComplexPortal</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#ComplexPortal"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.ComplexPortal" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Corum">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Corum</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Corum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Corum" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.GuideToPharmacology">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">GuideToPharmacology</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#GuideToPharmacology"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.GuideToPharmacology" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Havugimana">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Havugimana</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Havugimana"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Havugimana" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Hpmr">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Hpmr</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Hpmr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Hpmr" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Humap">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Humap</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Humap"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Humap" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Pdb">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Pdb</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Pdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="class">
<dt id="pypath.complex.Signor">
<em class="property">class </em><code class="descclassname">pypath.complex.</code><code class="descname">Signor</code><span class="sig-paren">(</span><em>input_args=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#Signor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.Signor" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pypath.complex.all_complexes">
<code class="descclassname">pypath.complex.</code><code class="descname">all_complexes</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#all_complexes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.all_complexes" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a set of all complexes in the database which serves as a
reference set for many methods, just like <code class="docutils literal notranslate"><span class="pre">uniprot_input.all_uniprots</span></code>
represents the proteome.</p>
</dd></dl>

<dl class="function">
<dt id="pypath.complex.get_db">
<code class="descclassname">pypath.complex.</code><code class="descname">get_db</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#get_db"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.get_db" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the current database instance and initializes it if does
not exist yet.</p>
</dd></dl>

<dl class="function">
<dt id="pypath.complex.init_db">
<code class="descclassname">pypath.complex.</code><code class="descname">init_db</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/complex.html#init_db"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.complex.init_db" title="Permalink to this definition">¶</a></dt>
<dd><p>Initializes or reloads the complex database.
The database will be assigned to the <code class="docutils literal notranslate"><span class="pre">db</span></code> attribute of this module.</p>
</dd></dl>

</div>
<div class="section" id="module-pypath.intera">
<span id="intera-module"></span><h2>intera module<a class="headerlink" href="#module-pypath.intera" title="Permalink to this headline">¶</a></h2>
<p>This module provides classes to represent and handle
structural details of protein interactions
i.e. residues, post-translational modifications,
short motifs, domains, domain-motifs and
domain-motif interactions, binding interfaces.</p>
</div>
<div class="section" id="module-pypath.intercell">
<span id="intercell-module"></span><h2>intercell module<a class="headerlink" href="#module-pypath.intercell" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pypath.intercell.IntercellRole">
<em class="property">class </em><code class="descclassname">pypath.intercell.</code><code class="descname">IntercellRole</code><span class="sig-paren">(</span><em>source</em>, <em>role</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.intercell.IntercellRole" title="Permalink to this definition">¶</a></dt>
<dd><dl class="attribute">
<dt id="pypath.intercell.IntercellRole.role">
<code class="descname">role</code><a class="headerlink" href="#pypath.intercell.IntercellRole.role" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 1</p>
</dd></dl>

<dl class="attribute">
<dt id="pypath.intercell.IntercellRole.source">
<code class="descname">source</code><a class="headerlink" href="#pypath.intercell.IntercellRole.source" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 0</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-pypath.main">
<span id="main-module"></span><h2>main module<a class="headerlink" href="#module-pypath.main" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pypath.main.PyPath">
<em class="property">class </em><code class="descclassname">pypath.main.</code><code class="descname">PyPath</code><span class="sig-paren">(</span><em>ncbi_tax_id=None</em>, <em>copy=None</em>, <em>name='unnamed'</em>, <em>cache_dir=None</em>, <em>outdir='results'</em>, <em>loglevel=0</em>, <em>loops=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath" title="Permalink to this definition">¶</a></dt>
<dd><p>Main network object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ncbi_tax_id</strong> (<em>int</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">9606</span></code> (Homo sapiens) by default. NCBI Taxonomic
identifier of the organism from which the data will be
downloaded.</p></li>
<li><p><strong>default_name_type</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{'protein':</span> <span class="pre">'uniprot',</span> <span class="pre">'mirna':</span> <span class="pre">'mirbase',</span> <span class="pre">'drug':</span>
<span class="pre">'chembl',</span> <span class="pre">'lncrna':</span> <span class="pre">'lncrna-genesymbol'}</span></code> by default. Contains
the default identifier types to which the downloaded data will
be converted. If others are used, user may need to provide the
format definitions for the conversion tables.</p></li>
<li><p><strong>copy</strong> (<a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><em>pypath.main.PyPath</em></a>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Other
<a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.main.PyPath</span></code></a> instance from which the data will
be copied.</p></li>
<li><p><strong>mysql</strong> (<em>tuple</em>) – DEPRECATED Optional, <code class="docutils literal notranslate"><span class="pre">(None,</span> <span class="pre">'mapping')</span></code> by default. Contains the MySQL
parameters used by the <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.mapping</span></code> module to load
the ID conversion tables.</p></li>
<li><p><strong>chembl_mysql</strong> (<em>tuple</em>) – DEPRECATED Optional, <code class="docutils literal notranslate"><span class="pre">(None,</span> <span class="pre">'chembl')</span></code> by default. Contains the MySQL
parameters used by the <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.mapping</span></code> module to load
the ChEMBL ID conversion tables.</p></li>
<li><p><strong>name</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'unnamed'</span></code> by default. Session or project name
(custom).</p></li>
<li><p><strong>outdir</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'results'</span></code> by default. Output directory where to
store all output files.</p></li>
<li><p><strong>loglevel</strong> (<em>int</em>) – Optional, 0 by default. Sets the level of the logger.
The higher the level the more messages will be written to the log.</p></li>
<li><p><strong>loops</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Determines if self-loop edges
are allowed in the graph.</p></li>
</ul>
</dd>
<dt class="field-even">Variables</dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>adjlist</strong> (<em>list</em>) – List of [set] containing the adjacency of each node. See
<a class="reference internal" href="#pypath.main.PyPath.update_adjlist" title="pypath.main.PyPath.update_adjlist"><code class="xref py py-meth docutils literal notranslate"><span class="pre">PyPath.update_adjlist()</span></code></a> method for more information.</p></li>
<li><p><strong>chembl</strong> (<em>pypath.chembl.Chembl</em>) – Contains the ChEMBL data. See <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.chembl</span></code> module
documentation for more information.</p></li>
<li><p><strong>chembl_mysql</strong> (<em>tuple</em>) – DEPRECATED Contains the MySQL parameters used by the
<code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.mapping</span></code> module to load the ChEMBL ID conversion
tables.</p></li>
<li><p><strong>data</strong> (<em>dict</em>) – Stores the loaded interaction and attribute table. See
<a class="reference internal" href="#pypath.main.PyPath.read_data_file" title="pypath.main.PyPath.read_data_file"><code class="xref py py-meth docutils literal notranslate"><span class="pre">PyPath.read_data_file()</span></code></a> method for more information.</p></li>
<li><p><strong>db_dict</strong> (<em>dict</em>) – Dictionary of dictionaries. Outer-level keys are <code class="docutils literal notranslate"><span class="pre">'nodes'</span></code> and
<code class="docutils literal notranslate"><span class="pre">'edges'</span></code>, corresponding values are [dict] whose keys are the
database sources with values of type [set] containing the
edge/node indexes for which that database provided some
information.</p></li>
<li><p><strong>dgraph</strong> (<em>igraph.Graph</em>) – Directed network graph object.</p></li>
<li><p><strong>disclaimer</strong> (<em>str</em>) – Disclaimer text.</p></li>
<li><p><strong>dlabDct</strong> (<em>dict</em>) – Maps the directed graph node labels [str] (keys) to their
indices [int] (values).</p></li>
<li><p><strong>dnodDct</strong> (<em>dict</em>) – Maps the directed graph node names [str] (keys) to their indices
[int] (values).</p></li>
<li><p><strong>dnodInd</strong> (<em>set</em>) – Stores the directed graph node names [str].</p></li>
<li><p><strong>dnodLab</strong> (<em>dict</em>) – Maps the directed graph node indices [int] (keys) to their
labels [str] (values).</p></li>
<li><p><strong>dnodNam</strong> (<em>dict</em>) – Maps the directed graph node indices [int] (keys) to their names
[str] (values).</p></li>
<li><p><strong>edgeAttrs</strong> (<em>dict</em>) – Stores the edge attribute names [str] as keys and their
corresponding types (e.g.: <code class="docutils literal notranslate"><span class="pre">set</span></code>, <code class="docutils literal notranslate"><span class="pre">list</span></code>, <code class="docutils literal notranslate"><span class="pre">str</span></code>, …) as
values.</p></li>
<li><p><strong>exp</strong> (<em>pandas.DataFrame</em>) – Stores the expression data for the nodes (if loaded).</p></li>
<li><p><strong>exp_prod</strong> (<em>pandas.DataFrame</em>) – Stores the edge expression data (as the product of the
normalized expression between the pair of nodes by default). For
more details see <a class="reference internal" href="#pypath.main.PyPath.edges_expression" title="pypath.main.PyPath.edges_expression"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.edges_expression()</span></code></a>.</p></li>
<li><p><strong>exp_samples</strong> (<em>set</em>) – Contains a list of tissues as downloaded by ProteomicsDB. See
<a class="reference internal" href="#pypath.main.PyPath.get_proteomicsdb" title="pypath.main.PyPath.get_proteomicsdb"><code class="xref py py-meth docutils literal notranslate"><span class="pre">PyPath.get_proteomicsdb()</span></code></a> for more information.</p></li>
<li><p><strong>failed_edges</strong> (<em>list</em>) – List of lists containing information about the failed edges.
Each failed edge sublist contains (in this order): [tuple] with
the node IDs, [str] names of nodes A and B, [int] IDs of nodes
A and B and [int] IDs of the edges in both directions.</p></li>
<li><p><strong>go</strong> (<em>dict</em>) – Contains the organism(s) NCBI taxonomy ID as key [int] and
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.go.GOAnnotation</span></code> object as value, which
contains the GO annotations for the nodes in the graph. See
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.go.GOAnnotation</span></code> for more information.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Undirected network graph object.</p></li>
<li><p><strong>gsea</strong> (<em>pypath.gsea.GSEA</em>) – Contains the loaded gene-sets from MSigDB. See
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.gsea.GSEA</span></code> for more information.</p></li>
<li><p><strong>has_cats</strong> (<em>set</em>) – Contains the categories (e.g.: resource types) [str] loaded in
the current network. Possible categories are: <code class="docutils literal notranslate"><span class="pre">'m'</span></code> for
PTM/enzyme-substrate resources, <code class="docutils literal notranslate"><span class="pre">'p'</span></code> for pathway/activity
flow resources, <code class="docutils literal notranslate"><span class="pre">'i'</span></code> for undirected/PPI resources, <code class="docutils literal notranslate"><span class="pre">'r'</span></code>
for process description/reaction resources and <code class="docutils literal notranslate"><span class="pre">'t'</span></code> for
transcription resources.</p></li>
<li><p><strong>htp</strong> (<em>dict</em>) – Contains information about high-throughput data of the network
for different thresholds [int] (keys). Values are [dict]
containing the number of references (<code class="docutils literal notranslate"><span class="pre">'rnum'</span></code>) [int], number
of edges (<code class="docutils literal notranslate"><span class="pre">'enum'</span></code>) [int], number of sources (<code class="docutils literal notranslate"><span class="pre">'snum'</span></code>)
[int] and list of PMIDs of the most common references above the
given threshold (<code class="docutils literal notranslate"><span class="pre">'htrefs'</span></code>) [set].</p></li>
<li><p><strong>labDct</strong> (<em>dict</em>) – Maps the undirected graph node labels [str] (keys) to their
indices [int] (values).</p></li>
<li><p><strong>lists</strong> (<em>dict</em>) – Contains specific lists of nodes (values) for different
categories [str] (keys). These can to be loaded from a file or
a resource. Some methods include
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.receptor_list()</span></code> (<code class="docutils literal notranslate"><span class="pre">'rec'</span></code>),
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.druggability_list()</span></code> (<code class="docutils literal notranslate"><span class="pre">'dgb'</span></code>),
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.kinases_list()</span></code> (<code class="docutils literal notranslate"><span class="pre">'kin'</span></code>),
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.tfs_list()</span></code> (<code class="docutils literal notranslate"><span class="pre">'tf'</span></code>),
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.disease_genes_list()</span></code> (<code class="docutils literal notranslate"><span class="pre">'dis'</span></code>),
<a class="reference internal" href="#pypath.main.PyPath.signaling_proteins_list" title="pypath.main.PyPath.signaling_proteins_list"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.signaling_proteins_list()</span></code></a>
(<code class="docutils literal notranslate"><span class="pre">'sig'</span></code>), <code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.proteome_list()</span></code>
(<code class="docutils literal notranslate"><span class="pre">'proteome'</span></code>) and
<a class="reference internal" href="#pypath.main.PyPath.cancer_drivers_list" title="pypath.main.PyPath.cancer_drivers_list"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.cancer_drivers_list()</span></code></a> (<code class="docutils literal notranslate"><span class="pre">'cdv'</span></code>).</p></li>
<li><p><strong>loglevel</strong> (<em>str</em>) – The level of the logger.</p></li>
<li><p><strong>loops</strong> (<em>bool</em>) – Whether if self-loop edges are allowed in the graph.</p></li>
<li><p><strong>mapper</strong> (<em>pypath.mapping.Mapper</em>) – <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.mapper.Mapper</span></code> object for ID conversion and
other ID-related operations across resources.</p></li>
<li><p><strong>mutation_samples</strong> (<em>list</em>) – DEPRECATED</p></li>
<li><p><strong>mysql_conf</strong> (<em>tuple</em>) – DEPRECATED Contains the MySQL parameters used by the
<code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.mapping</span></code> module to load the ID conversion
tables.</p></li>
<li><p><strong>name</strong> (<em>str</em>) – Session or project name (custom).</p></li>
<li><p><strong>ncbi_tax_id</strong> (<em>int</em>) – NCBI Taxonomic identifier of the organism from which the data
will be downloaded.</p></li>
<li><p><strong>negatives</strong> (<em>dict</em>) – Contains a list of negative interactions according to a given
source (e.g.: Negatome database). See
<a class="reference internal" href="#pypath.main.PyPath.apply_negative" title="pypath.main.PyPath.apply_negative"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.apply_negative()</span></code></a> for more
information.</p></li>
<li><p><strong>nodDct</strong> (<em>dict</em>) – Maps the undirected graph node names [str] (keys) to their
indices [int] (values).</p></li>
<li><p><strong>nodInd</strong> (<em>set</em>) – Stores the undirected graph node names [str].</p></li>
<li><p><strong>nodLab</strong> (<em>dict</em>) – Maps the undirected graph node indices [int] (keys) to their
labels [str] (values).</p></li>
<li><p><strong>nodNam</strong> (<em>dict</em>) – Maps the directed graph node indices [int] (keys) to their names
[str] (values).</p></li>
<li><p><strong>outdir</strong> (<em>str</em>) – Output directory where to store all output files.</p></li>
<li><p><strong>palette</strong> (<em>list</em>) – Contains a list of hexadecimal [str] of colors. Used for
plotting purposes.</p></li>
<li><p><strong>pathway_types</strong> (<em>list</em>) – Contains the names of all the loaded pathway resources [str].</p></li>
<li><p><strong>pathways</strong> (<em>dict</em>) – Contains the list of pathways (values) for each resource (keys)
loaded in the network.</p></li>
<li><p><strong>plots</strong> (<em>dict</em>) – DEPRECATED (?)</p></li>
<li><p><strong>proteomicsdb</strong> (<em>pypath.proteomicsdb.ProteomicsDB</em>) – Contains a <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.proteomicsdb.ProteomicsDB</span></code>
instance, see the class documentation for more information.</p></li>
<li><p><strong>raw_data</strong> (<em>list</em>) – Contains a list of loaded edges [dict] from a data file. See
<a class="reference internal" href="#pypath.main.PyPath.read_data_file" title="pypath.main.PyPath.read_data_file"><code class="xref py py-meth docutils literal notranslate"><span class="pre">PyPath.read_data_file()</span></code></a> for more information.</p></li>
<li><p><strong>seq</strong> (<em>dict</em>) – (?)</p></li>
<li><p><strong>session</strong> (<em>str</em>) – Session ID, a five random alphanumeric characters.</p></li>
<li><p><strong>session_name</strong> (<em>str</em>) – Session name and ID (e.g. <code class="docutils literal notranslate"><span class="pre">'unnamed-abc12'</span></code>).</p></li>
<li><p><strong>sourceNetEdges</strong> (<em>igraph.Graph</em>) – (?)</p></li>
<li><p><strong>sourceNetNodes</strong> (<em>igraph.Graph</em>) – (?)</p></li>
<li><p><strong>sources</strong> (<em>list</em>) – List contianing the names of the loaded resources [str].</p></li>
<li><p><strong>u_pfam</strong> (<em>dict</em>) – Dictionary of dictionaries, contains the mapping of UniProt IDs
to their respective protein families and other information.</p></li>
<li><p><strong>uniprot_mapped</strong> (<em>list</em>) – DEPRECATED (?)</p></li>
<li><p><strong>unmapped</strong> (<em>list</em>) – Contains the names of unmapped items [str]. See
<a class="reference internal" href="#pypath.main.PyPath.map_item" title="pypath.main.PyPath.map_item"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.map_item()</span></code></a> for more information.</p></li>
<li><p><strong>vertexAttrs</strong> (<em>dict</em>) – Stores the node attribute names [str] as keys and their
corresponding types (e.g.: <code class="docutils literal notranslate"><span class="pre">set</span></code>, <code class="docutils literal notranslate"><span class="pre">list</span></code>, <code class="docutils literal notranslate"><span class="pre">str</span></code>, …) as
values.</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="pypath.main.PyPath.acsn_effects">
<code class="descname">acsn_effects</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.acsn_effects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.acsn_effects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_genesets">
<code class="descname">add_genesets</code><span class="sig-paren">(</span><em>genesets</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_genesets"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_genesets" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_grouped_eattr">
<code class="descname">add_grouped_eattr</code><span class="sig-paren">(</span><em>edge</em>, <em>attr</em>, <em>group</em>, <em>value</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_grouped_eattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_grouped_eattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges (or creates) a given edge attribute as [dict] of [list]
values.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge</strong> (<em>int</em>) – Edge index where the given attribute value is to be merged
or created.</p></li>
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty [dict], <em>value</em> will only be assigned to the given
<em>edge</em> and <em>group</em>).</p></li>
<li><p><strong>group</strong> (<em>str</em>) – The key of the attribute dictionary where <em>value</em> is to be
assigned.</p></li>
<li><p><strong>value</strong> (<em>list</em>) – The value of the attribute to be assigned/merged.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_grouped_set_eattr">
<code class="descname">add_grouped_set_eattr</code><span class="sig-paren">(</span><em>edge</em>, <em>attr</em>, <em>group</em>, <em>value</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_grouped_set_eattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_grouped_set_eattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges (or creates) a given edge attribute as [dict] of [set]
values.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge</strong> (<em>int</em>) – Edge index where the given attribute value is to be merged
or created.</p></li>
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty [dict], <em>value</em> will only be assigned to the given
<em>edge</em> and <em>group</em>).</p></li>
<li><p><strong>group</strong> (<em>str</em>) – The key of the attribute dictionary where <em>value</em> is to be
assigned.</p></li>
<li><p><strong>value</strong> (<em>set</em>) – The value of the attribute to be assigned/merged.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_list_eattr">
<code class="descname">add_list_eattr</code><span class="sig-paren">(</span><em>edge</em>, <em>attr</em>, <em>value</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_list_eattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_list_eattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges (or creates) a given edge attribute as [list].</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge</strong> (<em>int</em>) – Edge index where the given attribute value is to be merged
or created.</p></li>
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty [list], <em>value</em> will only be assigned to the given
<em>edge</em>).</p></li>
<li><p><strong>value</strong> (<em>list</em>) – The value of the attribute to be assigned/merged.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_set_eattr">
<code class="descname">add_set_eattr</code><span class="sig-paren">(</span><em>edge</em>, <em>attr</em>, <em>value</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_set_eattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_set_eattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges (or creates) a given edge attribute as [set].</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge</strong> (<em>int</em>) – Edge index where the given attribute value is to be merged
or created.</p></li>
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty [set], <em>value</em> will only be assigned to the given
<em>edge</em>).</p></li>
<li><p><strong>value</strong> (<em>set</em>) – The value of the attribute to be assigned/merged.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_update_edge">
<code class="descname">add_update_edge</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em>, <em>source</em>, <em>is_directed</em>, <em>refs</em>, <em>stim</em>, <em>inh</em>, <em>taxon_a</em>, <em>taxon_b</em>, <em>typ</em>, <em>extra_attrs={}</em>, <em>add=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_update_edge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_update_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Updates the attributes of one edge in the (undirected) network.
Optionally it creates a new edge and sets the attributes, but it
is not efficient as <code class="xref py py-mod docutils literal notranslate"><span class="pre">igraph</span></code> needs to reindex edges
after this operation, so better to create new edges in batches.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – Name of the source node of the edge to be added/updated.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – Name of the source node of the edge to be added/updated.</p></li>
<li><p><strong>source</strong> (<em>set</em>) – Or [list], contains the names [str] of the resources
supporting that edge.</p></li>
<li><p><strong>is_directed</strong> (<em>bool</em>) – Whether if the edge is directed or not.</p></li>
<li><p><strong>refs</strong> (<em>set</em>) – Or [list], contains the instances of the references
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.refs.Reference</span></code> for that edge.</p></li>
<li><p><strong>stim</strong> (<em>bool</em>) – Whether the edge is stimulatory or not.</p></li>
<li><p><strong>inh</strong> (<em>bool</em>) – Whether the edge is inhibitory or note</p></li>
<li><p><strong>taxon_a</strong> (<em>int</em>) – NCBI Taxonomic identifier of the source molecule.</p></li>
<li><p><strong>taxon_b</strong> (<em>int</em>) – NCBI Taxonomic identifier of the target molecule.</p></li>
<li><p><strong>typ</strong> (<em>str</em>) – The type of interaction (e.g.: <code class="docutils literal notranslate"><span class="pre">'PPI'</span></code>)</p></li>
<li><p><strong>extra_attrs</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains any extra attributes
for the edge to be updated.</p></li>
<li><p><strong>add</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If set to <code class="docutils literal notranslate"><span class="pre">True</span></code> and the
edge is not in the network, it will be created. Otherwise,
in such case it will raise an error message.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.add_update_vertex">
<code class="descname">add_update_vertex</code><span class="sig-paren">(</span><em>default_attrs</em>, <em>original_name</em>, <em>original_name_type</em>, <em>extra_attrs={}</em>, <em>add=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.add_update_vertex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.add_update_vertex" title="Permalink to this definition">¶</a></dt>
<dd><p>Updates the attributes of one node in the (undirected) network.
Optionally it creates a new node and sets the attributes, but it
is not efficient as <code class="xref py py-mod docutils literal notranslate"><span class="pre">igraph</span></code> needs to reindex vertices
after this operation, so better to create new nodes in batches.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>default_attrs</strong> (<em>dict</em>) – The attribute dictionary of the node to be updated/created.</p></li>
<li><p><strong>original_name</strong> (<em>str</em>) – Original node name (e.g.: UniProt ID).</p></li>
<li><p><strong>original_name_type</strong> (<em>str</em>) – The original node name type (e.g.: for the previous example,
this would be <code class="docutils literal notranslate"><span class="pre">'uniprot'</span></code>).</p></li>
<li><p><strong>extra_attrs</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains any extra attributes
for the node to be updated.</p></li>
<li><p><strong>add</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If set to <code class="docutils literal notranslate"><span class="pre">True</span></code> and the
node is not in the network, it will be created. Otherwise,
in such case it will raise an error message.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.affects">
<code class="descname">affects</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.affects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.affects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.all_between">
<code class="descname">all_between</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.all_between"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.all_between" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks for any edges (in any direction) between the provided
nodes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – The name of the source node.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – The name of the target node.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>dict</em>) – Contains information on the directionality of
the requested edge. Keys are <code class="docutils literal notranslate"><span class="pre">'ab'</span></code> and <code class="docutils literal notranslate"><span class="pre">'ba'</span></code>, denoting
the straight/reverse directionalities respectively. Values
are [list] whose elements are the edge ID or <code class="docutils literal notranslate"><span class="pre">None</span></code>
according to the existance of that edge in the following
categories: undirected, straight and reverse (in that
order).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.all_neighbours">
<code class="descname">all_neighbours</code><span class="sig-paren">(</span><em>indices=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.all_neighbours"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.all_neighbours" title="Permalink to this definition">¶</a></dt>
<dd><p>Looks for the first neighbours of all the nodes and creates an
attribute (<code class="docutils literal notranslate"><span class="pre">'neighbours'</span></code>) on each one of them containing a
list of their UniProt IDs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>indices</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to list the
neighbour nodes indices or their UniProt IDs.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.apply_list">
<code class="descname">apply_list</code><span class="sig-paren">(</span><em>name</em>, <em>node_or_edge='node'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.apply_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.apply_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates vertex or edge attribute based on a list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> (<em>str</em>) – The name of the list to be added as attribute. Must have
been previously loaded with
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.load_list()</span></code> or other methods.
See description of <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code>
attribute for more information.</p></li>
<li><p><strong>node_or_edge</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'node'</span></code> by default. Whether the attribute list
is to be added to the nodes or to the edges.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.apply_negative">
<code class="descname">apply_negative</code><span class="sig-paren">(</span><em>settings</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.apply_negative"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.apply_negative" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads a negative interaction source (e.g.: Negatome) into the
current network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>settings</strong> (<em>pypath.input_formats.ReadSettings</em>) – <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings</span></code> instance
containing the detailed definition of the input format to
the downloaded file. For instance
<code class="xref py py-data docutils literal notranslate"><span class="pre">pypath.data_formats.negative['negatome']</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.attach_network">
<code class="descname">attach_network</code><span class="sig-paren">(</span><em>edge_list=False</em>, <em>regulator=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.attach_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.attach_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds edges to the network from <em>edge_list</em> obtained from file or
other input method. If none is passed, checks for such data in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.raw_data</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge_list</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. The source name of the list
of edges to be added. This must have been loaded previously
(e.g.: with <a class="reference internal" href="#pypath.main.PyPath.read_data_file" title="pypath.main.PyPath.read_data_file"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.read_data_file()</span></code></a>).
If none is passed, loads the data directly from
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.raw_data</span></code>.</p></li>
<li><p><strong>regulator</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If set to <code class="docutils literal notranslate"><span class="pre">True</span></code>, non
previously existing nodes, will not be added (and hence, the
edges involved).</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.basic_stats">
<code class="descname">basic_stats</code><span class="sig-paren">(</span><em>latex=False, caption='', latex_hdr=True, fontsize=8, font='HelveticaNeueLTStd-LtCn', fname=None, header_format='%s', row_order=None, by_category=True, use_cats=['p', 'm', 'i', 'r'], urls=True, annots=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.basic_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.basic_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns basic numbers about the network resources, e.g. edge and
node counts.</p>
<dl class="simple">
<dt>latex</dt><dd><p>Return table in a LaTeX document. This can be compiled by
PDFLaTeX:
latex stats.tex</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.basic_stats_intergroup">
<code class="descname">basic_stats_intergroup</code><span class="sig-paren">(</span><em>groupA</em>, <em>groupB</em>, <em>header=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.basic_stats_intergroup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.basic_stats_intergroup" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.cancer_drivers_list">
<code class="descname">cancer_drivers_list</code><span class="sig-paren">(</span><em>intogen_file=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.cancer_drivers_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.cancer_drivers_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the list of cancer drivers. Contains information from
COSMIC (needs user log in credentials) and IntOGen (if provided)
and adds the attribute to the undirected network nodes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>intogen_file</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Path to the data file. Can
also be [function] that provides the data. In general,
anything accepted by
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings.input</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.cancer_gene_census_list">
<code class="descname">cancer_gene_census_list</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.cancer_gene_census_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.cancer_gene_census_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the list of cancer driver proteins from the COSMIC Cancer
Gene Census.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.clean_graph">
<code class="descname">clean_graph</code><span class="sig-paren">(</span><em>organisms_allowed=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.clean_graph"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.clean_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes multiple edges, unknown molecules and those from wrong
taxon. Multiple edges will be combined by
<a class="reference internal" href="#pypath.main.PyPath.combine_attr" title="pypath.main.PyPath.combine_attr"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.combine_attr()</span></code></a> method.
Loops will be deleted unless the attribute
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.loops</span></code> is set to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>organisms_allowed</strong> (<em>set</em>) – NCBI Taxonomy identifiers [int] of the organisms allowed
in the network.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.collapse_by_name">
<code class="descname">collapse_by_name</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.collapse_by_name"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.collapse_by_name" title="Permalink to this definition">¶</a></dt>
<dd><p>Collapses nodes with the same name by copying and merging
all edges and attributes. Operates directly on the provided
network object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network for which the
nodes are to be collapsed. If none is provided, takes
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.graph</span></code> (undirected network) by
default.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.combine_attr">
<code class="descname">combine_attr</code><span class="sig-paren">(</span><em>lst</em>, <em>num_method=&lt;built-in function max&gt;</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.combine_attr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.combine_attr" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines multiple attributes into one. This method attempts
to find out which is the best way to combine attributes.</p>
<blockquote>
<div><ul class="simple">
<li><p>If there is only one value or one of them is None, then
returns the one available.</p></li>
<li><p>Lists: concatenates unique values of lists.</p></li>
<li><p>Numbers: returns the greater by default or calls
<em>num_method</em> if given.</p></li>
<li><p>Sets: returns the union.</p></li>
<li><p>Dictionaries: calls <code class="xref py py-func docutils literal notranslate"><span class="pre">pypath.common.merge_dicts()</span></code>.</p></li>
<li><p>Direction: calls their special
<a class="reference internal" href="#pypath.main.Direction.merge" title="pypath.main.Direction.merge"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.Direction.merge()</span></code></a> method.</p></li>
</ul>
</div></blockquote>
<p>Works on more than 2 attributes recursively.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>list</em>) – List of one or two attribute values.</p></li>
<li><p><strong>num_method</strong> (<em>function</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">max</span></code> by default. Method to merge numeric
attributes.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.communities">
<code class="descname">communities</code><span class="sig-paren">(</span><em>method</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.communities"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.communities" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.complex_comembership_network">
<code class="descname">complex_comembership_network</code><span class="sig-paren">(</span><em>graph=None</em>, <em>resources=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.complex_comembership_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.complex_comembership_network" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.complexes">
<code class="descname">complexes</code><span class="sig-paren">(</span><em>methods=['3dcomplexes', 'havugimana', 'corum', 'complexportal', 'compleat']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.complexes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.complexes" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.complexes_in_network">
<code class="descname">complexes_in_network</code><span class="sig-paren">(</span><em>csource='corum'</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.complexes_in_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.complexes_in_network" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.compounds_from_chembl">
<code class="descname">compounds_from_chembl</code><span class="sig-paren">(</span><em>chembl_mysql=None, nodes=None, crit=None, andor='or', assay_types=['B', 'F'], relationship_types=['D', 'H'], multi_query=False, **kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.compounds_from_chembl"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.compounds_from_chembl" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads compound data from ChEMBL to the network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>chebl_mysql</strong> (<em>tuple</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Contains the MySQL parameters
used by the <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.mapping</span></code> module to loadthe
ChEMBL ID conversion tables. If none is passed, takes the
current instance <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.chembl_mysql</span></code>
attribute.</p></li>
<li><p><strong>nodes</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. List of node indices for
which the information is to be loaded. If none is provided
calls the method <a class="reference internal" href="#pypath.main.PyPath.get_sub" title="pypath.main.PyPath.get_sub"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.get_sub()</span></code></a> with
the provided <em>crit</em> parameter.</p></li>
<li><p><strong>crit</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Defines the critical
attributes to generate a subnetwork to extract the nodes in
case <em>nodes</em> is not provided. Keys are <code class="docutils literal notranslate"><span class="pre">'edge'</span></code> and
<code class="docutils literal notranslate"><span class="pre">'node'</span></code> and values are [dict] containing the critical
attribute names [str] and values are [set] containing those
attributes of the nodes/edges that are to be kept in the
subnetwork. If none is provided, takes the whole network.</p></li>
<li><p><strong>andor</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'or'</span></code> by default. Determines the search mode
for the subnetwork generation (if <code class="docutils literal notranslate"><span class="pre">nodes=None</span></code>). See
<a class="reference internal" href="#pypath.main.PyPath.search_attr_or" title="pypath.main.PyPath.search_attr_or"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_or()</span></code></a> and
<a class="reference internal" href="#pypath.main.PyPath.search_attr_and" title="pypath.main.PyPath.search_attr_and"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_and()</span></code></a> for more
details.</p></li>
<li><p><strong>assay_types</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">['B',</span> <span class="pre">'F']</span></code> by default. Types of assay to query
Options are: <code class="docutils literal notranslate"><span class="pre">'A'</span></code> (ADME), <code class="docutils literal notranslate"><span class="pre">'B'</span></code> (Binding),``’F’``
(Functional), <code class="docutils literal notranslate"><span class="pre">'P'</span></code> (Physicochemical), <code class="docutils literal notranslate"><span class="pre">'T'</span></code> (Toxicity)
and/or <code class="docutils literal notranslate"><span class="pre">'U'</span></code> (Unassigned).</p></li>
<li><p><strong>relationship_types</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">['D',</span> <span class="pre">'H']</span></code> by default. Assay relationship
types to query. Possible values are: <code class="docutils literal notranslate"><span class="pre">'D'</span></code> (Direct protein
target assigned), <code class="docutils literal notranslate"><span class="pre">'H'</span></code> (Homologous protein target
assigned), <code class="docutils literal notranslate"><span class="pre">'M'</span></code> (Molecular target other than protein
assigned), <code class="docutils literal notranslate"><span class="pre">'N'</span></code> (Non-molecular target assigned), <code class="docutils literal notranslate"><span class="pre">'S'</span></code>
(Subcellular target assigned) and/or <code class="docutils literal notranslate"><span class="pre">'U'</span></code> (Default value,
target has yet to be curated).</p></li>
<li><p><strong>multi_query</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Not used.</p></li>
<li><p><strong>**kwargs</strong> – Additional keyword arguments for
<code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.chembl.Chembl.compounds_targets()</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.consistency">
<code class="descname">consistency</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.consistency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.consistency" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.copy">
<code class="descname">copy</code><span class="sig-paren">(</span><em>other</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.copy"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.copy" title="Permalink to this definition">¶</a></dt>
<dd><p>Copies another <a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.main.PyPath</span></code></a> instance into the
current one.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>other</strong> (<a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><em>pypath.main.PyPath</em></a>) – The instance to be copied from.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.copy_edges">
<code class="descname">copy_edges</code><span class="sig-paren">(</span><em>sources</em>, <em>target</em>, <em>move=False</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.copy_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.copy_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Copies edges from <em>sources</em> node(s) to another one (<em>target</em>),
keeping attributes and directions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sources</strong> (<em>list</em>) – Contains the vertex index(es) [int] of the node(s) to be
copied or moved.</p></li>
<li><p><strong>target</strong> (<em>int</em>) – Vertex index where edges and attributes are to be copied to.</p></li>
<li><p><strong>move</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to perform copy or
move (remove or keep the source edges).</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network graph object from
which the nodes are to be merged. If none is passed, takes
the undirected network graph.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.count_sol">
<code class="descname">count_sol</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.count_sol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.count_sol" title="Permalink to this definition">¶</a></dt>
<dd><p>Counts the number of nodes with zero degree.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>int</em>) – The number of nodes with zero degree.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.coverage">
<code class="descname">coverage</code><span class="sig-paren">(</span><em>lst</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.coverage"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.coverage" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the coverage (range [0, 1]) of a list of nodes against
the current (undirected) network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>lst</strong> (<em>set</em>) – Can also be [list] (will be converted to [set]) or [str]. In
the latter case it will retrieve the list with that name (if
such list exists in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code>).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.curation_effort">
<code class="descname">curation_effort</code><span class="sig-paren">(</span><em>sum_by_source=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.curation_effort"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.curation_effort" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the total number of reference-interactions pairs.</p>
<dl class="simple">
<dt>&#64;sum_by_source<span class="classifier">bool</span></dt><dd><p>If True, counts the refrence-interaction pairs by
sources, and returns the sum of these values.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.curation_stats">
<code class="descname">curation_stats</code><span class="sig-paren">(</span><em>by_category=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.curation_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.curation_stats" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.curation_tab">
<code class="descname">curation_tab</code><span class="sig-paren">(</span><em>fname='curation_stats.tex', by_category=True, use_cats=['p', 'm', 'i', 'r'], header_size='normalsize', **kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.curation_tab"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.curation_tab" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.curators_work">
<code class="descname">curators_work</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.curators_work"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.curators_work" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes and prints an estimation of how many years of curation
took to achieve the amount of information on the network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.databases_similarity">
<code class="descname">databases_similarity</code><span class="sig-paren">(</span><em>index='simpson'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.databases_similarity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.databases_similarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the similarity across databases according to a given
index metric. Computes the similarity across the loaded
resources (listed in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.sources</span></code> in
terms of nodes and edges separately.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>index</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'simpson'</span></code> by default. The type of index metric
to use to compute the similarity. Options are <code class="docutils literal notranslate"><span class="pre">'simpson'</span></code>,
<code class="docutils literal notranslate"><span class="pre">'sorensen'</span></code> and <code class="docutils literal notranslate"><span class="pre">'jaccard'</span></code>.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>dict</em>) – Nested dictionaries (three levels). First-level
keys are <code class="docutils literal notranslate"><span class="pre">'nodes'</span></code> and <code class="docutils literal notranslate"><span class="pre">'edges'</span></code>, then second and third
levels correspond to sources names which map to the
similarity index between those sources [float].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.degree_dist">
<code class="descname">degree_dist</code><span class="sig-paren">(</span><em>prefix</em>, <em>g=None</em>, <em>group=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.degree_dist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.degree_dist" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the degree distribution over all nodes of the network.
If <em>group</em> is provided, also across nodes of that group(s).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>prefix</strong> (<em>str</em>) – Prefix for the file name(s).</p></li>
<li><p><strong>g</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network over which to
compute the degree distribution. If none is passed, takes
the undirected network of the current instance.</p></li>
<li><p><strong>group</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Additional group(s) name(s)
[str] of node attributes to subset the network and compute
its degree distribution.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.degree_dists">
<code class="descname">degree_dists</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.degree_dists"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.degree_dists" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the degree distribution for all the different network
sources. This is, for each source, the subnetwork comprising all
interactions coming from it is extracted and the degree
distribution information is computed and saved into a file.
A file is created for each resource under the name
<code class="docutils literal notranslate"><span class="pre">'pwnet-&lt;session_id&gt;-degdist-&lt;resource&gt;''</span></code>. Files are stored
in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by
default).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.delete_by_organism">
<code class="descname">delete_by_organism</code><span class="sig-paren">(</span><em>organisms_allowed=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.delete_by_organism"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.delete_by_organism" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes the proteins of all organisms which are not given in
<em>tax</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>organisms_allowed</strong> (<em>list</em><em>,</em><em>set</em>) – List of NCBI Taxonomy IDs [int] of the organism(s) that are
to be kept.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.delete_by_source">
<code class="descname">delete_by_source</code><span class="sig-paren">(</span><em>source</em>, <em>vertexAttrsToDel=None</em>, <em>edgeAttrsToDel=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.delete_by_source"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.delete_by_source" title="Permalink to this definition">¶</a></dt>
<dd><p>Deletes nodes and edges from the network according to a provided
source name. Optionally can also remove the given list of
attributes from nodes and/or edges.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>source</strong> (<em>str</em>) – Name of the source from which the nodes and edges have to be
removed.</p></li>
<li><p><strong>vertexAttrsToDel</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Contains the names [str] of
the attributes to be removed from the nodes.</p></li>
<li><p><strong>edgeAttrsToDel</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Contains the names [str] of
the attributes to be removed from the edges.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.delete_unknown">
<code class="descname">delete_unknown</code><span class="sig-paren">(</span><em>organisms_allowed=None</em>, <em>entity_type='protein'</em>, <em>default_name_type=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.delete_unknown"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.delete_unknown" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes those items which are not in the list of all default
IDs of the organisms. By default, it means to remove all protein
nodes not having a human UniProt ID.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>typ</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'protein'</span></code> by default. Determines the molecule
type. These can be <code class="docutils literal notranslate"><span class="pre">'protein'</span></code>, <code class="docutils literal notranslate"><span class="pre">'drug'</span></code>, <code class="docutils literal notranslate"><span class="pre">'lncrna'</span></code>,
<code class="docutils literal notranslate"><span class="pre">'mirna'</span></code> or any other type defined in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.default_name_type</span></code>.</p></li>
<li><p><strong>default_name_type</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The default name type for the
given molecular species. If none is specified takes it from
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.default_name_type</span></code> (e.g.: for
<code class="docutils literal notranslate"><span class="pre">'protein'</span></code>, default is <code class="docutils literal notranslate"><span class="pre">'uniprot'</span></code>).</p></li>
<li><p><strong>organisms_allowed</strong> (<em>set</em>) – NCBI Taxonomy identifiers [int] of the organisms allowed in
the network.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.delete_unmapped">
<code class="descname">delete_unmapped</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.delete_unmapped"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.delete_unmapped" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks the network for any existing unmapped node and removes
it.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dgenesymbol">
<code class="descname">dgenesymbol</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.dgenesymbol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.dgenesymbol" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the GeneSymbol
can be found in the default directed network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;genesymbol<span class="classifier">str</span></dt><dd><p>GeneSymbol.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dgenesymbols">
<code class="descname">dgenesymbols</code><span class="sig-paren">(</span><em>genesymbols</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.dgenesymbols"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.dgenesymbols" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dgs">
<code class="descname">dgs</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the GeneSymbol
can be found in the default directed network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;genesymbol<span class="classifier">str</span></dt><dd><p>GeneSymbol.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dgss">
<code class="descname">dgss</code><span class="sig-paren">(</span><em>genesymbols</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dgss" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dneighbors">
<code class="descname">dneighbors</code><span class="sig-paren">(</span><em>identifier</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.dneighbors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.dneighbors" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dp">
<code class="descname">dp</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dp" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.get_node</span></code>, just for the directed graph.
Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dproteins">
<code class="descname">dproteins</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dproteins" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dps">
<code class="descname">dps</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dps" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.duniprot">
<code class="descname">duniprot</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.duniprot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.duniprot" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.uniprot(),</span> <span class="pre">just</span> <span class="pre">for</span> <span class="pre">directed</span> <span class="pre">graph.</span>
<span class="pre">Returns</span> <span class="pre">``igraph.Vertex()</span></code> object if the UniProt
can be found in the default directed network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;uniprot<span class="classifier">str</span></dt><dd><p>UniProt ID.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.duniprots">
<code class="descname">duniprots</code><span class="sig-paren">(</span><em>uniprots</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.duniprots"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.duniprots" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns list of <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object
for a list of UniProt IDs omitting those
could not be found in the default
directed graph.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dup">
<code class="descname">dup</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dup" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.uniprot(),</span> <span class="pre">just</span> <span class="pre">for</span> <span class="pre">directed</span> <span class="pre">graph.</span>
<span class="pre">Returns</span> <span class="pre">``igraph.Vertex()</span></code> object if the UniProt
can be found in the default directed network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;uniprot<span class="classifier">str</span></dt><dd><p>UniProt ID.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dups">
<code class="descname">dups</code><span class="sig-paren">(</span><em>uniprots</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dups" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns list of <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object
for a list of UniProt IDs omitting those
could not be found in the default
directed graph.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dv">
<code class="descname">dv</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dv" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.get_node</span></code>, just for the directed graph.
Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.dvs">
<code class="descname">dvs</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.dvs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edge_exists">
<code class="descname">edge_exists</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edge_exists"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edge_exists" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a tuple of vertex indices if edge doesn’t exist,
otherwise, the edge ID. Not sensitive to direction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – Name of the source node.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – Name of the target node.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>int</em>) – The edge index, if exists such edge. Otherwise,
[tuple] of [int] corresponding to the node IDs.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edge_loc">
<code class="descname">edge_loc</code><span class="sig-paren">(</span><em>graph=None</em>, <em>topn=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edge_loc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edge_loc" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edge_names">
<code class="descname">edge_names</code><span class="sig-paren">(</span><em>e</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edge_names"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edge_names" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the node names of a given edge.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>e</strong> (<em>int</em>) – The edge index.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>tuple</em>) – Contains the source and target node names of
the edge [str].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edges_3d">
<code class="descname">edges_3d</code><span class="sig-paren">(</span><em>methods=['dataio.get_instruct', 'dataio.get_i3d']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edges_3d"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edges_3d" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edges_between">
<code class="descname">edges_between</code><span class="sig-paren">(</span><em>group1</em>, <em>group2</em>, <em>directed=True</em>, <em>strict=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edges_between"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edges_between" title="Permalink to this definition">¶</a></dt>
<dd><p>Selects edges between two groups of vertex IDs.
Returns set of edge IDs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>group1</strong><strong>,</strong><strong>group2</strong> (<em>set</em>) – List, set or tuple of vertex IDs.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – Only edges with direction <cite>group1 -&gt; group2</cite> selected.</p></li>
<li><p><strong>strict</strong> (<em>bool</em>) – Edges with no direction information still selected even if
<code class="docutils literal notranslate"><span class="pre">directed</span></code> is <cite>False</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edges_expression">
<code class="descname">edges_expression</code><span class="sig-paren">(</span><em>func=&lt;function PyPath.&lt;lambda&gt;&gt;</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edges_expression"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edges_expression" title="Permalink to this definition">¶</a></dt>
<dd><p>Executes function <cite>func</cite> for each pairs of connected proteins in the
network, for every expression dataset. By default, <cite>func</cite> simply
gives the product the (normalized) expression values.</p>
<dl class="simple">
<dt>func<span class="classifier">callable</span></dt><dd><p>Function to handle 2 vectors (pandas.Series() objects), should
return one vector of the same length.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edges_in_comlexes">
<code class="descname">edges_in_comlexes</code><span class="sig-paren">(</span><em>csources=['corum'], graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edges_in_comlexes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edges_in_comlexes" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates edge attributes <code class="docutils literal notranslate"><span class="pre">complexes</span></code> and <code class="docutils literal notranslate"><span class="pre">in_complex</span></code>.
These are both dicts where the keys are complex resources.
The values in <code class="docutils literal notranslate"><span class="pre">complexes</span></code> are the list of complex names
both the source and the target vertices belong to.
The values <code class="docutils literal notranslate"><span class="pre">in_complex</span></code> are boolean values whether there
is at least one complex in the given resources both the
source and the target vertex of the edge belong to.</p>
<dl class="simple">
<dt>&#64;csources<span class="classifier">list</span></dt><dd><p>List of complex resources. Should be already loaded.</p>
</dd>
<dt>&#64;graph<span class="classifier">igraph.Graph()</span></dt><dd><p>The graph object to do the calculations on.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edges_ptms">
<code class="descname">edges_ptms</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edges_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edges_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.edgeseq_inverse">
<code class="descname">edgeseq_inverse</code><span class="sig-paren">(</span><em>edges</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.edgeseq_inverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.edgeseq_inverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of all edge indexes that are not in
the argument <em>edges</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>edges</strong> (<em>set</em>) – Sequence of edge indices [int] that will not be returned.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains all edge indices [int] of the
undirected network except the ones on <em>edges</em> argument.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.entities_by_resources">
<code class="descname">entities_by_resources</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.entities_by_resources"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.entities_by_resources" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a dict of sets with resources as keys and sets of entity IDs
as values.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_dot">
<code class="descname">export_dot</code><span class="sig-paren">(</span><em>nodes=None</em>, <em>edges=None</em>, <em>directed=True</em>, <em>labels='genesymbol'</em>, <em>edges_filter=&lt;function PyPath.&lt;lambda&gt;&gt;</em>, <em>nodes_filter=&lt;function PyPath.&lt;lambda&gt;&gt;</em>, <em>edge_sources=None</em>, <em>dir_sources=None</em>, <em>graph=None</em>, <em>return_object=False</em>, <em>save_dot=None</em>, <em>save_graphics=None</em>, <em>prog='neato'</em>, <em>format=None</em>, <em>hide=False</em>, <em>font=None</em>, <em>auto_edges=False</em>, <em>hide_nodes=[]</em>, <em>defaults={}</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_dot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_dot" title="Permalink to this definition">¶</a></dt>
<dd><p>Builds a pygraphviz.AGraph() object with filtering the edges
and vertices along arbitrary criteria.
Returns the Agraph object if requesred, or exports the dot
file, or saves the graphics.</p>
<p>&#64;nodes : list
List of vertex ids to be included.
&#64;edges : list
List of edge ids to be included.
&#64;directed : bool
Create a directed or undirected graph.
&#64;labels : str
Name type to be used as id/label in the dot format.
&#64;edges_filter : function
Function to filter edges, accepting igraph.Edge as argument.
&#64;nodes_filter : function
Function to filter vertices, accepting igraph.Vertex as argument.
&#64;edge_sources : list
Sources to be included.
&#64;dir_sources : list
Direction and effect sources to be included.
&#64;graph : igraph.Graph
The graph object to export.
&#64;return_object : bool
Whether to return the pygraphviz.AGraph object.
&#64;save_dot : str
Filename to export the dot file to.
&#64;save_graphics : str
Filename to export the graphics, the extension defines the format.
&#64;prog : str
The graphviz layout algorithm to use.
&#64;format : str
The graphics format passed to pygraphviz.AGrapg().draw().
&#64;hide : bool
Hide filtered edges instead of omit them.
&#64;hide nodes : list
Nodes to hide. List of vertex ids.
&#64;auto_edges : str
Automatic, built-in style for edges.
‘DIRECTIONS’ or ‘RESOURCE_CATEGORIES’ are supported.
&#64;font : str
Font to use for labels.
For using more than one fonts refer to graphviz attributes with constant values
or define callbacks or mapping dictionaries.
&#64;defaults : dict
Default values for graphviz attributes, labeled with the entity, e.g.
<cite>{‘edge_penwidth’: 0.2}</cite>.
&#64;**kwargs : constant, callable or dict
Graphviz attributes, labeled by the target entity. E.g. <cite>edge_penwidth</cite>,
‘vertex_shape` or <cite>graph_label</cite>.
If the value is constant, this value will be used.
If the value is dict, and has <cite>_name</cite> as key, for every instance of the
given entity, the value of the attribute defined by <cite>_name</cite> will be looked
up in the dict, and the corresponding value will be given to the graphviz
attribute. If the key <cite>_name</cite> is missing from the dict, igraph vertex and
edge indices will be looked up among the keys.
If the value is callable, it will be called with the current instance of
the entity and the returned value will be used for the graphviz attribute.
E.g. <cite>edge_arrowhead(edge)</cite> or <cite>vertex_fillcolor(vertex)</cite>
Example:</p>
<blockquote>
<div><p>import pypath
from pypath import data_formats
net = pypath.PyPath()
net.init_network(pfile = ‘cache/default.pickle’)
#net.init_network({‘arn’: data_formats.omnipath[‘arn’]})
tgf = [v.index for v in net.graph.vs if ‘TGF’ in v[‘slk_pathways’]]
dot = net.export_dot(nodes = tgf, save_graphics = ‘tgf_slk.pdf’, prog = ‘dot’,</p>
<blockquote>
<div><p>main_title = ‘TGF-beta pathway’, return_object = True,
label_font = ‘HelveticaNeueLTStd Med Cn’,
edge_sources = [‘SignaLink3’],
dir_sources = [‘SignaLink3’], hide = True)</p>
</div></blockquote>
</div></blockquote>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_edgelist">
<code class="descname">export_edgelist</code><span class="sig-paren">(</span><em>fname, graph=None, names=['name'], edge_attributes=[], sep='\t'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_edgelist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_edgelist" title="Permalink to this definition">¶</a></dt>
<dd><p>Write edge list to text file with attributes</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>fname</strong> – the name of the file or a stream to read from.</p></li>
<li><p><strong>graph</strong> – the igraph object containing the network</p></li>
<li><p><strong>names</strong> – list with the vertex attribute names to be printed
for source and target vertices</p></li>
<li><p><strong>edge_attributes</strong> – list with the edge attribute names
to be printed</p></li>
<li><p><strong>sep</strong> – string used to separate columns</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_graphml">
<code class="descname">export_graphml</code><span class="sig-paren">(</span><em>outfile=None</em>, <em>graph=None</em>, <em>name='main'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_graphml"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_graphml" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves the network in a <code class="docutils literal notranslate"><span class="pre">.graphml</span></code> file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Name/path of the output file.
If none is passed,
<code class="docutils literal notranslate"><span class="pre">'results/netrowk-&lt;session_id&gt;.graphml'</span></code> is used.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network object to be
saved. If none is passed, takes the undirected network of
the current instance.</p></li>
<li><p><strong>name</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'main'</span></code> by default. The graph name for the
output file.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_ptms_tab">
<code class="descname">export_ptms_tab</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_ptms_tab"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_ptms_tab" title="Permalink to this definition">¶</a></dt>
<dd><p>Exports a tab file containing the PTM interaction information
loaded in the network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The output file nama/path to
store the PTM information. If none is provided, the default
is <code class="docutils literal notranslate"><span class="pre">'results/network-&lt;session_id&gt;.tab'</span></code>.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the edge indices [int] of all PTM
interactions.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_sif">
<code class="descname">export_sif</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_sif"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_sif" title="Permalink to this definition">¶</a></dt>
<dd><p>Exports the network interactions in <code class="docutils literal notranslate"><span class="pre">.sif</span></code> format (Simple
Interaction Format).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Name/path of the output file.
If none is passed, <code class="docutils literal notranslate"><span class="pre">'results/netrowk-&lt;session_id&gt;.sif'</span></code>
is used.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_struct_tab">
<code class="descname">export_struct_tab</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_struct_tab"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_struct_tab" title="Permalink to this definition">¶</a></dt>
<dd><p>Exports a tab file containing the domain interaction information
and PTM regulation loaded in the network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The output file nama/path to
store the PTM information. If none is provided, the default
is <code class="docutils literal notranslate"><span class="pre">'results/network-&lt;session_id&gt;.tab'</span></code>.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the edge indices [int] of all PTM
interactions.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.export_tab">
<code class="descname">export_tab</code><span class="sig-paren">(</span><em>outfile=None</em>, <em>extra_node_attrs={}</em>, <em>extra_edge_attrs={}</em>, <em>unique_pairs=True</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.export_tab"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.export_tab" title="Permalink to this definition">¶</a></dt>
<dd><p>Exports the network in a tabular format. By default UniProt IDs,
Gene Symbols, source databases, literature references,
directionality and sign information and interaction type are
included.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Name/path of the output file.
If none is passed, <code class="docutils literal notranslate"><span class="pre">'results/netrowk-&lt;session_id&gt;.tab'</span></code>
is used.</p></li>
<li><p><strong>extra_node_attrs</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Additional node attributes to
be included in the exported table. Keys are column names
used in the header while values are names of vertex
attributes. In the header <code class="docutils literal notranslate"><span class="pre">'_A'</span></code> and <code class="docutils literal notranslate"><span class="pre">'_B'</span></code> suffixes
will be appended to the column names so the values can be
assigned to A and B side interaction partners.</p></li>
<li><p><strong>extra_edge_attrs</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Additional edge attributes to
be included in the exported table. Keys are column names
used in the header while values are names of edge
attributes.</p></li>
<li><p><strong>unique_pairs</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. If set to <code class="docutils literal notranslate"><span class="pre">True</span></code> each line
corresponds to a unique pair of molecules, all
directionality and sign information are covered in other
columns. If <code class="docutils literal notranslate"><span class="pre">False</span></code>, order of <code class="docutils literal notranslate"><span class="pre">'A'</span></code> and <code class="docutils literal notranslate"><span class="pre">'B'</span></code> IDs
corresponds to the direction while sign covered in further
columns.</p></li>
<li><p><strong>kwargs</strong> (<em>**</em>) – Additional keyword arguments passed to
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.export.Export</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.filters">
<code class="descname">filters</code><span class="sig-paren">(</span><em>line</em>, <em>positive_filters=[]</em>, <em>negative_filters=[]</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.filters"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.filters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.find_all_paths">
<code class="descname">find_all_paths</code><span class="sig-paren">(</span><em>start</em>, <em>end</em>, <em>attr=None</em>, <em>mode='OUT'</em>, <em>maxlen=2</em>, <em>graph=None</em>, <em>silent=False</em>, <em>update_adjlist=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.find_all_paths"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.find_all_paths" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds all paths up to length <cite>maxlen</cite> between groups of
vertices. This function is needed only becaues igraph`s
get_all_shortest_paths() finds only the shortest, not any
path up to a defined length.</p>
<dl class="simple">
<dt>start<span class="classifier">int or list</span></dt><dd><p>Indices of the starting node(s) of the paths.</p>
</dd>
<dt>end<span class="classifier">int or list</span></dt><dd><p>Indices of the target node(s) of the paths.</p>
</dd>
<dt>attr<span class="classifier">str</span></dt><dd><p>Name of the vertex attribute to identify the vertices by.
Necessary if <code class="docutils literal notranslate"><span class="pre">start</span></code> and <code class="docutils literal notranslate"><span class="pre">end</span></code> are not igraph vertex ids
but for example vertex names or labels.</p>
</dd>
<dt>mode<span class="classifier">‘IN’, ‘OUT’, ‘ALL’</span></dt><dd><p>Passed to igraph.Graph.neighbors()</p>
</dd>
<dt>maxlen<span class="classifier">int</span></dt><dd><p>Maximum length of paths in steps, i.e. if maxlen = 3, then
the longest path may consist of 3 edges and 4 nodes.</p>
</dd>
<dt>graph<span class="classifier">igraph.Graph object</span></dt><dd><p>The graph you want to find paths in. self.graph by default.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.find_all_paths2">
<code class="descname">find_all_paths2</code><span class="sig-paren">(</span><em>graph</em>, <em>start</em>, <em>end</em>, <em>mode='OUT'</em>, <em>maxlen=2</em>, <em>psize=100</em>, <em>update_adjlist=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.find_all_paths2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.find_all_paths2" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.find_complex">
<code class="descname">find_complex</code><span class="sig-paren">(</span><em>search</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.find_complex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.find_complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds complexes by their non standard names.
E.g. to find DNA polymerases you can use the search
term <cite>DNA pol</cite> which will be tested against complex names
in CORUM.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.first_neighbours">
<code class="descname">first_neighbours</code><span class="sig-paren">(</span><em>node</em>, <em>indices=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.first_neighbours"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.first_neighbours" title="Permalink to this definition">¶</a></dt>
<dd><p>Looks for the first neighbours of a given node and returns a
list of their UniProt IDs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>node</strong> (<em>str</em>) – The UniProt ID of the node of interest. Can also be the
index of such node [int].</p></li>
<li><p><strong>indices</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return the
neighbour nodes indices or their UniProt IDs.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – The list containing the first neighbours of the
queried node.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.fisher_enrichment">
<code class="descname">fisher_enrichment</code><span class="sig-paren">(</span><em>lst</em>, <em>attr</em>, <em>ref='proteome'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.fisher_enrichment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.fisher_enrichment" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes an enrichment analysis using Fisher’s exact test. The
contingency table is built as follows:
First row contains the number of nodes in the <em>ref</em> list (such
list is considered to be loaded in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code>) and the number of nodes in
the current (undirected) network. Second row contains the number
of nodes in <em>lst</em> list (also considered to be already loaded)
and the number of nodes in the network with a non-empty
attribute <em>attr</em>. Uses <code class="xref py py-func docutils literal notranslate"><span class="pre">scipy.stats.fisher_exact()</span></code>, see
the documentation of the corresponding package for more
information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>str</em>) – Name of the list in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code>
whose number of elements will be the first element in the
second row of the contingency table.</p></li>
<li><p><strong>attr</strong> (<em>str</em>) – The node attribute name for which the number of nodes in the
network with such attribute will be the second element of
the second row of the contingency table.</p></li>
<li><p><strong>ref</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'proteome'</span></code> by default. The name of the list in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code> whose number of elements
will be the first element of the first row of the
contingency table.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><ul class="simple">
<li><p>(<em>float</em>) – Prior odds ratio.</p></li>
<li><p>(<em>float</em>) – P-value or probability of obtaining a
distribution as extreme as the observed, assuming that the
null hypothesis is true.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.geneset_enrichment">
<code class="descname">geneset_enrichment</code><span class="sig-paren">(</span><em>proteins</em>, <em>all_proteins=None</em>, <em>geneset_ids=None</em>, <em>alpha=0.05</em>, <em>correction_method='hommel'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.geneset_enrichment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.geneset_enrichment" title="Permalink to this definition">¶</a></dt>
<dd><p>Does not work at the moment because cfisher module should be
replaced with scipy.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.genesymbol">
<code class="descname">genesymbol</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.genesymbol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.genesymbol" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the GeneSymbol
can be found in the default undirected network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;genesymbol<span class="classifier">str</span></dt><dd><p>GeneSymbol.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.genesymbol_labels">
<code class="descname">genesymbol_labels</code><span class="sig-paren">(</span><em>graph=None</em>, <em>remap_all=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.genesymbol_labels"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.genesymbol_labels" title="Permalink to this definition">¶</a></dt>
<dd><p>Creats vertex attribute <code class="docutils literal notranslate"><span class="pre">'label'</span></code> and fills up with the
corresponding GeneSymbols of all proteins where the GeneSymbol
can be looked up based on the default name of the protein
vertex (UniProt ID by default). If the attribute <code class="docutils literal notranslate"><span class="pre">'label'</span></code> has
been already initialized, updates this attribute or recreates if
<em>remap_all</em> is set to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network graph object
where the GeneSymbol labels are to be set/updated. If none
is passed, takes the current network undirected graph by
default (<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.graph</span></code>).</p></li>
<li><p><strong>remap_all</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to map anew the
GeneSymbol labels if those were already initialized.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.genesymbols">
<code class="descname">genesymbols</code><span class="sig-paren">(</span><em>genesymbols</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.genesymbols"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.genesymbols" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_attrs">
<code class="descname">get_attrs</code><span class="sig-paren">(</span><em>line</em>, <em>spec</em>, <em>lnum</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_attrs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_directed">
<code class="descname">get_directed</code><span class="sig-paren">(</span><em>graph=False</em>, <em>conv_edges=False</em>, <em>mutual=False</em>, <em>ret=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_directed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_directed" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a copy of <em>graph</em> undirected <em>igraph.Graph</em> object to a
directed one. By default it converts the current network
instance in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.graph</span></code> and places the
copy of the directed instance in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.dgraph</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Undirected graph object. If
none is passed, takes the current undirected network
instance and saves the directed network under the attribute
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.dgraph</span></code>. Otherwise, the
directed graph will be returned instead.</p></li>
<li><p><strong>conv_edges</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to convert
undirected edges (those without explicit direction
information) to an arbitrary direction edge or
a pair of opposite edges. Otherwise those will be deleted.</p></li>
<li><p><strong>mutual</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If <em>conv_edges</em> is <code class="docutils literal notranslate"><span class="pre">True</span></code>,
whether to convert the undirected edges to a single,
arbitrary directed edge, or a pair of opposite directed
edges.</p></li>
<li><p><strong>ret</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return the
directed graph instance, or not. If a <em>graph</em> is provided,
its directed version will be returned anyway.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>igraph.Graph</em>) – If <em>graph</em> is passed or <em>ret</em> is
<code class="docutils literal notranslate"><span class="pre">True</span></code>, returns the copy of the directed graph. otherwise
returns <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_dirs_signs">
<code class="descname">get_dirs_signs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_dirs_signs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_dirs_signs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_edge">
<code class="descname">get_edge</code><span class="sig-paren">(</span><em>source</em>, <em>target</em>, <em>directed=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_edge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Edge</span></code> object if an edge exist between
the 2 proteins, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>source</strong> (<em>int</em><em>,</em><em>str</em>) – Vertex index or UniProt ID or GeneSymbol or <code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code>
object.</p></li>
<li><p><strong>target</strong> (<em>int</em><em>,</em><em>str</em>) – Vertex index or UniProt ID or GeneSymbol or <code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code>
object.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – To be passed to igraph.Graph.get_eid()</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_edges">
<code class="descname">get_edges</code><span class="sig-paren">(</span><em>sources</em>, <em>targets</em>, <em>directed=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a generator with all edges between source and target vertices.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sources</strong> (<em>iterable</em>) – Source vertex IDs, names, labels, or any iterable yielding
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> objects.</p></li>
<li><p><strong>targets</strong> (<em>iterable</em>) – Target vertec IDs, names, labels, or any iterable yielding
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> objects.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – Passed to <code class="docutils literal notranslate"><span class="pre">igraph.get_eid()</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_function">
<code class="descname">get_function</code><span class="sig-paren">(</span><em>fun</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_function"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_function" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_giant">
<code class="descname">get_giant</code><span class="sig-paren">(</span><em>replace=False</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_giant"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_giant" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the giant component of the <em>graph</em>, or replaces the
<code class="xref py py-class docutils literal notranslate"><span class="pre">igraph.Graph</span></code> instance with only the giant component
if specified.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>replace</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to replace
the <code class="xref py py-class docutils literal notranslate"><span class="pre">igraph.Graph</span></code> instance. This can be either
the undirected network of the current
<a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.main.PyPath</span></code></a> instance (default) or the one
passed under the keyword argument <em>graph</em>.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object from which
the giant component is to be computed. If none is specified,
takes the undirected network of the current
<a class="reference internal" href="#pypath.main.PyPath" title="pypath.main.PyPath"><code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.main.PyPath</span></code></a> instance.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>igraph.Graph</em>) – If <code class="docutils literal notranslate"><span class="pre">replace=False</span></code>, returns a copy of
the giant component graph.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_go">
<code class="descname">get_go</code><span class="sig-paren">(</span><em>organism=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_go"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_go" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the <code class="docutils literal notranslate"><span class="pre">GOAnnotation</span></code> object for the organism requested
(or the default one).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_max">
<code class="descname">get_max</code><span class="sig-paren">(</span><em>attrList</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_max"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_max" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_network">
<code class="descname">get_network</code><span class="sig-paren">(</span><em>crit</em>, <em>andor='or'</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a subnetwork according to a set of user-defined
attributes. Basically applies
<a class="reference internal" href="#pypath.main.PyPath.get_sub" title="pypath.main.PyPath.get_sub"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.get_sub()</span></code></a> on a given <em>graph</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>crit</strong> (<em>dict</em>) – Defines the critical attributes to generate the subnetwork.
Keys are <code class="docutils literal notranslate"><span class="pre">'edge'</span></code> and <code class="docutils literal notranslate"><span class="pre">'node'</span></code> and values are [dict]
containing the critical attribute names [str] and values
are [set] containing those attributes of the nodes/edges
that are to be kept.</p></li>
<li><p><strong>andor</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'or'</span></code> by default. Determines the search mode.
See <a class="reference internal" href="#pypath.main.PyPath.search_attr_or" title="pypath.main.PyPath.search_attr_or"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_or()</span></code></a> and
<a class="reference internal" href="#pypath.main.PyPath.search_attr_and" title="pypath.main.PyPath.search_attr_and"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_and()</span></code></a> for more
details.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where to
extract the subnetwork. If none is passed, takes the current
network (undirected) graph
(<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.graph</span></code>).</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>igraph.Graph</em>) – The subgraph obtained from filtering
according to the attributes defined in <em>crit</em>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_node">
<code class="descname">get_node</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_node"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_node" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_node_d">
<code class="descname">get_node_d</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_node_d"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_node_d" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.get_node</span></code>, just for the directed graph.
Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_node_pair">
<code class="descname">get_node_pair</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em>, <em>directed=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_node_pair"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_node_pair" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the node IDs from a pair of node names.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – Name of the source node.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – Name of the target node.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return the node
indices from the directed or undirected graph.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>tuple</em>) – The pair of node IDs of the selected graph.
If not found, returns <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_nodes">
<code class="descname">get_nodes</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_nodes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_nodes" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_nodes_d">
<code class="descname">get_nodes_d</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_nodes_d"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_nodes_d" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_pathways">
<code class="descname">get_pathways</code><span class="sig-paren">(</span><em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_pathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_proteomicsdb">
<code class="descname">get_proteomicsdb</code><span class="sig-paren">(</span><em>user</em>, <em>passwd</em>, <em>tissues=None</em>, <em>pickle=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_proteomicsdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_proteomicsdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_sub">
<code class="descname">get_sub</code><span class="sig-paren">(</span><em>crit</em>, <em>andor='or'</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_sub"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_sub" title="Permalink to this definition">¶</a></dt>
<dd><p>Selects the nodes from <em>graph</em> (and edges to be removed)
according to a set of user-defined attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>crit</strong> (<em>dict</em>) – Defines the critical attributes to generate the subnetwork.
Keys are <code class="docutils literal notranslate"><span class="pre">'edge'</span></code> and <code class="docutils literal notranslate"><span class="pre">'node'</span></code> and values are [dict]
containing the critical attribute names [str] and values
are [set] containing those attributes of the nodes/edges
that are to be kept.</p></li>
<li><p><strong>andor</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'or'</span></code> by default. Determines the search mode.
See <a class="reference internal" href="#pypath.main.PyPath.search_attr_or" title="pypath.main.PyPath.search_attr_or"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_or()</span></code></a> and
<a class="reference internal" href="#pypath.main.PyPath.search_attr_and" title="pypath.main.PyPath.search_attr_and"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.search_attr_and()</span></code></a> for more
details.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where to
extract the subnetwork. If none is passed, takes the current
network (undirected) graph
(<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.graph</span></code>).</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>dict</em>) – Keys are <code class="docutils literal notranslate"><span class="pre">'nodes'</span></code> and <code class="docutils literal notranslate"><span class="pre">'edges'</span></code> whose
values are [lst] of elements (as indexes [int]). Nodes are
those to be kept and edges to be removed on the extracted
subnetwork.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.get_taxon">
<code class="descname">get_taxon</code><span class="sig-paren">(</span><em>tax_dict</em>, <em>fields</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.get_taxon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.get_taxon" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.go_annotate_graph">
<code class="descname">go_annotate_graph</code><span class="sig-paren">(</span><em>aspects=('C'</em>, <em>'F'</em>, <em>'P')</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.go_annotate_graph"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.go_annotate_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Annotates protein nodes with GO terms. In the <code class="docutils literal notranslate"><span class="pre">go</span></code> vertex
attribute each node is annotated by a dict of sets where keys are
one letter codes of GO aspects and values are sets of GO accessions.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.go_enrichment">
<code class="descname">go_enrichment</code><span class="sig-paren">(</span><em>proteins=None</em>, <em>aspect='P'</em>, <em>alpha=0.05</em>, <em>correction_method='hommel'</em>, <em>all_proteins=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.go_enrichment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.go_enrichment" title="Permalink to this definition">¶</a></dt>
<dd><p>Does not work at the moment because cfisher module should be
replaced with scipy.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs">
<code class="descname">gs</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.gs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the GeneSymbol
can be found in the default undirected network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;genesymbol<span class="classifier">str</span></dt><dd><p>GeneSymbol.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_affected_by">
<code class="descname">gs_affected_by</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_affected_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_affected_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_affects">
<code class="descname">gs_affects</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_affects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_affects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_edge">
<code class="descname">gs_edge</code><span class="sig-paren">(</span><em>source</em>, <em>target</em>, <em>directed=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_edge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Edge</span></code> object if an edge exist between
the 2 proteins, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;source<span class="classifier">str</span></dt><dd><p>GeneSymbol</p>
</dd>
<dt>&#64;target<span class="classifier">str</span></dt><dd><p>GeneSymbol</p>
</dd>
<dt>&#64;directed<span class="classifier">bool</span></dt><dd><p>To be passed to igraph.Graph.get_eid()</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_in_directed">
<code class="descname">gs_in_directed</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_in_directed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_in_directed" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_in_undirected">
<code class="descname">gs_in_undirected</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_in_undirected"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_in_undirected" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_inhibited_by">
<code class="descname">gs_inhibited_by</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_inhibited_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_inhibited_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_inhibits">
<code class="descname">gs_inhibits</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_inhibits"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_inhibits" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_neighborhood">
<code class="descname">gs_neighborhood</code><span class="sig-paren">(</span><em>genesymbols</em>, <em>order=1</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_neighborhood"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_neighborhood" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_neighbors">
<code class="descname">gs_neighbors</code><span class="sig-paren">(</span><em>genesymbol</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_neighbors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_neighbors" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_stimulated_by">
<code class="descname">gs_stimulated_by</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_stimulated_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_stimulated_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gs_stimulates">
<code class="descname">gs_stimulates</code><span class="sig-paren">(</span><em>genesymbol</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.gs_stimulates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.gs_stimulates" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.gss">
<code class="descname">gss</code><span class="sig-paren">(</span><em>genesymbols</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.gss" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.guide2pharma">
<code class="descname">guide2pharma</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.guide2pharma"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.guide2pharma" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.having_attr">
<code class="descname">having_attr</code><span class="sig-paren">(</span><em>attr</em>, <em>graph=None</em>, <em>index=True</em>, <em>edges=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.having_attr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.having_attr" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if edges or nodes of the network have a specific
attribute and returns an iterator of the indices (or the
edge/node instances) of edges/nodes having such attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute to look for.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.</p></li>
<li><p><strong>index</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to return the
iterator of the indices or the node/edge instances.</p></li>
<li><p><strong>edges</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to look for the
attribute in the networks edges or nodes instead.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>generator</em>) – Generator object containing the edge/node
indices (or instances) having the specified attribute.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.having_eattr">
<code class="descname">having_eattr</code><span class="sig-paren">(</span><em>attr</em>, <em>graph=None</em>, <em>index=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.having_eattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.having_eattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if edges of the network have a specific attribute and
returns an iterator of the indices (or the edge instances) of
edges having such attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute to look for.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.</p></li>
<li><p><strong>index</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to return the
iterator of the indices or the node/edge instances.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>generator</em>) – Generator object containing the edge
indices (or instances) having the specified attribute.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.having_ptm">
<code class="descname">having_ptm</code><span class="sig-paren">(</span><em>index=True</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.having_ptm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.having_ptm" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if edges of the network have the <code class="docutils literal notranslate"><span class="pre">'ptm'</span></code> attribute and
returns an iterator of the indices (or the edge instances) of
edges having such attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>index</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to return the
iterator of the indices or the node/edge instances.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>generator</em>) – Generator object containing the edge
indices (or instances) having the <code class="docutils literal notranslate"><span class="pre">ptm''</span></code> attribute.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.having_vattr">
<code class="descname">having_vattr</code><span class="sig-paren">(</span><em>attr</em>, <em>graph=None</em>, <em>index=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.having_vattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.having_vattr" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if nodes of the network have a specific attribute and
returns an iterator of the indices (or the node instances) of
nodes having such attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>attr</strong> (<em>str</em>) – The name of the attribute to look for.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.</p></li>
<li><p><strong>index</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to return the
iterator of the indices or the node/edge instances.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>generator</em>) – Generator object containing the node
indices (or instances) having the specified attribute.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.homology_translation">
<code class="descname">homology_translation</code><span class="sig-paren">(</span><em>target</em>, <em>source=None</em>, <em>only_swissprot=True</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.homology_translation" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates the current object to another organism by orthology.
Proteins without known ortholog will be deleted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>target</strong> (<em>int</em>) – NCBI Taxonomy ID of the target organism. E.g. 10090 for mouse.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.htp_stats">
<code class="descname">htp_stats</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.htp_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.htp_stats" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.in_complex">
<code class="descname">in_complex</code><span class="sig-paren">(</span><em>csources=['corum']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.in_complex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.in_complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Deprecated, will be removed.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.in_directed">
<code class="descname">in_directed</code><span class="sig-paren">(</span><em>vertex</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.in_directed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.in_directed" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.in_undirected">
<code class="descname">in_undirected</code><span class="sig-paren">(</span><em>vertex</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.in_undirected"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.in_undirected" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.info">
<code class="descname">info</code><span class="sig-paren">(</span><em>name</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.info"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.info" title="Permalink to this definition">¶</a></dt>
<dd><p>Given the name of a resource, prints out the information about
that source/database. You can check the list of available
resource descriptions in
<code class="xref py py-func docutils literal notranslate"><span class="pre">ypath.descriptions.descriptions.keys()</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>name</strong> (<em>str</em>) – The name of the resource from which to print the
information.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.init_complex_attr">
<code class="descname">init_complex_attr</code><span class="sig-paren">(</span><em>graph</em>, <em>name</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.init_complex_attr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.init_complex_attr" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.init_edge_attr">
<code class="descname">init_edge_attr</code><span class="sig-paren">(</span><em>attr</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.init_edge_attr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.init_edge_attr" title="Permalink to this definition">¶</a></dt>
<dd><p>Fills all edges attribute <em>attr</em> with its default type (if
such attribute value is <code class="docutils literal notranslate"><span class="pre">None</span></code>), creates [list] if in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.edgeAttrs</span></code> such attribute is
registered as [list].</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>attr</strong> (<em>str</em>) – The attribute name to be initialized on the network edges.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.init_gsea">
<code class="descname">init_gsea</code><span class="sig-paren">(</span><em>user</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.init_gsea"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.init_gsea" title="Permalink to this definition">¶</a></dt>
<dd><p>Initializes a <code class="docutils literal notranslate"><span class="pre">pypath.gsea.GSEA</span></code> object and shows the list of the
collections in MSigDB.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.init_network">
<code class="descname">init_network</code><span class="sig-paren">(</span><em>lst=None</em>, <em>exclude=[]</em>, <em>cache_files={}</em>, <em>pfile=False</em>, <em>save=False</em>, <em>reread=None</em>, <em>redownload=None</em>, <em>keep_raw=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.init_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.init_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the network data.</p>
<p>This is a lazy way to start the module, load data and build the
high confidence, literature curated part of the signaling
network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the data input
formats for the different resources (keys) [str]. Values
are <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings</span></code> instances
containing the information. By default uses the set of
resources of OmniPath.</p></li>
<li><p><strong>exclude</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">[]</span></code> by default. List of resources [str] to
exclude from the network.</p></li>
<li><p><strong>cache_files</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains the resource name(s)
[str] (keys) and the corresponding cached file name [str].
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.</p></li>
<li><p><strong>pfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If any, provides the file
name or path to a previously saved network pickle file.
If <code class="docutils literal notranslate"><span class="pre">True</span></code> is passed, takes the default path from
<a class="reference internal" href="#pypath.main.PyPath.save_network" title="pypath.main.PyPath.save_network"><code class="xref py py-meth docutils literal notranslate"><span class="pre">PyPath.save_network()</span></code></a>
(<code class="docutils literal notranslate"><span class="pre">'cache/default_network.pickle'</span></code>).</p></li>
<li><p><strong>save</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. If set to <code class="docutils literal notranslate"><span class="pre">True</span></code>, saves
the loaded network to its default location
(<code class="docutils literal notranslate"><span class="pre">'cache/default_network.pickle'</span></code>).</p></li>
<li><p><strong>reread</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to reread
the data files from the cache or omit them (similar to
<em>redownload</em>).</p></li>
<li><p><strong>redownload</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to
re-download the data and ignore the cache.</p></li>
<li><p><strong>**kwargs</strong> – Not used.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.init_vertex_attr">
<code class="descname">init_vertex_attr</code><span class="sig-paren">(</span><em>attr</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.init_vertex_attr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.init_vertex_attr" title="Permalink to this definition">¶</a></dt>
<dd><p>Fills all vertices attribute <em>attr</em> with its default type (if
such attribute value is <code class="docutils literal notranslate"><span class="pre">None</span></code>), creates [list] if in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.vertexAttrs</span></code> such attribute is
registered as [list].</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>attr</strong> (<em>str</em>) – The attribute name to be initialized on the network
vertices.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.intergroup_shortest_paths">
<code class="descname">intergroup_shortest_paths</code><span class="sig-paren">(</span><em>groupA</em>, <em>groupB</em>, <em>random=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.intergroup_shortest_paths"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.intergroup_shortest_paths" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.intogen_cancer_drivers_list">
<code class="descname">intogen_cancer_drivers_list</code><span class="sig-paren">(</span><em>intogen_file</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.intogen_cancer_drivers_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.intogen_cancer_drivers_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the list of cancer driver proteins from IntOGen data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>intogen_file</strong> (<em>str</em>) – Path to the data file. Can also be [function] that provides
the data. In general, anything accepted by
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings.input</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.iter_interactions">
<code class="descname">iter_interactions</code><span class="sig-paren">(</span><em>signs=True</em>, <em>all_undirected=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.iter_interactions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.iter_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Iterates over edges and yields interaction records.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>signs</strong> (<em>bool</em>) – Ignoring signs if <code class="docutils literal notranslate"><span class="pre">False</span></code>. This way each directed interaction
will yield a single record even if it’s ambiguously labeled
both positive and negative. The default behaviour is to yield
two records in this case, one with positive and one with negative
sign.</p></li>
<li><p><strong>all_undirected</strong> (<em>bool</em>) – Yield records for undirected interactions even if certain sources
provide direction. If <code class="docutils literal notranslate"><span class="pre">False</span></code> only the directed records will
be provided and the undirected sources and references will be
added to the directed records.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.jaccard_edges">
<code class="descname">jaccard_edges</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.jaccard_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.jaccard_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the Jaccard similarity index between the sets of first
neighbours of all node pairs. <strong>NOTE:</strong> this method can take a
while to compute, e.g.: if the network has 10K nodes, the total
number of possible pairs to compute is:</p>
<div class="math notranslate nohighlight">
\[\binom{10^4}{2} = 49995000\]</div>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>list</em>) – Large list of [tuple] elements containing the
node pair names [str] and their corresponding first
neighbours Jaccard index [float].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.jaccard_meta">
<code class="descname">jaccard_meta</code><span class="sig-paren">(</span><em>jedges</em>, <em>critical</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.jaccard_meta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.jaccard_meta" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a (undirected) graph from a list of edges filtering by
their Jaccard index.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>jedges</strong> (<em>list</em>) – List of [tuple] containing the edges node names [str] and
their Jaccard index. Basically, the output of
<a class="reference internal" href="#pypath.main.PyPath.jaccard_edges" title="pypath.main.PyPath.jaccard_edges"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.jaccard_edges()</span></code></a>.</p></li>
<li><p><strong>critical</strong> (<em>float</em>) – Specifies the threshold of the Jaccard index from above
which an edge will be included in the graph.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>igraph.Graph</em>) – The Undirected graph instance containing
only the edges whose Jaccard similarity index is above the
threshold specified by <em>critical</em>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.kegg_directions">
<code class="descname">kegg_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.kegg_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.kegg_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.kegg_pathways">
<code class="descname">kegg_pathways</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.kegg_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.kegg_pathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.kinase_stats">
<code class="descname">kinase_stats</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.kinase_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.kinase_stats" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.label">
<code class="descname">label</code><span class="sig-paren">(</span><em>label</em>, <em>idx</em>, <em>what='vertices'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.label"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.label" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a boolean attribute <code class="docutils literal notranslate"><span class="pre">label</span></code> True for the
vertex or edge IDs in the set <code class="docutils literal notranslate"><span class="pre">idx</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.label_by_go">
<code class="descname">label_by_go</code><span class="sig-paren">(</span><em>label</em>, <em>go_terms</em>, <em>method='ANY'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.label_by_go"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.label_by_go" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns a boolean vertex attribute to nodes which tells whether
the node is annotated by all or any of the GO terms.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.label_edges">
<code class="descname">label_edges</code><span class="sig-paren">(</span><em>label</em>, <em>edges</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.label_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.label_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a boolean edge attribute <code class="docutils literal notranslate"><span class="pre">label</span></code> True for the
edge IDs in the set <code class="docutils literal notranslate"><span class="pre">edges</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.label_vertices">
<code class="descname">label_vertices</code><span class="sig-paren">(</span><em>label</em>, <em>vertices</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.label_vertices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.label_vertices" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a boolean vertex attribute <code class="docutils literal notranslate"><span class="pre">label</span></code> True for the
vertex IDs in the set <code class="docutils literal notranslate"><span class="pre">vertices</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.laudanna_directions">
<code class="descname">laudanna_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.laudanna_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.laudanna_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.laudanna_effects">
<code class="descname">laudanna_effects</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.laudanna_effects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.laudanna_effects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pypath.main.PyPath.license">
<em class="property">static </em><code class="descname">license</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.license"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.license" title="Permalink to this definition">¶</a></dt>
<dd><p>Prints information about data licences.</p>
</dd></dl>

<dl class="staticmethod">
<dt id="pypath.main.PyPath.list_resources">
<em class="property">static </em><code class="descname">list_resources</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.list_resources"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.list_resources" title="Permalink to this definition">¶</a></dt>
<dd><p>Prints the list of resources through the standard output.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_3dcomplexes">
<code class="descname">load_3dcomplexes</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_3dcomplexes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_3dcomplexes" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_3did_ddi">
<code class="descname">load_3did_ddi</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_3did_ddi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_3did_ddi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_3did_ddi2">
<code class="descname">load_3did_ddi2</code><span class="sig-paren">(</span><em>ddi=True</em>, <em>interfaces=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_3did_ddi2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_3did_ddi2" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_3did_dmi">
<code class="descname">load_3did_dmi</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_3did_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_3did_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_3did_interfaces">
<code class="descname">load_3did_interfaces</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_3did_interfaces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_3did_interfaces" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_all_pathways">
<code class="descname">load_all_pathways</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_all_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_all_pathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_compleat">
<code class="descname">load_compleat</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_compleat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_compleat" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads complexes from Compleat. Loads data into vertex attribute
<cite>graph.vs[‘complexes’][‘compleat’]</cite>.
This resource is human only.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_complexportal">
<code class="descname">load_complexportal</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_complexportal"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_complexportal" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads complexes from ComplexPortal. Loads data into vertex attribute
<cite>graph.vs[‘complexes’][‘complexportal’]</cite>.
This resource is human only.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_comppi">
<code class="descname">load_comppi</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_comppi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_comppi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_corum">
<code class="descname">load_corum</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_corum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_corum" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads complexes from CORUM database. Loads data into vertex attribute
<cite>graph.vs[‘complexes’][‘corum’]</cite>.
This resource is human only.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_dbptm">
<code class="descname">load_dbptm</code><span class="sig-paren">(</span><em>non_matching=False</em>, <em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_dbptm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_dbptm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ddi">
<code class="descname">load_ddi</code><span class="sig-paren">(</span><em>ddi</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ddi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ddi" title="Permalink to this definition">¶</a></dt>
<dd><p>ddi is either a list of intera.DomainDomain objects,
or a function resulting this list</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ddis">
<code class="descname">load_ddis</code><span class="sig-paren">(</span><em>methods=['dataio.get_3dc_ddi', 'dataio.get_domino_ddi', 'self.load_3did_ddi2']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ddis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ddis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_depod_dmi">
<code class="descname">load_depod_dmi</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_depod_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_depod_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_disgenet">
<code class="descname">load_disgenet</code><span class="sig-paren">(</span><em>dataset='curated'</em>, <em>score=0.0</em>, <em>umls=False</em>, <em>full_data=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_disgenet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_disgenet" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns DisGeNet disease-gene associations to the proteins
in the network. Disease annotations will be added to the <cite>dis</cite>
vertex attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>score</strong> (<em>float</em>) – Confidence score from DisGeNet. Only associations
above the score provided will be considered.</p></li>
<li><p><strong>ulms</strong> (<em>bool</em>) – By default we assign a list of disease names to
each protein. To use Unified Medical Language System IDs instead
set this to <cite>True</cite>.</p></li>
<li><p><strong>full_data</strong> (<em>bool</em>) – By default we load only disease names. Set this
to <cite>True</cite> if you wish to load additional annotations like number
of PubMed IDs, number of SNPs and original sources.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_dmi">
<code class="descname">load_dmi</code><span class="sig-paren">(</span><em>dmi</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_dmi" title="Permalink to this definition">¶</a></dt>
<dd><p>dmi is either a list of intera.DomainMotif objects,
or a function resulting this list</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_dmis">
<code class="descname">load_dmis</code><span class="sig-paren">(</span><em>methods=['self.pfam_regions', 'self.load_depod_dmi', 'self.load_dbptm', 'self.load_mimp_dmi', 'self.load_pnetworks_dmi', 'self.load_domino_dmi', 'self.load_pepcyber', 'self.load_psite_reg', 'self.load_psite_phos', 'self.load_ielm', 'self.load_phosphoelm', 'self.load_elm', 'self.load_3did_dmi']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_dmis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_dmis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_domino_dmi">
<code class="descname">load_domino_dmi</code><span class="sig-paren">(</span><em>organism=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_domino_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_domino_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_elm">
<code class="descname">load_elm</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_elm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_elm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_exocarta_attrs">
<code class="descname">load_exocarta_attrs</code><span class="sig-paren">(</span><em>load_samples=False</em>, <em>load_refs=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_exocarta_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_exocarta_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates vertex attributes from ExoCarta data. Creates a boolean
attribute <code class="docutils literal notranslate"><span class="pre">exocarts_exosomal</span></code> which tells whether a protein is
in ExoCarta i.e. has been found in exosomes. Optionally creates
attributes <code class="docutils literal notranslate"><span class="pre">exocarta_samples</span></code> and <code class="docutils literal notranslate"><span class="pre">exocarta_refs</span></code> listing the
sample tissue and the PubMed references, respectively.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_expression">
<code class="descname">load_expression</code><span class="sig-paren">(</span><em>array=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_expression"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_expression" title="Permalink to this definition">¶</a></dt>
<dd><p>Expression data can be loaded into vertex attributes,
or into a pandas DataFrame – the latter offers faster
ways to process and use these huge matrices.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_go">
<code class="descname">load_go</code><span class="sig-paren">(</span><em>organism=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_go"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_go" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a <code class="docutils literal notranslate"><span class="pre">pypath.go.GOAnnotation</span></code> object for one organism in the
dict under <code class="docutils literal notranslate"><span class="pre">go</span></code> attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>organism</strong> (<em>int</em>) – NCBI Taxonomy ID of the organism.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_havugimana">
<code class="descname">load_havugimana</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_havugimana"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_havugimana" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads complexes from Havugimana 2012. Loads data into vertex attribute
<cite>graph.vs[‘complexes’][‘havugimana’]</cite>.
This resource is human only.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_hpa">
<code class="descname">load_hpa</code><span class="sig-paren">(</span><em>normal=True</em>, <em>pathology=True</em>, <em>cancer=True</em>, <em>summarize_pathology=True</em>, <em>tissues=None</em>, <em>quality={'Approved'</em>, <em>'Supported'}</em>, <em>levels={'High': 3</em>, <em>'Low': 1</em>, <em>'Medium': 2</em>, <em>'Not detected': 0}</em>, <em>graph=None</em>, <em>na_value=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_hpa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_hpa" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads Human Protein Atlas data into vertex attributes.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_hprd_ptms">
<code class="descname">load_hprd_ptms</code><span class="sig-paren">(</span><em>non_matching=False</em>, <em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_hprd_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_hprd_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ielm">
<code class="descname">load_ielm</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ielm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ielm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_interfaces">
<code class="descname">load_interfaces</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_interfaces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_interfaces" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_li2012_ptms">
<code class="descname">load_li2012_ptms</code><span class="sig-paren">(</span><em>non_matching=False</em>, <em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_li2012_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_li2012_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ligand_receptor_network">
<code class="descname">load_ligand_receptor_network</code><span class="sig-paren">(</span><em>lig_rec_resources=True</em>, <em>inference_from_go=True</em>, <em>sources=None</em>, <em>keep_undirected=False</em>, <em>keep_rec_rec=False</em>, <em>keep_lig_lig=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ligand_receptor_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ligand_receptor_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Initializes a ligand-receptor network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_lmpid">
<code class="descname">load_lmpid</code><span class="sig-paren">(</span><em>method</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_lmpid"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_lmpid" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_matrisome_attrs">
<code class="descname">load_matrisome_attrs</code><span class="sig-paren">(</span><em>organism=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_matrisome_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_matrisome_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads vertex attributes from MatrisomeDB 2.0. Attributes are
<code class="docutils literal notranslate"><span class="pre">matrisome_class</span></code>, <code class="docutils literal notranslate"><span class="pre">matrisome_subclass</span></code> and <code class="docutils literal notranslate"><span class="pre">matrisome_notes</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_membranome_attrs">
<code class="descname">load_membranome_attrs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_membranome_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_membranome_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads attributes from Membranome, a database of single-helix
transmembrane proteins.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_mimp_dmi">
<code class="descname">load_mimp_dmi</code><span class="sig-paren">(</span><em>non_matching=False</em>, <em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_mimp_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_mimp_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_mutations">
<code class="descname">load_mutations</code><span class="sig-paren">(</span><em>attributes=None</em>, <em>gdsc_datadir=None</em>, <em>mutation_file=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_mutations"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_mutations" title="Permalink to this definition">¶</a></dt>
<dd><p>Mutations are listed in vertex attributes. Mutation() objects
offers methods to identify residues and look up in Ptm(), Motif()
and Domain() objects, to check if those residues are
modified, or are in some short motif or domain.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_negatives">
<code class="descname">load_negatives</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_negatives"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_negatives" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_old_omnipath">
<code class="descname">load_old_omnipath</code><span class="sig-paren">(</span><em>kinase_substrate_extra=False</em>, <em>remove_htp=False</em>, <em>htp_threshold=1</em>, <em>keep_directed=False</em>, <em>min_refs_undirected=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_old_omnipath"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_old_omnipath" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the OmniPath network as it was before August 2016.
Furthermore it gives some more options.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_omnipath">
<code class="descname">load_omnipath</code><span class="sig-paren">(</span><em>omnipath=None</em>, <em>kinase_substrate_extra=False</em>, <em>ligand_receptor_extra=False</em>, <em>pathway_extra=False</em>, <em>remove_htp=True</em>, <em>htp_threshold=1</em>, <em>keep_directed=True</em>, <em>min_refs_undirected=2</em>, <em>old_omnipath_resources=False</em>, <em>exclude=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_omnipath"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_omnipath" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the OmniPath network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pathways">
<code class="descname">load_pathways</code><span class="sig-paren">(</span><em>source</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pathways" title="Permalink to this definition">¶</a></dt>
<dd><p>Generic method to load pathway annotations from a resource.
We don’t recommend calling this method but either specific
methods for a single source e.g. <cite>kegg_pathways()</cite>
or <cite>sirnor_pathways()</cite> or call <cite>load_all_pathways()</cite> to
load all resources.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>source</strong> (<em>str</em>) – Name of the source, this need to match a method in the dict
in <cite>get_pathways()</cite> method and the edge and vertex attributes
with pathway annotations will be called “&lt;source&gt;_pathways”.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – A graph, by default the default the <cite>graph</cite> attribute of the
current instance.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pdb">
<code class="descname">load_pdb</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the 3D structure information from PDB into the network.
Creates the node attribute <code class="docutils literal notranslate"><span class="pre">'pdb'</span></code> containing a [dict] whose
keys are the PDB identifier [str] and values are [tuple] of two
elements denoting the experimental method [str] (e.g.:
<code class="docutils literal notranslate"><span class="pre">'X-ray'</span></code>, <code class="docutils literal notranslate"><span class="pre">'NMR'</span></code>, …) and the resolution [float] (if
applicable).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pepcyber">
<code class="descname">load_pepcyber</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pepcyber"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pepcyber" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pfam">
<code class="descname">load_pfam</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pfam"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pfam" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the protein family information from UniProt into the
network. Creates the node attribute <code class="docutils literal notranslate"><span class="pre">'pfam'</span></code> containing a
[list] of protein family identifier(s) [str].</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pfam2">
<code class="descname">load_pfam2</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pfam2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pfam2" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the protein family information from Pfam into the network.
Creates the node attribute <code class="docutils literal notranslate"><span class="pre">'pfam'</span></code> containing a [list] of
[dict] whose keys are protein family identifier(s) [str] and
corresponding values are [list] of [dict] containing detailed
information about the protein family(ies) for regions and
isoforms of the protein.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pfam3">
<code class="descname">load_pfam3</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pfam3"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pfam3" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the protein domain information from Pfam into the network.
Creates the node attribute <code class="docutils literal notranslate"><span class="pre">'doms'</span></code> containing a [list] of
<code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.intera.Domain</span></code> instances with information
about each domain of the protein (see the corresponding class
documentation for more information).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_phospho_dmi">
<code class="descname">load_phospho_dmi</code><span class="sig-paren">(</span><em>source</em>, <em>trace=False</em>, <em>return_raw=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_phospho_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_phospho_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_phosphoelm">
<code class="descname">load_phosphoelm</code><span class="sig-paren">(</span><em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_phosphoelm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_phosphoelm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pisa">
<code class="descname">load_pisa</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pisa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pisa" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_pnetworks_dmi">
<code class="descname">load_pnetworks_dmi</code><span class="sig-paren">(</span><em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_pnetworks_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_pnetworks_dmi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_psite_phos">
<code class="descname">load_psite_phos</code><span class="sig-paren">(</span><em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_psite_phos"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_psite_phos" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_psite_reg">
<code class="descname">load_psite_reg</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_psite_reg"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_psite_reg" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ptms">
<code class="descname">load_ptms</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_ptms2">
<code class="descname">load_ptms2</code><span class="sig-paren">(</span><em>input_methods=None, map_by_homology_from=[9606], homology_only_swissprot=True, ptm_homology_strict=False, nonhuman_direct_lookup=True, inputargs={}, database=None, force_load=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_ptms2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_ptms2" title="Permalink to this definition">¶</a></dt>
<dd><p>This is a new method which will replace <cite>load_ptms</cite>.
It uses <cite>pypath.ptm.PtmAggregator</cite>, a newly introduced
module for combining enzyme-substrate data from multiple
resources using homology translation on users demand.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_methods</strong> (<em>list</em>) – Resources to collect enzyme-substrate
interactions from. E.g. <cite>[‘Signor’, ‘phosphoELM’]</cite>. By default
it contains Signor, PhosphoSitePlus, HPRD, phosphoELM, dbPTM,
PhosphoNetworks, Li2012 and MIMP.</p></li>
<li><p><strong>map_by_homology_from</strong> (<em>list</em>) – List of NCBI Taxonomy IDs of
source taxons used for homology translation of enzyme-substrate
interactions. If you have a human network and you add here
<cite>[10090, 10116]</cite> then mouse and rat interactions from the source
databases will be translated to human.</p></li>
<li><p><strong>homology_only_swissprot</strong> (<em>bool</em>) – <cite>True</cite> by default which means
only SwissProt IDs are accepted at homology translateion, Trembl
IDs will be dropped.</p></li>
<li><p><strong>ptm_homology_strict</strong> (<em>bool</em>) – For homology translation use
PhosphoSite’s PTM homology table. This guarantees that only
truely homologous sites will be included. Otherwise we only
check if at the same numeric offset in the homologous sequence
the appropriate residue can be find.</p></li>
<li><p><strong>nonhuman_direct_lookup</strong> (<em>bool</em>) – Fetch also directly nonhuman
data from the resources whereever it’s available. PhosphoSite
contains mouse enzyme-substrate interactions and it is possible
to extract these directly beside translating the human ones
to mouse.</p></li>
<li><p><strong>inputargs</strong> (<em>dict</em>) – Additional arguments passed to <cite>PtmProcessor</cite>.
A <cite>dict</cite> can be supplied for each resource, e.g.
<cite>{‘Signor’: {…}, ‘PhosphoSite’: {…}, …}</cite>.
Those not used by <cite>PtmProcessor</cite> are forwarded to the
<cite>pypath.dataio</cite> methods.</p></li>
<li><p><strong>database</strong> – A <code class="docutils literal notranslate"><span class="pre">PtmAggregator</span></code> object. If provided no new database will be
created.</p></li>
<li><p><strong>force_load</strong> (<em>bool</em>) – If <code class="docutils literal notranslate"><span class="pre">True</span></code> the database will be loaded with the parameters
provided here; otherwise if the <code class="docutils literal notranslate"><span class="pre">ptm</span></code> module already has a
database no new database will be created. This means the
parameters specified in other arguments might have no effect.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_resource">
<code class="descname">load_resource</code><span class="sig-paren">(</span><em>settings</em>, <em>clean=True</em>, <em>cache_files={}</em>, <em>reread=None</em>, <em>redownload=None</em>, <em>keep_raw=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_resource"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_resource" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the data from a single resource and attaches it to the
network</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>settings</strong> (<em>pypath.input_formats.ReadSettings</em>) – <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings</span></code> instance
containing the detailed definition of the input format to
the downloaded file.</p></li>
<li><p><strong>clean</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to clean the graph
after importing the data or not. See
<a class="reference internal" href="#pypath.main.PyPath.clean_graph" title="pypath.main.PyPath.clean_graph"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.clean_graph()</span></code></a> for more
information.</p></li>
<li><p><strong>cache_files</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains the resource name(s)
[str] (keys) and the corresponding cached file name [str].
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.</p></li>
<li><p><strong>reread</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to reread
the data files from the cache or omit them (similar to
<em>redownload</em>).</p></li>
<li><p><strong>redownload</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to
re-download the data and ignore the cache.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_resources">
<code class="descname">load_resources</code><span class="sig-paren">(</span><em>lst=None</em>, <em>exclude=[]</em>, <em>cache_files={}</em>, <em>reread=False</em>, <em>redownload=None</em>, <em>keep_raw=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_resources"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_resources" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads multiple resources, and cleans up after. Looks up ID
types, and loads all ID conversion tables from UniProt if
necessary. This is much faster than loading the ID conversion
and the resources one by one.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the data input
formats for the different resources (keys) [str]. Values
are <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings</span></code> instances
containing the information. By default uses the set of
resources of OmniPath.</p></li>
<li><p><strong>exclude</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">[]</span></code> by default. List of resources [str] to
exclude from the network.</p></li>
<li><p><strong>cache_files</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains the resource name(s)
[str] (keys) and the corresponding cached file name [str].
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.</p></li>
<li><p><strong>reread</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to reread
the data files from the cache or omit them (similar to
<em>redownload</em>).</p></li>
<li><p><strong>redownload</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to
re-download the data and ignore the cache.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_signor_ptms">
<code class="descname">load_signor_ptms</code><span class="sig-paren">(</span><em>non_matching=False</em>, <em>trace=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_signor_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_signor_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_surfaceome_attrs">
<code class="descname">load_surfaceome_attrs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_surfaceome_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_surfaceome_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads vertex attributes from the In Silico Human Surfaceome.
Attributes are <code class="docutils literal notranslate"><span class="pre">surfaceome_score</span></code>, <code class="docutils literal notranslate"><span class="pre">surfaceome_class</span></code> and
<code class="docutils literal notranslate"><span class="pre">surfaceome_subclass</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_tfregulons">
<code class="descname">load_tfregulons</code><span class="sig-paren">(</span><em>levels={'A'</em>, <em>'B'}</em>, <em>only_curated=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_tfregulons"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_tfregulons" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds TF-target interactions from TF regulons to the network.
TF regulons is a comprehensive resource of TF-target
interactions combining multiple lines of evidences: literature
curated databases, ChIP-Seq data, PWM based prediction using
HOCOMOCO and JASPAR matrices and prediction from GTEx expression
data by ARACNe.</p>
<p>For details see <a class="reference external" href="https://github.com/saezlab/DoRothEA">https://github.com/saezlab/DoRothEA</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>levels</strong> (<em>set</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{'A',</span> <span class="pre">'B'}</span></code> by default. Confidence levels to be
loaded (from A to E) [str].</p></li>
<li><p><strong>only_curated</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to retrieve only the
literature curated interactions or not.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.load_vesiclepedia_attrs">
<code class="descname">load_vesiclepedia_attrs</code><span class="sig-paren">(</span><em>load_samples=False</em>, <em>load_refs=False</em>, <em>load_vesicle_type=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.load_vesiclepedia_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.load_vesiclepedia_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates vertex attributes from Vesiclepedia data. Creates a boolean
attribute <code class="docutils literal notranslate"><span class="pre">vesiclepedia_in_vesicle</span></code> which tells whether a protein is
in ExoCarta i.e. has been found in exosomes. Optionally creates
attributes <code class="docutils literal notranslate"><span class="pre">vesiclepedia_samples</span></code>, <code class="docutils literal notranslate"><span class="pre">vesiclepedia_refs</span></code> and
<code class="docutils literal notranslate"><span class="pre">vesiclepedia_vesicles</span></code> listing the sample tissue, the PubMed
references and the vesicle types, respectively.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.lookup_cache">
<code class="descname">lookup_cache</code><span class="sig-paren">(</span><em>name</em>, <em>cache_files</em>, <em>int_cache</em>, <em>edges_cache</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.lookup_cache"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.lookup_cache" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks up the cache folder for the files of a given resource.
First checks if <em>name</em> is on the <em>cache_files</em> dictionary.
If so, loads either the interactions or edges otherwise. If
not, checks <em>edges_cache</em> or <em>int_cache</em> otherwise.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> (<em>str</em>) – Name of the resource (lower-case).</p></li>
<li><p><strong>cache_files</strong> (<em>dict</em>) – Contains the resource name(s) [str] (keys) and the
corresponding cached file name [str] (values).</p></li>
<li><p><strong>int_cache</strong> (<em>str</em>) – Path to the interactions cache file of the resource.</p></li>
<li><p><strong>edges_cache</strong> (<em>str</em>) – Path to the edges cache file of the resource.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><ul class="simple">
<li><p>(<em>file</em>) – The loaded pickle file from the cache if the
file is contains the interactions. <code class="docutils literal notranslate"><span class="pre">None</span></code> otherwise.</p></li>
<li><p>(<em>list</em>) – List of mapped edges if the file contains the
information from the edges. <code class="docutils literal notranslate"><span class="pre">[]</span></code> otherwise.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.loop_edges">
<code class="descname">loop_edges</code><span class="sig-paren">(</span><em>index=True</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.loop_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.loop_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an iterator of the indices (or the edge instances) of
the edges which represent a loop (whose source and target node
are the same).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>index</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to return the
iterator of the indices or the edge instances.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The graph object where the
edge loops are to be searched. If none is passed, takes the
undirected network of the current instance.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>generator</em>) – Generator object containing the edge
indices (or instances) containing loops.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.map_edge">
<code class="descname">map_edge</code><span class="sig-paren">(</span><em>edge</em>, <em>expand_complexes=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.map_edge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.map_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates the identifiers in <em>edge</em> representing an edge. Default
name types are defined in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.default_name_type</span></code> If the mapping
is unsuccessful, the item will be added to
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.unmapped</span></code> list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>edge</strong> (<em>dict</em>) – Item whose name is to be mapped to a default name type.</p></li>
<li><p><strong>expand_complexes</strong> (<em>bool</em>) – Expand complexes, i.e. create links between each member of
the complex and the interacting partner.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the edge(s) [dict] with default mapped
names.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.map_item">
<code class="descname">map_item</code><span class="sig-paren">(</span><em>item</em>, <em>expand_complexes=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.map_item"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.map_item" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates the name in <em>item</em> representing a molecule. Default
name types are defined in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.default_name_type</span></code> If the mapping
is unsuccessful, the item will be added to
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.unmapped</span></code> list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>item</strong> (<em>dict</em>) – Item whose name is to be mapped to a default name type.</p></li>
<li><p><strong>expand_complexes</strong> (<em>bool</em>) – Expand complexes, i.e. create links between each member of
the complex and the interacting partner.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – The default mapped name(s) [str] of <em>item</em>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.map_list">
<code class="descname">map_list</code><span class="sig-paren">(</span><em>lst</em>, <em>single_list=False</em>, <em>expand_complexes=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.map_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.map_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Maps the names from a list of edges or items (molecules).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>list</em>) – List of items or edge dictionaries whose names have to be
mapped.</p></li>
<li><p><strong>single_list</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Determines whether the
provided elements are items or edges. This is, either calls
<a class="reference internal" href="#pypath.main.PyPath.map_edge" title="pypath.main.PyPath.map_edge"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.map_edge()</span></code></a> or
<a class="reference internal" href="#pypath.main.PyPath.map_item" title="pypath.main.PyPath.map_item"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.map_item()</span></code></a> to map the item
names.</p></li>
<li><p><strong>expand_complexes</strong> (<em>bool</em>) – Expand complexes, i.e. create links between each member of
the complex and the interacting partner.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Copy of <em>lst</em> with their elements’ names mapped.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.mean_reference_per_interaction">
<code class="descname">mean_reference_per_interaction</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.mean_reference_per_interaction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.mean_reference_per_interaction" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the mean number of references per interaction of the
network.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>float</em>) – Mean number of interactions per edge.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.merge_lists">
<code class="descname">merge_lists</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em>, <em>name=None</em>, <em>and_or='and'</em>, <em>delete=False</em>, <em>func='max'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.merge_lists"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.merge_lists" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges two lists from <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypat.main.PyPath.lists</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – Name of the first list to be merged.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – Name of the second list to be merged.</p></li>
<li><p><strong>name</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies a new name for the
merged list. If none is passed, name will be set to
<em>id_a*_*id_b</em>.</p></li>
<li><p><strong>and_or</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'and'</span></code> by default. The logic operation perfomed
in the merging: <code class="docutils literal notranslate"><span class="pre">'and'</span></code> performs an union, <code class="docutils literal notranslate"><span class="pre">'or'</span></code> for
the intersection.</p></li>
<li><p><strong>delete</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to delete the
former lists or not.</p></li>
<li><p><strong>func</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'max'</span></code> by default. Not used.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.merge_nodes">
<code class="descname">merge_nodes</code><span class="sig-paren">(</span><em>nodes</em>, <em>primary=None</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.merge_nodes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.merge_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges all attributes and edges of selected nodes and assigns
them to the primary node (by default the one with lowest index).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>nodes</strong> (<em>list</em>) – List of node indexes [int] that are to be collapsed.</p></li>
<li><p><strong>primary</strong> (<em>int</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. ID of the primary edge, if
none is passed, the node with lowest index on <em>nodes</em> is
selected.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network graph object from
which the nodes are to be merged. If none is passed, takes
the undirected network graph.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.mimp_directions">
<code class="descname">mimp_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.mimp_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.mimp_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.mutated_edges">
<code class="descname">mutated_edges</code><span class="sig-paren">(</span><em>sample</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.mutated_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.mutated_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Compares the mutated residues and the modified residues in PTMs.
Interactions are marked as mutated if the target residue in the
underlying PTM is mutated.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.name_edgelist">
<code class="descname">name_edgelist</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.name_edgelist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.name_edgelist" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an edge list, i.e. a list with tuples of vertex names.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.names2vids">
<code class="descname">names2vids</code><span class="sig-paren">(</span><em>names</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.names2vids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.names2vids" title="Permalink to this definition">¶</a></dt>
<dd><p>From a list of node names, returns their corresponding indices.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>names</strong> (<em>list</em>) – Contains the node names [str] for which the IDs are to be
searched.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – The queried node IDs [int].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.negative_report">
<code class="descname">negative_report</code><span class="sig-paren">(</span><em>lst=True</em>, <em>outFile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.negative_report"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.negative_report" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a report file with the negative interactions (assumed
to be already loaded).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lst</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Whether to retun a list of
edges containing the edge instances which have negative
references.</p></li>
<li><p><strong>outFile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The output file name/path. If
none is passed, the default is
<code class="docutils literal notranslate"><span class="pre">'results/&lt;session_id&gt;-negatives'</span></code></p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – If <em>lst</em> is set to <code class="docutils literal notranslate"><span class="pre">True</span></code>, returns a [list]
is returned with the <code class="xref py py-class docutils literal notranslate"><span class="pre">igraph.Edge</span></code> instances that
contain at least a negative reference.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.neighborhood">
<code class="descname">neighborhood</code><span class="sig-paren">(</span><em>identifiers</em>, <em>order=1</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.neighborhood"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.neighborhood" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.neighbors">
<code class="descname">neighbors</code><span class="sig-paren">(</span><em>identifier</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.neighbors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.neighbors" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.neighbourhood_network">
<code class="descname">neighbourhood_network</code><span class="sig-paren">(</span><em>center</em>, <em>second=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.neighbourhood_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.neighbourhood_network" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.network_by_go">
<code class="descname">network_by_go</code><span class="sig-paren">(</span><em>node_categories</em>, <em>network_sources=None</em>, <em>include=None</em>, <em>exclude=None</em>, <em>directed=False</em>, <em>keep_undirected=False</em>, <em>prefix='GO'</em>, <em>delete=True</em>, <em>copy=False</em>, <em>vertex_attrs=True</em>, <em>edge_attrs=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.network_by_go"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.network_by_go" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates or filters a network based on Gene Ontology annotations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>node_categories</strong> (<em>dict</em>) – <p>A dict with custom category labels as keys and expressions of
GO terms as values. E.g.
<a href="#id1"><span class="problematic" id="id2">``</span></a>{‘extracell’: ‘<a class="reference external" href="GO:0005576">GO:0005576</a> and not <a class="reference external" href="GO:0070062">GO:0070062</a>’,</p>
<blockquote>
<div><p>’plasmamem’: ‘<a class="reference external" href="GO:0005887">GO:0005887</a>’}``.</p>
</div></blockquote>
</p></li>
<li><p><strong>network_sources</strong> (<em>dict</em>) – A dict with anything as keys and network input format definintions
(<code class="docutils literal notranslate"><span class="pre">input_formats.ReadSettings</span></code> instances) as values.</p></li>
<li><p><strong>include</strong> (<em>list</em>) – A list of tuples of category label pairs. By default we keep all
edges connecting proteins annotated with any of the defined
categories. If <code class="docutils literal notranslate"><span class="pre">include</span></code> is defined then only edges between
category pairs defined here will be kept and all others deleted.</p></li>
<li><p><strong>exclude</strong> (<em>list</em>) – Similarly to include, all edges will be kept but the ones listed
in <code class="docutils literal notranslate"><span class="pre">exclude</span></code> will be deleted.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – If True <code class="docutils literal notranslate"><span class="pre">include</span></code> and <code class="docutils literal notranslate"><span class="pre">exclude</span></code> relations will be processed
with directed (source, target) else direction won’t be considered.</p></li>
<li><p><strong>keep_undirected</strong> (<em>bool</em>) – If True the interactions without direction information will be
kept even if <code class="docutils literal notranslate"><span class="pre">directed</span></code> is True. Passed to <code class="docutils literal notranslate"><span class="pre">edges_between</span></code>
as <code class="docutils literal notranslate"><span class="pre">strict</span></code> argument.</p></li>
<li><p><strong>prefix</strong> (<em>str</em>) – Prefix for all vertex and edge attributes created in this
operation. E.g. if you have a category label ‘bar’ and prefix
is ‘foo’ then you will have a new vertex attribute ‘foo__bar’.</p></li>
<li><p><strong>delete</strong> (<em>bool</em>) – Delete the vertices and edges which don’t belong to any of the
categories.</p></li>
<li><p><strong>copy</strong> (<em>bool</em>) – Return a copy of the entire <code class="docutils literal notranslate"><span class="pre">PyPath</span></code> object with the graph
filtered by GO terms. By default the object is modified in place
and <code class="docutils literal notranslate"><span class="pre">None</span></code> is returned.</p></li>
<li><p><strong>vertex_attrs</strong> (<em>bool</em>) – Create vertex attributes.</p></li>
<li><p><strong>edge_attrs</strong> (<em>bool</em>) – Create edge attributes.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.network_filter">
<code class="descname">network_filter</code><span class="sig-paren">(</span><em>p=2.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.network_filter"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.network_filter" title="Permalink to this definition">¶</a></dt>
<dd><p>This function aims to cut the number of edges in the network,
without losing nodes, to make the network less connected,
less hairball-like, more usable for analysis.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.network_stats">
<code class="descname">network_stats</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.network_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.network_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates basic statistics for the whole network and each of
sources (node and edge counts, average node degree, graph
diameter, transitivity, adhesion and cohesion). Writes the
results in a tab file. File is stored in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by default).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the file name. If
none is specified, this will be
<code class="docutils literal notranslate"><span class="pre">'pwnet-&lt;session_id&gt;-stats'</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.new_edges">
<code class="descname">new_edges</code><span class="sig-paren">(</span><em>edges</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.new_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.new_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds new edges from any iterable of edges to the undirected
graph. Basically, calls <code class="xref py py-meth docutils literal notranslate"><span class="pre">igraph.Graph.add_edges()</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>edges</strong> (<em>list</em>) – Contains the edges that are to be added to the network.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.new_nodes">
<code class="descname">new_nodes</code><span class="sig-paren">(</span><em>nodes</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.new_nodes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.new_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds new nodes from any iterable of nodes to the undirected
graph. Basically, calls <code class="xref py py-meth docutils literal notranslate"><span class="pre">igraph.Graph.add_vertices()</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>nodes</strong> (<em>list</em>) – Contains the nodes that are to be added to the network.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.node_exists">
<code class="descname">node_exists</code><span class="sig-paren">(</span><em>name</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.node_exists"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.node_exists" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if a node exists in the (undirected) network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>name</strong> (<em>str</em>) – The name of the node to be searched.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – Whether the node exists in the network or not.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.numof_directed_edges">
<code class="descname">numof_directed_edges</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.numof_directed_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.numof_directed_edges" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.numof_reference_interaction_pairs">
<code class="descname">numof_reference_interaction_pairs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.numof_reference_interaction_pairs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.numof_reference_interaction_pairs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the total of unique references per interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>int</em>) – Total number of unique references per
interaction.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.numof_references">
<code class="descname">numof_references</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.numof_references"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.numof_references" title="Permalink to this definition">¶</a></dt>
<dd><p>Counts the number of reference on the network.</p>
<p>Counts the total number of unique references in the edges of the
network.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>int</em>) – Number of unique references in the network.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.numof_undirected_edges">
<code class="descname">numof_undirected_edges</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.numof_undirected_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.numof_undirected_edges" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.orthology_translation">
<code class="descname">orthology_translation</code><span class="sig-paren">(</span><em>target</em>, <em>source=None</em>, <em>only_swissprot=True</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.orthology_translation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.orthology_translation" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates the current object to another organism by orthology.
Proteins without known ortholog will be deleted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>target</strong> (<em>int</em>) – NCBI Taxonomy ID of the target organism. E.g. 10090 for mouse.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.p">
<code class="descname">p</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.p" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pathway_attributes">
<code class="descname">pathway_attributes</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pathway_attributes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pathway_attributes" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pathway_members">
<code class="descname">pathway_members</code><span class="sig-paren">(</span><em>pathway</em>, <em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pathway_members"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pathway_members" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an iterator with the members of a single pathway.
Apart from the pathway name you need to supply its source
database too.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pathway_names">
<code class="descname">pathway_names</code><span class="sig-paren">(</span><em>source</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pathway_names"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pathway_names" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the names of all pathways having at least one member
in the current graph.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pathway_similarity">
<code class="descname">pathway_similarity</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pathway_similarity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pathway_similarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the Sorensen’s similarity index across nodes and edges
for all the available pathway sources (already loaded in the
network) and saves them into table files. Files are stored in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by default).
See <a class="reference internal" href="#pypath.main.PyPath.sorensen_pathways" title="pypath.main.PyPath.sorensen_pathways"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.sorensen_pathways()</span></code></a> for more
information..</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the file name
prefix (suffixes will be <code class="docutils literal notranslate"><span class="pre">'-nodes'</span></code> and <code class="docutils literal notranslate"><span class="pre">'-edges'</span></code>). If
none is specified, this will be
<code class="docutils literal notranslate"><span class="pre">'pwnet-&lt;session_id&gt;-sim-pw'</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pathways_table">
<code class="descname">pathways_table</code><span class="sig-paren">(</span><em>filename='genes_pathways.list', pw_sources=['signalink', 'signor', 'netpath', 'kegg'], graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pathways_table"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pathways_table" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.pfam_regions">
<code class="descname">pfam_regions</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.pfam_regions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.pfam_regions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.phosphonetworks_directions">
<code class="descname">phosphonetworks_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.phosphonetworks_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.phosphonetworks_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.phosphopoint_directions">
<code class="descname">phosphopoint_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.phosphopoint_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.phosphopoint_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.phosphorylation_directions">
<code class="descname">phosphorylation_directions</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.phosphorylation_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.phosphorylation_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.phosphorylation_signs">
<code class="descname">phosphorylation_signs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.phosphorylation_signs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.phosphorylation_signs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.phosphosite_directions">
<code class="descname">phosphosite_directions</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.phosphosite_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.phosphosite_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.prdb_tissue_expr">
<code class="descname">prdb_tissue_expr</code><span class="sig-paren">(</span><em>tissue</em>, <em>prdb=None</em>, <em>graph=None</em>, <em>occurrence=1</em>, <em>group_function=&lt;function PyPath.&lt;lambda&gt;&gt;</em>, <em>na_value=0.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.prdb_tissue_expr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.prdb_tissue_expr" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.process_direction">
<code class="descname">process_direction</code><span class="sig-paren">(</span><em>line</em>, <em>dir_col</em>, <em>dir_val</em>, <em>dir_sep</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.process_direction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.process_direction" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes the direction information of an interaction according
to a data file from a source.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>line</strong> (<em>list</em>) – The stripped and separated line from the resource data file
containing the information of an interaction.</p></li>
<li><p><strong>dir_col</strong> (<em>int</em>) – The column/position number where the information about the
direction is to be found (on <em>line</em>).</p></li>
<li><p><strong>dir_val</strong> (<em>list</em>) – Contains the terms [str] for which that interaction is to be
considered directed.</p></li>
<li><p><strong>dir_sep</strong> (<em>str</em>) – Separator for the field in <em>line</em> containing the direction
information (if any).</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – Determines whether the given interaction is
directed or not.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.process_directions">
<code class="descname">process_directions</code><span class="sig-paren">(</span><em>dirs</em>, <em>name</em>, <em>directed=None</em>, <em>stimulation=None</em>, <em>inhibition=None</em>, <em>graph=None</em>, <em>id_type=None</em>, <em>dirs_only=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.process_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.process_directions" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.process_dmi">
<code class="descname">process_dmi</code><span class="sig-paren">(</span><em>source</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.process_dmi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.process_dmi" title="Permalink to this definition">¶</a></dt>
<dd><p>This is an universal function
for loading domain-motif objects
like load_phospho_dmi() for PTMs.
TODO this will replace load_elm, load_ielm, etc</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.process_sign">
<code class="descname">process_sign</code><span class="sig-paren">(</span><em>signData</em>, <em>signDef</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.process_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.process_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes the sign of an interaction, used when processing an
input file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>signData</strong> (<em>str</em>) – Data regarding the sign to be processed.</p></li>
<li><p><strong>signDef</strong> (<em>tuple</em>) – Contains information about how to process <em>signData</em>. This
is defined in <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.data_formats</span></code>. First element
determines the position on the direction information of each
line on the data file [int], second element is either [str]
or [list] and defines the terms for which an interaction is
defined as stimulation, third element is similar but for the
inhibition and third (optional) element determines the
separator for <em>signData</em> if contains more than one element.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><ul class="simple">
<li><p>(<em>bool</em>) – Determines whether the processed interaction
is considered stimulation or not.</p></li>
<li><p>(<em>bool</em>) – Determines whether the processed interaction
is considered inhibition or not.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.protein">
<code class="descname">protein</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.protein" title="Permalink to this definition">¶</a></dt>
<dd><p>Same as <code class="docutils literal notranslate"><span class="pre">PyPath.get_node</span></code>, just for the directed graph.
Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.protein_edge">
<code class="descname">protein_edge</code><span class="sig-paren">(</span><em>source</em>, <em>target</em>, <em>directed=True</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.protein_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Edge</span></code> object if an edge exist between
the 2 proteins, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>source</strong> (<em>int</em><em>,</em><em>str</em>) – Vertex index or UniProt ID or GeneSymbol or <code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code>
object.</p></li>
<li><p><strong>target</strong> (<em>int</em><em>,</em><em>str</em>) – Vertex index or UniProt ID or GeneSymbol or <code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code>
object.</p></li>
<li><p><strong>directed</strong> (<em>bool</em>) – To be passed to igraph.Graph.get_eid()</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.proteins">
<code class="descname">proteins</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.proteins" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.ps">
<code class="descname">ps</code><span class="sig-paren">(</span><em>identifiers</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.ps" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.random_walk_with_return">
<code class="descname">random_walk_with_return</code><span class="sig-paren">(</span><em>q</em>, <em>graph=None</em>, <em>c=0.5</em>, <em>niter=1000</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.random_walk_with_return"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.random_walk_with_return" title="Permalink to this definition">¶</a></dt>
<dd><p>Random walk with return (RWR) starting from one or more query nodes.
Returns affinity (probability) vector of all nodes in the graph.</p>
<blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">param int,list q</dt>
<dd class="field-odd"><p>Vertex IDs of query nodes.</p>
</dd>
<dt class="field-even">param igraph.Graph graph</dt>
<dd class="field-even"><p>An <cite>igraph.Graph</cite> object.</p>
</dd>
<dt class="field-odd">param float c</dt>
<dd class="field-odd"><p>Probability of restart.</p>
</dd>
<dt class="field-even">param int niter</dt>
<dd class="field-even"><p>Number of iterations.</p>
</dd>
</dl>
</div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">igraph</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">pypath</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pa</span> <span class="o">=</span> <span class="n">pypath</span><span class="o">.</span><span class="n">PyPath</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pa</span><span class="o">.</span><span class="n">init_network</span><span class="p">({</span>
<span class="go">        &#39;signor&#39;: pypath.data_formats.pathway[&#39;signor&#39;]</span>
<span class="go">    })</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">q</span> <span class="o">=</span> <span class="p">[</span>
<span class="go">        pa.gs(&#39;EGFR&#39;).index,</span>
<span class="go">        pa.gs(&#39;ATG4B&#39;).index</span>
<span class="go">    ]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rwr</span> <span class="o">=</span> <span class="n">pa</span><span class="o">.</span><span class="n">random_walk_with_return</span><span class="p">(</span><span class="n">q</span> <span class="o">=</span> <span class="n">q</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">palette</span> <span class="o">=</span> <span class="n">igraph</span><span class="o">.</span><span class="n">RainbowPalette</span><span class="p">(</span><span class="n">n</span> <span class="o">=</span> <span class="mi">100</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">colors</span>  <span class="o">=</span> <span class="p">[</span><span class="n">palette</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">i</span><span class="p">)))</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">rwr</span> <span class="o">/</span> <span class="nb">max</span><span class="p">(</span><span class="n">rwr</span><span class="p">)</span> <span class="o">*</span> <span class="mi">99</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">igraph</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">pa</span><span class="o">.</span><span class="n">graph</span><span class="p">,</span> <span class="n">vertex_color</span> <span class="o">=</span> <span class="n">colors</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.random_walk_with_return2">
<code class="descname">random_walk_with_return2</code><span class="sig-paren">(</span><em>q</em>, <em>c=0.5</em>, <em>niter=1000</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.random_walk_with_return2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.random_walk_with_return2" title="Permalink to this definition">¶</a></dt>
<dd><p>Literally does random walks.
Only for testing of the other method, to be deleted later.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.read_data_file">
<code class="descname">read_data_file</code><span class="sig-paren">(</span><em>param</em>, <em>keep_raw=False</em>, <em>cache_files={}</em>, <em>reread=None</em>, <em>redownload=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.read_data_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.read_data_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads interaction data file containing node and edge attributes
that can be read from simple text based files and adds it to the
networkdata. This function works not only with files, but with
lists as well. Any other function can be written to download and
preprocess data, and then give it to this function to finally
attach to the network.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>param</strong> (<em>pypath.input_formats.ReadSettings</em>) – <code class="xref py py-class docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings</span></code> instance
containing the detailed definition of the input format of
the file. Instead of the file name (on the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.input_formats.ReadSettings.input</span></code>
attribute) you can give a custom function name, which will
be executed, and the returned data will be used instead.</p></li>
<li><p><strong>keep_raw</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to keep the raw data
read by this function, in order for debugging purposes, or
further use.</p></li>
<li><p><strong>cache_files</strong> (<em>dict</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">{}</span></code> by default. Contains the resource name(s)
[str] (keys) and the corresponding cached file name [str].
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.</p></li>
<li><p><strong>reread</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to reread
the data files from the cache or omit them (similar to
<em>redownload</em>).</p></li>
<li><p><strong>redownload</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to
re-download the data and ignore the cache.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.read_from_cache">
<code class="descname">read_from_cache</code><span class="sig-paren">(</span><em>cache_file</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.read_from_cache"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.read_from_cache" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a pickle file from the cache and returns it. It is assumed
that the subfolder <code class="docutils literal notranslate"><span class="pre">cache/</span></code> is on the supplied path.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>cache_file</strong> (<em>str</em>) – Path to the cache file that is to be loaded.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>file</em>) – The loaded pickle file from the cache. Type will
depend on the file itself (e.g.: if the pickle was saved
from a dictionary, the type will be [dict]).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.read_list_file">
<code class="descname">read_list_file</code><span class="sig-paren">(</span><em>settings</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.read_list_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.read_list_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a list from a file and adds it to
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.lists</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>settings</strong> (<em>pypath.input_formats.ReadList</em>) – <code class="xref py py-class docutils literal notranslate"><span class="pre">python.data_formats.ReadList</span></code> instance specifying
the settings of the file to be read. See the class
documentation for more details.</p></li>
<li><p><strong>**kwargs</strong> – Extra arguments passed to the file reading function. Such
function name is outlined in the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">python.data_formats.ReadList.input</span></code> attribute and
defined in <code class="xref py py-mod docutils literal notranslate"><span class="pre">pypath.dataio</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.reference_edge_ratio">
<code class="descname">reference_edge_ratio</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.reference_edge_ratio"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.reference_edge_ratio" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the average number of references per edge (as in the
undirected graph).</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>float</em>) – Average number of references per edge.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.reference_hist">
<code class="descname">reference_hist</code><span class="sig-paren">(</span><em>filename=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.reference_hist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.reference_hist" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a file containing a table with information about the
network’s edges. First column contains the source node ID,
followed by the target’s ID, third column contains the number of
references for that interaction and finally the number of
sources. Writes the results in a tab file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>filename</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the file name and
path to save the table. If none is passed, file will be
saved in <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code>
by default) with the name <code class="docutils literal notranslate"><span class="pre">'&lt;session_id&gt;-refs-hist'</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.reload">
<code class="descname">reload</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.reload"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.reload" title="Permalink to this definition">¶</a></dt>
<dd><p>Reloads the object from the module level.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.remove_htp">
<code class="descname">remove_htp</code><span class="sig-paren">(</span><em>threshold=50</em>, <em>keep_directed=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.remove_htp"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.remove_htp" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.remove_undirected">
<code class="descname">remove_undirected</code><span class="sig-paren">(</span><em>min_refs=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.remove_undirected"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.remove_undirected" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.run_batch">
<code class="descname">run_batch</code><span class="sig-paren">(</span><em>methods</em>, <em>toCall=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.run_batch"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.run_batch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.save_network">
<code class="descname">save_network</code><span class="sig-paren">(</span><em>pfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.save_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.save_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves the network object.</p>
<p>Stores the instance into a pickle (binary) file which can be
reloaded in the future.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The path/file name where to
store the pcikle file. If not specified, saves the network
to its default location
(<code class="docutils literal notranslate"><span class="pre">'cache/default_network.pickle'</span></code>).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.save_session">
<code class="descname">save_session</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.save_session"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.save_session" title="Permalink to this definition">¶</a></dt>
<dd><p>Save the current session state into pickle dump. The file will
be saved in the current working directory as
<code class="docutils literal notranslate"><span class="pre">'pypath-&lt;session_id&gt;.pickle'</span></code>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.search_attr_and">
<code class="descname">search_attr_and</code><span class="sig-paren">(</span><em>obj</em>, <em>lst</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.search_attr_and"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.search_attr_and" title="Permalink to this definition">¶</a></dt>
<dd><p>Searches a given collection of attributes in a given object.
Only returns <code class="docutils literal notranslate"><span class="pre">True</span></code>, if all elements of <em>lst</em> can be found in
<em>obj</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>obj</strong> (<em>object</em>) – Object (dictionary-like) where to search for elements of
<em>lst</em>.</p></li>
<li><p><strong>lst</strong> (<em>dict</em>) – Keys are the attribute names [str] and values the collection
of elements to be searched in such attribute [set].</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> only if <em>lst</em> is empty or all of its
elements are found in <em>obj</em>. Returns <code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise (as
soon as one element of <em>lst</em> is not found).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.search_attr_or">
<code class="descname">search_attr_or</code><span class="sig-paren">(</span><em>obj</em>, <em>lst</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.search_attr_or"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.search_attr_or" title="Permalink to this definition">¶</a></dt>
<dd><p>Searches a given collection of attributes in a given object. As
soon as one item is found, returns <code class="docutils literal notranslate"><span class="pre">True</span></code>, if none could be
found then returns <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>obj</strong> (<em>object</em>) – Object (dictionary-like) where to search for elements of
<em>lst</em>.</p></li>
<li><p><strong>lst</strong> (<em>dict</em>) – Keys are the attribute names [str] and values the collection
of elements to be searched in such attribute [set].</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if <em>lst</em> is empty or any of its
elements is found in <em>obj</em>. Returns only <code class="docutils literal notranslate"><span class="pre">False</span></code> if cannot
find anything.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.second_neighbours">
<code class="descname">second_neighbours</code><span class="sig-paren">(</span><em>node</em>, <em>indices=False</em>, <em>with_first=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.second_neighbours"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.second_neighbours" title="Permalink to this definition">¶</a></dt>
<dd><p>Looks for the (first and) second neighbours of a given node and
returns a list of their UniProt IDs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>node</strong> (<em>str</em>) – The UniProt ID of the node of interest. Can also be the
index of such node [int].</p></li>
<li><p><strong>indices</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return the
neighbour nodes indices or their UniProt IDs.</p></li>
<li><p><strong>wit_first</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return also the
first neighbours or not.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – The list containing the second neighbours of the
queried node (including the first ones if specified).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.select_by_go">
<code class="descname">select_by_go</code><span class="sig-paren">(</span><em>go_terms</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.select_by_go"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.select_by_go" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the vertex IDs of all vertices annotated with any
Gene Ontology terms or their descendants, or evaluates string
expression (see <code class="docutils literal notranslate"><span class="pre">select_by_go_expr</span></code>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>go_terms</strong> (<em>str</em><em>,</em><em>set</em>) – A single GO term, a set of GO terms or an expression with
GO terms.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.select_by_go_all">
<code class="descname">select_by_go_all</code><span class="sig-paren">(</span><em>go_terms</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.select_by_go_all"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.select_by_go_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Selects the nodes annotated by all GO terms in <code class="docutils literal notranslate"><span class="pre">go_terms</span></code>.</p>
<p>Returns set of vertex IDs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>go_terms</strong> (<em>list</em>) – List, set or tuple of GO terms.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.select_by_go_expr">
<code class="descname">select_by_go_expr</code><span class="sig-paren">(</span><em>go_expr</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.select_by_go_expr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.select_by_go_expr" title="Permalink to this definition">¶</a></dt>
<dd><p>Selects vertices based on an expression of Gene Ontology terms.
Operator precedence not considered, please use parentheses.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>go_expr</strong> (<em>str</em>) – An expression of Gene Ontology terms. E.g.
<code class="docutils literal notranslate"><span class="pre">'(GO:0005576</span> <span class="pre">and</span> <span class="pre">not</span> <span class="pre">GO:0070062)</span> <span class="pre">or</span> <span class="pre">GO:0005887'</span></code>. Parentheses
and operators <code class="docutils literal notranslate"><span class="pre">and</span></code>, <code class="docutils literal notranslate"><span class="pre">or</span></code> and <code class="docutils literal notranslate"><span class="pre">not</span></code> can be used.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.separate">
<code class="descname">separate</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.separate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.separate" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates the undirected network according to the different
sources. Basically applies
<a class="reference internal" href="#pypath.main.PyPath.get_network" title="pypath.main.PyPath.get_network"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.get_network()</span></code></a> for each resource.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>dict</em>) – Keys are resource names [str] whose values are
the subnetwork [igraph.Graph] containing the elements of
that source.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.separate_by_category">
<code class="descname">separate_by_category</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.separate_by_category"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.separate_by_category" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates the undirected network according to resource
categories. Possible categories are:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">'m'</span></code>: PTM/enzyme-substrate resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'p'</span></code>: Pathway/activity flow resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'i'</span></code>: Undirected/PPI resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'r'</span></code>: Process description/reaction resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'t'</span></code>: Transcription resources.</p></li>
</ul>
</div></blockquote>
<p>Works in the same way as <a class="reference internal" href="#pypath.main.PyPath.separate" title="pypath.main.PyPath.separate"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.separate()</span></code></a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>dict</em>) – Keys are category names [str] whose values are
the subnetwork [igraph.Graph] containing the elements of
those resources corresponding to that category.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sequences">
<code class="descname">sequences</code><span class="sig-paren">(</span><em>isoforms=True</em>, <em>update=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sequences" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_boolean_vattr">
<code class="descname">set_boolean_vattr</code><span class="sig-paren">(</span><em>attr</em>, <em>vids</em>, <em>negate=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_boolean_vattr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_boolean_vattr" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_categories">
<code class="descname">set_categories</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_categories"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_categories" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the category attribute on the network nodes and edges
(<code class="docutils literal notranslate"><span class="pre">'cat'</span></code>) as well the edge attribute coercing the references
by category (<code class="docutils literal notranslate"><span class="pre">'refs_by_cat'</span></code>). The possible categories are
as follows:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">'m'</span></code>: PTM/enzyme-substrate resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'p'</span></code>: Pathway/activity flow resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'i'</span></code>: Undirected/PPI resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'r'</span></code>: Process description/reaction resources.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'t'</span></code>: Transcription resources.</p></li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_chembl_mysql">
<code class="descname">set_chembl_mysql</code><span class="sig-paren">(</span><em>title</em>, <em>config_file=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_chembl_mysql"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_chembl_mysql" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the ChEMBL MySQL configuration according to the <em>title</em>
section in <em>config_file</em> ini file configuration.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>title</strong> (<em>str</em>) – Section title of the ini file.</p></li>
<li><p><strong>config_file</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the configuration
file name if none is passed, <code class="docutils literal notranslate"><span class="pre">mysql_config/defaults.mysql</span></code>
will be used.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_disease_genes">
<code class="descname">set_disease_genes</code><span class="sig-paren">(</span><em>dataset='curated'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_disease_genes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_disease_genes" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute named <cite>dis</cite> with boolean values <em>True</em>
if the protein encoded by a disease related gene according to
DisGeNet.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>dataset</strong> (<em>str</em>) – Which dataset to use from DisGeNet. Default is <cite>curated</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_druggability">
<code class="descname">set_druggability</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_druggability"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_druggability" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>dgb</cite> with value <em>True</em> if
the protein is druggable, otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_drugtargets">
<code class="descname">set_drugtargets</code><span class="sig-paren">(</span><em>pchembl=5.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_drugtargets"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_drugtargets" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>dtg</cite> with value <em>True</em> if
the protein has at least one compound binding with
affinity higher than <cite>pchembl</cite>, otherwise <em>False</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pchembl</strong> (<em>float</em>) – Pchembl threshold.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_kinases">
<code class="descname">set_kinases</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_kinases"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_kinases" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>kin</cite> with value <em>True</em> if
the protein is a kinase, otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_plasma_membrane_proteins_cspa">
<code class="descname">set_plasma_membrane_proteins_cspa</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_plasma_membrane_proteins_cspa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_plasma_membrane_proteins_cspa" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>cspa</cite> with value <em>True</em> if
the protein is a plasma membrane protein according to CPSA,
otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_plasma_membrane_proteins_cspa_surfaceome">
<code class="descname">set_plasma_membrane_proteins_cspa_surfaceome</code><span class="sig-paren">(</span><em>score_threshold=0.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_plasma_membrane_proteins_cspa_surfaceome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_plasma_membrane_proteins_cspa_surfaceome" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <code class="docutils literal notranslate"><span class="pre">surf</span></code> with value <em>True</em> if
the protein is a plasma membrane protein according either to the
Cell Surface Protein Atlas or the In Silico Human Surfaceome.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_plasma_membrane_proteins_surfaceome">
<code class="descname">set_plasma_membrane_proteins_surfaceome</code><span class="sig-paren">(</span><em>score_threshold=0.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_plasma_membrane_proteins_surfaceome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_plasma_membrane_proteins_surfaceome" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>ishs</cite> with value <em>True</em> if
the protein is a plasma membrane protein according to the In Silico
Human Surfaceome, otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_receptors">
<code class="descname">set_receptors</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_receptors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_receptors" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>rec</cite> with value <em>True</em> if
the protein is a receptor, otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_signaling_proteins">
<code class="descname">set_signaling_proteins</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_signaling_proteins"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_signaling_proteins" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>kin</cite> with value <em>True</em> if
the protein is a kinase, otherwise <em>False</em>.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_tfs">
<code class="descname">set_tfs</code><span class="sig-paren">(</span><em>classes=['a', 'b', 'other']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_tfs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_tfs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.set_transcription_factors">
<code class="descname">set_transcription_factors</code><span class="sig-paren">(</span><em>classes=['a', 'b', 'other']</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.set_transcription_factors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.set_transcription_factors" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a vertex attribute <cite>tf</cite> with value <em>True</em> if
the protein is a transcription factor, otherwise <em>False</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>classes</strong> (<em>list</em>) – Classes to use from TF Census. Default is <cite>[‘a’, ‘b’, ‘other’]</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.shortest_path_dist">
<code class="descname">shortest_path_dist</code><span class="sig-paren">(</span><em>graph=None</em>, <em>subset=None</em>, <em>outfile=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.shortest_path_dist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.shortest_path_dist" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the distribution of shortest paths for each pair of
nodes in the network (or between group(s) of nodes if <em>subset</em>
is provided). <strong>NOTE:</strong> this method can take a while to compute,
e.g.: if the network has 10K nodes, the total number of possible
pairs to compute is:</p>
<div class="math notranslate nohighlight">
\[\binom{10^4}{2} = 49995000\]</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The network object for which
the shortest path distribution is to be computed. If none is
passed, takes the undirected network of the current
instance.</p></li>
<li><p><strong>susbet</strong> (<em>tuple</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Contains two lists of node
indices defining two groups between which the distribution
is to be computed. Can also be [list] if the shortest paths
are to be searched whithin the group. If none is passed, the
whole network is taken by default.</p></li>
<li><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. File name/path to save the
shortest path distribution. If none is passed, no file is
generated.</p></li>
<li><p><strong>**kwargs</strong> – Additional keyword arguments passed to
<code class="xref py py-meth docutils literal notranslate"><span class="pre">igraph.Graph.get_shortest_paths()</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – The length of the shortest paths for each pair
of nodes of the network (or whithin/between group/s if
<em>subset</em> is provided).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.signaling_proteins_list">
<code class="descname">signaling_proteins_list</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.signaling_proteins_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.signaling_proteins_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Compiles a list of signaling proteins (as opposed to other
proteins like metabolic enzymes, matrix proteins, etc), by
looking up a few simple keywords in short description of GO
terms.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.signor_pathways">
<code class="descname">signor_pathways</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.signor_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.signor_pathways" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.similarity_groups">
<code class="descname">similarity_groups</code><span class="sig-paren">(</span><em>groups</em>, <em>index='simpson'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.similarity_groups"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.similarity_groups" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the similarity index across the given <em>groups</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>groups</strong> (<em>dict</em>) – Contains the different group names [str] as keys and their
corresponding elements [set].</p></li>
<li><p><strong>index</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'simpson'</span></code> by default. The type of index metric
to use to compute the similarity. Options are <code class="docutils literal notranslate"><span class="pre">'simpson'</span></code>,
<code class="docutils literal notranslate"><span class="pre">'sorensen'</span></code> and <code class="docutils literal notranslate"><span class="pre">'jaccard'</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>dict</em>) – Dictionary of dictionaries containing the groups
names [str] as keys (for both inner and outer dictionaries)
and the index metric as inner value [float] between those
groups.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.small_plot">
<code class="descname">small_plot</code><span class="sig-paren">(</span><em>graph</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.small_plot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.small_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>This method is deprecated, do not use it.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sorensen_pathways">
<code class="descname">sorensen_pathways</code><span class="sig-paren">(</span><em>pwlist=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sorensen_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sorensen_pathways" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the Sorensen’s similarity index across nodes and edges
for the given list of pathway sources (all loaded pathway
sources by default).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pwlist</strong> (<em>list</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. The list of pathway sources
to be compared.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>dict</em>) – Nested dictionaries (three levels). First-level
keys are <code class="docutils literal notranslate"><span class="pre">'nodes'</span></code> and <code class="docutils literal notranslate"><span class="pre">'edges'</span></code>, then second and third
levels correspond to <code class="docutils literal notranslate"><span class="pre">&lt;source&gt;__&lt;patwhay&gt;</span></code> names which map
to the similarity index between those pathways [float].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.source_diagram">
<code class="descname">source_diagram</code><span class="sig-paren">(</span><em>outf=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.source_diagram"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.source_diagram" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.source_network">
<code class="descname">source_network</code><span class="sig-paren">(</span><em>font='HelveticaNeueLTStd'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.source_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.source_network" title="Permalink to this definition">¶</a></dt>
<dd><p>For EMBL branding, use Helvetica Neue Linotype Standard light</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.source_similarity">
<code class="descname">source_similarity</code><span class="sig-paren">(</span><em>outfile=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.source_similarity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.source_similarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the Sorensen’s similarity index across nodes and edges
for all the sources available (already loaded in the network)
and saves them into table files. Files are stored in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by default).
See <a class="reference internal" href="#pypath.main.PyPath.databases_similarity" title="pypath.main.PyPath.databases_similarity"><code class="xref py py-meth docutils literal notranslate"><span class="pre">pypath.main.PyPath.databases_similarity()</span></code></a> for more
information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>outfile</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the file name
prefix (suffixes will be <code class="docutils literal notranslate"><span class="pre">'-nodes'</span></code> and <code class="docutils literal notranslate"><span class="pre">'-edges'</span></code>). If
none is specified, this will be
<code class="docutils literal notranslate"><span class="pre">'pwnet-&lt;session_id&gt;-sim-src'</span></code>.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.source_stats">
<code class="descname">source_stats</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.source_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.source_stats" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sources_hist">
<code class="descname">sources_hist</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sources_hist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sources_hist" title="Permalink to this definition">¶</a></dt>
<dd><p>Counts the number of sources per interaction in the graph and
saves them into a file named <code class="docutils literal notranslate"><span class="pre">source_num</span></code>. File is stored in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by
default).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sources_overlap">
<code class="descname">sources_overlap</code><span class="sig-paren">(</span><em>diagonal=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sources_overlap"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sources_overlap" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sources_venn_data">
<code class="descname">sources_venn_data</code><span class="sig-paren">(</span><em>fname=None</em>, <em>return_data=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sources_venn_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sources_venn_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the overlap in number of interactions for all pairs of
sources.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>fname</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If provided, saves the
results into a table file. File is stored in
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code> (<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by
default).</p></li>
<li><p><strong>return_data</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Whether to return the
results as a [list].</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Only if <em>return_data</em> is set to <code class="docutils literal notranslate"><span class="pre">True</span></code>. List
of lists containing the counts for each pair of resources.
This is, for instance, number of interactions only in
resource A, number of interactions only in resource B and
number of common interactions between A and B.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.straight_between">
<code class="descname">straight_between</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.straight_between"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.straight_between" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds an edge between the provided node names.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – The name of the source node.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – The name of the target node.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>int</em>) – The edge ID. If the edge doesn’t exist, returns
[list] with the node indices [int].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.string_effects">
<code class="descname">string_effects</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.string_effects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.string_effects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.sum_in_complex">
<code class="descname">sum_in_complex</code><span class="sig-paren">(</span><em>csources=['corum'], graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.sum_in_complex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.sum_in_complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the total number of edges in the network falling
between two members of the same complex.
Returns as a dict by complex resources.
Calls :py:func:pypath.pypath.Pypath.edges_in_comlexes()
to do the calculations.</p>
<dl class="simple">
<dt>&#64;csources<span class="classifier">list</span></dt><dd><p>List of complex resources. Should be already loaded.</p>
</dd>
<dt>&#64;graph<span class="classifier">igraph.Graph()</span></dt><dd><p>The graph object to do the calculations on.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.table_latex">
<code class="descname">table_latex</code><span class="sig-paren">(</span><em>fname</em>, <em>header</em>, <em>data</em>, <em>sum_row=True</em>, <em>row_order=None</em>, <em>latex_hdr=True</em>, <em>caption=''</em>, <em>font='HelveticaNeueLTStd-LtCn'</em>, <em>fontsize=8</em>, <em>sum_label='Total'</em>, <em>sum_cols=None</em>, <em>header_format='%s'</em>, <em>by_category=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.table_latex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.table_latex" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.third_source_directions">
<code class="descname">third_source_directions</code><span class="sig-paren">(</span><em>graph=None</em>, <em>use_string_effects=False</em>, <em>use_laudanna_data=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.third_source_directions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.third_source_directions" title="Permalink to this definition">¶</a></dt>
<dd><p>This method calls a series of methods to get
additional direction &amp; effect information
from sources having no literature curated references,
but giving sufficient evidence about the directionality
for interactions already supported by literature
evidences from other sources.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.tissue_network">
<code class="descname">tissue_network</code><span class="sig-paren">(</span><em>tissue</em>, <em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.tissue_network"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.tissue_network" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a network which includes the proteins expressed in
certain tissue according to ProteomicsDB.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>tissue</strong> (<em>str</em>) – Tissue name as used in ProteomicsDB.</p></li>
<li><p><strong>graph</strong> (<em>igraph.Graph</em>) – A graph object, by default the <cite>graph</cite> attribute of
the current instance.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.transcription_factors">
<code class="descname">transcription_factors</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.transcription_factors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.transcription_factors" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.uniprot">
<code class="descname">uniprot</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.uniprot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.uniprot" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the UniProt
can be found in the default undirected network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;uniprot<span class="classifier">str</span></dt><dd><p>UniProt ID.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.uniprots">
<code class="descname">uniprots</code><span class="sig-paren">(</span><em>uniprots</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.uniprots"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.uniprots" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns list of <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object
for a list of UniProt IDs omitting those
could not be found in the default
undirected graph.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.uniq_node_list">
<code class="descname">uniq_node_list</code><span class="sig-paren">(</span><em>lst</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.uniq_node_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.uniq_node_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a given list of nodes containing only the unique
elements.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>lst</strong> (<em>list</em>) – List of nodes.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Copy of <em>lst</em> containing only unique nodes.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.uniq_ptm">
<code class="descname">uniq_ptm</code><span class="sig-paren">(</span><em>ptms</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.uniq_ptm"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.uniq_ptm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.uniq_ptms">
<code class="descname">uniq_ptms</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.uniq_ptms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.uniq_ptms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up">
<code class="descname">up</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.up" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the UniProt
can be found in the default undirected network,
otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;uniprot<span class="classifier">str</span></dt><dd><p>UniProt ID.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_affected_by">
<code class="descname">up_affected_by</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_affected_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_affected_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_affects">
<code class="descname">up_affects</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_affects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_affects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_edge">
<code class="descname">up_edge</code><span class="sig-paren">(</span><em>source</em>, <em>target</em>, <em>directed=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_edge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Edge</span></code> object if an edge exist between
the 2 proteins, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;source<span class="classifier">str</span></dt><dd><p>UniProt ID</p>
</dd>
<dt>&#64;target<span class="classifier">str</span></dt><dd><p>UniProt ID</p>
</dd>
<dt>&#64;directed<span class="classifier">bool</span></dt><dd><p>To be passed to igraph.Graph.get_eid()</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_in_directed">
<code class="descname">up_in_directed</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_in_directed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_in_directed" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_in_undirected">
<code class="descname">up_in_undirected</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_in_undirected"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_in_undirected" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_inhibited_by">
<code class="descname">up_inhibited_by</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_inhibited_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_inhibited_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_inhibits">
<code class="descname">up_inhibits</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_inhibits"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_inhibits" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_neighborhood">
<code class="descname">up_neighborhood</code><span class="sig-paren">(</span><em>uniprots</em>, <em>order=1</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_neighborhood"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_neighborhood" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_neighbors">
<code class="descname">up_neighbors</code><span class="sig-paren">(</span><em>uniprot</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_neighbors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_neighbors" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_stimulated_by">
<code class="descname">up_stimulated_by</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_stimulated_by"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_stimulated_by" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.up_stimulates">
<code class="descname">up_stimulates</code><span class="sig-paren">(</span><em>uniprot</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.up_stimulates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.up_stimulates" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_adjlist">
<code class="descname">update_adjlist</code><span class="sig-paren">(</span><em>graph=None</em>, <em>mode='ALL'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_adjlist"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_adjlist" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates an adjacency list in a list of sets format.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_attrs">
<code class="descname">update_attrs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_attrs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_attrs" title="Permalink to this definition">¶</a></dt>
<dd><p>Updates the node and edge attributes. Note that no data is
donwloaded, mainly updates the dictionaries of attributes
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.edgeAttrs</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.vertexAttrs</span></code> containing the
attributes names and their correspoding types and initializes
such attributes in the network nodes/edges if they weren’t.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_cats">
<code class="descname">update_cats</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_cats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_cats" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes sure that the <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.has_cats</span></code>
attribute is an up to date [set] of all categories in the
current network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_db_dict">
<code class="descname">update_db_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_db_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_db_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_pathway_types">
<code class="descname">update_pathway_types</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_pathway_types"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_pathway_types" title="Permalink to this definition">¶</a></dt>
<dd><p>Updates the pathway types attribute
(<code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.pathway_types</span></code>) according to the
loaded resources of the undirected network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_pathways">
<code class="descname">update_pathways</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_pathways" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes sure that the <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.pathways</span></code>
attribute is an up to date [dict] of all pathways and their
sources in the current network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_sources">
<code class="descname">update_sources</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_sources"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_sources" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes sure that the <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.sources</span></code>
attribute is an up to date [list] of all sources in the current
network.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_vertex_sources">
<code class="descname">update_vertex_sources</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_vertex_sources"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_vertex_sources" title="Permalink to this definition">¶</a></dt>
<dd><p>Updates the all the vertex attributes <code class="docutils literal notranslate"><span class="pre">'sources'</span></code> and
<code class="docutils literal notranslate"><span class="pre">'references'</span></code> according to their related edges (on the
undirected graph).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_vindex">
<code class="descname">update_vindex</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_vindex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_vindex" title="Permalink to this definition">¶</a></dt>
<dd><p>This is deprecated.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.update_vname">
<code class="descname">update_vname</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.update_vname"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.update_vname" title="Permalink to this definition">¶</a></dt>
<dd><p>Fast lookup of node names and indexes, these are hold in a
[list] and a [dict] as well. However, every time new nodes are
added, these should be updated. This function is automatically
called after all operations affecting node indices.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.ups">
<code class="descname">ups</code><span class="sig-paren">(</span><em>uniprots</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.ups" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns list of <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object
for a list of UniProt IDs omitting those
could not be found in the default
undirected graph.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.v">
<code class="descname">v</code><span class="sig-paren">(</span><em>identifier</em><span class="sig-paren">)</span><a class="headerlink" href="#pypath.main.PyPath.v" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns <code class="docutils literal notranslate"><span class="pre">igraph.Vertex()</span></code> object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p>
<dl class="simple">
<dt>&#64;identifier<span class="classifier">int, str</span></dt><dd><p>Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
<code class="docutils literal notranslate"><span class="pre">igraph.Vertex</span></code> object.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.vertex_pathways">
<code class="descname">vertex_pathways</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.vertex_pathways"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.vertex_pathways" title="Permalink to this definition">¶</a></dt>
<dd><p>Some resources assignes interactions some others proteins to
pathways. This function copies pathway annotations from edge
attributes to vertex attributes.</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.vsgs">
<code class="descname">vsgs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.vsgs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.vsgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a generator sequence of the node names as GeneSymbols
[str] (from the undirected graph).</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>generator</em>) – Sequence containing the node names as
GeneSymbols [str].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.vsup">
<code class="descname">vsup</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.vsup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.vsup" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a generator sequence of the node names as UniProt IDs
[str] (from the undirected graph).</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>generator</em>) – Sequence containing the node names as
UniProt IDs [str].</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.wang_effects">
<code class="descname">wang_effects</code><span class="sig-paren">(</span><em>graph=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.wang_effects"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.wang_effects" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pypath.main.PyPath.write_table">
<code class="descname">write_table</code><span class="sig-paren">(</span><em>tbl</em>, <em>outfile</em>, <em>sep='\t'</em>, <em>cut=None</em>, <em>colnames=True</em>, <em>rownames=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#PyPath.write_table"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.PyPath.write_table" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes a given table to a file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>tbl</strong> (<em>dict</em>) – Contains the data of the table. It is assumed that keys are
the row names [str] and the values, well, values. Column
names (if any) are defined with the key <code class="docutils literal notranslate"><span class="pre">'header'</span></code>.</p></li>
<li><p><strong>outfile</strong> (<em>str</em>) – File name where to save the table. The file will be saved
under the object’s <code class="xref py py-attr docutils literal notranslate"><span class="pre">pypath.main.PyPath.outdir</span></code>
(<code class="docutils literal notranslate"><span class="pre">'results'</span></code> by default).</p></li>
<li><p><strong>sep</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'</span>&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">'</span></code> (tab) by default. Specifies the separator
for the file.</p></li>
<li><p><strong>cut</strong> (<em>int</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Specifies the maximum number
of characters for the row names.</p></li>
<li><p><strong>colnames</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Specifies whether to write
the column names in the file or not.</p></li>
<li><p><strong>rownames</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">True</span></code> by default. Specifies whether to write
the row names in the file or not.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pypath.main.Direction">
<em class="property">class </em><code class="descclassname">pypath.main.</code><code class="descname">Direction</code><span class="sig-paren">(</span><em>id_a</em>, <em>id_b</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction" title="Permalink to this definition">¶</a></dt>
<dd><p>Object storing directionality information of an edge. Also includes
information about the reverse direction, mode of regulation and
sources of that information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id_a</strong> (<em>str</em>) – Name of the source node.</p></li>
<li><p><strong>id_b</strong> (<em>str</em>) – Name of the target node.</p></li>
</ul>
</dd>
<dt class="field-even">Variables</dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>dirs</strong> (<em>dict</em>) – Dictionary containing the presence of directionality of the
given edge. Keys are <em>straight</em>, <em>reverse</em> and <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code>
and their values denote the presence/absence [bool].</p></li>
<li><p><strong>negative</strong> (<em>dict</em>) – Dictionary contianing the presence/absence [bool] of negative
interactions for both <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> and <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
directions.</p></li>
<li><p><strong>negative_sources</strong> (<em>dict</em>) – Contains the resource names [str] supporting a negative
interaction on <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> and <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
directions.</p></li>
<li><p><strong>nodes</strong> (<em>list</em>) – Contains the node names [str] sorted alphabetically (<em>id_a</em>,
<em>id_b</em>).</p></li>
<li><p><strong>positive</strong> (<em>dict</em>) – Dictionary contianing the presence/absence [bool] of positive
interactions for both <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> and <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
directions.</p></li>
<li><p><strong>positive_sources</strong> (<em>dict</em>) – Contains the resource names [str] supporting a positive
interaction on <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> and <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
directions.</p></li>
<li><p><strong>reverse</strong> (<em>tuple</em>) – Contains the node names [str] in reverse order e.g. (<em>id_b</em>,
<em>id_a</em>).</p></li>
<li><p><strong>sources</strong> (<em>dict</em>) – Contains the resource names [str] of a given edge for each
directionality (<code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code>, <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> and
<code class="docutils literal notranslate"><span class="pre">'undirected'</span></code>). Values are sets containing the names of those
resources supporting such directionality.</p></li>
<li><p><strong>straight</strong> (<em>tuple</em>) – Contains the node names [str] in the original order e.g.
(<em>id_a</em>, <em>id_b</em>).</p></li>
</ul>
</dd>
</dl>
<dl class="method">
<dt id="pypath.main.Direction.check_nodes">
<code class="descname">check_nodes</code><span class="sig-paren">(</span><em>nodes</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.check_nodes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.check_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if <em>nodes</em> is contained in the edge.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>nodes</strong> (<em>list</em>) – Or [tuple], contains the names of the nodes to be checked.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if all elements in <em>nodes</em> are
contained in the object <code class="xref py py-attr docutils literal notranslate"><span class="pre">nodes</span></code> list.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.check_param">
<code class="descname">check_param</code><span class="sig-paren">(</span><em>di</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.check_param"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.check_param" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if <em>di</em> is <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code> or contains the nodes of
the current edge. Used internally to check that <em>di</em> is a valid
key for the object attributes declared on dictionaries.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>di</strong> (<em>tuple</em>) – Or [str], key to be tested for validity.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><dl class="simple">
<dt>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if <em>di</em> is <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code> or a tuple</dt><dd><p>of node names contained in the edge, <code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.consensus_edges">
<code class="descname">consensus_edges</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.consensus_edges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.consensus_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Infers the consensus edge(s) according to the number of
supporting sources. This includes direction and sign.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>list</em>) – Contains the consensus edge(s) along with the
consensus sign. If there is no major directionality, both
are returned. The structure is as follows:
<code class="docutils literal notranslate"><span class="pre">['&lt;source&gt;',</span> <span class="pre">'&lt;target&gt;',</span> <span class="pre">'&lt;(un)directed&gt;',</span> <span class="pre">'&lt;sign&gt;']</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.get_dir">
<code class="descname">get_dir</code><span class="sig-paren">(</span><em>direction</em>, <em>sources=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.get_dir"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.get_dir" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the state (or <em>sources</em> if specified) of the given
<em>direction</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – Or [str] (if <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code>). Pair of nodes from which
direction information is to be retrieved.</p></li>
<li><p><strong>sources</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">'False'</span></code> by default. Specifies if the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code> information of the given direction is to
be retrieved instead.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em> or <em>set</em>) – (if <code class="docutils literal notranslate"><span class="pre">sources=True</span></code>). Presence/absence
of the requested direction (or the list of sources if
specified). Returns <code class="docutils literal notranslate"><span class="pre">None</span></code> if <em>direction</em> is not valid.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.get_dirs">
<code class="descname">get_dirs</code><span class="sig-paren">(</span><em>src</em>, <em>tgt</em>, <em>sources=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.get_dirs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.get_dirs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all directions with boolean values or list of sources.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>src</strong> (<em>str</em>) – Source node.</p></li>
<li><p><strong>tgt</strong> (<em>str</em>) – Target node.</p></li>
<li><p><strong>sources</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to return
the <code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code> attribute instead of <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Contains the <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> (or <code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code> if
specified) of the given edge.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.get_sign">
<code class="descname">get_sign</code><span class="sig-paren">(</span><em>direction</em>, <em>sign=None</em>, <em>sources=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.get_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.get_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the sign information of the edge in the given
diretion. If specified in <em>sign</em>, only that sign’s information
will be retrieved. If specified in <em>sources</em>, the sources of
that information will be retrieved instead.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – Contains the pair of nodes specifying the directionality of
the edge from which th information is to be retrieved.</p></li>
<li><p><strong>sign</strong> (<em>str</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. Denotes whether to retrieve
the <code class="docutils literal notranslate"><span class="pre">'positive'</span></code> or <code class="docutils literal notranslate"><span class="pre">'negative'</span></code> specific information.</p></li>
<li><p><strong>sources</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default. Specifies whether to return
the sources instead of sign.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – If <code class="docutils literal notranslate"><span class="pre">sign=None</span></code> containing [bool] values
denoting the presence of positive and negative sign on that
direction, if <code class="docutils literal notranslate"><span class="pre">sources=True</span></code> the [set] of sources for each
of them will be returned instead. If <em>sign</em> is specified,
returns [bool] or [set] (if <code class="docutils literal notranslate"><span class="pre">sources=True</span></code>) of that
specific direction and sign.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.has_sign">
<code class="descname">has_sign</code><span class="sig-paren">(</span><em>direction=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.has_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.has_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks whether the edge (or for a specific <em>direction</em>) has
any signed information (about positive/negative interactions).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>direction</strong> (<em>tuple</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If specified, only the
information of that direction is checked for sign.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><dl class="simple">
<dt>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if there exist any information on the</dt><dd><p>sign of the interaction, <code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.is_directed">
<code class="descname">is_directed</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.is_directed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.is_directed" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if edge has any directionality information.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>bool</em>) – Returns <code class="docutils literal notranslate"><span class="pre">True</span></code> if any of the <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code>
attribute values is <code class="docutils literal notranslate"><span class="pre">True</span></code> (except <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code>),
<code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.is_inhibition">
<code class="descname">is_inhibition</code><span class="sig-paren">(</span><em>direction=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.is_inhibition"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.is_inhibition" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if any (or for a specific <em>direction</em>) interaction is
inhibition (negative interaction).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>direction</strong> (<em>tuple</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If specified, checks the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative</span></code> attribute of that specific
directionality. If not specified, checks both.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if any interaction (or the specified
<em>direction</em>) is inhibitory (negative).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.is_mutual">
<code class="descname">is_mutual</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.is_mutual"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.is_mutual" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the edge has mutual directions (both A–&gt;B and B–&gt;A).</p>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.is_stimulation">
<code class="descname">is_stimulation</code><span class="sig-paren">(</span><em>direction=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.is_stimulation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.is_stimulation" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if any (or for a specific <em>direction</em>) interaction is
activation (positive interaction).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>direction</strong> (<em>tuple</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If specified, checks the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">positive</span></code> attribute of that specific
directionality. If not specified, checks both.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if any interaction (or the specified
<em>direction</em>) is activatory (positive).</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.majority_dir">
<code class="descname">majority_dir</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.majority_dir"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.majority_dir" title="Permalink to this definition">¶</a></dt>
<dd><p>Infers which is the major directionality of the edge by number
of supporting sources.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>tuple</em>) – Contains the pair of nodes denoting the
consensus directionality. If the number of sources on both
directions is equal, <code class="docutils literal notranslate"><span class="pre">None</span></code> is returned. If there is no
directionality information, <code class="docutils literal notranslate"><span class="pre">'undirected'`</span></code> will be
returned.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.majority_sign">
<code class="descname">majority_sign</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.majority_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.majority_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Infers which is the major sign (activation/inhibition) of the
edge by number of supporting sources on both directions.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>dict</em>) – Keys are the node tuples on both directions
(<code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code>/<code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>) and values can be
either <code class="docutils literal notranslate"><span class="pre">None</span></code> if that direction has no sign information or
a list of two [bool] elements corresponding to majority of
positive and majority of negative support. In case both
elements of the list are <code class="docutils literal notranslate"><span class="pre">True</span></code>, this means the number of
supporting sources for both signs in that direction is
equal.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.merge">
<code class="descname">merge</code><span class="sig-paren">(</span><em>other</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.merge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.merge" title="Permalink to this definition">¶</a></dt>
<dd><p>Merges current edge with another (if and only if they are the
same class and contain the same nodes). Updates the attributes
<code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code>, <code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code>, <code class="xref py py-attr docutils literal notranslate"><span class="pre">positive</span></code>,
<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative</span></code>, <code class="xref py py-attr docutils literal notranslate"><span class="pre">positive_sources</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative_sources</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>other</strong> (<a class="reference internal" href="#pypath.main.Direction" title="pypath.main.Direction"><em>pypath.main.Direction</em></a>) – The new edge object to be merged with the current one.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.negative_reverse">
<code class="descname">negative_reverse</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.negative_reverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.negative_reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> directionality is a negative
interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if there is supporting information on
the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> direction of the edge as inhibition.
<code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.negative_sources_reverse">
<code class="descname">negative_sources_reverse</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.negative_sources_reverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.negative_sources_reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
direction and negative sign.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> directionality of the edge with a
negative sign.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.negative_sources_straight">
<code class="descname">negative_sources_straight</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.negative_sources_straight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.negative_sources_straight" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code>
direction and negative sign.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> directionality of the edge with a
negative sign.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.negative_straight">
<code class="descname">negative_straight</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.negative_straight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.negative_straight" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> directionality is a negative
interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if there is supporting information on
the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> direction of the edge as inhibition.
<code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.positive_reverse">
<code class="descname">positive_reverse</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.positive_reverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.positive_reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> directionality is a positive
interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if there is supporting information on
the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> direction of the edge as activation.
<code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.positive_sources_reverse">
<code class="descname">positive_sources_reverse</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.positive_sources_reverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.positive_sources_reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code>
direction and positive sign.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> directionality of the edge with a
positive sign.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.positive_sources_straight">
<code class="descname">positive_sources_straight</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.positive_sources_straight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.positive_sources_straight" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code>
direction and positive sign.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> directionality of the edge with a
positive sign.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.positive_straight">
<code class="descname">positive_straight</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.positive_straight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.positive_straight" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> directionality is a positive
interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>bool</em>) – <code class="docutils literal notranslate"><span class="pre">True</span></code> if there is supporting information on
the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> direction of the edge as activation.
<code class="docutils literal notranslate"><span class="pre">False</span></code> otherwise.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.set_dir">
<code class="descname">set_dir</code><span class="sig-paren">(</span><em>direction</em>, <em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.set_dir"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.set_dir" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds directionality information with the corresponding data
source named. Modifies self attributes <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – Or [str], the directionality key for which the value on
<code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> has to be set <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>source</strong> (<em>set</em>) – Contains the name(s) of the source(s) from which such
information was obtained.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.set_sign">
<code class="descname">set_sign</code><span class="sig-paren">(</span><em>direction</em>, <em>sign</em>, <em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.set_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.set_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets sign and source information on a given direction of the
edge. Modifies the attributes <code class="xref py py-attr docutils literal notranslate"><span class="pre">positive</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">positive_sources</span></code> or <code class="xref py py-attr docutils literal notranslate"><span class="pre">negative</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative_sources</span></code> depending on the sign. Direction is
also updated accordingly, which also modifies the attributes
<code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> and <code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – Pair of edge nodes specifying the direction from which the
information is to be set/updated.</p></li>
<li><p><strong>sign</strong> (<em>str</em>) – Specifies the type of interaction. If <code class="docutils literal notranslate"><span class="pre">'positive'</span></code>, is
considered activation, otherwise, is assumed to be negative
(inhibition).</p></li>
<li><p><strong>source</strong> (<em>set</em>) – Contains the name(s) of the source(s) from which the
information was obtained.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.sources_reverse">
<code class="descname">sources_reverse</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.sources_reverse"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.sources_reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> direction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">reverse</span></code> directionality of the edge.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.sources_straight">
<code class="descname">sources_straight</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.sources_straight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.sources_straight" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources for the <code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code>
direction.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
<code class="xref py py-attr docutils literal notranslate"><span class="pre">straight</span></code> directionality of the edge.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.sources_undirected">
<code class="descname">sources_undirected</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.sources_undirected"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.sources_undirected" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of sources without directed information.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>set</em>) – Contains the names of the sources supporting the
edge presence but without specific directionality
information.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.src">
<code class="descname">src</code><span class="sig-paren">(</span><em>undirected=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.src"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.src" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the name(s) of the source node(s) for each existing
direction on the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>undirected</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the name(s) for the source node(s).
This means if the interaction is bidirectional, the list
will contain both identifiers on the edge. If the
interaction is undirected, an empty list will be returned.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.src_by_source">
<code class="descname">src_by_source</code><span class="sig-paren">(</span><em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.src_by_source"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.src_by_source" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the name(s) of the source node(s) for each existing
direction on the interaction for a specific <em>source</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>source</strong> (<em>str</em>) – Name of the source according to which the information is to
be retrieved.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the name(s) for the source node(s)
according to the specified <em>source</em>. This means if the
interaction is bidirectional, the list will contain both
identifiers on the edge. If the specified <em>source</em> is not
found or invalid, an empty list will be returned.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.tgt">
<code class="descname">tgt</code><span class="sig-paren">(</span><em>undirected=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.tgt"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.tgt" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the name(s) of the target node(s) for each existing
direction on the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>undirected</strong> (<em>bool</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">False</span></code> by default.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the name(s) for the target node(s).
This means if the interaction is bidirectional, the list
will contain both identifiers on the edge. If the
interaction is undirected, an empty list will be returned.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.tgt_by_source">
<code class="descname">tgt_by_source</code><span class="sig-paren">(</span><em>source</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.tgt_by_source"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.tgt_by_source" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the name(s) of the target node(s) for each existing
direction on the interaction for a specific <em>source</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>source</strong> (<em>str</em>) – Name of the source according to which the information is to
be retrieved.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>list</em>) – Contains the name(s) for the target node(s)
according to the specified <em>source</em>. This means if the
interaction is bidirectional, the list will contain both
identifiers on the edge. If the specified <em>source</em> is not
found or invalid, an empty list will be returned.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.translate">
<code class="descname">translate</code><span class="sig-paren">(</span><em>ids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.translate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates the node names/identifiers according to the
dictionary <em>ids</em>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>ids</strong> (<em>dict</em>) – Dictionary containing (at least) the current names of the
nodes as keys and their translation as values.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>(<em>pypath.main.Direction</em>) – The copy of current edge object
with translated node names.</p>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.unset_dir">
<code class="descname">unset_dir</code><span class="sig-paren">(</span><em>direction</em>, <em>source=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.unset_dir"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.unset_dir" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes directionality and/or source information of the
specified <em>direction</em>. Modifies attribute <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – Or [str] (if <code class="docutils literal notranslate"><span class="pre">'undirected'</span></code>) the pair of nodes specifying
the directionality from which the information is to be
removed.</p></li>
<li><p><strong>source</strong> (<em>set</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If specified, determines
which specific source(s) is(are) to be removed from
<code class="xref py py-attr docutils literal notranslate"><span class="pre">sources</span></code> attribute in the specified <em>direction</em>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.unset_sign">
<code class="descname">unset_sign</code><span class="sig-paren">(</span><em>direction</em>, <em>sign</em>, <em>source=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.unset_sign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.unset_sign" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes sign and/or source information of the specified
<em>direction</em> and <em>sign</em>. Modifies attribute <code class="xref py py-attr docutils literal notranslate"><span class="pre">positive</span></code>
and <code class="xref py py-attr docutils literal notranslate"><span class="pre">positive_sources</span></code> or <code class="xref py py-attr docutils literal notranslate"><span class="pre">negative</span></code> and
<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative_sources</span></code> (or
<code class="xref py py-attr docutils literal notranslate"><span class="pre">positive_attributes</span></code>/<code class="xref py py-attr docutils literal notranslate"><span class="pre">negative_sources</span></code>
only if <code class="docutils literal notranslate"><span class="pre">source=True</span></code>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>direction</strong> (<em>tuple</em>) – The pair of nodes specifying the directionality from which
the information is to be removed.</p></li>
<li><p><strong>sign</strong> (<em>str</em>) – Sign from which the information is to be removed. Must be
either <code class="docutils literal notranslate"><span class="pre">'positive'</span></code> or <code class="docutils literal notranslate"><span class="pre">'negative'</span></code>.</p></li>
<li><p><strong>source</strong> (<em>set</em>) – Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default. If specified, determines
which source(s) is(are) to be removed from the sources in
the specified <em>direction</em> and <em>sign</em>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pypath.main.Direction.which_dirs">
<code class="descname">which_dirs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#Direction.which_dirs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.Direction.which_dirs" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the pair(s) of nodes for which there is information
about their directionality.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<em>list</em>) – List of tuples containing the nodes for which
their attribute <code class="xref py py-attr docutils literal notranslate"><span class="pre">dirs</span></code> is <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pypath.main.AttrHelper">
<em class="property">class </em><code class="descclassname">pypath.main.</code><code class="descname">AttrHelper</code><span class="sig-paren">(</span><em>value</em>, <em>name=None</em>, <em>defaults={}</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pypath/main.html#AttrHelper"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pypath.main.AttrHelper" title="Permalink to this definition">¶</a></dt>
<dd><p>Attribute helper class.</p>
<ul class="simple">
<li><dl class="simple">
<dt>Initialization arguments:</dt><dd><ul>
<li><p><em>value</em> [dict/str]?:</p></li>
<li><p><em>name</em> [str]?: Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default.</p></li>
<li><p><em>defaults</em> [dict]:</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Attributes:</dt><dd><ul>
<li><p><em>value</em> [dict]?:</p></li>
<li><p><em>name</em> [str]?:</p></li>
<li><p><em>defaults</em> [dict]:</p></li>
<li><p><em>id_type</em> [type]:</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Call arguments:</dt><dd><ul>
<li><p><em>instance</em> []:</p></li>
<li><p><em>this_directed</em> [tuple?]: Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default.</p></li>
<li><p><em>thisSign</em> []: Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default.</p></li>
<li><p><em>this_directedSources</em> []: Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default.</p></li>
<li><p><em>thisSources</em> []: Optional, <code class="docutils literal notranslate"><span class="pre">None</span></code> by default.</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Returns:</dt><dd><ul>
<li></li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd></dl>

</div>
<div class="section" id="module-pypath.ptm">
<span id="ptm-module"></span><h2>ptm module<a class="headerlink" href="#module-pypath.ptm" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Reference</a><ul>
<li><a class="reference internal" href="#module-pypath.annot">annot module</a></li>
<li><a class="reference internal" href="#bel-module">bel module</a></li>
<li><a class="reference internal" href="#module-pypath.complex">complex module</a></li>
<li><a class="reference internal" href="#module-pypath.intera">intera module</a></li>
<li><a class="reference internal" href="#module-pypath.intercell">intercell module</a></li>
<li><a class="reference internal" href="#module-pypath.main">main module</a></li>
<li><a class="reference internal" href="#module-pypath.ptm">ptm module</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="installation.html"
                        title="previous chapter">Installation</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="webservice.html"
                        title="next chapter">Webservice</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/reference.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="webservice.html" title="Webservice"
             >next</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Installation"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pypath 0.8.34 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Dénes Türei, Nicolàs Palacio, Olga Ivanova. 
European Molecular Biology Laboratory, Heidelberg Germany
European Bioinformatics Institute, Hinxton UK
University Hospital RWTH, Aachen Germany
University Hospital Heidelberg Germany.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.0.1.
    </div>
  </body>
</html>