%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{4}
\setcounter{secnumdepth}{4}


\title{pypath Documentation}
\date{Oct 14, 2019}
\release{0.8.34}
\author{Dénes Türei}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxstylestrong{pypath} is a Python package built around igraph to work with molecular
network representations e.g. protein, miRNA and drug compound interaction
networks.
\begin{quote}\begin{description}
\item[{note}] \leavevmode
\sphinxcode{\sphinxupquote{pypath}} supports both Python 2.7 and Python 3.6+. In the beginning,
pypath has been developed only for Python 2.7. Then the code have been
adjusted to Py3 however we can not guarantee no incompatibilities
remained. If you find any method does not work please submit an issue on
github. For few years I develop and test \sphinxcode{\sphinxupquote{pypath}} in Python 3. Therefore
this is the better supported Python variant.

\item[{documentation}] \leavevmode
\sphinxurl{http://saezlab.github.io/pypath}

\item[{issues}] \leavevmode
\sphinxurl{https://github.com/saezlab/pypath/issues}

\end{description}\end{quote}


\chapter{Installation}
\label{\detokenize{installation:installation}}\label{\detokenize{installation::doc}}

\section{Linux}
\label{\detokenize{installation:linux}}
In almost any up-to-date Linux distribution the dependencies of \sphinxstylestrong{pypath} are
built-in, or provided by the distributors. You only need to install a couple
of things in your package manager (cairo, py(2)cairo, igraph,
python(2)-igraph, graphviz, pygraphviz), and after install \sphinxstylestrong{pypath} by \sphinxstyleemphasis{pip}
(see below). If any module still missing, you can install them the usual way
by \sphinxstyleemphasis{pip} or your package manager.


\subsection{igraph C library, cairo and pycairo}
\label{\detokenize{installation:igraph-c-library-cairo-and-pycairo}}
\sphinxstyleemphasis{python(2)-igraph} is a Python interface to use the igraph C library. The
C library must be installed. The same goes for \sphinxstyleemphasis{cairo}, \sphinxstyleemphasis{py(2)cairo} and
\sphinxstyleemphasis{graphviz}.


\subsection{Directly from git}
\label{\detokenize{installation:directly-from-git}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pip install git+https://github.com/saezlab/pypath.git
\end{sphinxVerbatim}


\subsection{With pip}
\label{\detokenize{installation:with-pip}}
Download the package from /dist, and install with pip:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pip install pypath\PYGZhy{}x.y.z.tar.gz
\end{sphinxVerbatim}


\subsection{Build source distribution}
\label{\detokenize{installation:build-source-distribution}}
Clone the git repo, and run setup.py:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
python setup.py sdist
\end{sphinxVerbatim}


\section{Mac OS X}
\label{\detokenize{installation:mac-os-x}}
On OS X installation is not straightforward primarily because cairo needs to
be compiled from source. We provide 2 scripts here: the
\sphinxstylestrong{mac-install-brew.sh} installs everything with HomeBrew, and
\sphinxstylestrong{mac-install-conda.sh} installs from Anaconda distribution. With these
scripts installation of igraph, cairo and graphviz goes smoothly most of the
time, and options are available for omitting the 2 latter. To know more see
the description in the script header. There is a third script
\sphinxstylestrong{mac-install-source.sh} which compiles everything from source and presumes
only Python 2.7 and Xcode installed. We do not recommend this as it is time
consuming and troubleshooting requires expertise.


\subsection{Troubleshooting}
\label{\detokenize{installation:troubleshooting}}\begin{itemize}
\item {} 
\sphinxcode{\sphinxupquote{no module named ...}} when you try to load a module in Python. Did
theinstallation of the module run without error? Try to run again the specific
part from the mac install shell script to see if any error comes up. Is the
path where the module has been installed in your \sphinxcode{\sphinxupquote{\$PYTHONPATH}}? Try \sphinxcode{\sphinxupquote{echo
\$PYTHONPATH}} to see the current paths. Add your local install directories if
those are not there, e.g.
\sphinxcode{\sphinxupquote{export PYTHONPATH="/Users/me/local/python2.7/site-packages:\$PYTHONPATH"}}.
If it works afterwards, don’t forget to append these export path statements to
your \sphinxcode{\sphinxupquote{\textasciitilde{}/.bash\_profile}}, so these will be set every time you launch a new
shell.

\item {} 
\sphinxcode{\sphinxupquote{pkgconfig}} not found. Check if the \sphinxcode{\sphinxupquote{\$PKG\_CONFIG\_PATH}} variable is
set correctly, and pointing on a directory where pkgconfig really can be
found.

\item {} 
Error while trying to install py(2)cairo by pip. py(2)cairo could not be
installed by pip, but only by waf. Please set the \sphinxcode{\sphinxupquote{\$PKG\_CONFIG\_PATH}} before.
See \sphinxstylestrong{mac-install-source.sh} on how to install with waf.

\item {} 
Error at pygraphviz build: \sphinxcode{\sphinxupquote{graphviz/cgraph.h file not found}}. This is
because the directory of graphviz detected wrong by pkgconfig. See
\sphinxstylestrong{mac-install-source.sh} how to set include dirs and library dirs by
\sphinxcode{\sphinxupquote{-{-}global-option}} parameters.

\item {} 
Can not install bioservices, because installation of jurko-suds fails. Ok,
this fails because pip is not able to install the recent version of
setuptools, because a very old version present in the system path. The
development version of jurko-suds does not require setuptools, so you can
install it directly from git as it is done in \sphinxstylestrong{mac-install-source.sh}.

\item {} 
In \sphinxstylestrong{Anaconda}, \sphinxstyleemphasis{pypath} can be imported, but the modules and classes are
missing. Apparently Anaconda has some built-in stuff called \sphinxstyleemphasis{pypath}. This
has nothing to do with this module. Please be aware that Anaconda installs a
completely separated Python distribution, and does not detect modules in the
main Python installation. You need to install all modules within Anaconda’s
directory. \sphinxstylestrong{mac-install-conda.sh} does exactly this. If you still
experience issues, please contact us.

\end{itemize}


\section{Microsoft Windows}
\label{\detokenize{installation:microsoft-windows}}
Not many people have used \sphinxstyleemphasis{pypath} on Microsoft computers so far. Please share
your experiences and contact us if you encounter any issue. We appreciate
your feedback, and it would be nice to have better support for other computer
systems.


\subsection{With Anaconda}
\label{\detokenize{installation:with-anaconda}}
The same workflow like you see in \sphinxcode{\sphinxupquote{mac-install-conda.sh}} should work for
Anaconda on Windows. The only problem you certainly will encounter is that not
all the channels have packages for all platforms. If certain channel provides
no package for Windows, or for your Python version, you just need to find an
other one. For this, do a search:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
anaconda search \PYGZhy{}t conda \PYGZlt{}package name\PYGZgt{}
\end{sphinxVerbatim}

For example, if you search for \sphinxstyleemphasis{pycairo}, you will find out that \sphinxstyleemphasis{vgauther}
provides it for osx-64, but only for Python 3.4, while \sphinxstyleemphasis{richlewis} provides
also for Python 3.5. And for win-64 platform, there is the channel of
\sphinxstyleemphasis{KristanAmstrong}. Go along all the commands in \sphinxcode{\sphinxupquote{mac-install-conda.sh}}, and
modify the channel if necessary, until all packages install successfully.


\subsection{With other Python distributions}
\label{\detokenize{installation:with-other-python-distributions}}
Here the basic principles are the same as everywhere: first try to install all
external dependencies, after \sphinxstyleemphasis{pip} install should work. On Windows certain
packages can not be installed by compiled from source by \sphinxstyleemphasis{pip}, instead the
easiest to install them precompiled. These are in our case \sphinxstyleemphasis{fisher, lxml,
numpy (mkl version), pycairo, igraph, pygraphviz, scipy and statsmodels}. The
precompiled packages are available here:
\sphinxurl{http://www.lfd.uci.edu/~gohlke/pythonlibs/}. We tested the setup with Python
3.4.3 and Python 2.7.11. The former should just work fine, while with the
latter we have issues to be resolved.


\subsection{Known issues}
\label{\detokenize{installation:known-issues}}\begin{itemize}
\item {} 
\sphinxstyleemphasis{“No module fabric available.”} \textendash{} or \sphinxstyleemphasis{pysftp} missing: this is not

\end{itemize}

important, only certain data download methods rely on these modules, but
likely you won’t call those at all.
* Progress indicator floods terminal: sorry about that, will be fixed soon.
* Encoding related exceptions in Python2: these might occur at some points in
the module, please send the traceback if you encounter one, and we will fix
as soon as possible.

\sphinxstyleemphasis{Special thanks to Jorge Ferreira for testing pypath on Windows!}


\chapter{Reference}
\label{\detokenize{reference:reference}}\label{\detokenize{reference::doc}}

\section{annot module}
\label{\detokenize{reference:module-pypath.annot}}\label{\detokenize{reference:annot-module}}\index{pypath.annot (module)@\spxentry{pypath.annot}\spxextra{module}}\index{Adhesome (class in pypath.annot)@\spxentry{Adhesome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Adhesome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Adhesome}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{AnnotationBase (class in pypath.annot)@\spxentry{AnnotationBase}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{AnnotationBase}}}{\emph{name}, \emph{ncbi\_tax\_id=9606}, \emph{input\_method=None}, \emph{input\_args=None}, \emph{entity\_type='protein'}, \emph{swissprot\_only=True}, \emph{proteins=()}, \emph{complexes=()}, \emph{reference\_set=()}, \emph{infer\_complexes=True}, \emph{dump=None}, \emph{**kwargs}}{}~\index{add\_complexes\_by\_inference() (pypath.annot.AnnotationBase method)@\spxentry{add\_complexes\_by\_inference()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.add_complexes_by_inference}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_complexes\_by\_inference}}}{\emph{complexes=None}}{}
Creates complex annotations by in silico inference and adds them
to this annotation set.

\end{fulllineitems}

\index{all\_proteins() (pypath.annot.AnnotationBase method)@\spxentry{all\_proteins()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.all_proteins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{all\_proteins}}}{}{}
All UniProt IDs annotated in this resource.

\end{fulllineitems}

\index{annotate\_complex() (pypath.annot.AnnotationBase method)@\spxentry{annotate\_complex()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.annotate_complex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{annotate\_complex}}}{\emph{cplex}}{}
Infers annotations for a single complex.

\end{fulllineitems}

\index{complex\_inference() (pypath.annot.AnnotationBase method)@\spxentry{complex\_inference()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.complex_inference}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{complex\_inference}}}{\emph{complexes=None}}{}
Annotates all complexes in \sphinxtitleref{complexes}, by default in the default
complex database (existing in the \sphinxtitleref{complex} module or generated
on demand according to the module’s current settings).

Dict with complexes as keys and sets of annotations as values.
Complexes with no valid information in this annotation resource
won’t be in the dict.
\begin{description}
\item[{complexes}] \leavevmode{[}iterable{]}
Iterable yielding complexes.

\end{description}

\end{fulllineitems}

\index{get\_subset() (pypath.annot.AnnotationBase method)@\spxentry{get\_subset()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.get_subset}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_subset}}}{\emph{method=None}, \emph{**kwargs}}{}
Retrieves a subset by filtering based on \sphinxcode{\sphinxupquote{kwargs}}.
Each argument should be a name and a value or set of values.
Elements having the provided values in the annotation will be
returned.
Returns a set of UniProt IDs.

\end{fulllineitems}

\index{load\_proteins() (pypath.annot.AnnotationBase method)@\spxentry{load\_proteins()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.load_proteins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_proteins}}}{}{}
Retrieves a set of all UniProt IDs to have a base set of the entire
proteome.

\end{fulllineitems}

\index{reload() (pypath.annot.AnnotationBase method)@\spxentry{reload()}\spxextra{pypath.annot.AnnotationBase method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.AnnotationBase.reload}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reload}}}{}{}
Reloads the object from the module level.

\end{fulllineitems}


\end{fulllineitems}

\index{CancerGeneCensus (class in pypath.annot)@\spxentry{CancerGeneCensus}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CancerGeneCensus}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CancerGeneCensus}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Cancersea (class in pypath.annot)@\spxentry{Cancersea}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Cancersea}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Cancersea}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{CellPhoneDB (class in pypath.annot)@\spxentry{CellPhoneDB}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CellPhoneDB}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CellPhoneDB}}}{\emph{**kwargs}}{}~\index{record (pypath.annot.CellPhoneDB attribute)@\spxentry{record}\spxextra{pypath.annot.CellPhoneDB attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CellPhoneDB.record}}\pysigline{\sphinxbfcode{\sphinxupquote{record}}}
alias of \sphinxcode{\sphinxupquote{pypath.dataio.CellPhoneDBAnnotation}}

\end{fulllineitems}


\end{fulllineitems}

\index{CellPhoneDBComplex (class in pypath.annot)@\spxentry{CellPhoneDBComplex}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CellPhoneDBComplex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CellPhoneDBComplex}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{CellSurfaceProteinAtlas (class in pypath.annot)@\spxentry{CellSurfaceProteinAtlas}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CellSurfaceProteinAtlas}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CellSurfaceProteinAtlas}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Comppi (class in pypath.annot)@\spxentry{Comppi}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Comppi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Comppi}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Corum (class in pypath.annot)@\spxentry{Corum}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Corum}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Corum}}}{\emph{name}, \emph{annot\_attr}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{CorumFuncat (class in pypath.annot)@\spxentry{CorumFuncat}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CorumFuncat}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CorumFuncat}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{CorumGO (class in pypath.annot)@\spxentry{CorumGO}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.CorumGO}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{CorumGO}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Cpad (class in pypath.annot)@\spxentry{Cpad}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Cpad}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Cpad}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Dgidb (class in pypath.annot)@\spxentry{Dgidb}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Dgidb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Dgidb}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Disgenet (class in pypath.annot)@\spxentry{Disgenet}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Disgenet}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Disgenet}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Exocarta (class in pypath.annot)@\spxentry{Exocarta}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Exocarta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Exocarta}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{GOIntercell (class in pypath.annot)@\spxentry{GOIntercell}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.GOIntercell}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{GOIntercell}}}{\emph{categories=None}, \emph{go\_annot=None}, \emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{GuideToPharmacology (class in pypath.annot)@\spxentry{GuideToPharmacology}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.GuideToPharmacology}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{GuideToPharmacology}}}{\emph{load\_sources=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Hgnc (class in pypath.annot)@\spxentry{Hgnc}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Hgnc}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Hgnc}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{HpmrComplex (class in pypath.annot)@\spxentry{HpmrComplex}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.HpmrComplex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{HpmrComplex}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{HumanPlasmaMembraneReceptome (class in pypath.annot)@\spxentry{HumanPlasmaMembraneReceptome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.HumanPlasmaMembraneReceptome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{HumanPlasmaMembraneReceptome}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{HumanProteinAtlas (class in pypath.annot)@\spxentry{HumanProteinAtlas}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.HumanProteinAtlas}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{HumanProteinAtlas}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Integrins (class in pypath.annot)@\spxentry{Integrins}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Integrins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Integrins}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Intogen (class in pypath.annot)@\spxentry{Intogen}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Intogen}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Intogen}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{KeggPathways (class in pypath.annot)@\spxentry{KeggPathways}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.KeggPathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{KeggPathways}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Kinasedotcom (class in pypath.annot)@\spxentry{Kinasedotcom}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Kinasedotcom}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Kinasedotcom}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Kirouac2010 (class in pypath.annot)@\spxentry{Kirouac2010}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Kirouac2010}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Kirouac2010}}}{\emph{load\_sources=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{LigandReceptor (class in pypath.annot)@\spxentry{LigandReceptor}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.LigandReceptor}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{LigandReceptor}}}{\emph{name}, \emph{ligand\_col=None}, \emph{receptor\_col=None}, \emph{ligand\_id\_type=None}, \emph{receptor\_id\_type=None}, \emph{record\_processor\_method=None}, \emph{record\_extra\_fields=None}, \emph{record\_defaults=None}, \emph{extra\_fields\_methods=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Locate (class in pypath.annot)@\spxentry{Locate}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Locate}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Locate}}}{\emph{ncbi\_tax\_id=9606}, \emph{literature=True}, \emph{external=True}, \emph{predictions=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Matrisome (class in pypath.annot)@\spxentry{Matrisome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Matrisome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Matrisome}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Matrixdb (class in pypath.annot)@\spxentry{Matrixdb}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Matrixdb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Matrixdb}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Membranome (class in pypath.annot)@\spxentry{Membranome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Membranome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Membranome}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{NetpathPathways (class in pypath.annot)@\spxentry{NetpathPathways}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.NetpathPathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{NetpathPathways}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Opm (class in pypath.annot)@\spxentry{Opm}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Opm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Opm}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Phosphatome (class in pypath.annot)@\spxentry{Phosphatome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Phosphatome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Phosphatome}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Ramilowski2015 (class in pypath.annot)@\spxentry{Ramilowski2015}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Ramilowski2015}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Ramilowski2015}}}{\emph{load\_sources=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Ramilowski2015Location (class in pypath.annot)@\spxentry{Ramilowski2015Location}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Ramilowski2015Location}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Ramilowski2015Location}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{SignalinkPathways (class in pypath.annot)@\spxentry{SignalinkPathways}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.SignalinkPathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{SignalinkPathways}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{SignorPathways (class in pypath.annot)@\spxentry{SignorPathways}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.SignorPathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{SignorPathways}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Surfaceome (class in pypath.annot)@\spxentry{Surfaceome}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Surfaceome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Surfaceome}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Tfcensus (class in pypath.annot)@\spxentry{Tfcensus}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Tfcensus}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Tfcensus}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Topdb (class in pypath.annot)@\spxentry{Topdb}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Topdb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Topdb}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Vesiclepedia (class in pypath.annot)@\spxentry{Vesiclepedia}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Vesiclepedia}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Vesiclepedia}}}{\emph{ncbi\_tax\_id=9606}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Zhong2015 (class in pypath.annot)@\spxentry{Zhong2015}\spxextra{class in pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.Zhong2015}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{Zhong2015}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{get\_db() (in module pypath.annot)@\spxentry{get\_db()}\spxextra{in module pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.get_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{get\_db}}}{\emph{keep\_annotators=True}, \emph{create\_dataframe=False}, \emph{use\_complexes=True}, \emph{**kwargs}}{}
Retrieves the current database instance and initializes it if does
not exist yet.

\end{fulllineitems}

\index{init\_db() (in module pypath.annot)@\spxentry{init\_db()}\spxextra{in module pypath.annot}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.annot.init_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pypath.annot.}}\sphinxbfcode{\sphinxupquote{init\_db}}}{\emph{keep\_annotators=True}, \emph{create\_dataframe=False}, \emph{use\_complexes=True}, \emph{**kwargs}}{}
Initializes or reloads the annotation database.
The database will be assigned to the \sphinxcode{\sphinxupquote{db}} attribute of this module.

\end{fulllineitems}



\section{bel module}
\label{\detokenize{reference:bel-module}}

\section{complex module}
\label{\detokenize{reference:module-pypath.complex}}\label{\detokenize{reference:complex-module}}\index{pypath.complex (module)@\spxentry{pypath.complex}\spxextra{module}}\index{AbstractComplexResource (class in pypath.complex)@\spxentry{AbstractComplexResource}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.AbstractComplexResource}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{AbstractComplexResource}}}{\emph{name}, \emph{ncbi\_tax\_id=9606}, \emph{input\_method=None}, \emph{input\_args=None}, \emph{dump=None}, \emph{**kwargs}}{}
A resource which provides information about molecular complexes.

\end{fulllineitems}

\index{CellPhoneDB (class in pypath.complex)@\spxentry{CellPhoneDB}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.CellPhoneDB}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{CellPhoneDB}}}{\emph{**kwargs}}{}
\end{fulllineitems}

\index{Compleat (class in pypath.complex)@\spxentry{Compleat}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Compleat}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Compleat}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{ComplexAggregator (class in pypath.complex)@\spxentry{ComplexAggregator}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.ComplexAggregator}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{ComplexAggregator}}}{\emph{resources=None}, \emph{pickle\_file=None}}{}~\index{reload() (pypath.complex.ComplexAggregator method)@\spxentry{reload()}\spxextra{pypath.complex.ComplexAggregator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.ComplexAggregator.reload}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reload}}}{}{}
Reloads the object from the module level.

\end{fulllineitems}


\end{fulllineitems}

\index{ComplexPortal (class in pypath.complex)@\spxentry{ComplexPortal}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.ComplexPortal}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{ComplexPortal}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Corum (class in pypath.complex)@\spxentry{Corum}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Corum}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Corum}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{GuideToPharmacology (class in pypath.complex)@\spxentry{GuideToPharmacology}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.GuideToPharmacology}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{GuideToPharmacology}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Havugimana (class in pypath.complex)@\spxentry{Havugimana}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Havugimana}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Havugimana}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Hpmr (class in pypath.complex)@\spxentry{Hpmr}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Hpmr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Hpmr}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Humap (class in pypath.complex)@\spxentry{Humap}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Humap}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Humap}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Pdb (class in pypath.complex)@\spxentry{Pdb}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Pdb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Pdb}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{Signor (class in pypath.complex)@\spxentry{Signor}\spxextra{class in pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.Signor}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{Signor}}}{\emph{input\_args=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{all\_complexes() (in module pypath.complex)@\spxentry{all\_complexes()}\spxextra{in module pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.all_complexes}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{all\_complexes}}}{}{}
Returns a set of all complexes in the database which serves as a
reference set for many methods, just like \sphinxcode{\sphinxupquote{uniprot\_input.all\_uniprots}}
represents the proteome.

\end{fulllineitems}

\index{get\_db() (in module pypath.complex)@\spxentry{get\_db()}\spxextra{in module pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.get_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{get\_db}}}{\emph{**kwargs}}{}
Retrieves the current database instance and initializes it if does
not exist yet.

\end{fulllineitems}

\index{init\_db() (in module pypath.complex)@\spxentry{init\_db()}\spxextra{in module pypath.complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.complex.init_db}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pypath.complex.}}\sphinxbfcode{\sphinxupquote{init\_db}}}{\emph{**kwargs}}{}
Initializes or reloads the complex database.
The database will be assigned to the \sphinxcode{\sphinxupquote{db}} attribute of this module.

\end{fulllineitems}



\section{intera module}
\label{\detokenize{reference:module-pypath.intera}}\label{\detokenize{reference:intera-module}}\index{pypath.intera (module)@\spxentry{pypath.intera}\spxextra{module}}
This module provides classes to represent and handle
structural details of protein interactions
i.e. residues, post-translational modifications,
short motifs, domains, domain-motifs and
domain-motif interactions, binding interfaces.


\section{intercell module}
\label{\detokenize{reference:module-pypath.intercell}}\label{\detokenize{reference:intercell-module}}\index{pypath.intercell (module)@\spxentry{pypath.intercell}\spxextra{module}}\index{IntercellRole (class in pypath.intercell)@\spxentry{IntercellRole}\spxextra{class in pypath.intercell}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.intercell.IntercellRole}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.intercell.}}\sphinxbfcode{\sphinxupquote{IntercellRole}}}{\emph{source}, \emph{role}}{}~\index{role (pypath.intercell.IntercellRole attribute)@\spxentry{role}\spxextra{pypath.intercell.IntercellRole attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.intercell.IntercellRole.role}}\pysigline{\sphinxbfcode{\sphinxupquote{role}}}
Alias for field number 1

\end{fulllineitems}

\index{source (pypath.intercell.IntercellRole attribute)@\spxentry{source}\spxextra{pypath.intercell.IntercellRole attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.intercell.IntercellRole.source}}\pysigline{\sphinxbfcode{\sphinxupquote{source}}}
Alias for field number 0

\end{fulllineitems}


\end{fulllineitems}



\section{main module}
\label{\detokenize{reference:module-pypath.main}}\label{\detokenize{reference:main-module}}\index{pypath.main (module)@\spxentry{pypath.main}\spxextra{module}}\index{PyPath (class in pypath.main)@\spxentry{PyPath}\spxextra{class in pypath.main}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.main.}}\sphinxbfcode{\sphinxupquote{PyPath}}}{\emph{ncbi\_tax\_id=None}, \emph{copy=None}, \emph{name='unnamed'}, \emph{cache\_dir=None}, \emph{outdir='results'}, \emph{loglevel=0}, \emph{loops=False}}{}
Main network object.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ncbi\_tax\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{9606}} (Homo sapiens) by default. NCBI Taxonomic
identifier of the organism from which the data will be
downloaded.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{default\_name\_type}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{'protein': 'uniprot', 'mirna': 'mirbase', 'drug':
'chembl', 'lncrna': 'lncrna-genesymbol'\}}} by default. Contains
the default identifier types to which the downloaded data will
be converted. If others are used, user may need to provide the
format definitions for the conversion tables.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{copy}} ({\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{pypath.main.PyPath}}}}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Other
{\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath}}}}} instance from which the data will
be copied.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mysql}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} DEPRECATED Optional, \sphinxcode{\sphinxupquote{(None, 'mapping')}} by default. Contains the MySQL
parameters used by the \sphinxcode{\sphinxupquote{pypath.mapping}} module to load
the ID conversion tables.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{chembl\_mysql}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} DEPRECATED Optional, \sphinxcode{\sphinxupquote{(None, 'chembl')}} by default. Contains the MySQL
parameters used by the \sphinxcode{\sphinxupquote{pypath.mapping}} module to load
the ChEMBL ID conversion tables.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'unnamed'}} by default. Session or project name
(custom).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outdir}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'results'}} by default. Output directory where to
store all output files.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{loglevel}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Optional, 0 by default. Sets the level of the logger.
The higher the level the more messages will be written to the log.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{loops}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Determines if self-loop edges
are allowed in the graph.

\end{itemize}

\item[{Variables}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{adjlist}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of {[}set{]} containing the adjacency of each node. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.update_adjlist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{PyPath.update\_adjlist()}}}}} method for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{chembl}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.chembl.Chembl}}) \textendash{} Contains the ChEMBL data. See \sphinxcode{\sphinxupquote{pypath.chembl}} module
documentation for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{chembl\_mysql}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} DEPRECATED Contains the MySQL parameters used by the
\sphinxcode{\sphinxupquote{pypath.mapping}} module to load the ChEMBL ID conversion
tables.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{data}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Stores the loaded interaction and attribute table. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.read_data_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{PyPath.read\_data\_file()}}}}} method for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{db\_dict}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary of dictionaries. Outer-level keys are \sphinxcode{\sphinxupquote{'nodes'}} and
\sphinxcode{\sphinxupquote{'edges'}}, corresponding values are {[}dict{]} whose keys are the
database sources with values of type {[}set{]} containing the
edge/node indexes for which that database provided some
information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dgraph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Directed network graph object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{disclaimer}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Disclaimer text.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dlabDct}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the directed graph node labels {[}str{]} (keys) to their
indices {[}int{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dnodDct}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the directed graph node names {[}str{]} (keys) to their indices
{[}int{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dnodInd}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Stores the directed graph node names {[}str{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dnodLab}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the directed graph node indices {[}int{]} (keys) to their
labels {[}str{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dnodNam}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the directed graph node indices {[}int{]} (keys) to their names
{[}str{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edgeAttrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Stores the edge attribute names {[}str{]} as keys and their
corresponding types (e.g.: \sphinxcode{\sphinxupquote{set}}, \sphinxcode{\sphinxupquote{list}}, \sphinxcode{\sphinxupquote{str}}, …) as
values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exp}} (\sphinxstyleliteralemphasis{\sphinxupquote{pandas.DataFrame}}) \textendash{} Stores the expression data for the nodes (if loaded).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exp\_prod}} (\sphinxstyleliteralemphasis{\sphinxupquote{pandas.DataFrame}}) \textendash{} Stores the edge expression data (as the product of the
normalized expression between the pair of nodes by default). For
more details see {\hyperref[\detokenize{reference:pypath.main.PyPath.edges_expression}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.edges\_expression()}}}}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exp\_samples}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Contains a list of tissues as downloaded by ProteomicsDB. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.get_proteomicsdb}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{PyPath.get\_proteomicsdb()}}}}} for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{failed\_edges}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of lists containing information about the failed edges.
Each failed edge sublist contains (in this order): {[}tuple{]} with
the node IDs, {[}str{]} names of nodes A and B, {[}int{]} IDs of nodes
A and B and {[}int{]} IDs of the edges in both directions.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{go}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the organism(s) NCBI taxonomy ID as key {[}int{]} and
\sphinxcode{\sphinxupquote{pypath.go.GOAnnotation}} object as value, which
contains the GO annotations for the nodes in the graph. See
\sphinxcode{\sphinxupquote{pypath.go.GOAnnotation}} for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Undirected network graph object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{gsea}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.gsea.GSEA}}) \textendash{} Contains the loaded gene-sets from MSigDB. See
\sphinxcode{\sphinxupquote{pypath.gsea.GSEA}} for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{has\_cats}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Contains the categories (e.g.: resource types) {[}str{]} loaded in
the current network. Possible categories are: \sphinxcode{\sphinxupquote{'m'}} for
PTM/enzyme-substrate resources, \sphinxcode{\sphinxupquote{'p'}} for pathway/activity
flow resources, \sphinxcode{\sphinxupquote{'i'}} for undirected/PPI resources, \sphinxcode{\sphinxupquote{'r'}}
for process description/reaction resources and \sphinxcode{\sphinxupquote{'t'}} for
transcription resources.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{htp}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains information about high-throughput data of the network
for different thresholds {[}int{]} (keys). Values are {[}dict{]}
containing the number of references (\sphinxcode{\sphinxupquote{'rnum'}}) {[}int{]}, number
of edges (\sphinxcode{\sphinxupquote{'enum'}}) {[}int{]}, number of sources (\sphinxcode{\sphinxupquote{'snum'}})
{[}int{]} and list of PMIDs of the most common references above the
given threshold (\sphinxcode{\sphinxupquote{'htrefs'}}) {[}set{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{labDct}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the undirected graph node labels {[}str{]} (keys) to their
indices {[}int{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lists}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains specific lists of nodes (values) for different
categories {[}str{]} (keys). These can to be loaded from a file or
a resource. Some methods include
\sphinxcode{\sphinxupquote{pypath.main.PyPath.receptor\_list()}} (\sphinxcode{\sphinxupquote{'rec'}}),
\sphinxcode{\sphinxupquote{pypath.main.PyPath.druggability\_list()}} (\sphinxcode{\sphinxupquote{'dgb'}}),
\sphinxcode{\sphinxupquote{pypath.main.PyPath.kinases\_list()}} (\sphinxcode{\sphinxupquote{'kin'}}),
\sphinxcode{\sphinxupquote{pypath.main.PyPath.tfs\_list()}} (\sphinxcode{\sphinxupquote{'tf'}}),
\sphinxcode{\sphinxupquote{pypath.main.PyPath.disease\_genes\_list()}} (\sphinxcode{\sphinxupquote{'dis'}}),
{\hyperref[\detokenize{reference:pypath.main.PyPath.signaling_proteins_list}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.signaling\_proteins\_list()}}}}}
(\sphinxcode{\sphinxupquote{'sig'}}), \sphinxcode{\sphinxupquote{pypath.main.PyPath.proteome\_list()}}
(\sphinxcode{\sphinxupquote{'proteome'}}) and
{\hyperref[\detokenize{reference:pypath.main.PyPath.cancer_drivers_list}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.cancer\_drivers\_list()}}}}} (\sphinxcode{\sphinxupquote{'cdv'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{loglevel}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The level of the logger.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{loops}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Whether if self-loop edges are allowed in the graph.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mapper}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.mapping.Mapper}}) \textendash{} \sphinxcode{\sphinxupquote{pypath.mapper.Mapper}} object for ID conversion and
other ID-related operations across resources.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mutation\_samples}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} DEPRECATED

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mysql\_conf}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} DEPRECATED Contains the MySQL parameters used by the
\sphinxcode{\sphinxupquote{pypath.mapping}} module to load the ID conversion
tables.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Session or project name (custom).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ncbi\_tax\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomic identifier of the organism from which the data
will be downloaded.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{negatives}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains a list of negative interactions according to a given
source (e.g.: Negatome database). See
{\hyperref[\detokenize{reference:pypath.main.PyPath.apply_negative}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.apply\_negative()}}}}} for more
information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodDct}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the undirected graph node names {[}str{]} (keys) to their
indices {[}int{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodInd}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Stores the undirected graph node names {[}str{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodLab}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the undirected graph node indices {[}int{]} (keys) to their
labels {[}str{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodNam}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Maps the directed graph node indices {[}int{]} (keys) to their names
{[}str{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outdir}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Output directory where to store all output files.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{palette}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains a list of hexadecimal {[}str{]} of colors. Used for
plotting purposes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pathway\_types}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the names of all the loaded pathway resources {[}str{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pathways}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the list of pathways (values) for each resource (keys)
loaded in the network.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{plots}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} DEPRECATED (?)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{proteomicsdb}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.proteomicsdb.ProteomicsDB}}) \textendash{} Contains a \sphinxcode{\sphinxupquote{pypath.proteomicsdb.ProteomicsDB}}
instance, see the class documentation for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{raw\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains a list of loaded edges {[}dict{]} from a data file. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.read_data_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{PyPath.read\_data\_file()}}}}} for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{seq}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} (?)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{session}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Session ID, a five random alphanumeric characters.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{session\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Session name and ID (e.g. \sphinxcode{\sphinxupquote{'unnamed-abc12'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sourceNetEdges}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} (?)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sourceNetNodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} (?)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List contianing the names of the loaded resources {[}str{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{u\_pfam}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary of dictionaries, contains the mapping of UniProt IDs
to their respective protein families and other information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{uniprot\_mapped}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} DEPRECATED (?)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{unmapped}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the names of unmapped items {[}str{]}. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.map_item}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.map\_item()}}}}} for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{vertexAttrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Stores the node attribute names {[}str{]} as keys and their
corresponding types (e.g.: \sphinxcode{\sphinxupquote{set}}, \sphinxcode{\sphinxupquote{list}}, \sphinxcode{\sphinxupquote{str}}, …) as
values.

\end{itemize}

\end{description}\end{quote}
\index{acsn\_effects() (pypath.main.PyPath method)@\spxentry{acsn\_effects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.acsn_effects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{acsn\_effects}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{add\_genesets() (pypath.main.PyPath method)@\spxentry{add\_genesets()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_genesets}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_genesets}}}{\emph{genesets}}{}
\end{fulllineitems}

\index{add\_grouped\_eattr() (pypath.main.PyPath method)@\spxentry{add\_grouped\_eattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_grouped_eattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_grouped\_eattr}}}{\emph{edge}, \emph{attr}, \emph{group}, \emph{value}}{}
Merges (or creates) a given edge attribute as {[}dict{]} of {[}list{]}
values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Edge index where the given attribute value is to be merged
or created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty {[}dict{]}, \sphinxstyleemphasis{value} will only be assigned to the given
\sphinxstyleemphasis{edge} and \sphinxstyleemphasis{group}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The key of the attribute dictionary where \sphinxstyleemphasis{value} is to be
assigned.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} The value of the attribute to be assigned/merged.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_grouped\_set\_eattr() (pypath.main.PyPath method)@\spxentry{add\_grouped\_set\_eattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_grouped_set_eattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_grouped\_set\_eattr}}}{\emph{edge}, \emph{attr}, \emph{group}, \emph{value}}{}
Merges (or creates) a given edge attribute as {[}dict{]} of {[}set{]}
values.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Edge index where the given attribute value is to be merged
or created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty {[}dict{]}, \sphinxstyleemphasis{value} will only be assigned to the given
\sphinxstyleemphasis{edge} and \sphinxstyleemphasis{group}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The key of the attribute dictionary where \sphinxstyleemphasis{value} is to be
assigned.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} The value of the attribute to be assigned/merged.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_list\_eattr() (pypath.main.PyPath method)@\spxentry{add\_list\_eattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_list_eattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_list\_eattr}}}{\emph{edge}, \emph{attr}, \emph{value}}{}
Merges (or creates) a given edge attribute as {[}list{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Edge index where the given attribute value is to be merged
or created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty {[}list{]}, \sphinxstyleemphasis{value} will only be assigned to the given
\sphinxstyleemphasis{edge}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} The value of the attribute to be assigned/merged.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_set\_eattr() (pypath.main.PyPath method)@\spxentry{add\_set\_eattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_set_eattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_set\_eattr}}}{\emph{edge}, \emph{attr}, \emph{value}}{}
Merges (or creates) a given edge attribute as {[}set{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Edge index where the given attribute value is to be merged
or created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute. If such attribute does not exist
in the network edges, it will be created on all edges (as an
empty {[}set{]}, \sphinxstyleemphasis{value} will only be assigned to the given
\sphinxstyleemphasis{edge}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{value}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} The value of the attribute to be assigned/merged.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_update\_edge() (pypath.main.PyPath method)@\spxentry{add\_update\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_update_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_update\_edge}}}{\emph{id\_a}, \emph{id\_b}, \emph{source}, \emph{is\_directed}, \emph{refs}, \emph{stim}, \emph{inh}, \emph{taxon\_a}, \emph{taxon\_b}, \emph{typ}, \emph{extra\_attrs=\{\}}, \emph{add=False}}{}
Updates the attributes of one edge in the (undirected) network.
Optionally it creates a new edge and sets the attributes, but it
is not efficient as \sphinxcode{\sphinxupquote{igraph}} needs to reindex edges
after this operation, so better to create new edges in batches.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source node of the edge to be added/updated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source node of the edge to be added/updated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Or {[}list{]}, contains the names {[}str{]} of the resources
supporting that edge.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{is\_directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Whether if the edge is directed or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{refs}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Or {[}list{]}, contains the instances of the references
\sphinxcode{\sphinxupquote{pypath.refs.Reference}} for that edge.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{stim}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Whether the edge is stimulatory or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{inh}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Whether the edge is inhibitory or note

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{taxon\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomic identifier of the source molecule.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{taxon\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomic identifier of the target molecule.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{typ}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The type of interaction (e.g.: \sphinxcode{\sphinxupquote{'PPI'}})

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains any extra attributes
for the edge to be updated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{add}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If set to \sphinxcode{\sphinxupquote{True}} and the
edge is not in the network, it will be created. Otherwise,
in such case it will raise an error message.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_update\_vertex() (pypath.main.PyPath method)@\spxentry{add\_update\_vertex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.add_update_vertex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_update\_vertex}}}{\emph{default\_attrs}, \emph{original\_name}, \emph{original\_name\_type}, \emph{extra\_attrs=\{\}}, \emph{add=False}}{}
Updates the attributes of one node in the (undirected) network.
Optionally it creates a new node and sets the attributes, but it
is not efficient as \sphinxcode{\sphinxupquote{igraph}} needs to reindex vertices
after this operation, so better to create new nodes in batches.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{default\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} The attribute dictionary of the node to be updated/created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{original\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Original node name (e.g.: UniProt ID).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{original\_name\_type}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The original node name type (e.g.: for the previous example,
this would be \sphinxcode{\sphinxupquote{'uniprot'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains any extra attributes
for the node to be updated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{add}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If set to \sphinxcode{\sphinxupquote{True}} and the
node is not in the network, it will be created. Otherwise,
in such case it will raise an error message.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{affects() (pypath.main.PyPath method)@\spxentry{affects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.affects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{affects}}}{\emph{identifier}}{}
\end{fulllineitems}

\index{all\_between() (pypath.main.PyPath method)@\spxentry{all\_between()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.all_between}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{all\_between}}}{\emph{id\_a}, \emph{id\_b}}{}
Checks for any edges (in any direction) between the provided
nodes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the target node.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Contains information on the directionality of
the requested edge. Keys are \sphinxcode{\sphinxupquote{'ab'}} and \sphinxcode{\sphinxupquote{'ba'}}, denoting
the straight/reverse directionalities respectively. Values
are {[}list{]} whose elements are the edge ID or \sphinxcode{\sphinxupquote{None}}
according to the existance of that edge in the following
categories: undirected, straight and reverse (in that
order).

\end{description}\end{quote}

\end{fulllineitems}

\index{all\_neighbours() (pypath.main.PyPath method)@\spxentry{all\_neighbours()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.all_neighbours}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{all\_neighbours}}}{\emph{indices=False}}{}
Looks for the first neighbours of all the nodes and creates an
attribute (\sphinxcode{\sphinxupquote{'neighbours'}}) on each one of them containing a
list of their UniProt IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{indices}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to list the
neighbour nodes indices or their UniProt IDs.

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_list() (pypath.main.PyPath method)@\spxentry{apply\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.apply_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_list}}}{\emph{name}, \emph{node\_or\_edge='node'}}{}
Creates vertex or edge attribute based on a list.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the list to be added as attribute. Must have
been previously loaded with
\sphinxcode{\sphinxupquote{pypath.main.PyPath.load\_list()}} or other methods.
See description of \sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}}
attribute for more information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node\_or\_edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'node'}} by default. Whether the attribute list
is to be added to the nodes or to the edges.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_negative() (pypath.main.PyPath method)@\spxentry{apply\_negative()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.apply_negative}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_negative}}}{\emph{settings}}{}
Loads a negative interaction source (e.g.: Negatome) into the
current network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{settings}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.input\_formats.ReadSettings}}) \textendash{} \sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings}} instance
containing the detailed definition of the input format to
the downloaded file. For instance
\sphinxcode{\sphinxupquote{pypath.data\_formats.negative{[}'negatome'{]}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{attach\_network() (pypath.main.PyPath method)@\spxentry{attach\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.attach_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{attach\_network}}}{\emph{edge\_list=False}, \emph{regulator=False}}{}
Adds edges to the network from \sphinxstyleemphasis{edge\_list} obtained from file or
other input method. If none is passed, checks for such data in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.raw\_data}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. The source name of the list
of edges to be added. This must have been loaded previously
(e.g.: with {\hyperref[\detokenize{reference:pypath.main.PyPath.read_data_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.read\_data\_file()}}}}}).
If none is passed, loads the data directly from
\sphinxcode{\sphinxupquote{pypath.main.PyPath.raw\_data}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{regulator}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If set to \sphinxcode{\sphinxupquote{True}}, non
previously existing nodes, will not be added (and hence, the
edges involved).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{basic\_stats() (pypath.main.PyPath method)@\spxentry{basic\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.basic_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{basic\_stats}}}{\emph{latex=False, caption='', latex\_hdr=True, fontsize=8, font='HelveticaNeueLTStd-LtCn', fname=None, header\_format='\%s', row\_order=None, by\_category=True, use\_cats={[}'p', 'm', 'i', 'r'{]}, urls=True, annots=False}}{}
Returns basic numbers about the network resources, e.g. edge and
node counts.
\begin{description}
\item[{latex}] \leavevmode
Return table in a LaTeX document. This can be compiled by
PDFLaTeX:
latex stats.tex

\end{description}

\end{fulllineitems}

\index{basic\_stats\_intergroup() (pypath.main.PyPath method)@\spxentry{basic\_stats\_intergroup()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.basic_stats_intergroup}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{basic\_stats\_intergroup}}}{\emph{groupA}, \emph{groupB}, \emph{header=None}}{}
\end{fulllineitems}

\index{cancer\_drivers\_list() (pypath.main.PyPath method)@\spxentry{cancer\_drivers\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.cancer_drivers_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{cancer\_drivers\_list}}}{\emph{intogen\_file=None}}{}
Loads the list of cancer drivers. Contains information from
COSMIC (needs user log in credentials) and IntOGen (if provided)
and adds the attribute to the undirected network nodes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{intogen\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Path to the data file. Can
also be {[}function{]} that provides the data. In general,
anything accepted by
\sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings.input}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{cancer\_gene\_census\_list() (pypath.main.PyPath method)@\spxentry{cancer\_gene\_census\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.cancer_gene_census_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{cancer\_gene\_census\_list}}}{}{}
Loads the list of cancer driver proteins from the COSMIC Cancer
Gene Census.

\end{fulllineitems}

\index{clean\_graph() (pypath.main.PyPath method)@\spxentry{clean\_graph()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.clean_graph}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{clean\_graph}}}{\emph{organisms\_allowed=None}}{}
Removes multiple edges, unknown molecules and those from wrong
taxon. Multiple edges will be combined by
{\hyperref[\detokenize{reference:pypath.main.PyPath.combine_attr}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.combine\_attr()}}}}} method.
Loops will be deleted unless the attribute
\sphinxcode{\sphinxupquote{pypath.main.PyPath.loops}} is set to \sphinxcode{\sphinxupquote{True}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{organisms\_allowed}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} NCBI Taxonomy identifiers {[}int{]} of the organisms allowed
in the network.

\end{description}\end{quote}

\end{fulllineitems}

\index{collapse\_by\_name() (pypath.main.PyPath method)@\spxentry{collapse\_by\_name()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.collapse_by_name}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{collapse\_by\_name}}}{\emph{graph=None}}{}
Collapses nodes with the same name by copying and merging
all edges and attributes. Operates directly on the provided
network object.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network for which the
nodes are to be collapsed. If none is provided, takes
\sphinxcode{\sphinxupquote{pypath.main.PyPath.graph}} (undirected network) by
default.

\end{description}\end{quote}

\end{fulllineitems}

\index{combine\_attr() (pypath.main.PyPath method)@\spxentry{combine\_attr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.combine_attr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{combine\_attr}}}{\emph{lst}, \emph{num\_method=\textless{}built-in function max\textgreater{}}}{}
Combines multiple attributes into one. This method attempts
to find out which is the best way to combine attributes.
\begin{itemize}
\item {} 
If there is only one value or one of them is None, then
returns the one available.

\item {} 
Lists: concatenates unique values of lists.

\item {} 
Numbers: returns the greater by default or calls
\sphinxstyleemphasis{num\_method} if given.

\item {} 
Sets: returns the union.

\item {} 
Dictionaries: calls \sphinxcode{\sphinxupquote{pypath.common.merge\_dicts()}}.

\item {} 
Direction: calls their special
{\hyperref[\detokenize{reference:pypath.main.Direction.merge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.Direction.merge()}}}}} method.

\end{itemize}

Works on more than 2 attributes recursively.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of one or two attribute values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{num\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{function}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{max}} by default. Method to merge numeric
attributes.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{communities() (pypath.main.PyPath method)@\spxentry{communities()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.communities}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{communities}}}{\emph{method}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{complex\_comembership\_network() (pypath.main.PyPath method)@\spxentry{complex\_comembership\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.complex_comembership_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{complex\_comembership\_network}}}{\emph{graph=None}, \emph{resources=None}}{}
\end{fulllineitems}

\index{complexes() (pypath.main.PyPath method)@\spxentry{complexes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.complexes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{complexes}}}{\emph{methods={[}'3dcomplexes', 'havugimana', 'corum', 'complexportal', 'compleat'{]}}}{}
\end{fulllineitems}

\index{complexes\_in\_network() (pypath.main.PyPath method)@\spxentry{complexes\_in\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.complexes_in_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{complexes\_in\_network}}}{\emph{csource='corum'}, \emph{graph=None}}{}
\end{fulllineitems}

\index{compounds\_from\_chembl() (pypath.main.PyPath method)@\spxentry{compounds\_from\_chembl()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.compounds_from_chembl}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{compounds\_from\_chembl}}}{\emph{chembl\_mysql=None, nodes=None, crit=None, andor='or', assay\_types={[}'B', 'F'{]}, relationship\_types={[}'D', 'H'{]}, multi\_query=False, **kwargs}}{}
Loads compound data from ChEMBL to the network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{chebl\_mysql}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Contains the MySQL parameters
used by the \sphinxcode{\sphinxupquote{pypath.mapping}} module to loadthe
ChEMBL ID conversion tables. If none is passed, takes the
current instance \sphinxcode{\sphinxupquote{pypath.main.PyPath.chembl\_mysql}}
attribute.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. List of node indices for
which the information is to be loaded. If none is provided
calls the method {\hyperref[\detokenize{reference:pypath.main.PyPath.get_sub}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.get\_sub()}}}}} with
the provided \sphinxstyleemphasis{crit} parameter.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{crit}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Defines the critical
attributes to generate a subnetwork to extract the nodes in
case \sphinxstyleemphasis{nodes} is not provided. Keys are \sphinxcode{\sphinxupquote{'edge'}} and
\sphinxcode{\sphinxupquote{'node'}} and values are {[}dict{]} containing the critical
attribute names {[}str{]} and values are {[}set{]} containing those
attributes of the nodes/edges that are to be kept in the
subnetwork. If none is provided, takes the whole network.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{andor}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'or'}} by default. Determines the search mode
for the subnetwork generation (if \sphinxcode{\sphinxupquote{nodes=None}}). See
{\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_or}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_or()}}}}} and
{\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_and}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_and()}}}}} for more
details.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{assay\_types}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{{[}'B', 'F'{]}}} by default. Types of assay to query
Options are: \sphinxcode{\sphinxupquote{'A'}} (ADME), \sphinxcode{\sphinxupquote{'B'}} (Binding),{}`{}`’F’{}`{}`
(Functional), \sphinxcode{\sphinxupquote{'P'}} (Physicochemical), \sphinxcode{\sphinxupquote{'T'}} (Toxicity)
and/or \sphinxcode{\sphinxupquote{'U'}} (Unassigned).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{relationship\_types}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{{[}'D', 'H'{]}}} by default. Assay relationship
types to query. Possible values are: \sphinxcode{\sphinxupquote{'D'}} (Direct protein
target assigned), \sphinxcode{\sphinxupquote{'H'}} (Homologous protein target
assigned), \sphinxcode{\sphinxupquote{'M'}} (Molecular target other than protein
assigned), \sphinxcode{\sphinxupquote{'N'}} (Non-molecular target assigned), \sphinxcode{\sphinxupquote{'S'}}
(Subcellular target assigned) and/or \sphinxcode{\sphinxupquote{'U'}} (Default value,
target has yet to be curated).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{multi\_query}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Not used.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{**kwargs}} \textendash{} Additional keyword arguments for
\sphinxcode{\sphinxupquote{pypath.chembl.Chembl.compounds\_targets()}}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{consistency() (pypath.main.PyPath method)@\spxentry{consistency()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.consistency}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{consistency}}}{}{}
\end{fulllineitems}

\index{copy() (pypath.main.PyPath method)@\spxentry{copy()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.copy}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{copy}}}{\emph{other}}{}
Copies another {\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath}}}}} instance into the
current one.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{other}} ({\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{pypath.main.PyPath}}}}}) \textendash{} The instance to be copied from.

\end{description}\end{quote}

\end{fulllineitems}

\index{copy\_edges() (pypath.main.PyPath method)@\spxentry{copy\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.copy_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{copy\_edges}}}{\emph{sources}, \emph{target}, \emph{move=False}, \emph{graph=None}}{}
Copies edges from \sphinxstyleemphasis{sources} node(s) to another one (\sphinxstyleemphasis{target}),
keeping attributes and directions.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the vertex index(es) {[}int{]} of the node(s) to be
copied or moved.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{target}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Vertex index where edges and attributes are to be copied to.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{move}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to perform copy or
move (remove or keep the source edges).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network graph object from
which the nodes are to be merged. If none is passed, takes
the undirected network graph.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{count\_sol() (pypath.main.PyPath method)@\spxentry{count\_sol()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.count_sol}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_sol}}}{}{}
Counts the number of nodes with zero degree.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{int}) \textendash{} The number of nodes with zero degree.

\end{description}\end{quote}

\end{fulllineitems}

\index{coverage() (pypath.main.PyPath method)@\spxentry{coverage()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.coverage}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{coverage}}}{\emph{lst}}{}
Computes the coverage (range {[}0, 1{]}) of a list of nodes against
the current (undirected) network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Can also be {[}list{]} (will be converted to {[}set{]}) or {[}str{]}. In
the latter case it will retrieve the list with that name (if
such list exists in \sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}}).

\end{description}\end{quote}

\end{fulllineitems}

\index{curation\_effort() (pypath.main.PyPath method)@\spxentry{curation\_effort()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.curation_effort}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{curation\_effort}}}{\emph{sum\_by\_source=False}}{}
Returns the total number of reference-interactions pairs.
\begin{description}
\item[{@sum\_by\_source}] \leavevmode{[}bool{]}
If True, counts the refrence-interaction pairs by
sources, and returns the sum of these values.

\end{description}

\end{fulllineitems}

\index{curation\_stats() (pypath.main.PyPath method)@\spxentry{curation\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.curation_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{curation\_stats}}}{\emph{by\_category=True}}{}
\end{fulllineitems}

\index{curation\_tab() (pypath.main.PyPath method)@\spxentry{curation\_tab()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.curation_tab}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{curation\_tab}}}{\emph{fname='curation\_stats.tex', by\_category=True, use\_cats={[}'p', 'm', 'i', 'r'{]}, header\_size='normalsize', **kwargs}}{}
\end{fulllineitems}

\index{curators\_work() (pypath.main.PyPath method)@\spxentry{curators\_work()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.curators_work}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{curators\_work}}}{}{}
Computes and prints an estimation of how many years of curation
took to achieve the amount of information on the network.

\end{fulllineitems}

\index{databases\_similarity() (pypath.main.PyPath method)@\spxentry{databases\_similarity()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.databases_similarity}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{databases\_similarity}}}{\emph{index='simpson'}}{}
Computes the similarity across databases according to a given
index metric. Computes the similarity across the loaded
resources (listed in \sphinxcode{\sphinxupquote{pypath.main.PyPath.sources}} in
terms of nodes and edges separately.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'simpson'}} by default. The type of index metric
to use to compute the similarity. Options are \sphinxcode{\sphinxupquote{'simpson'}},
\sphinxcode{\sphinxupquote{'sorensen'}} and \sphinxcode{\sphinxupquote{'jaccard'}}.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Nested dictionaries (three levels). First-level
keys are \sphinxcode{\sphinxupquote{'nodes'}} and \sphinxcode{\sphinxupquote{'edges'}}, then second and third
levels correspond to sources names which map to the
similarity index between those sources {[}float{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{degree\_dist() (pypath.main.PyPath method)@\spxentry{degree\_dist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.degree_dist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{degree\_dist}}}{\emph{prefix}, \emph{g=None}, \emph{group=None}}{}
Computes the degree distribution over all nodes of the network.
If \sphinxstyleemphasis{group} is provided, also across nodes of that group(s).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{prefix}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Prefix for the file name(s).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{g}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network over which to
compute the degree distribution. If none is passed, takes
the undirected network of the current instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Additional group(s) name(s)
{[}str{]} of node attributes to subset the network and compute
its degree distribution.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{degree\_dists() (pypath.main.PyPath method)@\spxentry{degree\_dists()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.degree_dists}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{degree\_dists}}}{}{}
Computes the degree distribution for all the different network
sources. This is, for each source, the subnetwork comprising all
interactions coming from it is extracted and the degree
distribution information is computed and saved into a file.
A file is created for each resource under the name
\sphinxcode{\sphinxupquote{'pwnet-\textless{}session\_id\textgreater{}-degdist-\textless{}resource\textgreater{}'{'}}}. Files are stored
in \sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by
default).

\end{fulllineitems}

\index{delete\_by\_organism() (pypath.main.PyPath method)@\spxentry{delete\_by\_organism()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.delete_by_organism}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_by\_organism}}}{\emph{organisms\_allowed=None}}{}
Removes the proteins of all organisms which are not given in
\sphinxstyleemphasis{tax}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{organisms\_allowed}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} List of NCBI Taxonomy IDs {[}int{]} of the organism(s) that are
to be kept.

\end{description}\end{quote}

\end{fulllineitems}

\index{delete\_by\_source() (pypath.main.PyPath method)@\spxentry{delete\_by\_source()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.delete_by_source}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_by\_source}}}{\emph{source}, \emph{vertexAttrsToDel=None}, \emph{edgeAttrsToDel=None}}{}
Deletes nodes and edges from the network according to a provided
source name. Optionally can also remove the given list of
attributes from nodes and/or edges.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source from which the nodes and edges have to be
removed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{vertexAttrsToDel}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Contains the names {[}str{]} of
the attributes to be removed from the nodes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edgeAttrsToDel}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Contains the names {[}str{]} of
the attributes to be removed from the edges.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{delete\_unknown() (pypath.main.PyPath method)@\spxentry{delete\_unknown()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.delete_unknown}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_unknown}}}{\emph{organisms\_allowed=None}, \emph{entity\_type='protein'}, \emph{default\_name\_type=None}}{}
Removes those items which are not in the list of all default
IDs of the organisms. By default, it means to remove all protein
nodes not having a human UniProt ID.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{typ}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'protein'}} by default. Determines the molecule
type. These can be \sphinxcode{\sphinxupquote{'protein'}}, \sphinxcode{\sphinxupquote{'drug'}}, \sphinxcode{\sphinxupquote{'lncrna'}},
\sphinxcode{\sphinxupquote{'mirna'}} or any other type defined in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.default\_name\_type}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{default\_name\_type}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The default name type for the
given molecular species. If none is specified takes it from
\sphinxcode{\sphinxupquote{pypath.main.PyPath.default\_name\_type}} (e.g.: for
\sphinxcode{\sphinxupquote{'protein'}}, default is \sphinxcode{\sphinxupquote{'uniprot'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{organisms\_allowed}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} NCBI Taxonomy identifiers {[}int{]} of the organisms allowed in
the network.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{delete\_unmapped() (pypath.main.PyPath method)@\spxentry{delete\_unmapped()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.delete_unmapped}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_unmapped}}}{}{}
Checks the network for any existing unmapped node and removes
it.

\end{fulllineitems}

\index{dgenesymbol() (pypath.main.PyPath method)@\spxentry{dgenesymbol()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dgenesymbol}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dgenesymbol}}}{\emph{genesymbol}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the GeneSymbol
can be found in the default directed network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@genesymbol}] \leavevmode{[}str{]}
GeneSymbol.

\end{description}

\end{fulllineitems}

\index{dgenesymbols() (pypath.main.PyPath method)@\spxentry{dgenesymbols()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dgenesymbols}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dgenesymbols}}}{\emph{genesymbols}}{}
\end{fulllineitems}

\index{dgs() (pypath.main.PyPath method)@\spxentry{dgs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dgs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dgs}}}{\emph{genesymbol}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the GeneSymbol
can be found in the default directed network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@genesymbol}] \leavevmode{[}str{]}
GeneSymbol.

\end{description}

\end{fulllineitems}

\index{dgss() (pypath.main.PyPath method)@\spxentry{dgss()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dgss}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dgss}}}{\emph{genesymbols}}{}
\end{fulllineitems}

\index{dneighbors() (pypath.main.PyPath method)@\spxentry{dneighbors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dneighbors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dneighbors}}}{\emph{identifier}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{dp() (pypath.main.PyPath method)@\spxentry{dp()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dp}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dp}}}{\emph{identifier}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.get\_node}}, just for the directed graph.
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{dproteins() (pypath.main.PyPath method)@\spxentry{dproteins()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dproteins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dproteins}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{dps() (pypath.main.PyPath method)@\spxentry{dps()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dps}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dps}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{duniprot() (pypath.main.PyPath method)@\spxentry{duniprot()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.duniprot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{duniprot}}}{\emph{uniprot}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.uniprot(), just for directed graph.
Returns {}`{}`igraph.Vertex()}} object if the UniProt
can be found in the default directed network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@uniprot}] \leavevmode{[}str{]}
UniProt ID.

\end{description}

\end{fulllineitems}

\index{duniprots() (pypath.main.PyPath method)@\spxentry{duniprots()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.duniprots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{duniprots}}}{\emph{uniprots}}{}
Returns list of \sphinxcode{\sphinxupquote{igraph.Vertex()}} object
for a list of UniProt IDs omitting those
could not be found in the default
directed graph.

\end{fulllineitems}

\index{dup() (pypath.main.PyPath method)@\spxentry{dup()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dup}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dup}}}{\emph{uniprot}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.uniprot(), just for directed graph.
Returns {}`{}`igraph.Vertex()}} object if the UniProt
can be found in the default directed network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@uniprot}] \leavevmode{[}str{]}
UniProt ID.

\end{description}

\end{fulllineitems}

\index{dups() (pypath.main.PyPath method)@\spxentry{dups()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dups}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dups}}}{\emph{uniprots}}{}
Returns list of \sphinxcode{\sphinxupquote{igraph.Vertex()}} object
for a list of UniProt IDs omitting those
could not be found in the default
directed graph.

\end{fulllineitems}

\index{dv() (pypath.main.PyPath method)@\spxentry{dv()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dv}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dv}}}{\emph{identifier}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.get\_node}}, just for the directed graph.
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{dvs() (pypath.main.PyPath method)@\spxentry{dvs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.dvs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dvs}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{edge\_exists() (pypath.main.PyPath method)@\spxentry{edge\_exists()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edge_exists}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edge\_exists}}}{\emph{id\_a}, \emph{id\_b}}{}
Returns a tuple of vertex indices if edge doesn’t exist,
otherwise, the edge ID. Not sensitive to direction.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the target node.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{int}) \textendash{} The edge index, if exists such edge. Otherwise,
{[}tuple{]} of {[}int{]} corresponding to the node IDs.

\end{description}\end{quote}

\end{fulllineitems}

\index{edge\_loc() (pypath.main.PyPath method)@\spxentry{edge\_loc()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edge_loc}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edge\_loc}}}{\emph{graph=None}, \emph{topn=2}}{}
\end{fulllineitems}

\index{edge\_names() (pypath.main.PyPath method)@\spxentry{edge\_names()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edge_names}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edge\_names}}}{\emph{e}}{}
Returns the node names of a given edge.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{e}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The edge index.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{tuple}) \textendash{} Contains the source and target node names of
the edge {[}str{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{edges\_3d() (pypath.main.PyPath method)@\spxentry{edges\_3d()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edges_3d}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edges\_3d}}}{\emph{methods={[}'dataio.get\_instruct', 'dataio.get\_i3d'{]}}}{}
\end{fulllineitems}

\index{edges\_between() (pypath.main.PyPath method)@\spxentry{edges\_between()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edges_between}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edges\_between}}}{\emph{group1}, \emph{group2}, \emph{directed=True}, \emph{strict=False}}{}
Selects edges between two groups of vertex IDs.
Returns set of edge IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{group1}}\sphinxstyleliteralstrong{\sphinxupquote{,}}\sphinxstyleliteralstrong{\sphinxupquote{group2}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} List, set or tuple of vertex IDs.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Only edges with direction \sphinxtitleref{group1 -\textgreater{} group2} selected.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{strict}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Edges with no direction information still selected even if
\sphinxcode{\sphinxupquote{directed}} is \sphinxtitleref{False}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{edges\_expression() (pypath.main.PyPath method)@\spxentry{edges\_expression()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edges_expression}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edges\_expression}}}{\emph{func=\textless{}function PyPath.\textless{}lambda\textgreater{}\textgreater{}}}{}
Executes function \sphinxtitleref{func} for each pairs of connected proteins in the
network, for every expression dataset. By default, \sphinxtitleref{func} simply
gives the product the (normalized) expression values.
\begin{description}
\item[{func}] \leavevmode{[}callable{]}
Function to handle 2 vectors (pandas.Series() objects), should
return one vector of the same length.

\end{description}

\end{fulllineitems}

\index{edges\_in\_comlexes() (pypath.main.PyPath method)@\spxentry{edges\_in\_comlexes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edges_in_comlexes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edges\_in\_comlexes}}}{\emph{csources={[}'corum'{]}, graph=None}}{}
Creates edge attributes \sphinxcode{\sphinxupquote{complexes}} and \sphinxcode{\sphinxupquote{in\_complex}}.
These are both dicts where the keys are complex resources.
The values in \sphinxcode{\sphinxupquote{complexes}} are the list of complex names
both the source and the target vertices belong to.
The values \sphinxcode{\sphinxupquote{in\_complex}} are boolean values whether there
is at least one complex in the given resources both the
source and the target vertex of the edge belong to.
\begin{description}
\item[{@csources}] \leavevmode{[}list{]}
List of complex resources. Should be already loaded.

\item[{@graph}] \leavevmode{[}igraph.Graph(){]}
The graph object to do the calculations on.

\end{description}

\end{fulllineitems}

\index{edges\_ptms() (pypath.main.PyPath method)@\spxentry{edges\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edges_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edges\_ptms}}}{}{}
\end{fulllineitems}

\index{edgeseq\_inverse() (pypath.main.PyPath method)@\spxentry{edgeseq\_inverse()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.edgeseq_inverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{edgeseq\_inverse}}}{\emph{edges}}{}
Returns the sequence of all edge indexes that are not in
the argument \sphinxstyleemphasis{edges}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{edges}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Sequence of edge indices {[}int{]} that will not be returned.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains all edge indices {[}int{]} of the
undirected network except the ones on \sphinxstyleemphasis{edges} argument.

\end{description}\end{quote}

\end{fulllineitems}

\index{entities\_by\_resources() (pypath.main.PyPath method)@\spxentry{entities\_by\_resources()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.entities_by_resources}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{entities\_by\_resources}}}{}{}
Returns a dict of sets with resources as keys and sets of entity IDs
as values.

\end{fulllineitems}

\index{export\_dot() (pypath.main.PyPath method)@\spxentry{export\_dot()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_dot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_dot}}}{\emph{nodes=None}, \emph{edges=None}, \emph{directed=True}, \emph{labels='genesymbol'}, \emph{edges\_filter=\textless{}function PyPath.\textless{}lambda\textgreater{}\textgreater{}}, \emph{nodes\_filter=\textless{}function PyPath.\textless{}lambda\textgreater{}\textgreater{}}, \emph{edge\_sources=None}, \emph{dir\_sources=None}, \emph{graph=None}, \emph{return\_object=False}, \emph{save\_dot=None}, \emph{save\_graphics=None}, \emph{prog='neato'}, \emph{format=None}, \emph{hide=False}, \emph{font=None}, \emph{auto\_edges=False}, \emph{hide\_nodes={[}{]}}, \emph{defaults=\{\}}, \emph{**kwargs}}{}
Builds a pygraphviz.AGraph() object with filtering the edges
and vertices along arbitrary criteria.
Returns the Agraph object if requesred, or exports the dot
file, or saves the graphics.

@nodes : list
List of vertex ids to be included.
@edges : list
List of edge ids to be included.
@directed : bool
Create a directed or undirected graph.
@labels : str
Name type to be used as id/label in the dot format.
@edges\_filter : function
Function to filter edges, accepting igraph.Edge as argument.
@nodes\_filter : function
Function to filter vertices, accepting igraph.Vertex as argument.
@edge\_sources : list
Sources to be included.
@dir\_sources : list
Direction and effect sources to be included.
@graph : igraph.Graph
The graph object to export.
@return\_object : bool
Whether to return the pygraphviz.AGraph object.
@save\_dot : str
Filename to export the dot file to.
@save\_graphics : str
Filename to export the graphics, the extension defines the format.
@prog : str
The graphviz layout algorithm to use.
@format : str
The graphics format passed to pygraphviz.AGrapg().draw().
@hide : bool
Hide filtered edges instead of omit them.
@hide nodes : list
Nodes to hide. List of vertex ids.
@auto\_edges : str
Automatic, built-in style for edges.
‘DIRECTIONS’ or ‘RESOURCE\_CATEGORIES’ are supported.
@font : str
Font to use for labels.
For using more than one fonts refer to graphviz attributes with constant values
or define callbacks or mapping dictionaries.
@defaults : dict
Default values for graphviz attributes, labeled with the entity, e.g.
\sphinxtitleref{\{‘edge\_penwidth’: 0.2\}}.
@**kwargs : constant, callable or dict
Graphviz attributes, labeled by the target entity. E.g. \sphinxtitleref{edge\_penwidth},
‘vertex\_shape{}` or \sphinxtitleref{graph\_label}.
If the value is constant, this value will be used.
If the value is dict, and has \sphinxtitleref{\_name} as key, for every instance of the
given entity, the value of the attribute defined by \sphinxtitleref{\_name} will be looked
up in the dict, and the corresponding value will be given to the graphviz
attribute. If the key \sphinxtitleref{\_name} is missing from the dict, igraph vertex and
edge indices will be looked up among the keys.
If the value is callable, it will be called with the current instance of
the entity and the returned value will be used for the graphviz attribute.
E.g. \sphinxtitleref{edge\_arrowhead(edge)} or \sphinxtitleref{vertex\_fillcolor(vertex)}
Example:
\begin{quote}

import pypath
from pypath import data\_formats
net = pypath.PyPath()
net.init\_network(pfile = ‘cache/default.pickle’)
\#net.init\_network(\{‘arn’: data\_formats.omnipath{[}‘arn’{]}\})
tgf = {[}v.index for v in net.graph.vs if ‘TGF’ in v{[}‘slk\_pathways’{]}{]}
dot = net.export\_dot(nodes = tgf, save\_graphics = ‘tgf\_slk.pdf’, prog = ‘dot’,
\begin{quote}

main\_title = ‘TGF-beta pathway’, return\_object = True,
label\_font = ‘HelveticaNeueLTStd Med Cn’,
edge\_sources = {[}‘SignaLink3’{]},
dir\_sources = {[}‘SignaLink3’{]}, hide = True)
\end{quote}
\end{quote}

\end{fulllineitems}

\index{export\_edgelist() (pypath.main.PyPath method)@\spxentry{export\_edgelist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_edgelist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_edgelist}}}{\emph{fname, graph=None, names={[}'name'{]}, edge\_attributes={[}{]}, sep='\textbackslash{}t'}}{}
Write edge list to text file with attributes
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fname}} \textendash{} the name of the file or a stream to read from.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} \textendash{} the igraph object containing the network

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{names}} \textendash{} list with the vertex attribute names to be printed
for source and target vertices

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge\_attributes}} \textendash{} list with the edge attribute names
to be printed

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sep}} \textendash{} string used to separate columns

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{export\_graphml() (pypath.main.PyPath method)@\spxentry{export\_graphml()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_graphml}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_graphml}}}{\emph{outfile=None}, \emph{graph=None}, \emph{name='main'}}{}
Saves the network in a \sphinxcode{\sphinxupquote{.graphml}} file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Name/path of the output file.
If none is passed,
\sphinxcode{\sphinxupquote{'results/netrowk-\textless{}session\_id\textgreater{}.graphml'}} is used.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network object to be
saved. If none is passed, takes the undirected network of
the current instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'main'}} by default. The graph name for the
output file.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{export\_ptms\_tab() (pypath.main.PyPath method)@\spxentry{export\_ptms\_tab()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_ptms_tab}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_ptms\_tab}}}{\emph{outfile=None}}{}
Exports a tab file containing the PTM interaction information
loaded in the network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The output file nama/path to
store the PTM information. If none is provided, the default
is \sphinxcode{\sphinxupquote{'results/network-\textless{}session\_id\textgreater{}.tab'}}.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the edge indices {[}int{]} of all PTM
interactions.

\end{description}\end{quote}

\end{fulllineitems}

\index{export\_sif() (pypath.main.PyPath method)@\spxentry{export\_sif()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_sif}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_sif}}}{\emph{outfile=None}}{}
Exports the network interactions in \sphinxcode{\sphinxupquote{.sif}} format (Simple
Interaction Format).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Name/path of the output file.
If none is passed, \sphinxcode{\sphinxupquote{'results/netrowk-\textless{}session\_id\textgreater{}.sif'}}
is used.

\end{description}\end{quote}

\end{fulllineitems}

\index{export\_struct\_tab() (pypath.main.PyPath method)@\spxentry{export\_struct\_tab()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_struct_tab}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_struct\_tab}}}{\emph{outfile=None}}{}
Exports a tab file containing the domain interaction information
and PTM regulation loaded in the network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The output file nama/path to
store the PTM information. If none is provided, the default
is \sphinxcode{\sphinxupquote{'results/network-\textless{}session\_id\textgreater{}.tab'}}.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the edge indices {[}int{]} of all PTM
interactions.

\end{description}\end{quote}

\end{fulllineitems}

\index{export\_tab() (pypath.main.PyPath method)@\spxentry{export\_tab()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.export_tab}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_tab}}}{\emph{outfile=None}, \emph{extra\_node\_attrs=\{\}}, \emph{extra\_edge\_attrs=\{\}}, \emph{unique\_pairs=True}, \emph{**kwargs}}{}
Exports the network in a tabular format. By default UniProt IDs,
Gene Symbols, source databases, literature references,
directionality and sign information and interaction type are
included.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Name/path of the output file.
If none is passed, \sphinxcode{\sphinxupquote{'results/netrowk-\textless{}session\_id\textgreater{}.tab'}}
is used.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra\_node\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Additional node attributes to
be included in the exported table. Keys are column names
used in the header while values are names of vertex
attributes. In the header \sphinxcode{\sphinxupquote{'\_A'}} and \sphinxcode{\sphinxupquote{'\_B'}} suffixes
will be appended to the column names so the values can be
assigned to A and B side interaction partners.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{extra\_edge\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Additional edge attributes to
be included in the exported table. Keys are column names
used in the header while values are names of edge
attributes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{unique\_pairs}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. If set to \sphinxcode{\sphinxupquote{True}} each line
corresponds to a unique pair of molecules, all
directionality and sign information are covered in other
columns. If \sphinxcode{\sphinxupquote{False}}, order of \sphinxcode{\sphinxupquote{'A'}} and \sphinxcode{\sphinxupquote{'B'}} IDs
corresponds to the direction while sign covered in further
columns.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{kwargs}} (\sphinxstyleliteralemphasis{\sphinxupquote{**}}) \textendash{} Additional keyword arguments passed to
\sphinxcode{\sphinxupquote{pypath.export.Export}}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{filters() (pypath.main.PyPath method)@\spxentry{filters()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.filters}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{filters}}}{\emph{line}, \emph{positive\_filters={[}{]}}, \emph{negative\_filters={[}{]}}}{}
\end{fulllineitems}

\index{find\_all\_paths() (pypath.main.PyPath method)@\spxentry{find\_all\_paths()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.find_all_paths}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find\_all\_paths}}}{\emph{start}, \emph{end}, \emph{attr=None}, \emph{mode='OUT'}, \emph{maxlen=2}, \emph{graph=None}, \emph{silent=False}, \emph{update\_adjlist=True}}{}
Finds all paths up to length \sphinxtitleref{maxlen} between groups of
vertices. This function is needed only becaues igraph{}`s
get\_all\_shortest\_paths() finds only the shortest, not any
path up to a defined length.
\begin{description}
\item[{start}] \leavevmode{[}int or list{]}
Indices of the starting node(s) of the paths.

\item[{end}] \leavevmode{[}int or list{]}
Indices of the target node(s) of the paths.

\item[{attr}] \leavevmode{[}str{]}
Name of the vertex attribute to identify the vertices by.
Necessary if \sphinxcode{\sphinxupquote{start}} and \sphinxcode{\sphinxupquote{end}} are not igraph vertex ids
but for example vertex names or labels.

\item[{mode}] \leavevmode{[}‘IN’, ‘OUT’, ‘ALL’{]}
Passed to igraph.Graph.neighbors()

\item[{maxlen}] \leavevmode{[}int{]}
Maximum length of paths in steps, i.e. if maxlen = 3, then
the longest path may consist of 3 edges and 4 nodes.

\item[{graph}] \leavevmode{[}igraph.Graph object{]}
The graph you want to find paths in. self.graph by default.

\end{description}

\end{fulllineitems}

\index{find\_all\_paths2() (pypath.main.PyPath method)@\spxentry{find\_all\_paths2()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.find_all_paths2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find\_all\_paths2}}}{\emph{graph}, \emph{start}, \emph{end}, \emph{mode='OUT'}, \emph{maxlen=2}, \emph{psize=100}, \emph{update\_adjlist=True}}{}
\end{fulllineitems}

\index{find\_complex() (pypath.main.PyPath method)@\spxentry{find\_complex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.find_complex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find\_complex}}}{\emph{search}}{}
Finds complexes by their non standard names.
E.g. to find DNA polymerases you can use the search
term \sphinxtitleref{DNA pol} which will be tested against complex names
in CORUM.

\end{fulllineitems}

\index{first\_neighbours() (pypath.main.PyPath method)@\spxentry{first\_neighbours()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.first_neighbours}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{first\_neighbours}}}{\emph{node}, \emph{indices=False}}{}
Looks for the first neighbours of a given node and returns a
list of their UniProt IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The UniProt ID of the node of interest. Can also be the
index of such node {[}int{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{indices}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return the
neighbour nodes indices or their UniProt IDs.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} The list containing the first neighbours of the
queried node.

\end{description}\end{quote}

\end{fulllineitems}

\index{fisher\_enrichment() (pypath.main.PyPath method)@\spxentry{fisher\_enrichment()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.fisher_enrichment}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fisher\_enrichment}}}{\emph{lst}, \emph{attr}, \emph{ref='proteome'}}{}
Computes an enrichment analysis using Fisher’s exact test. The
contingency table is built as follows:
First row contains the number of nodes in the \sphinxstyleemphasis{ref} list (such
list is considered to be loaded in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}}) and the number of nodes in
the current (undirected) network. Second row contains the number
of nodes in \sphinxstyleemphasis{lst} list (also considered to be already loaded)
and the number of nodes in the network with a non-empty
attribute \sphinxstyleemphasis{attr}. Uses \sphinxcode{\sphinxupquote{scipy.stats.fisher\_exact()}}, see
the documentation of the corresponding package for more
information.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the list in \sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}}
whose number of elements will be the first element in the
second row of the contingency table.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The node attribute name for which the number of nodes in the
network with such attribute will be the second element of
the second row of the contingency table.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ref}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'proteome'}} by default. The name of the list in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}} whose number of elements
will be the first element of the first row of the
contingency table.

\end{itemize}

\item[{Returns}] \leavevmode
\begin{itemize}
\item {} 
(\sphinxstyleemphasis{float}) \textendash{} Prior odds ratio.

\item {} 
(\sphinxstyleemphasis{float}) \textendash{} P-value or probability of obtaining a
distribution as extreme as the observed, assuming that the
null hypothesis is true.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{geneset\_enrichment() (pypath.main.PyPath method)@\spxentry{geneset\_enrichment()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.geneset_enrichment}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{geneset\_enrichment}}}{\emph{proteins}, \emph{all\_proteins=None}, \emph{geneset\_ids=None}, \emph{alpha=0.05}, \emph{correction\_method='hommel'}}{}
Does not work at the moment because cfisher module should be
replaced with scipy.

\end{fulllineitems}

\index{genesymbol() (pypath.main.PyPath method)@\spxentry{genesymbol()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.genesymbol}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{genesymbol}}}{\emph{genesymbol}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the GeneSymbol
can be found in the default undirected network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@genesymbol}] \leavevmode{[}str{]}
GeneSymbol.

\end{description}

\end{fulllineitems}

\index{genesymbol\_labels() (pypath.main.PyPath method)@\spxentry{genesymbol\_labels()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.genesymbol_labels}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{genesymbol\_labels}}}{\emph{graph=None}, \emph{remap\_all=False}}{}
Creats vertex attribute \sphinxcode{\sphinxupquote{'label'}} and fills up with the
corresponding GeneSymbols of all proteins where the GeneSymbol
can be looked up based on the default name of the protein
vertex (UniProt ID by default). If the attribute \sphinxcode{\sphinxupquote{'label'}} has
been already initialized, updates this attribute or recreates if
\sphinxstyleemphasis{remap\_all} is set to \sphinxcode{\sphinxupquote{True}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network graph object
where the GeneSymbol labels are to be set/updated. If none
is passed, takes the current network undirected graph by
default (\sphinxcode{\sphinxupquote{pypath.main.PyPath.graph}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{remap\_all}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to map anew the
GeneSymbol labels if those were already initialized.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{genesymbols() (pypath.main.PyPath method)@\spxentry{genesymbols()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.genesymbols}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{genesymbols}}}{\emph{genesymbols}}{}
\end{fulllineitems}

\index{get\_attrs() (pypath.main.PyPath method)@\spxentry{get\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_attrs}}}{\emph{line}, \emph{spec}, \emph{lnum}}{}
\end{fulllineitems}

\index{get\_directed() (pypath.main.PyPath method)@\spxentry{get\_directed()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_directed}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_directed}}}{\emph{graph=False}, \emph{conv\_edges=False}, \emph{mutual=False}, \emph{ret=False}}{}
Converts a copy of \sphinxstyleemphasis{graph} undirected \sphinxstyleemphasis{igraph.Graph} object to a
directed one. By default it converts the current network
instance in \sphinxcode{\sphinxupquote{pypath.main.PyPath.graph}} and places the
copy of the directed instance in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.dgraph}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Undirected graph object. If
none is passed, takes the current undirected network
instance and saves the directed network under the attribute
\sphinxcode{\sphinxupquote{pypath.main.PyPath.dgraph}}. Otherwise, the
directed graph will be returned instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{conv\_edges}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to convert
undirected edges (those without explicit direction
information) to an arbitrary direction edge or
a pair of opposite edges. Otherwise those will be deleted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{mutual}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If \sphinxstyleemphasis{conv\_edges} is \sphinxcode{\sphinxupquote{True}},
whether to convert the undirected edges to a single,
arbitrary directed edge, or a pair of opposite directed
edges.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ret}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return the
directed graph instance, or not. If a \sphinxstyleemphasis{graph} is provided,
its directed version will be returned anyway.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{igraph.Graph}) \textendash{} If \sphinxstyleemphasis{graph} is passed or \sphinxstyleemphasis{ret} is
\sphinxcode{\sphinxupquote{True}}, returns the copy of the directed graph. otherwise
returns \sphinxcode{\sphinxupquote{None}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dirs\_signs() (pypath.main.PyPath method)@\spxentry{get\_dirs\_signs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_dirs_signs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dirs\_signs}}}{}{}
\end{fulllineitems}

\index{get\_edge() (pypath.main.PyPath method)@\spxentry{get\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_edge}}}{\emph{source}, \emph{target}, \emph{directed=True}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Edge}} object if an edge exist between
the 2 proteins, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Vertex index or UniProt ID or GeneSymbol or \sphinxcode{\sphinxupquote{igraph.Vertex}}
object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{target}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Vertex index or UniProt ID or GeneSymbol or \sphinxcode{\sphinxupquote{igraph.Vertex}}
object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} To be passed to igraph.Graph.get\_eid()

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_edges() (pypath.main.PyPath method)@\spxentry{get\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_edges}}}{\emph{sources}, \emph{targets}, \emph{directed=True}}{}
Returns a generator with all edges between source and target vertices.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{iterable}}) \textendash{} Source vertex IDs, names, labels, or any iterable yielding
\sphinxcode{\sphinxupquote{igraph.Vertex}} objects.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{targets}} (\sphinxstyleliteralemphasis{\sphinxupquote{iterable}}) \textendash{} Target vertec IDs, names, labels, or any iterable yielding
\sphinxcode{\sphinxupquote{igraph.Vertex}} objects.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Passed to \sphinxcode{\sphinxupquote{igraph.get\_eid()}}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_function() (pypath.main.PyPath method)@\spxentry{get\_function()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_function}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_function}}}{\emph{fun}}{}
\end{fulllineitems}

\index{get\_giant() (pypath.main.PyPath method)@\spxentry{get\_giant()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_giant}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_giant}}}{\emph{replace=False}, \emph{graph=None}}{}
Returns the giant component of the \sphinxstyleemphasis{graph}, or replaces the
\sphinxcode{\sphinxupquote{igraph.Graph}} instance with only the giant component
if specified.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{replace}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to replace
the \sphinxcode{\sphinxupquote{igraph.Graph}} instance. This can be either
the undirected network of the current
{\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath}}}}} instance (default) or the one
passed under the keyword argument \sphinxstyleemphasis{graph}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object from which
the giant component is to be computed. If none is specified,
takes the undirected network of the current
{\hyperref[\detokenize{reference:pypath.main.PyPath}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath}}}}} instance.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{igraph.Graph}) \textendash{} If \sphinxcode{\sphinxupquote{replace=False}}, returns a copy of
the giant component graph.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_go() (pypath.main.PyPath method)@\spxentry{get\_go()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_go}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_go}}}{\emph{organism=None}}{}
Returns the \sphinxcode{\sphinxupquote{GOAnnotation}} object for the organism requested
(or the default one).

\end{fulllineitems}

\index{get\_max() (pypath.main.PyPath method)@\spxentry{get\_max()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_max}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_max}}}{\emph{attrList}}{}
\end{fulllineitems}

\index{get\_network() (pypath.main.PyPath method)@\spxentry{get\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_network}}}{\emph{crit}, \emph{andor='or'}, \emph{graph=None}}{}
Retrieves a subnetwork according to a set of user-defined
attributes. Basically applies
{\hyperref[\detokenize{reference:pypath.main.PyPath.get_sub}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.get\_sub()}}}}} on a given \sphinxstyleemphasis{graph}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{crit}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Defines the critical attributes to generate the subnetwork.
Keys are \sphinxcode{\sphinxupquote{'edge'}} and \sphinxcode{\sphinxupquote{'node'}} and values are {[}dict{]}
containing the critical attribute names {[}str{]} and values
are {[}set{]} containing those attributes of the nodes/edges
that are to be kept.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{andor}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'or'}} by default. Determines the search mode.
See {\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_or}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_or()}}}}} and
{\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_and}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_and()}}}}} for more
details.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where to
extract the subnetwork. If none is passed, takes the current
network (undirected) graph
(\sphinxcode{\sphinxupquote{pypath.main.PyPath.graph}}).

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{igraph.Graph}) \textendash{} The subgraph obtained from filtering
according to the attributes defined in \sphinxstyleemphasis{crit}.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_node() (pypath.main.PyPath method)@\spxentry{get\_node()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_node}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_node}}}{\emph{identifier}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{get\_node\_d() (pypath.main.PyPath method)@\spxentry{get\_node\_d()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_node_d}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_node\_d}}}{\emph{identifier}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.get\_node}}, just for the directed graph.
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{get\_node\_pair() (pypath.main.PyPath method)@\spxentry{get\_node\_pair()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_node_pair}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_node\_pair}}}{\emph{id\_a}, \emph{id\_b}, \emph{directed=False}}{}
Retrieves the node IDs from a pair of node names.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the target node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return the node
indices from the directed or undirected graph.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{tuple}) \textendash{} The pair of node IDs of the selected graph.
If not found, returns \sphinxcode{\sphinxupquote{False}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_nodes() (pypath.main.PyPath method)@\spxentry{get\_nodes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_nodes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nodes}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{get\_nodes\_d() (pypath.main.PyPath method)@\spxentry{get\_nodes\_d()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_nodes_d}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nodes\_d}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{get\_pathways() (pypath.main.PyPath method)@\spxentry{get\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_pathways}}}{\emph{source}}{}
\end{fulllineitems}

\index{get\_proteomicsdb() (pypath.main.PyPath method)@\spxentry{get\_proteomicsdb()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_proteomicsdb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_proteomicsdb}}}{\emph{user}, \emph{passwd}, \emph{tissues=None}, \emph{pickle=None}}{}
\end{fulllineitems}

\index{get\_sub() (pypath.main.PyPath method)@\spxentry{get\_sub()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_sub}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_sub}}}{\emph{crit}, \emph{andor='or'}, \emph{graph=None}}{}
Selects the nodes from \sphinxstyleemphasis{graph} (and edges to be removed)
according to a set of user-defined attributes.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{crit}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Defines the critical attributes to generate the subnetwork.
Keys are \sphinxcode{\sphinxupquote{'edge'}} and \sphinxcode{\sphinxupquote{'node'}} and values are {[}dict{]}
containing the critical attribute names {[}str{]} and values
are {[}set{]} containing those attributes of the nodes/edges
that are to be kept.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{andor}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'or'}} by default. Determines the search mode.
See {\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_or}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_or()}}}}} and
{\hyperref[\detokenize{reference:pypath.main.PyPath.search_attr_and}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.search\_attr\_and()}}}}} for more
details.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where to
extract the subnetwork. If none is passed, takes the current
network (undirected) graph
(\sphinxcode{\sphinxupquote{pypath.main.PyPath.graph}}).

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Keys are \sphinxcode{\sphinxupquote{'nodes'}} and \sphinxcode{\sphinxupquote{'edges'}} whose
values are {[}lst{]} of elements (as indexes {[}int{]}). Nodes are
those to be kept and edges to be removed on the extracted
subnetwork.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_taxon() (pypath.main.PyPath method)@\spxentry{get\_taxon()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.get_taxon}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_taxon}}}{\emph{tax\_dict}, \emph{fields}}{}
\end{fulllineitems}

\index{go\_annotate\_graph() (pypath.main.PyPath method)@\spxentry{go\_annotate\_graph()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.go_annotate_graph}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{go\_annotate\_graph}}}{\emph{aspects=('C'}, \emph{'F'}, \emph{'P')}}{}
Annotates protein nodes with GO terms. In the \sphinxcode{\sphinxupquote{go}} vertex
attribute each node is annotated by a dict of sets where keys are
one letter codes of GO aspects and values are sets of GO accessions.

\end{fulllineitems}

\index{go\_enrichment() (pypath.main.PyPath method)@\spxentry{go\_enrichment()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.go_enrichment}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{go\_enrichment}}}{\emph{proteins=None}, \emph{aspect='P'}, \emph{alpha=0.05}, \emph{correction\_method='hommel'}, \emph{all\_proteins=None}}{}
Does not work at the moment because cfisher module should be
replaced with scipy.

\end{fulllineitems}

\index{gs() (pypath.main.PyPath method)@\spxentry{gs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs}}}{\emph{genesymbol}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the GeneSymbol
can be found in the default undirected network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@genesymbol}] \leavevmode{[}str{]}
GeneSymbol.

\end{description}

\end{fulllineitems}

\index{gs\_affected\_by() (pypath.main.PyPath method)@\spxentry{gs\_affected\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_affected_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_affected\_by}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_affects() (pypath.main.PyPath method)@\spxentry{gs\_affects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_affects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_affects}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_edge() (pypath.main.PyPath method)@\spxentry{gs\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_edge}}}{\emph{source}, \emph{target}, \emph{directed=True}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Edge}} object if an edge exist between
the 2 proteins, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@source}] \leavevmode{[}str{]}
GeneSymbol

\item[{@target}] \leavevmode{[}str{]}
GeneSymbol

\item[{@directed}] \leavevmode{[}bool{]}
To be passed to igraph.Graph.get\_eid()

\end{description}

\end{fulllineitems}

\index{gs\_in\_directed() (pypath.main.PyPath method)@\spxentry{gs\_in\_directed()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_in_directed}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_in\_directed}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_in\_undirected() (pypath.main.PyPath method)@\spxentry{gs\_in\_undirected()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_in_undirected}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_in\_undirected}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_inhibited\_by() (pypath.main.PyPath method)@\spxentry{gs\_inhibited\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_inhibited_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_inhibited\_by}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_inhibits() (pypath.main.PyPath method)@\spxentry{gs\_inhibits()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_inhibits}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_inhibits}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_neighborhood() (pypath.main.PyPath method)@\spxentry{gs\_neighborhood()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_neighborhood}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_neighborhood}}}{\emph{genesymbols}, \emph{order=1}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{gs\_neighbors() (pypath.main.PyPath method)@\spxentry{gs\_neighbors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_neighbors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_neighbors}}}{\emph{genesymbol}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{gs\_stimulated\_by() (pypath.main.PyPath method)@\spxentry{gs\_stimulated\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_stimulated_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_stimulated\_by}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gs\_stimulates() (pypath.main.PyPath method)@\spxentry{gs\_stimulates()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gs_stimulates}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gs\_stimulates}}}{\emph{genesymbol}}{}
\end{fulllineitems}

\index{gss() (pypath.main.PyPath method)@\spxentry{gss()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.gss}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{gss}}}{\emph{genesymbols}}{}
\end{fulllineitems}

\index{guide2pharma() (pypath.main.PyPath method)@\spxentry{guide2pharma()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.guide2pharma}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{guide2pharma}}}{}{}
\end{fulllineitems}

\index{having\_attr() (pypath.main.PyPath method)@\spxentry{having\_attr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.having_attr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{having\_attr}}}{\emph{attr}, \emph{graph=None}, \emph{index=True}, \emph{edges=True}}{}
Checks if edges or nodes of the network have a specific
attribute and returns an iterator of the indices (or the
edge/node instances) of edges/nodes having such attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute to look for.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to return the
iterator of the indices or the node/edge instances.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edges}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to look for the
attribute in the networks edges or nodes instead.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Generator object containing the edge/node
indices (or instances) having the specified attribute.

\end{description}\end{quote}

\end{fulllineitems}

\index{having\_eattr() (pypath.main.PyPath method)@\spxentry{having\_eattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.having_eattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{having\_eattr}}}{\emph{attr}, \emph{graph=None}, \emph{index=True}}{}
Checks if edges of the network have a specific attribute and
returns an iterator of the indices (or the edge instances) of
edges having such attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute to look for.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to return the
iterator of the indices or the node/edge instances.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Generator object containing the edge
indices (or instances) having the specified attribute.

\end{description}\end{quote}

\end{fulllineitems}

\index{having\_ptm() (pypath.main.PyPath method)@\spxentry{having\_ptm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.having_ptm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{having\_ptm}}}{\emph{index=True}, \emph{graph=None}}{}
Checks if edges of the network have the \sphinxcode{\sphinxupquote{'ptm'}} attribute and
returns an iterator of the indices (or the edge instances) of
edges having such attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to return the
iterator of the indices or the node/edge instances.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Generator object containing the edge
indices (or instances) having the \sphinxcode{\sphinxupquote{ptm'{'}}} attribute.

\end{description}\end{quote}

\end{fulllineitems}

\index{having\_vattr() (pypath.main.PyPath method)@\spxentry{having\_vattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.having_vattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{having\_vattr}}}{\emph{attr}, \emph{graph=None}, \emph{index=True}}{}
Checks if nodes of the network have a specific attribute and
returns an iterator of the indices (or the node instances) of
nodes having such attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the attribute to look for.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where the
edge/node attribute is to be searched. If none is passed,
takes the undirected network of the current instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to return the
iterator of the indices or the node/edge instances.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Generator object containing the node
indices (or instances) having the specified attribute.

\end{description}\end{quote}

\end{fulllineitems}

\index{homology\_translation() (pypath.main.PyPath method)@\spxentry{homology\_translation()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.homology_translation}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{homology\_translation}}}{\emph{target}, \emph{source=None}, \emph{only\_swissprot=True}, \emph{graph=None}}{}
Translates the current object to another organism by orthology.
Proteins without known ortholog will be deleted.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{target}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomy ID of the target organism. E.g. 10090 for mouse.

\end{description}\end{quote}

\end{fulllineitems}

\index{htp\_stats() (pypath.main.PyPath method)@\spxentry{htp\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.htp_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{htp\_stats}}}{}{}
\end{fulllineitems}

\index{in\_complex() (pypath.main.PyPath method)@\spxentry{in\_complex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.in_complex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{in\_complex}}}{\emph{csources={[}'corum'{]}}}{}
Deprecated, will be removed.

\end{fulllineitems}

\index{in\_directed() (pypath.main.PyPath method)@\spxentry{in\_directed()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.in_directed}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{in\_directed}}}{\emph{vertex}}{}
\end{fulllineitems}

\index{in\_undirected() (pypath.main.PyPath method)@\spxentry{in\_undirected()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.in_undirected}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{in\_undirected}}}{\emph{vertex}}{}
\end{fulllineitems}

\index{info() (pypath.main.PyPath method)@\spxentry{info()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.info}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{info}}}{\emph{name}}{}
Given the name of a resource, prints out the information about
that source/database. You can check the list of available
resource descriptions in
\sphinxcode{\sphinxupquote{ypath.descriptions.descriptions.keys()}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the resource from which to print the
information.

\end{description}\end{quote}

\end{fulllineitems}

\index{init\_complex\_attr() (pypath.main.PyPath method)@\spxentry{init\_complex\_attr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.init_complex_attr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{init\_complex\_attr}}}{\emph{graph}, \emph{name}}{}
\end{fulllineitems}

\index{init\_edge\_attr() (pypath.main.PyPath method)@\spxentry{init\_edge\_attr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.init_edge_attr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{init\_edge\_attr}}}{\emph{attr}}{}
Fills all edges attribute \sphinxstyleemphasis{attr} with its default type (if
such attribute value is \sphinxcode{\sphinxupquote{None}}), creates {[}list{]} if in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.edgeAttrs}} such attribute is
registered as {[}list{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The attribute name to be initialized on the network edges.

\end{description}\end{quote}

\end{fulllineitems}

\index{init\_gsea() (pypath.main.PyPath method)@\spxentry{init\_gsea()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.init_gsea}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{init\_gsea}}}{\emph{user}}{}
Initializes a \sphinxcode{\sphinxupquote{pypath.gsea.GSEA}} object and shows the list of the
collections in MSigDB.

\end{fulllineitems}

\index{init\_network() (pypath.main.PyPath method)@\spxentry{init\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.init_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{init\_network}}}{\emph{lst=None}, \emph{exclude={[}{]}}, \emph{cache\_files=\{\}}, \emph{pfile=False}, \emph{save=False}, \emph{reread=None}, \emph{redownload=None}, \emph{keep\_raw=False}, \emph{**kwargs}}{}
Loads the network data.

This is a lazy way to start the module, load data and build the
high confidence, literature curated part of the signaling
network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the data input
formats for the different resources (keys) {[}str{]}. Values
are \sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings}} instances
containing the information. By default uses the set of
resources of OmniPath.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exclude}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{{[}{]}}} by default. List of resources {[}str{]} to
exclude from the network.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cache\_files}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains the resource name(s)
{[}str{]} (keys) and the corresponding cached file name {[}str{]}.
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If any, provides the file
name or path to a previously saved network pickle file.
If \sphinxcode{\sphinxupquote{True}} is passed, takes the default path from
{\hyperref[\detokenize{reference:pypath.main.PyPath.save_network}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{PyPath.save\_network()}}}}}
(\sphinxcode{\sphinxupquote{'cache/default\_network.pickle'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{save}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. If set to \sphinxcode{\sphinxupquote{True}}, saves
the loaded network to its default location
(\sphinxcode{\sphinxupquote{'cache/default\_network.pickle'}}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reread}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to reread
the data files from the cache or omit them (similar to
\sphinxstyleemphasis{redownload}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{redownload}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to
re-download the data and ignore the cache.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{**kwargs}} \textendash{} Not used.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{init\_vertex\_attr() (pypath.main.PyPath method)@\spxentry{init\_vertex\_attr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.init_vertex_attr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{init\_vertex\_attr}}}{\emph{attr}}{}
Fills all vertices attribute \sphinxstyleemphasis{attr} with its default type (if
such attribute value is \sphinxcode{\sphinxupquote{None}}), creates {[}list{]} if in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.vertexAttrs}} such attribute is
registered as {[}list{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{attr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The attribute name to be initialized on the network
vertices.

\end{description}\end{quote}

\end{fulllineitems}

\index{intergroup\_shortest\_paths() (pypath.main.PyPath method)@\spxentry{intergroup\_shortest\_paths()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.intergroup_shortest_paths}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{intergroup\_shortest\_paths}}}{\emph{groupA}, \emph{groupB}, \emph{random=False}}{}
\end{fulllineitems}

\index{intogen\_cancer\_drivers\_list() (pypath.main.PyPath method)@\spxentry{intogen\_cancer\_drivers\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.intogen_cancer_drivers_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{intogen\_cancer\_drivers\_list}}}{\emph{intogen\_file}}{}
Loads the list of cancer driver proteins from IntOGen data.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{intogen\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the data file. Can also be {[}function{]} that provides
the data. In general, anything accepted by
\sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings.input}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{iter\_interactions() (pypath.main.PyPath method)@\spxentry{iter\_interactions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.iter_interactions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{iter\_interactions}}}{\emph{signs=True}, \emph{all\_undirected=True}}{}
Iterates over edges and yields interaction records.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{signs}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Ignoring signs if \sphinxcode{\sphinxupquote{False}}. This way each directed interaction
will yield a single record even if it’s ambiguously labeled
both positive and negative. The default behaviour is to yield
two records in this case, one with positive and one with negative
sign.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{all\_undirected}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Yield records for undirected interactions even if certain sources
provide direction. If \sphinxcode{\sphinxupquote{False}} only the directed records will
be provided and the undirected sources and references will be
added to the directed records.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{jaccard\_edges() (pypath.main.PyPath method)@\spxentry{jaccard\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.jaccard_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{jaccard\_edges}}}{}{}
Computes the Jaccard similarity index between the sets of first
neighbours of all node pairs. \sphinxstylestrong{NOTE:} this method can take a
while to compute, e.g.: if the network has 10K nodes, the total
number of possible pairs to compute is:
\begin{equation*}
\begin{split}\binom{10^4}{2} = 49995000\end{split}
\end{equation*}\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Large list of {[}tuple{]} elements containing the
node pair names {[}str{]} and their corresponding first
neighbours Jaccard index {[}float{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{jaccard\_meta() (pypath.main.PyPath method)@\spxentry{jaccard\_meta()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.jaccard_meta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{jaccard\_meta}}}{\emph{jedges}, \emph{critical}}{}
Creates a (undirected) graph from a list of edges filtering by
their Jaccard index.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{jedges}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of {[}tuple{]} containing the edges node names {[}str{]} and
their Jaccard index. Basically, the output of
{\hyperref[\detokenize{reference:pypath.main.PyPath.jaccard_edges}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.jaccard\_edges()}}}}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{critical}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Specifies the threshold of the Jaccard index from above
which an edge will be included in the graph.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{igraph.Graph}) \textendash{} The Undirected graph instance containing
only the edges whose Jaccard similarity index is above the
threshold specified by \sphinxstyleemphasis{critical}.

\end{description}\end{quote}

\end{fulllineitems}

\index{kegg\_directions() (pypath.main.PyPath method)@\spxentry{kegg\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.kegg_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{kegg\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{kegg\_pathways() (pypath.main.PyPath method)@\spxentry{kegg\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.kegg_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{kegg\_pathways}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{kinase\_stats() (pypath.main.PyPath method)@\spxentry{kinase\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.kinase_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{kinase\_stats}}}{}{}
\end{fulllineitems}

\index{label() (pypath.main.PyPath method)@\spxentry{label()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.label}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{label}}}{\emph{label}, \emph{idx}, \emph{what='vertices'}}{}
Creates a boolean attribute \sphinxcode{\sphinxupquote{label}} True for the
vertex or edge IDs in the set \sphinxcode{\sphinxupquote{idx}}.

\end{fulllineitems}

\index{label\_by\_go() (pypath.main.PyPath method)@\spxentry{label\_by\_go()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.label_by_go}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{label\_by\_go}}}{\emph{label}, \emph{go\_terms}, \emph{method='ANY'}}{}
Assigns a boolean vertex attribute to nodes which tells whether
the node is annotated by all or any of the GO terms.

\end{fulllineitems}

\index{label\_edges() (pypath.main.PyPath method)@\spxentry{label\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.label_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{label\_edges}}}{\emph{label}, \emph{edges}}{}
Creates a boolean edge attribute \sphinxcode{\sphinxupquote{label}} True for the
edge IDs in the set \sphinxcode{\sphinxupquote{edges}}.

\end{fulllineitems}

\index{label\_vertices() (pypath.main.PyPath method)@\spxentry{label\_vertices()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.label_vertices}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{label\_vertices}}}{\emph{label}, \emph{vertices}}{}
Creates a boolean vertex attribute \sphinxcode{\sphinxupquote{label}} True for the
vertex IDs in the set \sphinxcode{\sphinxupquote{vertices}}.

\end{fulllineitems}

\index{laudanna\_directions() (pypath.main.PyPath method)@\spxentry{laudanna\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.laudanna_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{laudanna\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{laudanna\_effects() (pypath.main.PyPath method)@\spxentry{laudanna\_effects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.laudanna_effects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{laudanna\_effects}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{license() (pypath.main.PyPath static method)@\spxentry{license()}\spxextra{pypath.main.PyPath static method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.license}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{static }}\sphinxbfcode{\sphinxupquote{license}}}{\emph{self}}{}
Prints information about data licences.

\end{fulllineitems}

\index{list\_resources() (pypath.main.PyPath static method)@\spxentry{list\_resources()}\spxextra{pypath.main.PyPath static method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.list_resources}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{static }}\sphinxbfcode{\sphinxupquote{list\_resources}}}{}{}
Prints the list of resources through the standard output.

\end{fulllineitems}

\index{load\_3dcomplexes() (pypath.main.PyPath method)@\spxentry{load\_3dcomplexes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_3dcomplexes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_3dcomplexes}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{load\_3did\_ddi() (pypath.main.PyPath method)@\spxentry{load\_3did\_ddi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_3did_ddi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_3did\_ddi}}}{}{}
\end{fulllineitems}

\index{load\_3did\_ddi2() (pypath.main.PyPath method)@\spxentry{load\_3did\_ddi2()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_3did_ddi2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_3did\_ddi2}}}{\emph{ddi=True}, \emph{interfaces=False}}{}
\end{fulllineitems}

\index{load\_3did\_dmi() (pypath.main.PyPath method)@\spxentry{load\_3did\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_3did_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_3did\_dmi}}}{}{}
\end{fulllineitems}

\index{load\_3did\_interfaces() (pypath.main.PyPath method)@\spxentry{load\_3did\_interfaces()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_3did_interfaces}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_3did\_interfaces}}}{}{}
\end{fulllineitems}

\index{load\_all\_pathways() (pypath.main.PyPath method)@\spxentry{load\_all\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_all_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_all\_pathways}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{load\_compleat() (pypath.main.PyPath method)@\spxentry{load\_compleat()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_compleat}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_compleat}}}{\emph{graph=None}}{}
Loads complexes from Compleat. Loads data into vertex attribute
\sphinxtitleref{graph.vs{[}‘complexes’{]}{[}‘compleat’{]}}.
This resource is human only.

\end{fulllineitems}

\index{load\_complexportal() (pypath.main.PyPath method)@\spxentry{load\_complexportal()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_complexportal}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_complexportal}}}{\emph{graph=None}}{}
Loads complexes from ComplexPortal. Loads data into vertex attribute
\sphinxtitleref{graph.vs{[}‘complexes’{]}{[}‘complexportal’{]}}.
This resource is human only.

\end{fulllineitems}

\index{load\_comppi() (pypath.main.PyPath method)@\spxentry{load\_comppi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_comppi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_comppi}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{load\_corum() (pypath.main.PyPath method)@\spxentry{load\_corum()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_corum}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_corum}}}{\emph{graph=None}}{}
Loads complexes from CORUM database. Loads data into vertex attribute
\sphinxtitleref{graph.vs{[}‘complexes’{]}{[}‘corum’{]}}.
This resource is human only.

\end{fulllineitems}

\index{load\_dbptm() (pypath.main.PyPath method)@\spxentry{load\_dbptm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_dbptm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dbptm}}}{\emph{non\_matching=False}, \emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_ddi() (pypath.main.PyPath method)@\spxentry{load\_ddi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ddi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ddi}}}{\emph{ddi}}{}
ddi is either a list of intera.DomainDomain objects,
or a function resulting this list

\end{fulllineitems}

\index{load\_ddis() (pypath.main.PyPath method)@\spxentry{load\_ddis()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ddis}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ddis}}}{\emph{methods={[}'dataio.get\_3dc\_ddi', 'dataio.get\_domino\_ddi', 'self.load\_3did\_ddi2'{]}}}{}
\end{fulllineitems}

\index{load\_depod\_dmi() (pypath.main.PyPath method)@\spxentry{load\_depod\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_depod_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_depod\_dmi}}}{}{}
\end{fulllineitems}

\index{load\_disgenet() (pypath.main.PyPath method)@\spxentry{load\_disgenet()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_disgenet}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_disgenet}}}{\emph{dataset='curated'}, \emph{score=0.0}, \emph{umls=False}, \emph{full\_data=False}}{}
Assigns DisGeNet disease-gene associations to the proteins
in the network. Disease annotations will be added to the \sphinxtitleref{dis}
vertex attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{score}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Confidence score from DisGeNet. Only associations
above the score provided will be considered.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ulms}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} By default we assign a list of disease names to
each protein. To use Unified Medical Language System IDs instead
set this to \sphinxtitleref{True}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{full\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} By default we load only disease names. Set this
to \sphinxtitleref{True} if you wish to load additional annotations like number
of PubMed IDs, number of SNPs and original sources.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_dmi() (pypath.main.PyPath method)@\spxentry{load\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dmi}}}{\emph{dmi}}{}
dmi is either a list of intera.DomainMotif objects,
or a function resulting this list

\end{fulllineitems}

\index{load\_dmis() (pypath.main.PyPath method)@\spxentry{load\_dmis()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_dmis}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_dmis}}}{\emph{methods={[}'self.pfam\_regions', 'self.load\_depod\_dmi', 'self.load\_dbptm', 'self.load\_mimp\_dmi', 'self.load\_pnetworks\_dmi', 'self.load\_domino\_dmi', 'self.load\_pepcyber', 'self.load\_psite\_reg', 'self.load\_psite\_phos', 'self.load\_ielm', 'self.load\_phosphoelm', 'self.load\_elm', 'self.load\_3did\_dmi'{]}}}{}
\end{fulllineitems}

\index{load\_domino\_dmi() (pypath.main.PyPath method)@\spxentry{load\_domino\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_domino_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_domino\_dmi}}}{\emph{organism=None}}{}
\end{fulllineitems}

\index{load\_elm() (pypath.main.PyPath method)@\spxentry{load\_elm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_elm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_elm}}}{}{}
\end{fulllineitems}

\index{load\_exocarta\_attrs() (pypath.main.PyPath method)@\spxentry{load\_exocarta\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_exocarta_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_exocarta\_attrs}}}{\emph{load\_samples=False}, \emph{load\_refs=False}}{}
Creates vertex attributes from ExoCarta data. Creates a boolean
attribute \sphinxcode{\sphinxupquote{exocarts\_exosomal}} which tells whether a protein is
in ExoCarta i.e. has been found in exosomes. Optionally creates
attributes \sphinxcode{\sphinxupquote{exocarta\_samples}} and \sphinxcode{\sphinxupquote{exocarta\_refs}} listing the
sample tissue and the PubMed references, respectively.

\end{fulllineitems}

\index{load\_expression() (pypath.main.PyPath method)@\spxentry{load\_expression()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_expression}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_expression}}}{\emph{array=False}}{}
Expression data can be loaded into vertex attributes,
or into a pandas DataFrame \textendash{} the latter offers faster
ways to process and use these huge matrices.

\end{fulllineitems}

\index{load\_go() (pypath.main.PyPath method)@\spxentry{load\_go()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_go}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_go}}}{\emph{organism=None}}{}
Creates a \sphinxcode{\sphinxupquote{pypath.go.GOAnnotation}} object for one organism in the
dict under \sphinxcode{\sphinxupquote{go}} attribute.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{organism}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomy ID of the organism.

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_havugimana() (pypath.main.PyPath method)@\spxentry{load\_havugimana()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_havugimana}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_havugimana}}}{\emph{graph=None}}{}
Loads complexes from Havugimana 2012. Loads data into vertex attribute
\sphinxtitleref{graph.vs{[}‘complexes’{]}{[}‘havugimana’{]}}.
This resource is human only.

\end{fulllineitems}

\index{load\_hpa() (pypath.main.PyPath method)@\spxentry{load\_hpa()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_hpa}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_hpa}}}{\emph{normal=True}, \emph{pathology=True}, \emph{cancer=True}, \emph{summarize\_pathology=True}, \emph{tissues=None}, \emph{quality=\{'Approved'}, \emph{'Supported'\}}, \emph{levels=\{'High': 3}, \emph{'Low': 1}, \emph{'Medium': 2}, \emph{'Not detected': 0\}}, \emph{graph=None}, \emph{na\_value=0}}{}
Loads Human Protein Atlas data into vertex attributes.

\end{fulllineitems}

\index{load\_hprd\_ptms() (pypath.main.PyPath method)@\spxentry{load\_hprd\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_hprd_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_hprd\_ptms}}}{\emph{non\_matching=False}, \emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_ielm() (pypath.main.PyPath method)@\spxentry{load\_ielm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ielm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ielm}}}{}{}
\end{fulllineitems}

\index{load\_interfaces() (pypath.main.PyPath method)@\spxentry{load\_interfaces()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_interfaces}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_interfaces}}}{}{}
\end{fulllineitems}

\index{load\_li2012\_ptms() (pypath.main.PyPath method)@\spxentry{load\_li2012\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_li2012_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_li2012\_ptms}}}{\emph{non\_matching=False}, \emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_ligand\_receptor\_network() (pypath.main.PyPath method)@\spxentry{load\_ligand\_receptor\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ligand_receptor_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ligand\_receptor\_network}}}{\emph{lig\_rec\_resources=True}, \emph{inference\_from\_go=True}, \emph{sources=None}, \emph{keep\_undirected=False}, \emph{keep\_rec\_rec=False}, \emph{keep\_lig\_lig=False}}{}
Initializes a ligand-receptor network.

\end{fulllineitems}

\index{load\_lmpid() (pypath.main.PyPath method)@\spxentry{load\_lmpid()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_lmpid}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_lmpid}}}{\emph{method}}{}
\end{fulllineitems}

\index{load\_matrisome\_attrs() (pypath.main.PyPath method)@\spxentry{load\_matrisome\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_matrisome_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_matrisome\_attrs}}}{\emph{organism=None}}{}
Loads vertex attributes from MatrisomeDB 2.0. Attributes are
\sphinxcode{\sphinxupquote{matrisome\_class}}, \sphinxcode{\sphinxupquote{matrisome\_subclass}} and \sphinxcode{\sphinxupquote{matrisome\_notes}}.

\end{fulllineitems}

\index{load\_membranome\_attrs() (pypath.main.PyPath method)@\spxentry{load\_membranome\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_membranome_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_membranome\_attrs}}}{}{}
Loads attributes from Membranome, a database of single-helix
transmembrane proteins.

\end{fulllineitems}

\index{load\_mimp\_dmi() (pypath.main.PyPath method)@\spxentry{load\_mimp\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_mimp_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_mimp\_dmi}}}{\emph{non\_matching=False}, \emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_mutations() (pypath.main.PyPath method)@\spxentry{load\_mutations()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_mutations}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_mutations}}}{\emph{attributes=None}, \emph{gdsc\_datadir=None}, \emph{mutation\_file=None}}{}
Mutations are listed in vertex attributes. Mutation() objects
offers methods to identify residues and look up in Ptm(), Motif()
and Domain() objects, to check if those residues are
modified, or are in some short motif or domain.

\end{fulllineitems}

\index{load\_negatives() (pypath.main.PyPath method)@\spxentry{load\_negatives()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_negatives}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_negatives}}}{}{}
\end{fulllineitems}

\index{load\_old\_omnipath() (pypath.main.PyPath method)@\spxentry{load\_old\_omnipath()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_old_omnipath}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_old\_omnipath}}}{\emph{kinase\_substrate\_extra=False}, \emph{remove\_htp=False}, \emph{htp\_threshold=1}, \emph{keep\_directed=False}, \emph{min\_refs\_undirected=2}}{}
Loads the OmniPath network as it was before August 2016.
Furthermore it gives some more options.

\end{fulllineitems}

\index{load\_omnipath() (pypath.main.PyPath method)@\spxentry{load\_omnipath()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_omnipath}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_omnipath}}}{\emph{omnipath=None}, \emph{kinase\_substrate\_extra=False}, \emph{ligand\_receptor\_extra=False}, \emph{pathway\_extra=False}, \emph{remove\_htp=True}, \emph{htp\_threshold=1}, \emph{keep\_directed=True}, \emph{min\_refs\_undirected=2}, \emph{old\_omnipath\_resources=False}, \emph{exclude=None}}{}
Loads the OmniPath network.

\end{fulllineitems}

\index{load\_pathways() (pypath.main.PyPath method)@\spxentry{load\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pathways}}}{\emph{source}, \emph{graph=None}}{}
Generic method to load pathway annotations from a resource.
We don’t recommend calling this method but either specific
methods for a single source e.g. \sphinxtitleref{kegg\_pathways()}
or \sphinxtitleref{sirnor\_pathways()} or call \sphinxtitleref{load\_all\_pathways()} to
load all resources.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source, this need to match a method in the dict
in \sphinxtitleref{get\_pathways()} method and the edge and vertex attributes
with pathway annotations will be called “\textless{}source\textgreater{}\_pathways”.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} A graph, by default the default the \sphinxtitleref{graph} attribute of the
current instance.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_pdb() (pypath.main.PyPath method)@\spxentry{load\_pdb()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pdb}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pdb}}}{\emph{graph=None}}{}
Loads the 3D structure information from PDB into the network.
Creates the node attribute \sphinxcode{\sphinxupquote{'pdb'}} containing a {[}dict{]} whose
keys are the PDB identifier {[}str{]} and values are {[}tuple{]} of two
elements denoting the experimental method {[}str{]} (e.g.:
\sphinxcode{\sphinxupquote{'X-ray'}}, \sphinxcode{\sphinxupquote{'NMR'}}, …) and the resolution {[}float{]} (if
applicable).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_pepcyber() (pypath.main.PyPath method)@\spxentry{load\_pepcyber()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pepcyber}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pepcyber}}}{}{}
\end{fulllineitems}

\index{load\_pfam() (pypath.main.PyPath method)@\spxentry{load\_pfam()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pfam}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pfam}}}{\emph{graph=None}}{}
Loads the protein family information from UniProt into the
network. Creates the node attribute \sphinxcode{\sphinxupquote{'pfam'}} containing a
{[}list{]} of protein family identifier(s) {[}str{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_pfam2() (pypath.main.PyPath method)@\spxentry{load\_pfam2()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pfam2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pfam2}}}{}{}
Loads the protein family information from Pfam into the network.
Creates the node attribute \sphinxcode{\sphinxupquote{'pfam'}} containing a {[}list{]} of
{[}dict{]} whose keys are protein family identifier(s) {[}str{]} and
corresponding values are {[}list{]} of {[}dict{]} containing detailed
information about the protein family(ies) for regions and
isoforms of the protein.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network object for which
the information is to be loaded. If none is passed, takes
the undirected network of the current instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_pfam3() (pypath.main.PyPath method)@\spxentry{load\_pfam3()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pfam3}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pfam3}}}{}{}
Loads the protein domain information from Pfam into the network.
Creates the node attribute \sphinxcode{\sphinxupquote{'doms'}} containing a {[}list{]} of
\sphinxcode{\sphinxupquote{pypath.intera.Domain}} instances with information
about each domain of the protein (see the corresponding class
documentation for more information).

\end{fulllineitems}

\index{load\_phospho\_dmi() (pypath.main.PyPath method)@\spxentry{load\_phospho\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_phospho_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_phospho\_dmi}}}{\emph{source}, \emph{trace=False}, \emph{return\_raw=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_phosphoelm() (pypath.main.PyPath method)@\spxentry{load\_phosphoelm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_phosphoelm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_phosphoelm}}}{\emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_pisa() (pypath.main.PyPath method)@\spxentry{load\_pisa()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pisa}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pisa}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{load\_pnetworks\_dmi() (pypath.main.PyPath method)@\spxentry{load\_pnetworks\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_pnetworks_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_pnetworks\_dmi}}}{\emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_psite\_phos() (pypath.main.PyPath method)@\spxentry{load\_psite\_phos()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_psite_phos}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_psite\_phos}}}{\emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_psite\_reg() (pypath.main.PyPath method)@\spxentry{load\_psite\_reg()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_psite_reg}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_psite\_reg}}}{}{}
\end{fulllineitems}

\index{load\_ptms() (pypath.main.PyPath method)@\spxentry{load\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ptms}}}{}{}
\end{fulllineitems}

\index{load\_ptms2() (pypath.main.PyPath method)@\spxentry{load\_ptms2()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_ptms2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_ptms2}}}{\emph{input\_methods=None, map\_by\_homology\_from={[}9606{]}, homology\_only\_swissprot=True, ptm\_homology\_strict=False, nonhuman\_direct\_lookup=True, inputargs=\{\}, database=None, force\_load=False}}{}
This is a new method which will replace \sphinxtitleref{load\_ptms}.
It uses \sphinxtitleref{pypath.ptm.PtmAggregator}, a newly introduced
module for combining enzyme-substrate data from multiple
resources using homology translation on users demand.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{input\_methods}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Resources to collect enzyme-substrate
interactions from. E.g. \sphinxtitleref{{[}‘Signor’, ‘phosphoELM’{]}}. By default
it contains Signor, PhosphoSitePlus, HPRD, phosphoELM, dbPTM,
PhosphoNetworks, Li2012 and MIMP.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{map\_by\_homology\_from}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of NCBI Taxonomy IDs of
source taxons used for homology translation of enzyme-substrate
interactions. If you have a human network and you add here
\sphinxtitleref{{[}10090, 10116{]}} then mouse and rat interactions from the source
databases will be translated to human.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{homology\_only\_swissprot}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} \sphinxtitleref{True} by default which means
only SwissProt IDs are accepted at homology translateion, Trembl
IDs will be dropped.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{ptm\_homology\_strict}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} For homology translation use
PhosphoSite’s PTM homology table. This guarantees that only
truely homologous sites will be included. Otherwise we only
check if at the same numeric offset in the homologous sequence
the appropriate residue can be find.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nonhuman\_direct\_lookup}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Fetch also directly nonhuman
data from the resources whereever it’s available. PhosphoSite
contains mouse enzyme-substrate interactions and it is possible
to extract these directly beside translating the human ones
to mouse.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{inputargs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Additional arguments passed to \sphinxtitleref{PtmProcessor}.
A \sphinxtitleref{dict} can be supplied for each resource, e.g.
\sphinxtitleref{\{‘Signor’: \{…\}, ‘PhosphoSite’: \{…\}, …\}}.
Those not used by \sphinxtitleref{PtmProcessor} are forwarded to the
\sphinxtitleref{pypath.dataio} methods.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{database}} \textendash{} A \sphinxcode{\sphinxupquote{PtmAggregator}} object. If provided no new database will be
created.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{force\_load}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If \sphinxcode{\sphinxupquote{True}} the database will be loaded with the parameters
provided here; otherwise if the \sphinxcode{\sphinxupquote{ptm}} module already has a
database no new database will be created. This means the
parameters specified in other arguments might have no effect.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_resource() (pypath.main.PyPath method)@\spxentry{load\_resource()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_resource}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_resource}}}{\emph{settings}, \emph{clean=True}, \emph{cache\_files=\{\}}, \emph{reread=None}, \emph{redownload=None}, \emph{keep\_raw=False}}{}
Loads the data from a single resource and attaches it to the
network
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{settings}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.input\_formats.ReadSettings}}) \textendash{} \sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings}} instance
containing the detailed definition of the input format to
the downloaded file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{clean}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to clean the graph
after importing the data or not. See
{\hyperref[\detokenize{reference:pypath.main.PyPath.clean_graph}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.clean\_graph()}}}}} for more
information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cache\_files}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains the resource name(s)
{[}str{]} (keys) and the corresponding cached file name {[}str{]}.
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reread}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to reread
the data files from the cache or omit them (similar to
\sphinxstyleemphasis{redownload}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{redownload}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to
re-download the data and ignore the cache.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_resources() (pypath.main.PyPath method)@\spxentry{load\_resources()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_resources}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_resources}}}{\emph{lst=None}, \emph{exclude={[}{]}}, \emph{cache\_files=\{\}}, \emph{reread=False}, \emph{redownload=None}, \emph{keep\_raw=False}}{}
Loads multiple resources, and cleans up after. Looks up ID
types, and loads all ID conversion tables from UniProt if
necessary. This is much faster than loading the ID conversion
and the resources one by one.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the data input
formats for the different resources (keys) {[}str{]}. Values
are \sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings}} instances
containing the information. By default uses the set of
resources of OmniPath.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exclude}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{{[}{]}}} by default. List of resources {[}str{]} to
exclude from the network.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cache\_files}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains the resource name(s)
{[}str{]} (keys) and the corresponding cached file name {[}str{]}.
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reread}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to reread
the data files from the cache or omit them (similar to
\sphinxstyleemphasis{redownload}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{redownload}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to
re-download the data and ignore the cache.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_signor\_ptms() (pypath.main.PyPath method)@\spxentry{load\_signor\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_signor_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_signor\_ptms}}}{\emph{non\_matching=False}, \emph{trace=False}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{load\_surfaceome\_attrs() (pypath.main.PyPath method)@\spxentry{load\_surfaceome\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_surfaceome_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_surfaceome\_attrs}}}{}{}
Loads vertex attributes from the In Silico Human Surfaceome.
Attributes are \sphinxcode{\sphinxupquote{surfaceome\_score}}, \sphinxcode{\sphinxupquote{surfaceome\_class}} and
\sphinxcode{\sphinxupquote{surfaceome\_subclass}}.

\end{fulllineitems}

\index{load\_tfregulons() (pypath.main.PyPath method)@\spxentry{load\_tfregulons()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_tfregulons}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_tfregulons}}}{\emph{levels=\{'A'}, \emph{'B'\}}, \emph{only\_curated=False}}{}
Adds TF-target interactions from TF regulons to the network.
TF regulons is a comprehensive resource of TF-target
interactions combining multiple lines of evidences: literature
curated databases, ChIP-Seq data, PWM based prediction using
HOCOMOCO and JASPAR matrices and prediction from GTEx expression
data by ARACNe.

For details see \sphinxurl{https://github.com/saezlab/DoRothEA}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{levels}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{'A', 'B'\}}} by default. Confidence levels to be
loaded (from A to E) {[}str{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{only\_curated}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to retrieve only the
literature curated interactions or not.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_vesiclepedia\_attrs() (pypath.main.PyPath method)@\spxentry{load\_vesiclepedia\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.load_vesiclepedia_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_vesiclepedia\_attrs}}}{\emph{load\_samples=False}, \emph{load\_refs=False}, \emph{load\_vesicle\_type=False}}{}
Creates vertex attributes from Vesiclepedia data. Creates a boolean
attribute \sphinxcode{\sphinxupquote{vesiclepedia\_in\_vesicle}} which tells whether a protein is
in ExoCarta i.e. has been found in exosomes. Optionally creates
attributes \sphinxcode{\sphinxupquote{vesiclepedia\_samples}}, \sphinxcode{\sphinxupquote{vesiclepedia\_refs}} and
\sphinxcode{\sphinxupquote{vesiclepedia\_vesicles}} listing the sample tissue, the PubMed
references and the vesicle types, respectively.

\end{fulllineitems}

\index{lookup\_cache() (pypath.main.PyPath method)@\spxentry{lookup\_cache()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.lookup_cache}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{lookup\_cache}}}{\emph{name}, \emph{cache\_files}, \emph{int\_cache}, \emph{edges\_cache}}{}
Checks up the cache folder for the files of a given resource.
First checks if \sphinxstyleemphasis{name} is on the \sphinxstyleemphasis{cache\_files} dictionary.
If so, loads either the interactions or edges otherwise. If
not, checks \sphinxstyleemphasis{edges\_cache} or \sphinxstyleemphasis{int\_cache} otherwise.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the resource (lower-case).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cache\_files}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the resource name(s) {[}str{]} (keys) and the
corresponding cached file name {[}str{]} (values).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{int\_cache}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the interactions cache file of the resource.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edges\_cache}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the edges cache file of the resource.

\end{itemize}

\item[{Returns}] \leavevmode
\begin{itemize}
\item {} 
(\sphinxstyleemphasis{file}) \textendash{} The loaded pickle file from the cache if the
file is contains the interactions. \sphinxcode{\sphinxupquote{None}} otherwise.

\item {} 
(\sphinxstyleemphasis{list}) \textendash{} List of mapped edges if the file contains the
information from the edges. \sphinxcode{\sphinxupquote{{[}{]}}} otherwise.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{loop\_edges() (pypath.main.PyPath method)@\spxentry{loop\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.loop_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{loop\_edges}}}{\emph{index=True}, \emph{graph=None}}{}
Returns an iterator of the indices (or the edge instances) of
the edges which represent a loop (whose source and target node
are the same).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to return the
iterator of the indices or the edge instances.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The graph object where the
edge loops are to be searched. If none is passed, takes the
undirected network of the current instance.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Generator object containing the edge
indices (or instances) containing loops.

\end{description}\end{quote}

\end{fulllineitems}

\index{map\_edge() (pypath.main.PyPath method)@\spxentry{map\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.map_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{map\_edge}}}{\emph{edge}, \emph{expand\_complexes=True}}{}
Translates the identifiers in \sphinxstyleemphasis{edge} representing an edge. Default
name types are defined in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.default\_name\_type}} If the mapping
is unsuccessful, the item will be added to
\sphinxcode{\sphinxupquote{pypath.main.PyPath.unmapped}} list.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Item whose name is to be mapped to a default name type.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expand\_complexes}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Expand complexes, i.e. create links between each member of
the complex and the interacting partner.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the edge(s) {[}dict{]} with default mapped
names.

\end{description}\end{quote}

\end{fulllineitems}

\index{map\_item() (pypath.main.PyPath method)@\spxentry{map\_item()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.map_item}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{map\_item}}}{\emph{item}, \emph{expand\_complexes=True}}{}
Translates the name in \sphinxstyleemphasis{item} representing a molecule. Default
name types are defined in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.default\_name\_type}} If the mapping
is unsuccessful, the item will be added to
\sphinxcode{\sphinxupquote{pypath.main.PyPath.unmapped}} list.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{item}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Item whose name is to be mapped to a default name type.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expand\_complexes}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Expand complexes, i.e. create links between each member of
the complex and the interacting partner.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} The default mapped name(s) {[}str{]} of \sphinxstyleemphasis{item}.

\end{description}\end{quote}

\end{fulllineitems}

\index{map\_list() (pypath.main.PyPath method)@\spxentry{map\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.map_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{map\_list}}}{\emph{lst}, \emph{single\_list=False}, \emph{expand\_complexes=True}}{}
Maps the names from a list of edges or items (molecules).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of items or edge dictionaries whose names have to be
mapped.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{single\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Determines whether the
provided elements are items or edges. This is, either calls
{\hyperref[\detokenize{reference:pypath.main.PyPath.map_edge}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.map\_edge()}}}}} or
{\hyperref[\detokenize{reference:pypath.main.PyPath.map_item}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.map\_item()}}}}} to map the item
names.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{expand\_complexes}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Expand complexes, i.e. create links between each member of
the complex and the interacting partner.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Copy of \sphinxstyleemphasis{lst} with their elements’ names mapped.

\end{description}\end{quote}

\end{fulllineitems}

\index{mean\_reference\_per\_interaction() (pypath.main.PyPath method)@\spxentry{mean\_reference\_per\_interaction()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.mean_reference_per_interaction}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{mean\_reference\_per\_interaction}}}{}{}
Computes the mean number of references per interaction of the
network.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{float}) \textendash{} Mean number of interactions per edge.

\end{description}\end{quote}

\end{fulllineitems}

\index{merge\_lists() (pypath.main.PyPath method)@\spxentry{merge\_lists()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.merge_lists}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge\_lists}}}{\emph{id\_a}, \emph{id\_b}, \emph{name=None}, \emph{and\_or='and'}, \emph{delete=False}, \emph{func='max'}}{}
Merges two lists from \sphinxcode{\sphinxupquote{pypat.main.PyPath.lists}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the first list to be merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the second list to be merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies a new name for the
merged list. If none is passed, name will be set to
\sphinxstyleemphasis{id\_a*\_*id\_b}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{and\_or}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'and'}} by default. The logic operation perfomed
in the merging: \sphinxcode{\sphinxupquote{'and'}} performs an union, \sphinxcode{\sphinxupquote{'or'}} for
the intersection.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{delete}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to delete the
former lists or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{func}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'max'}} by default. Not used.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{merge\_nodes() (pypath.main.PyPath method)@\spxentry{merge\_nodes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.merge_nodes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge\_nodes}}}{\emph{nodes}, \emph{primary=None}, \emph{graph=None}}{}
Merges all attributes and edges of selected nodes and assigns
them to the primary node (by default the one with lowest index).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of node indexes {[}int{]} that are to be collapsed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{primary}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. ID of the primary edge, if
none is passed, the node with lowest index on \sphinxstyleemphasis{nodes} is
selected.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network graph object from
which the nodes are to be merged. If none is passed, takes
the undirected network graph.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{mimp\_directions() (pypath.main.PyPath method)@\spxentry{mimp\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.mimp_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{mimp\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{mutated\_edges() (pypath.main.PyPath method)@\spxentry{mutated\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.mutated_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{mutated\_edges}}}{\emph{sample}}{}
Compares the mutated residues and the modified residues in PTMs.
Interactions are marked as mutated if the target residue in the
underlying PTM is mutated.

\end{fulllineitems}

\index{name\_edgelist() (pypath.main.PyPath method)@\spxentry{name\_edgelist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.name_edgelist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{name\_edgelist}}}{\emph{graph=None}}{}
Returns an edge list, i.e. a list with tuples of vertex names.

\end{fulllineitems}

\index{names2vids() (pypath.main.PyPath method)@\spxentry{names2vids()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.names2vids}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{names2vids}}}{\emph{names}}{}
From a list of node names, returns their corresponding indices.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{names}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the node names {[}str{]} for which the IDs are to be
searched.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} The queried node IDs {[}int{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{negative\_report() (pypath.main.PyPath method)@\spxentry{negative\_report()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.negative_report}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{negative\_report}}}{\emph{lst=True}, \emph{outFile=None}}{}
Generates a report file with the negative interactions (assumed
to be already loaded).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Whether to retun a list of
edges containing the edge instances which have negative
references.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outFile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The output file name/path. If
none is passed, the default is
\sphinxcode{\sphinxupquote{'results/\textless{}session\_id\textgreater{}-negatives'}}

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} If \sphinxstyleemphasis{lst} is set to \sphinxcode{\sphinxupquote{True}}, returns a {[}list{]}
is returned with the \sphinxcode{\sphinxupquote{igraph.Edge}} instances that
contain at least a negative reference.

\end{description}\end{quote}

\end{fulllineitems}

\index{neighborhood() (pypath.main.PyPath method)@\spxentry{neighborhood()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.neighborhood}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{neighborhood}}}{\emph{identifiers}, \emph{order=1}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{neighbors() (pypath.main.PyPath method)@\spxentry{neighbors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.neighbors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{neighbors}}}{\emph{identifier}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{neighbourhood\_network() (pypath.main.PyPath method)@\spxentry{neighbourhood\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.neighbourhood_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{neighbourhood\_network}}}{\emph{center}, \emph{second=False}}{}
\end{fulllineitems}

\index{network\_by\_go() (pypath.main.PyPath method)@\spxentry{network\_by\_go()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.network_by_go}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{network\_by\_go}}}{\emph{node\_categories}, \emph{network\_sources=None}, \emph{include=None}, \emph{exclude=None}, \emph{directed=False}, \emph{keep\_undirected=False}, \emph{prefix='GO'}, \emph{delete=True}, \emph{copy=False}, \emph{vertex\_attrs=True}, \emph{edge\_attrs=True}}{}
Creates or filters a network based on Gene Ontology annotations.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node\_categories}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} 
A dict with custom category labels as keys and expressions of
GO terms as values. E.g.
{\color{red}\bfseries{}{}`{}`}\{‘extracell’: ‘\sphinxurl{GO:0005576} and not \sphinxurl{GO:0070062}’,
\begin{quote}

’plasmamem’: ‘\sphinxurl{GO:0005887}’\}{}`{}`.
\end{quote}


\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{network\_sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} A dict with anything as keys and network input format definintions
(\sphinxcode{\sphinxupquote{input\_formats.ReadSettings}} instances) as values.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{include}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} A list of tuples of category label pairs. By default we keep all
edges connecting proteins annotated with any of the defined
categories. If \sphinxcode{\sphinxupquote{include}} is defined then only edges between
category pairs defined here will be kept and all others deleted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{exclude}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Similarly to include, all edges will be kept but the ones listed
in \sphinxcode{\sphinxupquote{exclude}} will be deleted.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If True \sphinxcode{\sphinxupquote{include}} and \sphinxcode{\sphinxupquote{exclude}} relations will be processed
with directed (source, target) else direction won’t be considered.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{keep\_undirected}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If True the interactions without direction information will be
kept even if \sphinxcode{\sphinxupquote{directed}} is True. Passed to \sphinxcode{\sphinxupquote{edges\_between}}
as \sphinxcode{\sphinxupquote{strict}} argument.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{prefix}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Prefix for all vertex and edge attributes created in this
operation. E.g. if you have a category label ‘bar’ and prefix
is ‘foo’ then you will have a new vertex attribute ‘foo\_\_bar’.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{delete}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Delete the vertices and edges which don’t belong to any of the
categories.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{copy}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Return a copy of the entire \sphinxcode{\sphinxupquote{PyPath}} object with the graph
filtered by GO terms. By default the object is modified in place
and \sphinxcode{\sphinxupquote{None}} is returned.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{vertex\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Create vertex attributes.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{edge\_attrs}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Create edge attributes.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{network\_filter() (pypath.main.PyPath method)@\spxentry{network\_filter()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.network_filter}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{network\_filter}}}{\emph{p=2.0}}{}
This function aims to cut the number of edges in the network,
without losing nodes, to make the network less connected,
less hairball-like, more usable for analysis.

\end{fulllineitems}

\index{network\_stats() (pypath.main.PyPath method)@\spxentry{network\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.network_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{network\_stats}}}{\emph{outfile=None}}{}
Calculates basic statistics for the whole network and each of
sources (node and edge counts, average node degree, graph
diameter, transitivity, adhesion and cohesion). Writes the
results in a tab file. File is stored in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by default).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the file name. If
none is specified, this will be
\sphinxcode{\sphinxupquote{'pwnet-\textless{}session\_id\textgreater{}-stats'}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{new\_edges() (pypath.main.PyPath method)@\spxentry{new\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.new_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{new\_edges}}}{\emph{edges}}{}
Adds new edges from any iterable of edges to the undirected
graph. Basically, calls \sphinxcode{\sphinxupquote{igraph.Graph.add\_edges()}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{edges}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the edges that are to be added to the network.

\end{description}\end{quote}

\end{fulllineitems}

\index{new\_nodes() (pypath.main.PyPath method)@\spxentry{new\_nodes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.new_nodes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{new\_nodes}}}{\emph{nodes}}{}
Adds new nodes from any iterable of nodes to the undirected
graph. Basically, calls \sphinxcode{\sphinxupquote{igraph.Graph.add\_vertices()}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{nodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the nodes that are to be added to the network.

\end{description}\end{quote}

\end{fulllineitems}

\index{node\_exists() (pypath.main.PyPath method)@\spxentry{node\_exists()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.node_exists}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{node\_exists}}}{\emph{name}}{}
Checks if a node exists in the (undirected) network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the node to be searched.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} Whether the node exists in the network or not.

\end{description}\end{quote}

\end{fulllineitems}

\index{numof\_directed\_edges() (pypath.main.PyPath method)@\spxentry{numof\_directed\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.numof_directed_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{numof\_directed\_edges}}}{}{}
\end{fulllineitems}

\index{numof\_reference\_interaction\_pairs() (pypath.main.PyPath method)@\spxentry{numof\_reference\_interaction\_pairs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.numof_reference_interaction_pairs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{numof\_reference\_interaction\_pairs}}}{}{}
Returns the total of unique references per interaction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{int}) \textendash{} Total number of unique references per
interaction.

\end{description}\end{quote}

\end{fulllineitems}

\index{numof\_references() (pypath.main.PyPath method)@\spxentry{numof\_references()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.numof_references}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{numof\_references}}}{}{}
Counts the number of reference on the network.

Counts the total number of unique references in the edges of the
network.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{int}) \textendash{} Number of unique references in the network.

\end{description}\end{quote}

\end{fulllineitems}

\index{numof\_undirected\_edges() (pypath.main.PyPath method)@\spxentry{numof\_undirected\_edges()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.numof_undirected_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{numof\_undirected\_edges}}}{}{}
\end{fulllineitems}

\index{orthology\_translation() (pypath.main.PyPath method)@\spxentry{orthology\_translation()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.orthology_translation}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{orthology\_translation}}}{\emph{target}, \emph{source=None}, \emph{only\_swissprot=True}, \emph{graph=None}}{}
Translates the current object to another organism by orthology.
Proteins without known ortholog will be deleted.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{target}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} NCBI Taxonomy ID of the target organism. E.g. 10090 for mouse.

\end{description}\end{quote}

\end{fulllineitems}

\index{p() (pypath.main.PyPath method)@\spxentry{p()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.p}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{p}}}{\emph{identifier}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{pathway\_attributes() (pypath.main.PyPath method)@\spxentry{pathway\_attributes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pathway_attributes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pathway\_attributes}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{pathway\_members() (pypath.main.PyPath method)@\spxentry{pathway\_members()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pathway_members}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pathway\_members}}}{\emph{pathway}, \emph{source}}{}
Returns an iterator with the members of a single pathway.
Apart from the pathway name you need to supply its source
database too.

\end{fulllineitems}

\index{pathway\_names() (pypath.main.PyPath method)@\spxentry{pathway\_names()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pathway_names}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pathway\_names}}}{\emph{source}, \emph{graph=None}}{}
Returns the names of all pathways having at least one member
in the current graph.

\end{fulllineitems}

\index{pathway\_similarity() (pypath.main.PyPath method)@\spxentry{pathway\_similarity()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pathway_similarity}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pathway\_similarity}}}{\emph{outfile=None}}{}
Computes the Sorensen’s similarity index across nodes and edges
for all the available pathway sources (already loaded in the
network) and saves them into table files. Files are stored in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by default).
See {\hyperref[\detokenize{reference:pypath.main.PyPath.sorensen_pathways}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.sorensen\_pathways()}}}}} for more
information..
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the file name
prefix (suffixes will be \sphinxcode{\sphinxupquote{'-nodes'}} and \sphinxcode{\sphinxupquote{'-edges'}}). If
none is specified, this will be
\sphinxcode{\sphinxupquote{'pwnet-\textless{}session\_id\textgreater{}-sim-pw'}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{pathways\_table() (pypath.main.PyPath method)@\spxentry{pathways\_table()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pathways_table}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pathways\_table}}}{\emph{filename='genes\_pathways.list', pw\_sources={[}'signalink', 'signor', 'netpath', 'kegg'{]}, graph=None}}{}
\end{fulllineitems}

\index{pfam\_regions() (pypath.main.PyPath method)@\spxentry{pfam\_regions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.pfam_regions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pfam\_regions}}}{}{}
\end{fulllineitems}

\index{phosphonetworks\_directions() (pypath.main.PyPath method)@\spxentry{phosphonetworks\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.phosphonetworks_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{phosphonetworks\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{phosphopoint\_directions() (pypath.main.PyPath method)@\spxentry{phosphopoint\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.phosphopoint_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{phosphopoint\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{phosphorylation\_directions() (pypath.main.PyPath method)@\spxentry{phosphorylation\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.phosphorylation_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{phosphorylation\_directions}}}{}{}
\end{fulllineitems}

\index{phosphorylation\_signs() (pypath.main.PyPath method)@\spxentry{phosphorylation\_signs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.phosphorylation_signs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{phosphorylation\_signs}}}{}{}
\end{fulllineitems}

\index{phosphosite\_directions() (pypath.main.PyPath method)@\spxentry{phosphosite\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.phosphosite_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{phosphosite\_directions}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{prdb\_tissue\_expr() (pypath.main.PyPath method)@\spxentry{prdb\_tissue\_expr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.prdb_tissue_expr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{prdb\_tissue\_expr}}}{\emph{tissue}, \emph{prdb=None}, \emph{graph=None}, \emph{occurrence=1}, \emph{group\_function=\textless{}function PyPath.\textless{}lambda\textgreater{}\textgreater{}}, \emph{na\_value=0.0}}{}
\end{fulllineitems}

\index{process\_direction() (pypath.main.PyPath method)@\spxentry{process\_direction()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.process_direction}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_direction}}}{\emph{line}, \emph{dir\_col}, \emph{dir\_val}, \emph{dir\_sep}}{}
Processes the direction information of an interaction according
to a data file from a source.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{line}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} The stripped and separated line from the resource data file
containing the information of an interaction.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dir\_col}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The column/position number where the information about the
direction is to be found (on \sphinxstyleemphasis{line}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dir\_val}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the terms {[}str{]} for which that interaction is to be
considered directed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dir\_sep}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Separator for the field in \sphinxstyleemphasis{line} containing the direction
information (if any).

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} Determines whether the given interaction is
directed or not.

\end{description}\end{quote}

\end{fulllineitems}

\index{process\_directions() (pypath.main.PyPath method)@\spxentry{process\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.process_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_directions}}}{\emph{dirs}, \emph{name}, \emph{directed=None}, \emph{stimulation=None}, \emph{inhibition=None}, \emph{graph=None}, \emph{id\_type=None}, \emph{dirs\_only=False}}{}
\end{fulllineitems}

\index{process\_dmi() (pypath.main.PyPath method)@\spxentry{process\_dmi()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.process_dmi}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_dmi}}}{\emph{source}, \emph{**kwargs}}{}
This is an universal function
for loading domain-motif objects
like load\_phospho\_dmi() for PTMs.
TODO this will replace load\_elm, load\_ielm, etc

\end{fulllineitems}

\index{process\_sign() (pypath.main.PyPath method)@\spxentry{process\_sign()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.process_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{process\_sign}}}{\emph{signData}, \emph{signDef}}{}
Processes the sign of an interaction, used when processing an
input file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{signData}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Data regarding the sign to be processed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{signDef}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Contains information about how to process \sphinxstyleemphasis{signData}. This
is defined in \sphinxcode{\sphinxupquote{pypath.data\_formats}}. First element
determines the position on the direction information of each
line on the data file {[}int{]}, second element is either {[}str{]}
or {[}list{]} and defines the terms for which an interaction is
defined as stimulation, third element is similar but for the
inhibition and third (optional) element determines the
separator for \sphinxstyleemphasis{signData} if contains more than one element.

\end{itemize}

\item[{Returns}] \leavevmode
\begin{itemize}
\item {} 
(\sphinxstyleemphasis{bool}) \textendash{} Determines whether the processed interaction
is considered stimulation or not.

\item {} 
(\sphinxstyleemphasis{bool}) \textendash{} Determines whether the processed interaction
is considered inhibition or not.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{protein() (pypath.main.PyPath method)@\spxentry{protein()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.protein}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{protein}}}{\emph{identifier}}{}
Same as \sphinxcode{\sphinxupquote{PyPath.get\_node}}, just for the directed graph.
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default directed graph,
or a UniProt ID or GeneSymbol which can be found in the
default directed network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{protein\_edge() (pypath.main.PyPath method)@\spxentry{protein\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.protein_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{protein\_edge}}}{\emph{source}, \emph{target}, \emph{directed=True}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Edge}} object if an edge exist between
the 2 proteins, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Vertex index or UniProt ID or GeneSymbol or \sphinxcode{\sphinxupquote{igraph.Vertex}}
object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{target}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Vertex index or UniProt ID or GeneSymbol or \sphinxcode{\sphinxupquote{igraph.Vertex}}
object.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{directed}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} To be passed to igraph.Graph.get\_eid()

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{proteins() (pypath.main.PyPath method)@\spxentry{proteins()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.proteins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{proteins}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{ps() (pypath.main.PyPath method)@\spxentry{ps()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.ps}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{ps}}}{\emph{identifiers}}{}
\end{fulllineitems}

\index{random\_walk\_with\_return() (pypath.main.PyPath method)@\spxentry{random\_walk\_with\_return()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.random_walk_with_return}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{random\_walk\_with\_return}}}{\emph{q}, \emph{graph=None}, \emph{c=0.5}, \emph{niter=1000}}{}
Random walk with return (RWR) starting from one or more query nodes.
Returns affinity (probability) vector of all nodes in the graph.
\begin{quote}
\begin{quote}\begin{description}
\item[{param int,list q}] \leavevmode
Vertex IDs of query nodes.

\item[{param igraph.Graph graph}] \leavevmode
An \sphinxtitleref{igraph.Graph} object.

\item[{param float c}] \leavevmode
Probability of restart.

\item[{param int niter}] \leavevmode
Number of iterations.

\end{description}\end{quote}
\end{quote}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{import} \PYG{n+nn}{igraph}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{import} \PYG{n+nn}{pypath}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{pa} \PYG{o}{=} \PYG{n}{pypath}\PYG{o}{.}\PYG{n}{PyPath}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{pa}\PYG{o}{.}\PYG{n}{init\PYGZus{}network}\PYG{p}{(}\PYG{p}{\PYGZob{}}
\PYG{g+go}{        \PYGZsq{}signor\PYGZsq{}: pypath.data\PYGZus{}formats.pathway[\PYGZsq{}signor\PYGZsq{}]}
\PYG{g+go}{    \PYGZcb{})}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{q} \PYG{o}{=} \PYG{p}{[}
\PYG{g+go}{        pa.gs(\PYGZsq{}EGFR\PYGZsq{}).index,}
\PYG{g+go}{        pa.gs(\PYGZsq{}ATG4B\PYGZsq{}).index}
\PYG{g+go}{    ]}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{rwr} \PYG{o}{=} \PYG{n}{pa}\PYG{o}{.}\PYG{n}{random\PYGZus{}walk\PYGZus{}with\PYGZus{}return}\PYG{p}{(}\PYG{n}{q} \PYG{o}{=} \PYG{n}{q}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{palette} \PYG{o}{=} \PYG{n}{igraph}\PYG{o}{.}\PYG{n}{RainbowPalette}\PYG{p}{(}\PYG{n}{n} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{colors}  \PYG{o}{=} \PYG{p}{[}\PYG{n}{palette}\PYG{o}{.}\PYG{n}{get}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{i}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)} \PYG{k}{for} \PYG{n}{i} \PYG{o+ow}{in} \PYG{n}{rwr} \PYG{o}{/} \PYG{n+nb}{max}\PYG{p}{(}\PYG{n}{rwr}\PYG{p}{)} \PYG{o}{*} \PYG{l+m+mi}{99}\PYG{p}{]}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{igraph}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{pa}\PYG{o}{.}\PYG{n}{graph}\PYG{p}{,} \PYG{n}{vertex\PYGZus{}color} \PYG{o}{=} \PYG{n}{colors}\PYG{p}{)}
\end{sphinxVerbatim}

\end{fulllineitems}

\index{random\_walk\_with\_return2() (pypath.main.PyPath method)@\spxentry{random\_walk\_with\_return2()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.random_walk_with_return2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{random\_walk\_with\_return2}}}{\emph{q}, \emph{c=0.5}, \emph{niter=1000}}{}
Literally does random walks.
Only for testing of the other method, to be deleted later.

\end{fulllineitems}

\index{read\_data\_file() (pypath.main.PyPath method)@\spxentry{read\_data\_file()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.read_data_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{read\_data\_file}}}{\emph{param}, \emph{keep\_raw=False}, \emph{cache\_files=\{\}}, \emph{reread=None}, \emph{redownload=None}}{}
Reads interaction data file containing node and edge attributes
that can be read from simple text based files and adds it to the
networkdata. This function works not only with files, but with
lists as well. Any other function can be written to download and
preprocess data, and then give it to this function to finally
attach to the network.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{param}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.input\_formats.ReadSettings}}) \textendash{} \sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings}} instance
containing the detailed definition of the input format of
the file. Instead of the file name (on the
\sphinxcode{\sphinxupquote{pypath.input\_formats.ReadSettings.input}}
attribute) you can give a custom function name, which will
be executed, and the returned data will be used instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{keep\_raw}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to keep the raw data
read by this function, in order for debugging purposes, or
further use.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cache\_files}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{\{\}}} by default. Contains the resource name(s)
{[}str{]} (keys) and the corresponding cached file name {[}str{]}.
If provided (and file exists) bypasses the download of the
data for that resource and uses the cache file instead.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reread}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to reread
the data files from the cache or omit them (similar to
\sphinxstyleemphasis{redownload}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{redownload}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to
re-download the data and ignore the cache.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{read\_from\_cache() (pypath.main.PyPath method)@\spxentry{read\_from\_cache()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.read_from_cache}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{read\_from\_cache}}}{\emph{cache\_file}}{}
Reads a pickle file from the cache and returns it. It is assumed
that the subfolder \sphinxcode{\sphinxupquote{cache/}} is on the supplied path.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{cache\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the cache file that is to be loaded.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{file}) \textendash{} The loaded pickle file from the cache. Type will
depend on the file itself (e.g.: if the pickle was saved
from a dictionary, the type will be {[}dict{]}).

\end{description}\end{quote}

\end{fulllineitems}

\index{read\_list\_file() (pypath.main.PyPath method)@\spxentry{read\_list\_file()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.read_list_file}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{read\_list\_file}}}{\emph{settings}, \emph{**kwargs}}{}
Reads a list from a file and adds it to
\sphinxcode{\sphinxupquote{pypath.main.PyPath.lists}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{settings}} (\sphinxstyleliteralemphasis{\sphinxupquote{pypath.input\_formats.ReadList}}) \textendash{} \sphinxcode{\sphinxupquote{python.data\_formats.ReadList}} instance specifying
the settings of the file to be read. See the class
documentation for more details.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{**kwargs}} \textendash{} Extra arguments passed to the file reading function. Such
function name is outlined in the
\sphinxcode{\sphinxupquote{python.data\_formats.ReadList.input}} attribute and
defined in \sphinxcode{\sphinxupquote{pypath.dataio}}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{reference\_edge\_ratio() (pypath.main.PyPath method)@\spxentry{reference\_edge\_ratio()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.reference_edge_ratio}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reference\_edge\_ratio}}}{}{}
Computes the average number of references per edge (as in the
undirected graph).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{float}) \textendash{} Average number of references per edge.

\end{description}\end{quote}

\end{fulllineitems}

\index{reference\_hist() (pypath.main.PyPath method)@\spxentry{reference\_hist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.reference_hist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reference\_hist}}}{\emph{filename=None}}{}
Generates a file containing a table with information about the
network’s edges. First column contains the source node ID,
followed by the target’s ID, third column contains the number of
references for that interaction and finally the number of
sources. Writes the results in a tab file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the file name and
path to save the table. If none is passed, file will be
saved in \sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}}
by default) with the name \sphinxcode{\sphinxupquote{'\textless{}session\_id\textgreater{}-refs-hist'}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{reload() (pypath.main.PyPath method)@\spxentry{reload()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.reload}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{reload}}}{}{}
Reloads the object from the module level.

\end{fulllineitems}

\index{remove\_htp() (pypath.main.PyPath method)@\spxentry{remove\_htp()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.remove_htp}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_htp}}}{\emph{threshold=50}, \emph{keep\_directed=False}}{}
\end{fulllineitems}

\index{remove\_undirected() (pypath.main.PyPath method)@\spxentry{remove\_undirected()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.remove_undirected}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_undirected}}}{\emph{min\_refs=None}}{}
\end{fulllineitems}

\index{run\_batch() (pypath.main.PyPath method)@\spxentry{run\_batch()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.run_batch}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run\_batch}}}{\emph{methods}, \emph{toCall=None}}{}
\end{fulllineitems}

\index{save\_network() (pypath.main.PyPath method)@\spxentry{save\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.save_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_network}}}{\emph{pfile=None}}{}
Saves the network object.

Stores the instance into a pickle (binary) file which can be
reloaded in the future.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{pfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The path/file name where to
store the pcikle file. If not specified, saves the network
to its default location
(\sphinxcode{\sphinxupquote{'cache/default\_network.pickle'}}).

\end{description}\end{quote}

\end{fulllineitems}

\index{save\_session() (pypath.main.PyPath method)@\spxentry{save\_session()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.save_session}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_session}}}{}{}
Save the current session state into pickle dump. The file will
be saved in the current working directory as
\sphinxcode{\sphinxupquote{'pypath-\textless{}session\_id\textgreater{}.pickle'}}.

\end{fulllineitems}

\index{search\_attr\_and() (pypath.main.PyPath method)@\spxentry{search\_attr\_and()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.search_attr_and}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{search\_attr\_and}}}{\emph{obj}, \emph{lst}}{}
Searches a given collection of attributes in a given object.
Only returns \sphinxcode{\sphinxupquote{True}}, if all elements of \sphinxstyleemphasis{lst} can be found in
\sphinxstyleemphasis{obj}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{obj}} (\sphinxstyleliteralemphasis{\sphinxupquote{object}}) \textendash{} Object (dictionary-like) where to search for elements of
\sphinxstyleemphasis{lst}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Keys are the attribute names {[}str{]} and values the collection
of elements to be searched in such attribute {[}set{]}.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} only if \sphinxstyleemphasis{lst} is empty or all of its
elements are found in \sphinxstyleemphasis{obj}. Returns \sphinxcode{\sphinxupquote{False}} otherwise (as
soon as one element of \sphinxstyleemphasis{lst} is not found).

\end{description}\end{quote}

\end{fulllineitems}

\index{search\_attr\_or() (pypath.main.PyPath method)@\spxentry{search\_attr\_or()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.search_attr_or}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{search\_attr\_or}}}{\emph{obj}, \emph{lst}}{}
Searches a given collection of attributes in a given object. As
soon as one item is found, returns \sphinxcode{\sphinxupquote{True}}, if none could be
found then returns \sphinxcode{\sphinxupquote{False}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{obj}} (\sphinxstyleliteralemphasis{\sphinxupquote{object}}) \textendash{} Object (dictionary-like) where to search for elements of
\sphinxstyleemphasis{lst}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Keys are the attribute names {[}str{]} and values the collection
of elements to be searched in such attribute {[}set{]}.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if \sphinxstyleemphasis{lst} is empty or any of its
elements is found in \sphinxstyleemphasis{obj}. Returns only \sphinxcode{\sphinxupquote{False}} if cannot
find anything.

\end{description}\end{quote}

\end{fulllineitems}

\index{second\_neighbours() (pypath.main.PyPath method)@\spxentry{second\_neighbours()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.second_neighbours}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{second\_neighbours}}}{\emph{node}, \emph{indices=False}, \emph{with\_first=False}}{}
Looks for the (first and) second neighbours of a given node and
returns a list of their UniProt IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{node}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The UniProt ID of the node of interest. Can also be the
index of such node {[}int{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{indices}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return the
neighbour nodes indices or their UniProt IDs.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{wit\_first}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return also the
first neighbours or not.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} The list containing the second neighbours of the
queried node (including the first ones if specified).

\end{description}\end{quote}

\end{fulllineitems}

\index{select\_by\_go() (pypath.main.PyPath method)@\spxentry{select\_by\_go()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.select_by_go}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{select\_by\_go}}}{\emph{go\_terms}}{}
Retrieves the vertex IDs of all vertices annotated with any
Gene Ontology terms or their descendants, or evaluates string
expression (see \sphinxcode{\sphinxupquote{select\_by\_go\_expr}}).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{go\_terms}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} A single GO term, a set of GO terms or an expression with
GO terms.

\end{description}\end{quote}

\end{fulllineitems}

\index{select\_by\_go\_all() (pypath.main.PyPath method)@\spxentry{select\_by\_go\_all()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.select_by_go_all}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{select\_by\_go\_all}}}{\emph{go\_terms}}{}
Selects the nodes annotated by all GO terms in \sphinxcode{\sphinxupquote{go\_terms}}.

Returns set of vertex IDs.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{go\_terms}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List, set or tuple of GO terms.

\end{description}\end{quote}

\end{fulllineitems}

\index{select\_by\_go\_expr() (pypath.main.PyPath method)@\spxentry{select\_by\_go\_expr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.select_by_go_expr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{select\_by\_go\_expr}}}{\emph{go\_expr}}{}
Selects vertices based on an expression of Gene Ontology terms.
Operator precedence not considered, please use parentheses.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{go\_expr}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} An expression of Gene Ontology terms. E.g.
\sphinxcode{\sphinxupquote{'(GO:0005576 and not GO:0070062) or GO:0005887'}}. Parentheses
and operators \sphinxcode{\sphinxupquote{and}}, \sphinxcode{\sphinxupquote{or}} and \sphinxcode{\sphinxupquote{not}} can be used.

\end{description}\end{quote}

\end{fulllineitems}

\index{separate() (pypath.main.PyPath method)@\spxentry{separate()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.separate}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{separate}}}{}{}
Separates the undirected network according to the different
sources. Basically applies
{\hyperref[\detokenize{reference:pypath.main.PyPath.get_network}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.get\_network()}}}}} for each resource.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Keys are resource names {[}str{]} whose values are
the subnetwork {[}igraph.Graph{]} containing the elements of
that source.

\end{description}\end{quote}

\end{fulllineitems}

\index{separate\_by\_category() (pypath.main.PyPath method)@\spxentry{separate\_by\_category()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.separate_by_category}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{separate\_by\_category}}}{}{}
Separates the undirected network according to resource
categories. Possible categories are:
\begin{itemize}
\item {} 
\sphinxcode{\sphinxupquote{'m'}}: PTM/enzyme-substrate resources.

\item {} 
\sphinxcode{\sphinxupquote{'p'}}: Pathway/activity flow resources.

\item {} 
\sphinxcode{\sphinxupquote{'i'}}: Undirected/PPI resources.

\item {} 
\sphinxcode{\sphinxupquote{'r'}}: Process description/reaction resources.

\item {} 
\sphinxcode{\sphinxupquote{'t'}}: Transcription resources.

\end{itemize}

Works in the same way as {\hyperref[\detokenize{reference:pypath.main.PyPath.separate}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.separate()}}}}}.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Keys are category names {[}str{]} whose values are
the subnetwork {[}igraph.Graph{]} containing the elements of
those resources corresponding to that category.

\end{description}\end{quote}

\end{fulllineitems}

\index{sequences() (pypath.main.PyPath method)@\spxentry{sequences()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sequences}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sequences}}}{\emph{isoforms=True}, \emph{update=False}}{}
\end{fulllineitems}

\index{set\_boolean\_vattr() (pypath.main.PyPath method)@\spxentry{set\_boolean\_vattr()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_boolean_vattr}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_boolean\_vattr}}}{\emph{attr}, \emph{vids}, \emph{negate=False}}{}
\end{fulllineitems}

\index{set\_categories() (pypath.main.PyPath method)@\spxentry{set\_categories()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_categories}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_categories}}}{}{}
Sets the category attribute on the network nodes and edges
(\sphinxcode{\sphinxupquote{'cat'}}) as well the edge attribute coercing the references
by category (\sphinxcode{\sphinxupquote{'refs\_by\_cat'}}). The possible categories are
as follows:
\begin{itemize}
\item {} 
\sphinxcode{\sphinxupquote{'m'}}: PTM/enzyme-substrate resources.

\item {} 
\sphinxcode{\sphinxupquote{'p'}}: Pathway/activity flow resources.

\item {} 
\sphinxcode{\sphinxupquote{'i'}}: Undirected/PPI resources.

\item {} 
\sphinxcode{\sphinxupquote{'r'}}: Process description/reaction resources.

\item {} 
\sphinxcode{\sphinxupquote{'t'}}: Transcription resources.

\end{itemize}

\end{fulllineitems}

\index{set\_chembl\_mysql() (pypath.main.PyPath method)@\spxentry{set\_chembl\_mysql()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_chembl_mysql}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_chembl\_mysql}}}{\emph{title}, \emph{config\_file=None}}{}
Sets the ChEMBL MySQL configuration according to the \sphinxstyleemphasis{title}
section in \sphinxstyleemphasis{config\_file} ini file configuration.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{title}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Section title of the ini file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{config\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the configuration
file name if none is passed, \sphinxcode{\sphinxupquote{mysql\_config/defaults.mysql}}
will be used.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_disease\_genes() (pypath.main.PyPath method)@\spxentry{set\_disease\_genes()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_disease_genes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_disease\_genes}}}{\emph{dataset='curated'}}{}
Creates a vertex attribute named \sphinxtitleref{dis} with boolean values \sphinxstyleemphasis{True}
if the protein encoded by a disease related gene according to
DisGeNet.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{dataset}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Which dataset to use from DisGeNet. Default is \sphinxtitleref{curated}.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_druggability() (pypath.main.PyPath method)@\spxentry{set\_druggability()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_druggability}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_druggability}}}{}{}
Creates a vertex attribute \sphinxtitleref{dgb} with value \sphinxstyleemphasis{True} if
the protein is druggable, otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_drugtargets() (pypath.main.PyPath method)@\spxentry{set\_drugtargets()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_drugtargets}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_drugtargets}}}{\emph{pchembl=5.0}}{}
Creates a vertex attribute \sphinxtitleref{dtg} with value \sphinxstyleemphasis{True} if
the protein has at least one compound binding with
affinity higher than \sphinxtitleref{pchembl}, otherwise \sphinxstyleemphasis{False}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{pchembl}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Pchembl threshold.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_kinases() (pypath.main.PyPath method)@\spxentry{set\_kinases()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_kinases}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_kinases}}}{}{}
Creates a vertex attribute \sphinxtitleref{kin} with value \sphinxstyleemphasis{True} if
the protein is a kinase, otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_plasma\_membrane\_proteins\_cspa() (pypath.main.PyPath method)@\spxentry{set\_plasma\_membrane\_proteins\_cspa()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_plasma_membrane_proteins_cspa}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_plasma\_membrane\_proteins\_cspa}}}{}{}
Creates a vertex attribute \sphinxtitleref{cspa} with value \sphinxstyleemphasis{True} if
the protein is a plasma membrane protein according to CPSA,
otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_plasma\_membrane\_proteins\_cspa\_surfaceome() (pypath.main.PyPath method)@\spxentry{set\_plasma\_membrane\_proteins\_cspa\_surfaceome()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_plasma_membrane_proteins_cspa_surfaceome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_plasma\_membrane\_proteins\_cspa\_surfaceome}}}{\emph{score\_threshold=0.0}}{}
Creates a vertex attribute \sphinxcode{\sphinxupquote{surf}} with value \sphinxstyleemphasis{True} if
the protein is a plasma membrane protein according either to the
Cell Surface Protein Atlas or the In Silico Human Surfaceome.

\end{fulllineitems}

\index{set\_plasma\_membrane\_proteins\_surfaceome() (pypath.main.PyPath method)@\spxentry{set\_plasma\_membrane\_proteins\_surfaceome()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_plasma_membrane_proteins_surfaceome}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_plasma\_membrane\_proteins\_surfaceome}}}{\emph{score\_threshold=0.0}}{}
Creates a vertex attribute \sphinxtitleref{ishs} with value \sphinxstyleemphasis{True} if
the protein is a plasma membrane protein according to the In Silico
Human Surfaceome, otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_receptors() (pypath.main.PyPath method)@\spxentry{set\_receptors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_receptors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_receptors}}}{}{}
Creates a vertex attribute \sphinxtitleref{rec} with value \sphinxstyleemphasis{True} if
the protein is a receptor, otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_signaling\_proteins() (pypath.main.PyPath method)@\spxentry{set\_signaling\_proteins()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_signaling_proteins}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_signaling\_proteins}}}{}{}
Creates a vertex attribute \sphinxtitleref{kin} with value \sphinxstyleemphasis{True} if
the protein is a kinase, otherwise \sphinxstyleemphasis{False}.

\end{fulllineitems}

\index{set\_tfs() (pypath.main.PyPath method)@\spxentry{set\_tfs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_tfs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_tfs}}}{\emph{classes={[}'a', 'b', 'other'{]}}}{}
\end{fulllineitems}

\index{set\_transcription\_factors() (pypath.main.PyPath method)@\spxentry{set\_transcription\_factors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.set_transcription_factors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_transcription\_factors}}}{\emph{classes={[}'a', 'b', 'other'{]}}}{}
Creates a vertex attribute \sphinxtitleref{tf} with value \sphinxstyleemphasis{True} if
the protein is a transcription factor, otherwise \sphinxstyleemphasis{False}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{classes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Classes to use from TF Census. Default is \sphinxtitleref{{[}‘a’, ‘b’, ‘other’{]}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{shortest\_path\_dist() (pypath.main.PyPath method)@\spxentry{shortest\_path\_dist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.shortest_path_dist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shortest\_path\_dist}}}{\emph{graph=None}, \emph{subset=None}, \emph{outfile=None}, \emph{**kwargs}}{}
Computes the distribution of shortest paths for each pair of
nodes in the network (or between group(s) of nodes if \sphinxstyleemphasis{subset}
is provided). \sphinxstylestrong{NOTE:} this method can take a while to compute,
e.g.: if the network has 10K nodes, the total number of possible
pairs to compute is:
\begin{equation*}
\begin{split}\binom{10^4}{2} = 49995000\end{split}
\end{equation*}\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The network object for which
the shortest path distribution is to be computed. If none is
passed, takes the undirected network of the current
instance.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{susbet}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Contains two lists of node
indices defining two groups between which the distribution
is to be computed. Can also be {[}list{]} if the shortest paths
are to be searched whithin the group. If none is passed, the
whole network is taken by default.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. File name/path to save the
shortest path distribution. If none is passed, no file is
generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{**kwargs}} \textendash{} Additional keyword arguments passed to
\sphinxcode{\sphinxupquote{igraph.Graph.get\_shortest\_paths()}}.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} The length of the shortest paths for each pair
of nodes of the network (or whithin/between group/s if
\sphinxstyleemphasis{subset} is provided).

\end{description}\end{quote}

\end{fulllineitems}

\index{signaling\_proteins\_list() (pypath.main.PyPath method)@\spxentry{signaling\_proteins\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.signaling_proteins_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{signaling\_proteins\_list}}}{}{}
Compiles a list of signaling proteins (as opposed to other
proteins like metabolic enzymes, matrix proteins, etc), by
looking up a few simple keywords in short description of GO
terms.

\end{fulllineitems}

\index{signor\_pathways() (pypath.main.PyPath method)@\spxentry{signor\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.signor_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{signor\_pathways}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{similarity\_groups() (pypath.main.PyPath method)@\spxentry{similarity\_groups()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.similarity_groups}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{similarity\_groups}}}{\emph{groups}, \emph{index='simpson'}}{}
Computes the similarity index across the given \sphinxstyleemphasis{groups}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{groups}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the different group names {[}str{]} as keys and their
corresponding elements {[}set{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{index}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'simpson'}} by default. The type of index metric
to use to compute the similarity. Options are \sphinxcode{\sphinxupquote{'simpson'}},
\sphinxcode{\sphinxupquote{'sorensen'}} and \sphinxcode{\sphinxupquote{'jaccard'}}.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Dictionary of dictionaries containing the groups
names {[}str{]} as keys (for both inner and outer dictionaries)
and the index metric as inner value {[}float{]} between those
groups.

\end{description}\end{quote}

\end{fulllineitems}

\index{small\_plot() (pypath.main.PyPath method)@\spxentry{small\_plot()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.small_plot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{small\_plot}}}{\emph{graph}, \emph{**kwargs}}{}
This method is deprecated, do not use it.

\end{fulllineitems}

\index{sorensen\_pathways() (pypath.main.PyPath method)@\spxentry{sorensen\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sorensen_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sorensen\_pathways}}}{\emph{pwlist=None}}{}
Computes the Sorensen’s similarity index across nodes and edges
for the given list of pathway sources (all loaded pathway
sources by default).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{pwlist}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. The list of pathway sources
to be compared.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Nested dictionaries (three levels). First-level
keys are \sphinxcode{\sphinxupquote{'nodes'}} and \sphinxcode{\sphinxupquote{'edges'}}, then second and third
levels correspond to \sphinxcode{\sphinxupquote{\textless{}source\textgreater{}\_\_\textless{}patwhay\textgreater{}}} names which map
to the similarity index between those pathways {[}float{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{source\_diagram() (pypath.main.PyPath method)@\spxentry{source\_diagram()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.source_diagram}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{source\_diagram}}}{\emph{outf=None}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{source\_network() (pypath.main.PyPath method)@\spxentry{source\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.source_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{source\_network}}}{\emph{font='HelveticaNeueLTStd'}}{}
For EMBL branding, use Helvetica Neue Linotype Standard light

\end{fulllineitems}

\index{source\_similarity() (pypath.main.PyPath method)@\spxentry{source\_similarity()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.source_similarity}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{source\_similarity}}}{\emph{outfile=None}}{}
Computes the Sorensen’s similarity index across nodes and edges
for all the sources available (already loaded in the network)
and saves them into table files. Files are stored in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by default).
See {\hyperref[\detokenize{reference:pypath.main.PyPath.databases_similarity}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{pypath.main.PyPath.databases\_similarity()}}}}} for more
information.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the file name
prefix (suffixes will be \sphinxcode{\sphinxupquote{'-nodes'}} and \sphinxcode{\sphinxupquote{'-edges'}}). If
none is specified, this will be
\sphinxcode{\sphinxupquote{'pwnet-\textless{}session\_id\textgreater{}-sim-src'}}.

\end{description}\end{quote}

\end{fulllineitems}

\index{source\_stats() (pypath.main.PyPath method)@\spxentry{source\_stats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.source_stats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{source\_stats}}}{}{}
\end{fulllineitems}

\index{sources\_hist() (pypath.main.PyPath method)@\spxentry{sources\_hist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sources_hist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_hist}}}{}{}
Counts the number of sources per interaction in the graph and
saves them into a file named \sphinxcode{\sphinxupquote{source\_num}}. File is stored in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by
default).

\end{fulllineitems}

\index{sources\_overlap() (pypath.main.PyPath method)@\spxentry{sources\_overlap()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sources_overlap}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_overlap}}}{\emph{diagonal=False}}{}
\end{fulllineitems}

\index{sources\_venn\_data() (pypath.main.PyPath method)@\spxentry{sources\_venn\_data()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sources_venn_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_venn\_data}}}{\emph{fname=None}, \emph{return\_data=False}}{}
Computes the overlap in number of interactions for all pairs of
sources.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{fname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If provided, saves the
results into a table file. File is stored in
\sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}} (\sphinxcode{\sphinxupquote{'results'}} by
default).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{return\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Whether to return the
results as a {[}list{]}.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Only if \sphinxstyleemphasis{return\_data} is set to \sphinxcode{\sphinxupquote{True}}. List
of lists containing the counts for each pair of resources.
This is, for instance, number of interactions only in
resource A, number of interactions only in resource B and
number of common interactions between A and B.

\end{description}\end{quote}

\end{fulllineitems}

\index{straight\_between() (pypath.main.PyPath method)@\spxentry{straight\_between()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.straight_between}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{straight\_between}}}{\emph{id\_a}, \emph{id\_b}}{}
Finds an edge between the provided node names.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the target node.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{int}) \textendash{} The edge ID. If the edge doesn’t exist, returns
{[}list{]} with the node indices {[}int{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{string\_effects() (pypath.main.PyPath method)@\spxentry{string\_effects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.string_effects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{string\_effects}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{sum\_in\_complex() (pypath.main.PyPath method)@\spxentry{sum\_in\_complex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.sum_in_complex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sum\_in\_complex}}}{\emph{csources={[}'corum'{]}, graph=None}}{}
Returns the total number of edges in the network falling
between two members of the same complex.
Returns as a dict by complex resources.
Calls :py:func:pypath.pypath.Pypath.edges\_in\_comlexes()
to do the calculations.
\begin{description}
\item[{@csources}] \leavevmode{[}list{]}
List of complex resources. Should be already loaded.

\item[{@graph}] \leavevmode{[}igraph.Graph(){]}
The graph object to do the calculations on.

\end{description}

\end{fulllineitems}

\index{table\_latex() (pypath.main.PyPath method)@\spxentry{table\_latex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.table_latex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{table\_latex}}}{\emph{fname}, \emph{header}, \emph{data}, \emph{sum\_row=True}, \emph{row\_order=None}, \emph{latex\_hdr=True}, \emph{caption=''}, \emph{font='HelveticaNeueLTStd-LtCn'}, \emph{fontsize=8}, \emph{sum\_label='Total'}, \emph{sum\_cols=None}, \emph{header\_format='\%s'}, \emph{by\_category=True}}{}
\end{fulllineitems}

\index{third\_source\_directions() (pypath.main.PyPath method)@\spxentry{third\_source\_directions()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.third_source_directions}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{third\_source\_directions}}}{\emph{graph=None}, \emph{use\_string\_effects=False}, \emph{use\_laudanna\_data=False}}{}
This method calls a series of methods to get
additional direction \& effect information
from sources having no literature curated references,
but giving sufficient evidence about the directionality
for interactions already supported by literature
evidences from other sources.

\end{fulllineitems}

\index{tissue\_network() (pypath.main.PyPath method)@\spxentry{tissue\_network()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.tissue_network}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{tissue\_network}}}{\emph{tissue}, \emph{graph=None}}{}
Returns a network which includes the proteins expressed in
certain tissue according to ProteomicsDB.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{tissue}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Tissue name as used in ProteomicsDB.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{graph}} (\sphinxstyleliteralemphasis{\sphinxupquote{igraph.Graph}}) \textendash{} A graph object, by default the \sphinxtitleref{graph} attribute of
the current instance.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{transcription\_factors() (pypath.main.PyPath method)@\spxentry{transcription\_factors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.transcription_factors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{transcription\_factors}}}{}{}
\end{fulllineitems}

\index{uniprot() (pypath.main.PyPath method)@\spxentry{uniprot()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.uniprot}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{uniprot}}}{\emph{uniprot}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the UniProt
can be found in the default undirected network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@uniprot}] \leavevmode{[}str{]}
UniProt ID.

\end{description}

\end{fulllineitems}

\index{uniprots() (pypath.main.PyPath method)@\spxentry{uniprots()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.uniprots}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{uniprots}}}{\emph{uniprots}}{}
Returns list of \sphinxcode{\sphinxupquote{igraph.Vertex()}} object
for a list of UniProt IDs omitting those
could not be found in the default
undirected graph.

\end{fulllineitems}

\index{uniq\_node\_list() (pypath.main.PyPath method)@\spxentry{uniq\_node\_list()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.uniq_node_list}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{uniq\_node\_list}}}{\emph{lst}}{}
Returns a given list of nodes containing only the unique
elements.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{lst}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of nodes.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Copy of \sphinxstyleemphasis{lst} containing only unique nodes.

\end{description}\end{quote}

\end{fulllineitems}

\index{uniq\_ptm() (pypath.main.PyPath method)@\spxentry{uniq\_ptm()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.uniq_ptm}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{uniq\_ptm}}}{\emph{ptms}}{}
\end{fulllineitems}

\index{uniq\_ptms() (pypath.main.PyPath method)@\spxentry{uniq\_ptms()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.uniq_ptms}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{uniq\_ptms}}}{}{}
\end{fulllineitems}

\index{up() (pypath.main.PyPath method)@\spxentry{up()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up}}}{\emph{uniprot}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the UniProt
can be found in the default undirected network,
otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@uniprot}] \leavevmode{[}str{]}
UniProt ID.

\end{description}

\end{fulllineitems}

\index{up\_affected\_by() (pypath.main.PyPath method)@\spxentry{up\_affected\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_affected_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_affected\_by}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_affects() (pypath.main.PyPath method)@\spxentry{up\_affects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_affects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_affects}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_edge() (pypath.main.PyPath method)@\spxentry{up\_edge()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_edge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_edge}}}{\emph{source}, \emph{target}, \emph{directed=True}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Edge}} object if an edge exist between
the 2 proteins, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@source}] \leavevmode{[}str{]}
UniProt ID

\item[{@target}] \leavevmode{[}str{]}
UniProt ID

\item[{@directed}] \leavevmode{[}bool{]}
To be passed to igraph.Graph.get\_eid()

\end{description}

\end{fulllineitems}

\index{up\_in\_directed() (pypath.main.PyPath method)@\spxentry{up\_in\_directed()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_in_directed}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_in\_directed}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_in\_undirected() (pypath.main.PyPath method)@\spxentry{up\_in\_undirected()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_in_undirected}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_in\_undirected}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_inhibited\_by() (pypath.main.PyPath method)@\spxentry{up\_inhibited\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_inhibited_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_inhibited\_by}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_inhibits() (pypath.main.PyPath method)@\spxentry{up\_inhibits()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_inhibits}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_inhibits}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_neighborhood() (pypath.main.PyPath method)@\spxentry{up\_neighborhood()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_neighborhood}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_neighborhood}}}{\emph{uniprots}, \emph{order=1}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{up\_neighbors() (pypath.main.PyPath method)@\spxentry{up\_neighbors()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_neighbors}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_neighbors}}}{\emph{uniprot}, \emph{mode='ALL'}}{}
\end{fulllineitems}

\index{up\_stimulated\_by() (pypath.main.PyPath method)@\spxentry{up\_stimulated\_by()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_stimulated_by}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_stimulated\_by}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{up\_stimulates() (pypath.main.PyPath method)@\spxentry{up\_stimulates()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.up_stimulates}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{up\_stimulates}}}{\emph{uniprot}}{}
\end{fulllineitems}

\index{update\_adjlist() (pypath.main.PyPath method)@\spxentry{update\_adjlist()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_adjlist}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_adjlist}}}{\emph{graph=None}, \emph{mode='ALL'}}{}
Creates an adjacency list in a list of sets format.

\end{fulllineitems}

\index{update\_attrs() (pypath.main.PyPath method)@\spxentry{update\_attrs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_attrs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_attrs}}}{}{}
Updates the node and edge attributes. Note that no data is
donwloaded, mainly updates the dictionaries of attributes
\sphinxcode{\sphinxupquote{pypath.main.PyPath.edgeAttrs}} and
\sphinxcode{\sphinxupquote{pypath.main.PyPath.vertexAttrs}} containing the
attributes names and their correspoding types and initializes
such attributes in the network nodes/edges if they weren’t.

\end{fulllineitems}

\index{update\_cats() (pypath.main.PyPath method)@\spxentry{update\_cats()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_cats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_cats}}}{}{}
Makes sure that the \sphinxcode{\sphinxupquote{pypath.main.PyPath.has\_cats}}
attribute is an up to date {[}set{]} of all categories in the
current network.

\end{fulllineitems}

\index{update\_db\_dict() (pypath.main.PyPath method)@\spxentry{update\_db\_dict()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_db_dict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_db\_dict}}}{}{}
\end{fulllineitems}

\index{update\_pathway\_types() (pypath.main.PyPath method)@\spxentry{update\_pathway\_types()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_pathway_types}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_pathway\_types}}}{}{}
Updates the pathway types attribute
(\sphinxcode{\sphinxupquote{pypath.main.PyPath.pathway\_types}}) according to the
loaded resources of the undirected network.

\end{fulllineitems}

\index{update\_pathways() (pypath.main.PyPath method)@\spxentry{update\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_pathways}}}{}{}
Makes sure that the \sphinxcode{\sphinxupquote{pypath.main.PyPath.pathways}}
attribute is an up to date {[}dict{]} of all pathways and their
sources in the current network.

\end{fulllineitems}

\index{update\_sources() (pypath.main.PyPath method)@\spxentry{update\_sources()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_sources}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_sources}}}{}{}
Makes sure that the \sphinxcode{\sphinxupquote{pypath.main.PyPath.sources}}
attribute is an up to date {[}list{]} of all sources in the current
network.

\end{fulllineitems}

\index{update\_vertex\_sources() (pypath.main.PyPath method)@\spxentry{update\_vertex\_sources()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_vertex_sources}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_vertex\_sources}}}{}{}
Updates the all the vertex attributes \sphinxcode{\sphinxupquote{'sources'}} and
\sphinxcode{\sphinxupquote{'references'}} according to their related edges (on the
undirected graph).

\end{fulllineitems}

\index{update\_vindex() (pypath.main.PyPath method)@\spxentry{update\_vindex()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_vindex}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_vindex}}}{}{}
This is deprecated.

\end{fulllineitems}

\index{update\_vname() (pypath.main.PyPath method)@\spxentry{update\_vname()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.update_vname}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_vname}}}{}{}
Fast lookup of node names and indexes, these are hold in a
{[}list{]} and a {[}dict{]} as well. However, every time new nodes are
added, these should be updated. This function is automatically
called after all operations affecting node indices.

\end{fulllineitems}

\index{ups() (pypath.main.PyPath method)@\spxentry{ups()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.ups}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{ups}}}{\emph{uniprots}}{}
Returns list of \sphinxcode{\sphinxupquote{igraph.Vertex()}} object
for a list of UniProt IDs omitting those
could not be found in the default
undirected graph.

\end{fulllineitems}

\index{v() (pypath.main.PyPath method)@\spxentry{v()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.v}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{v}}}{\emph{identifier}}{}
Returns \sphinxcode{\sphinxupquote{igraph.Vertex()}} object if the identifier
is a valid vertex index in the default undirected graph,
or a UniProt ID or GeneSymbol which can be found in the
default undirected network, otherwise \sphinxcode{\sphinxupquote{None}}.
\begin{description}
\item[{@identifier}] \leavevmode{[}int, str{]}
Vertex index (int) or GeneSymbol (str) or UniProt ID (str) or
\sphinxcode{\sphinxupquote{igraph.Vertex}} object.

\end{description}

\end{fulllineitems}

\index{vertex\_pathways() (pypath.main.PyPath method)@\spxentry{vertex\_pathways()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.vertex_pathways}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{vertex\_pathways}}}{}{}
Some resources assignes interactions some others proteins to
pathways. This function copies pathway annotations from edge
attributes to vertex attributes.

\end{fulllineitems}

\index{vsgs() (pypath.main.PyPath method)@\spxentry{vsgs()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.vsgs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{vsgs}}}{}{}
Returns a generator sequence of the node names as GeneSymbols
{[}str{]} (from the undirected graph).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Sequence containing the node names as
GeneSymbols {[}str{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{vsup() (pypath.main.PyPath method)@\spxentry{vsup()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.vsup}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{vsup}}}{}{}
Returns a generator sequence of the node names as UniProt IDs
{[}str{]} (from the undirected graph).
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{generator}) \textendash{} Sequence containing the node names as
UniProt IDs {[}str{]}.

\end{description}\end{quote}

\end{fulllineitems}

\index{wang\_effects() (pypath.main.PyPath method)@\spxentry{wang\_effects()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.wang_effects}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{wang\_effects}}}{\emph{graph=None}}{}
\end{fulllineitems}

\index{write\_table() (pypath.main.PyPath method)@\spxentry{write\_table()}\spxextra{pypath.main.PyPath method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.PyPath.write_table}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_table}}}{\emph{tbl}, \emph{outfile}, \emph{sep='\textbackslash{}t'}, \emph{cut=None}, \emph{colnames=True}, \emph{rownames=True}}{}
Writes a given table to a file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{tbl}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the data of the table. It is assumed that keys are
the row names {[}str{]} and the values, well, values. Column
names (if any) are defined with the key \sphinxcode{\sphinxupquote{'header'}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{outfile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} File name where to save the table. The file will be saved
under the object’s \sphinxcode{\sphinxupquote{pypath.main.PyPath.outdir}}
(\sphinxcode{\sphinxupquote{'results'}} by default).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sep}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'       '}} (tab) by default. Specifies the separator
for the file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{cut}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Specifies the maximum number
of characters for the row names.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{colnames}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Specifies whether to write
the column names in the file or not.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{rownames}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{True}} by default. Specifies whether to write
the row names in the file or not.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}

\index{Direction (class in pypath.main)@\spxentry{Direction}\spxextra{class in pypath.main}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.main.}}\sphinxbfcode{\sphinxupquote{Direction}}}{\emph{id\_a}, \emph{id\_b}}{}
Object storing directionality information of an edge. Also includes
information about the reverse direction, mode of regulation and
sources of that information.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_a}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{id\_b}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the target node.

\end{itemize}

\item[{Variables}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{dirs}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing the presence of directionality of the
given edge. Keys are \sphinxstyleemphasis{straight}, \sphinxstyleemphasis{reverse} and \sphinxcode{\sphinxupquote{'undirected'}}
and their values denote the presence/absence {[}bool{]}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{negative}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary contianing the presence/absence {[}bool{]} of negative
interactions for both \sphinxcode{\sphinxupquote{straight}} and \sphinxcode{\sphinxupquote{reverse}}
directions.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{negative\_sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the resource names {[}str{]} supporting a negative
interaction on \sphinxcode{\sphinxupquote{straight}} and \sphinxcode{\sphinxupquote{reverse}}
directions.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{nodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Contains the node names {[}str{]} sorted alphabetically (\sphinxstyleemphasis{id\_a},
\sphinxstyleemphasis{id\_b}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{positive}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary contianing the presence/absence {[}bool{]} of positive
interactions for both \sphinxcode{\sphinxupquote{straight}} and \sphinxcode{\sphinxupquote{reverse}}
directions.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{positive\_sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the resource names {[}str{]} supporting a positive
interaction on \sphinxcode{\sphinxupquote{straight}} and \sphinxcode{\sphinxupquote{reverse}}
directions.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{reverse}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Contains the node names {[}str{]} in reverse order e.g. (\sphinxstyleemphasis{id\_b},
\sphinxstyleemphasis{id\_a}).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Contains the resource names {[}str{]} of a given edge for each
directionality (\sphinxcode{\sphinxupquote{straight}}, \sphinxcode{\sphinxupquote{reverse}} and
\sphinxcode{\sphinxupquote{'undirected'}}). Values are sets containing the names of those
resources supporting such directionality.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{straight}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Contains the node names {[}str{]} in the original order e.g.
(\sphinxstyleemphasis{id\_a}, \sphinxstyleemphasis{id\_b}).

\end{itemize}

\end{description}\end{quote}
\index{check\_nodes() (pypath.main.Direction method)@\spxentry{check\_nodes()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.check_nodes}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_nodes}}}{\emph{nodes}}{}
Checks if \sphinxstyleemphasis{nodes} is contained in the edge.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{nodes}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} Or {[}tuple{]}, contains the names of the nodes to be checked.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if all elements in \sphinxstyleemphasis{nodes} are
contained in the object \sphinxcode{\sphinxupquote{nodes}} list.

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_param() (pypath.main.Direction method)@\spxentry{check\_param()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.check_param}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_param}}}{\emph{di}}{}
Checks if \sphinxstyleemphasis{di} is \sphinxcode{\sphinxupquote{'undirected'}} or contains the nodes of
the current edge. Used internally to check that \sphinxstyleemphasis{di} is a valid
key for the object attributes declared on dictionaries.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{di}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Or {[}str{]}, key to be tested for validity.

\item[{Returns}] \leavevmode
\begin{description}
\item[{(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if \sphinxstyleemphasis{di} is \sphinxcode{\sphinxupquote{'undirected'}} or a tuple}] \leavevmode
of node names contained in the edge, \sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}


\end{description}\end{quote}

\end{fulllineitems}

\index{consensus\_edges() (pypath.main.Direction method)@\spxentry{consensus\_edges()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.consensus_edges}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{consensus\_edges}}}{}{}
Infers the consensus edge(s) according to the number of
supporting sources. This includes direction and sign.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the consensus edge(s) along with the
consensus sign. If there is no major directionality, both
are returned. The structure is as follows:
\sphinxcode{\sphinxupquote{{[}'\textless{}source\textgreater{}', '\textless{}target\textgreater{}', '\textless{}(un)directed\textgreater{}', '\textless{}sign\textgreater{}'{]}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dir() (pypath.main.Direction method)@\spxentry{get\_dir()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.get_dir}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dir}}}{\emph{direction}, \emph{sources=False}}{}
Returns the state (or \sphinxstyleemphasis{sources} if specified) of the given
\sphinxstyleemphasis{direction}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Or {[}str{]} (if \sphinxcode{\sphinxupquote{'undirected'}}). Pair of nodes from which
direction information is to be retrieved.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{'False'}} by default. Specifies if the
\sphinxcode{\sphinxupquote{sources}} information of the given direction is to
be retrieved instead.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool} or \sphinxstyleemphasis{set}) \textendash{} (if \sphinxcode{\sphinxupquote{sources=True}}). Presence/absence
of the requested direction (or the list of sources if
specified). Returns \sphinxcode{\sphinxupquote{None}} if \sphinxstyleemphasis{direction} is not valid.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dirs() (pypath.main.Direction method)@\spxentry{get\_dirs()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.get_dirs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dirs}}}{\emph{src}, \emph{tgt}, \emph{sources=False}}{}
Returns all directions with boolean values or list of sources.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{src}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Source node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{tgt}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Target node.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to return
the \sphinxcode{\sphinxupquote{sources}} attribute instead of \sphinxcode{\sphinxupquote{dirs}}.

\end{itemize}

\item[{Returns}] \leavevmode
Contains the \sphinxcode{\sphinxupquote{dirs}} (or \sphinxcode{\sphinxupquote{sources}} if
specified) of the given edge.

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_sign() (pypath.main.Direction method)@\spxentry{get\_sign()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.get_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_sign}}}{\emph{direction}, \emph{sign=None}, \emph{sources=False}}{}
Retrieves the sign information of the edge in the given
diretion. If specified in \sphinxstyleemphasis{sign}, only that sign’s information
will be retrieved. If specified in \sphinxstyleemphasis{sources}, the sources of
that information will be retrieved instead.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Contains the pair of nodes specifying the directionality of
the edge from which th information is to be retrieved.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sign}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. Denotes whether to retrieve
the \sphinxcode{\sphinxupquote{'positive'}} or \sphinxcode{\sphinxupquote{'negative'}} specific information.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sources}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default. Specifies whether to return
the sources instead of sign.

\end{itemize}

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} If \sphinxcode{\sphinxupquote{sign=None}} containing {[}bool{]} values
denoting the presence of positive and negative sign on that
direction, if \sphinxcode{\sphinxupquote{sources=True}} the {[}set{]} of sources for each
of them will be returned instead. If \sphinxstyleemphasis{sign} is specified,
returns {[}bool{]} or {[}set{]} (if \sphinxcode{\sphinxupquote{sources=True}}) of that
specific direction and sign.

\end{description}\end{quote}

\end{fulllineitems}

\index{has\_sign() (pypath.main.Direction method)@\spxentry{has\_sign()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.has_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{has\_sign}}}{\emph{direction=None}}{}
Checks whether the edge (or for a specific \sphinxstyleemphasis{direction}) has
any signed information (about positive/negative interactions).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If specified, only the
information of that direction is checked for sign.

\item[{Returns}] \leavevmode
\begin{description}
\item[{(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if there exist any information on the}] \leavevmode
sign of the interaction, \sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}


\end{description}\end{quote}

\end{fulllineitems}

\index{is\_directed() (pypath.main.Direction method)@\spxentry{is\_directed()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.is_directed}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{is\_directed}}}{}{}
Checks if edge has any directionality information.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} Returns \sphinxcode{\sphinxupquote{True}} if any of the \sphinxcode{\sphinxupquote{dirs}}
attribute values is \sphinxcode{\sphinxupquote{True}} (except \sphinxcode{\sphinxupquote{'undirected'}}),
\sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}\end{quote}

\end{fulllineitems}

\index{is\_inhibition() (pypath.main.Direction method)@\spxentry{is\_inhibition()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.is_inhibition}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{is\_inhibition}}}{\emph{direction=None}}{}
Checks if any (or for a specific \sphinxstyleemphasis{direction}) interaction is
inhibition (negative interaction).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If specified, checks the
\sphinxcode{\sphinxupquote{negative}} attribute of that specific
directionality. If not specified, checks both.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if any interaction (or the specified
\sphinxstyleemphasis{direction}) is inhibitory (negative).

\end{description}\end{quote}

\end{fulllineitems}

\index{is\_mutual() (pypath.main.Direction method)@\spxentry{is\_mutual()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.is_mutual}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{is\_mutual}}}{}{}
Checks if the edge has mutual directions (both A\textendash{}\textgreater{}B and B\textendash{}\textgreater{}A).

\end{fulllineitems}

\index{is\_stimulation() (pypath.main.Direction method)@\spxentry{is\_stimulation()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.is_stimulation}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{is\_stimulation}}}{\emph{direction=None}}{}
Checks if any (or for a specific \sphinxstyleemphasis{direction}) interaction is
activation (positive interaction).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If specified, checks the
\sphinxcode{\sphinxupquote{positive}} attribute of that specific
directionality. If not specified, checks both.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if any interaction (or the specified
\sphinxstyleemphasis{direction}) is activatory (positive).

\end{description}\end{quote}

\end{fulllineitems}

\index{majority\_dir() (pypath.main.Direction method)@\spxentry{majority\_dir()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.majority_dir}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{majority\_dir}}}{}{}
Infers which is the major directionality of the edge by number
of supporting sources.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{tuple}) \textendash{} Contains the pair of nodes denoting the
consensus directionality. If the number of sources on both
directions is equal, \sphinxcode{\sphinxupquote{None}} is returned. If there is no
directionality information, \sphinxcode{\sphinxupquote{'undirected'{}`}} will be
returned.

\end{description}\end{quote}

\end{fulllineitems}

\index{majority\_sign() (pypath.main.Direction method)@\spxentry{majority\_sign()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.majority_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{majority\_sign}}}{}{}
Infers which is the major sign (activation/inhibition) of the
edge by number of supporting sources on both directions.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{dict}) \textendash{} Keys are the node tuples on both directions
(\sphinxcode{\sphinxupquote{straight}}/\sphinxcode{\sphinxupquote{reverse}}) and values can be
either \sphinxcode{\sphinxupquote{None}} if that direction has no sign information or
a list of two {[}bool{]} elements corresponding to majority of
positive and majority of negative support. In case both
elements of the list are \sphinxcode{\sphinxupquote{True}}, this means the number of
supporting sources for both signs in that direction is
equal.

\end{description}\end{quote}

\end{fulllineitems}

\index{merge() (pypath.main.Direction method)@\spxentry{merge()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.merge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge}}}{\emph{other}}{}
Merges current edge with another (if and only if they are the
same class and contain the same nodes). Updates the attributes
\sphinxcode{\sphinxupquote{dirs}}, \sphinxcode{\sphinxupquote{sources}}, \sphinxcode{\sphinxupquote{positive}},
\sphinxcode{\sphinxupquote{negative}}, \sphinxcode{\sphinxupquote{positive\_sources}} and
\sphinxcode{\sphinxupquote{negative\_sources}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{other}} ({\hyperref[\detokenize{reference:pypath.main.Direction}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{pypath.main.Direction}}}}}) \textendash{} The new edge object to be merged with the current one.

\end{description}\end{quote}

\end{fulllineitems}

\index{negative\_reverse() (pypath.main.Direction method)@\spxentry{negative\_reverse()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.negative_reverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{negative\_reverse}}}{}{}
Checks if the \sphinxcode{\sphinxupquote{reverse}} directionality is a negative
interaction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if there is supporting information on
the \sphinxcode{\sphinxupquote{reverse}} direction of the edge as inhibition.
\sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}\end{quote}

\end{fulllineitems}

\index{negative\_sources\_reverse() (pypath.main.Direction method)@\spxentry{negative\_sources\_reverse()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.negative_sources_reverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{negative\_sources\_reverse}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{reverse}}
direction and negative sign.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{reverse}} directionality of the edge with a
negative sign.

\end{description}\end{quote}

\end{fulllineitems}

\index{negative\_sources\_straight() (pypath.main.Direction method)@\spxentry{negative\_sources\_straight()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.negative_sources_straight}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{negative\_sources\_straight}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{straight}}
direction and negative sign.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{straight}} directionality of the edge with a
negative sign.

\end{description}\end{quote}

\end{fulllineitems}

\index{negative\_straight() (pypath.main.Direction method)@\spxentry{negative\_straight()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.negative_straight}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{negative\_straight}}}{}{}
Checks if the \sphinxcode{\sphinxupquote{straight}} directionality is a negative
interaction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if there is supporting information on
the \sphinxcode{\sphinxupquote{straight}} direction of the edge as inhibition.
\sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}\end{quote}

\end{fulllineitems}

\index{positive\_reverse() (pypath.main.Direction method)@\spxentry{positive\_reverse()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.positive_reverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{positive\_reverse}}}{}{}
Checks if the \sphinxcode{\sphinxupquote{reverse}} directionality is a positive
interaction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if there is supporting information on
the \sphinxcode{\sphinxupquote{reverse}} direction of the edge as activation.
\sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}\end{quote}

\end{fulllineitems}

\index{positive\_sources\_reverse() (pypath.main.Direction method)@\spxentry{positive\_sources\_reverse()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.positive_sources_reverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{positive\_sources\_reverse}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{reverse}}
direction and positive sign.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{reverse}} directionality of the edge with a
positive sign.

\end{description}\end{quote}

\end{fulllineitems}

\index{positive\_sources\_straight() (pypath.main.Direction method)@\spxentry{positive\_sources\_straight()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.positive_sources_straight}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{positive\_sources\_straight}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{straight}}
direction and positive sign.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{straight}} directionality of the edge with a
positive sign.

\end{description}\end{quote}

\end{fulllineitems}

\index{positive\_straight() (pypath.main.Direction method)@\spxentry{positive\_straight()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.positive_straight}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{positive\_straight}}}{}{}
Checks if the \sphinxcode{\sphinxupquote{straight}} directionality is a positive
interaction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{bool}) \textendash{} \sphinxcode{\sphinxupquote{True}} if there is supporting information on
the \sphinxcode{\sphinxupquote{straight}} direction of the edge as activation.
\sphinxcode{\sphinxupquote{False}} otherwise.

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_dir() (pypath.main.Direction method)@\spxentry{set\_dir()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.set_dir}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_dir}}}{\emph{direction}, \emph{source}}{}
Adds directionality information with the corresponding data
source named. Modifies self attributes \sphinxcode{\sphinxupquote{dirs}} and
\sphinxcode{\sphinxupquote{sources}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Or {[}str{]}, the directionality key for which the value on
\sphinxcode{\sphinxupquote{dirs}} has to be set \sphinxcode{\sphinxupquote{True}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Contains the name(s) of the source(s) from which such
information was obtained.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{set\_sign() (pypath.main.Direction method)@\spxentry{set\_sign()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.set_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_sign}}}{\emph{direction}, \emph{sign}, \emph{source}}{}
Sets sign and source information on a given direction of the
edge. Modifies the attributes \sphinxcode{\sphinxupquote{positive}} and
\sphinxcode{\sphinxupquote{positive\_sources}} or \sphinxcode{\sphinxupquote{negative}} and
\sphinxcode{\sphinxupquote{negative\_sources}} depending on the sign. Direction is
also updated accordingly, which also modifies the attributes
\sphinxcode{\sphinxupquote{dirs}} and \sphinxcode{\sphinxupquote{sources}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Pair of edge nodes specifying the direction from which the
information is to be set/updated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sign}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Specifies the type of interaction. If \sphinxcode{\sphinxupquote{'positive'}}, is
considered activation, otherwise, is assumed to be negative
(inhibition).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Contains the name(s) of the source(s) from which the
information was obtained.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{sources\_reverse() (pypath.main.Direction method)@\spxentry{sources\_reverse()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.sources_reverse}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_reverse}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{reverse}} direction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{reverse}} directionality of the edge.

\end{description}\end{quote}

\end{fulllineitems}

\index{sources\_straight() (pypath.main.Direction method)@\spxentry{sources\_straight()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.sources_straight}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_straight}}}{}{}
Retrieves the list of sources for the \sphinxcode{\sphinxupquote{straight}}
direction.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
\sphinxcode{\sphinxupquote{straight}} directionality of the edge.

\end{description}\end{quote}

\end{fulllineitems}

\index{sources\_undirected() (pypath.main.Direction method)@\spxentry{sources\_undirected()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.sources_undirected}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sources\_undirected}}}{}{}
Retrieves the list of sources without directed information.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{set}) \textendash{} Contains the names of the sources supporting the
edge presence but without specific directionality
information.

\end{description}\end{quote}

\end{fulllineitems}

\index{src() (pypath.main.Direction method)@\spxentry{src()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.src}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{src}}}{\emph{undirected=False}}{}
Returns the name(s) of the source node(s) for each existing
direction on the interaction.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{undirected}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the name(s) for the source node(s).
This means if the interaction is bidirectional, the list
will contain both identifiers on the edge. If the
interaction is undirected, an empty list will be returned.

\end{description}\end{quote}

\end{fulllineitems}

\index{src\_by\_source() (pypath.main.Direction method)@\spxentry{src\_by\_source()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.src_by_source}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{src\_by\_source}}}{\emph{source}}{}
Returns the name(s) of the source node(s) for each existing
direction on the interaction for a specific \sphinxstyleemphasis{source}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source according to which the information is to
be retrieved.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the name(s) for the source node(s)
according to the specified \sphinxstyleemphasis{source}. This means if the
interaction is bidirectional, the list will contain both
identifiers on the edge. If the specified \sphinxstyleemphasis{source} is not
found or invalid, an empty list will be returned.

\end{description}\end{quote}

\end{fulllineitems}

\index{tgt() (pypath.main.Direction method)@\spxentry{tgt()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.tgt}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{tgt}}}{\emph{undirected=False}}{}
Returns the name(s) of the target node(s) for each existing
direction on the interaction.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{undirected}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{False}} by default.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the name(s) for the target node(s).
This means if the interaction is bidirectional, the list
will contain both identifiers on the edge. If the
interaction is undirected, an empty list will be returned.

\end{description}\end{quote}

\end{fulllineitems}

\index{tgt\_by\_source() (pypath.main.Direction method)@\spxentry{tgt\_by\_source()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.tgt_by_source}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{tgt\_by\_source}}}{\emph{source}}{}
Returns the name(s) of the target node(s) for each existing
direction on the interaction for a specific \sphinxstyleemphasis{source}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the source according to which the information is to
be retrieved.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} Contains the name(s) for the target node(s)
according to the specified \sphinxstyleemphasis{source}. This means if the
interaction is bidirectional, the list will contain both
identifiers on the edge. If the specified \sphinxstyleemphasis{source} is not
found or invalid, an empty list will be returned.

\end{description}\end{quote}

\end{fulllineitems}

\index{translate() (pypath.main.Direction method)@\spxentry{translate()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.translate}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{translate}}}{\emph{ids}}{}
Translates the node names/identifiers according to the
dictionary \sphinxstyleemphasis{ids}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{ids}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing (at least) the current names of the
nodes as keys and their translation as values.

\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{pypath.main.Direction}) \textendash{} The copy of current edge object
with translated node names.

\end{description}\end{quote}

\end{fulllineitems}

\index{unset\_dir() (pypath.main.Direction method)@\spxentry{unset\_dir()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.unset_dir}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{unset\_dir}}}{\emph{direction}, \emph{source=None}}{}
Removes directionality and/or source information of the
specified \sphinxstyleemphasis{direction}. Modifies attribute \sphinxcode{\sphinxupquote{dirs}} and
\sphinxcode{\sphinxupquote{sources}}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} Or {[}str{]} (if \sphinxcode{\sphinxupquote{'undirected'}}) the pair of nodes specifying
the directionality from which the information is to be
removed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If specified, determines
which specific source(s) is(are) to be removed from
\sphinxcode{\sphinxupquote{sources}} attribute in the specified \sphinxstyleemphasis{direction}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{unset\_sign() (pypath.main.Direction method)@\spxentry{unset\_sign()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.unset_sign}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{unset\_sign}}}{\emph{direction}, \emph{sign}, \emph{source=None}}{}
Removes sign and/or source information of the specified
\sphinxstyleemphasis{direction} and \sphinxstyleemphasis{sign}. Modifies attribute \sphinxcode{\sphinxupquote{positive}}
and \sphinxcode{\sphinxupquote{positive\_sources}} or \sphinxcode{\sphinxupquote{negative}} and
\sphinxcode{\sphinxupquote{negative\_sources}} (or
\sphinxcode{\sphinxupquote{positive\_attributes}}/\sphinxcode{\sphinxupquote{negative\_sources}}
only if \sphinxcode{\sphinxupquote{source=True}}).
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{direction}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} The pair of nodes specifying the directionality from which
the information is to be removed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{sign}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Sign from which the information is to be removed. Must be
either \sphinxcode{\sphinxupquote{'positive'}} or \sphinxcode{\sphinxupquote{'negative'}}.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{source}} (\sphinxstyleliteralemphasis{\sphinxupquote{set}}) \textendash{} Optional, \sphinxcode{\sphinxupquote{None}} by default. If specified, determines
which source(s) is(are) to be removed from the sources in
the specified \sphinxstyleemphasis{direction} and \sphinxstyleemphasis{sign}.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{which\_dirs() (pypath.main.Direction method)@\spxentry{which\_dirs()}\spxextra{pypath.main.Direction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.Direction.which_dirs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{which\_dirs}}}{}{}
Returns the pair(s) of nodes for which there is information
about their directionality.
\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
(\sphinxstyleemphasis{list}) \textendash{} List of tuples containing the nodes for which
their attribute \sphinxcode{\sphinxupquote{dirs}} is \sphinxcode{\sphinxupquote{True}}.

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}

\index{AttrHelper (class in pypath.main)@\spxentry{AttrHelper}\spxextra{class in pypath.main}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{reference:pypath.main.AttrHelper}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pypath.main.}}\sphinxbfcode{\sphinxupquote{AttrHelper}}}{\emph{value}, \emph{name=None}, \emph{defaults=\{\}}}{}
Attribute helper class.
\begin{itemize}
\item {} \begin{description}
\item[{Initialization arguments:}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleemphasis{value} {[}dict/str{]}?:

\item {} 
\sphinxstyleemphasis{name} {[}str{]}?: Optional, \sphinxcode{\sphinxupquote{None}} by default.

\item {} 
\sphinxstyleemphasis{defaults} {[}dict{]}:

\end{itemize}

\end{description}

\item {} \begin{description}
\item[{Attributes:}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleemphasis{value} {[}dict{]}?:

\item {} 
\sphinxstyleemphasis{name} {[}str{]}?:

\item {} 
\sphinxstyleemphasis{defaults} {[}dict{]}:

\item {} 
\sphinxstyleemphasis{id\_type} {[}type{]}:

\end{itemize}

\end{description}

\item {} \begin{description}
\item[{Call arguments:}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleemphasis{instance} {[}{]}:

\item {} 
\sphinxstyleemphasis{this\_directed} {[}tuple?{]}: Optional, \sphinxcode{\sphinxupquote{None}} by default.

\item {} 
\sphinxstyleemphasis{thisSign} {[}{]}: Optional, \sphinxcode{\sphinxupquote{None}} by default.

\item {} 
\sphinxstyleemphasis{this\_directedSources} {[}{]}: Optional, \sphinxcode{\sphinxupquote{None}} by default.

\item {} 
\sphinxstyleemphasis{thisSources} {[}{]}: Optional, \sphinxcode{\sphinxupquote{None}} by default.

\end{itemize}

\end{description}

\item {} \begin{description}
\item[{Returns:}] \leavevmode\begin{itemize}
\item {} 
\end{itemize}

\end{description}

\end{itemize}

\end{fulllineitems}



\section{ptm module}
\label{\detokenize{reference:module-pypath.ptm}}\label{\detokenize{reference:ptm-module}}\index{pypath.ptm (module)@\spxentry{pypath.ptm}\spxextra{module}}

\chapter{Webservice}
\label{\detokenize{webservice:webservice}}\label{\detokenize{webservice::doc}}
\sphinxstylestrong{New webservice} from 14 June 2018: the queries slightly changed, have been
largely extended. See the examples below.

One instance of the pypath webservice runs at the domain
\sphinxurl{http://omnipathdb.org/}, serving not only the OmniPath data but other datasets:
TF-target interactions from TF Regulons, a large collection additional
enzyme-substrate interactions, and literature curated miRNA-mRNA interacions
combined from 4 databases. The webservice implements a very simple REST style
API, you can make requests by HTTP protocol (browser, wget, curl or whatever).

The webservice currently recognizes 3 types of queries: \sphinxcode{\sphinxupquote{interactions}},
\sphinxcode{\sphinxupquote{ptms}} and \sphinxcode{\sphinxupquote{info}}. The query types \sphinxcode{\sphinxupquote{resources}}, \sphinxcode{\sphinxupquote{network}} and
\sphinxcode{\sphinxupquote{about}} have not been implemented yet in the new webservice.


\section{Mouse and rat}
\label{\detokenize{webservice:mouse-and-rat}}
Except the miRNA interactions all interactions are available for human, mouse
and rat. The rodent data has been translated from human using the NCBI
Homologene database. Many human proteins have no known homolog in rodents
hence rodent datasets are smaller than their human counterparts. Note, if you
work with mouse omics data you might do better to translate your dataset to
human (for example using the \sphinxcode{\sphinxupquote{pypath.homology}} module) and use human
interaction data.


\section{Examples}
\label{\detokenize{webservice:examples}}
A request without any parameter, gives some basic numbers about the actual
loaded dataset:
\begin{quote}

\sphinxurl{http://omnipathdb.org}
\end{quote}

The \sphinxcode{\sphinxupquote{info}} returns a HTML page with comprehensive information about the
resources:
\begin{quote}

\sphinxurl{http://omnipathdb.org/info}
\end{quote}

The \sphinxcode{\sphinxupquote{interactions}} query accepts some parameters and returns interactions in
tabular format. This example returns all interactions of EGFR (P00533), with
sources and references listed.
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?partners=P00533\&fields=sources,references}
\end{quote}

By default only the OmniPath dataset used, to query the TF Regulons or add the
extra enzyme-substrate interactions you need to set additional parameters. For
example to query the transcriptional regulators of EGFR:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?targets=EGFR\&types=TF}
\end{quote}

The TF Regulons database assigns confidence levels to the interactions. You
might want to select only the highest confidence, \sphinxstyleemphasis{A} category:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?targets=EGFR\&types=TF\&tfregulons\_levels=A}
\end{quote}

Show the transcriptional targets of Smad2 homology translated to rat including
the confidence levels from TF Regulons:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&fields=type,ncbi\_tax\_id,tfregulons\_level\&organisms=10116\&sources=Smad2\&types=TF}
\end{quote}

Query interactions from PhosphoNetworks which is part of the \sphinxstyleemphasis{kinaseextra}
dataset:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&fields=sources\&databases=PhosphoNetworks\&datasets=kinaseextra}
\end{quote}

Get the interactions from Signor, SPIKE and SignaLink3:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&fields=sources,references\&databases=Signor,SPIKE,SignaLink3}
\end{quote}

All interactions of MAP1LC3B:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&partners=MAP1LC3B}
\end{quote}

By default \sphinxcode{\sphinxupquote{partners}} queries the interaction where either the source or the
arget is among the partners. If you set the \sphinxcode{\sphinxupquote{source\_target}} parameter to
\sphinxcode{\sphinxupquote{AND}} both the source and the target must be in the queried set:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&fields=sources,references\&sources=ATG3,ATG7,ATG4B,SQSTM1\&targets=MAP1LC3B,MAP1LC3A,MAP1LC3C,Q9H0R8,GABARAP,GABARAPL2\&source\_target=AND}
\end{quote}

As you see above you can use UniProt IDs and Gene Symbols in the queries and
also mix them. Get the miRNA regulating NOTCH1:
\begin{quote}

\sphinxurl{http://omnipathdb.org/interactions/?genesymbols=1\&fields=sources,references\&datasets=mirnatarget\&targets=NOTCH1}
\end{quote}

Note: with the exception of mandatory fields and genesymbols, the columns
appear exactly in the order you provided in your query.

Another query type available is \sphinxcode{\sphinxupquote{ptms}} which provides enzyme-substrate
interactions. It is very similar to the \sphinxcode{\sphinxupquote{interactions}}:
\begin{quote}

\sphinxurl{http://omnipathdb.org/ptms?genesymbols=1\&fields=sources,references,isoforms\&enzymes=FYN}
\end{quote}

Is there any ubiquitination reaction?
\begin{quote}

\sphinxurl{http://omnipathdb.org/ptms?genesymbols=1\&fields=sources,references\&types=ubiquitination}
\end{quote}

And acetylation in mouse?
\begin{quote}

\sphinxurl{http://omnipathdb.org/ptms?genesymbols=1\&fields=sources,references\&types=acetylation\&organisms=10090}
\end{quote}

Rat interactions, both directly from rat and homology translated from human,
from the PhosphoSite database:
\begin{quote}

\sphinxurl{http://omnipathdb.org/ptms?genesymbols=1\&fields=sources,references\&organisms=10116\&databases=PhosphoSite,PhosphoSite\_noref}
\end{quote}


\chapter{Release history}
\label{\detokenize{changelog:release-history}}\label{\detokenize{changelog::doc}}
Main improvements in the past releases:


\section{0.1.0}
\label{\detokenize{changelog:id1}}\begin{itemize}
\item {} 
First release of pypath, for initial testing.

\end{itemize}


\section{0.2.0}
\label{\detokenize{changelog:id2}}\begin{itemize}
\item {} 
Lots of small improvements in almost every module

\item {} 
Networks can be read from local files, remote files, lists or provided by
any function

\item {} 
Almost all redistributed data have been removed, every source downloaded
from the original provider.

\end{itemize}


\section{0.3.0}
\label{\detokenize{changelog:id3}}\begin{itemize}
\item {} 
First version with partial Python 3 support.

\end{itemize}


\section{0.4.0}
\label{\detokenize{changelog:id4}}\begin{itemize}
\item {} 
\sphinxstylestrong{pyreact} module with \sphinxstylestrong{BioPaxReader} and \sphinxstylestrong{PyReact} classes added

\item {} 
Process description databases, BioPax and PathwayCommons SIF conversion
rules are supported

\item {} 
Format definitions for 6 process description databases included.

\end{itemize}


\section{0.5.0}
\label{\detokenize{changelog:id5}}\begin{itemize}
\item {} 
Many classes have been added to the \sphinxstylestrong{plot} module

\item {} 
All figures and tables in the manuscript can be generated automatically

\item {} 
This is supported by a new module, \sphinxstylestrong{analysis}, which implements a generic
workflow in its \sphinxstylestrong{Workflow} class.

\end{itemize}


\section{0.7.74}
\label{\detokenize{changelog:id6}}\begin{itemize}
\item {} 
\sphinxstylestrong{homology} module: finds the homologs of proteins using the NCBI
Homologene database and the homologs of PTM sites using UniProt sequences
and PhosphoSitePlus homology table

\item {} 
\sphinxstylestrong{ptm} module: fully integrated way of processing enzyme-substrate
interactions from many databases and their translation by homology to other
species

\item {} 
\sphinxstylestrong{export} module: creates \sphinxcode{\sphinxupquote{pandas.DataFrame}} or exports the network into
tabular file

\item {} 
New webservice

\item {} 
TF Regulons database included and provides much more comprehensive
transcriptional regulation resources, including literature curated, in silico
predicted, ChIP-Seq and expression pattern based approaches

\item {} 
Many network resources added, including miRNA-mRNA and TF-miRNA interactions

\end{itemize}


\section{Upcoming}
\label{\detokenize{changelog:upcoming}}\begin{itemize}
\item {} 
New, more flexible network reader class

\item {} 
Full support for multi-species molecular interaction networks
(e.g. pathogene-host)

\item {} 
Better support for not protein only molecular interaction networks
(metabolites, drug compounds, RNA)

\item {} 
ChEMBL webservice interface, interface for PubChem and eventually
forDrugBank

\item {} 
Silent mode: a way to suppress messages and progress bars

\end{itemize}


\chapter{Features}
\label{\detokenize{index:features}}
The primary aim of \sphinxstylestrong{pypath} is to build up networks from multiple sources on
one igraph object. \sphinxstylestrong{pypath} handles ambiguous ID conversion, reads custom
edge and node attributes from text files and \sphinxstylestrong{MySQL}.

Submodules perform various features, e.g. graph visualization, working with
rug compound data, searching drug targets and compounds in \sphinxstylestrong{ChEMBL}.


\section{ID conversion}
\label{\detokenize{index:id-conversion}}
The ID conversion module \sphinxcode{\sphinxupquote{mapping}} can be used independently. It has the
feature to translate secondary UniProt IDs to primaries, and Trembl IDs to
SwissProt, using primary Gene Symbols to find the connections. This module
automatically loads and stores the necessary conversion tables. Many tables
are predefined, such as all the IDs in \sphinxstylestrong{UniProt mapping service,} while
users are able to load any table from \sphinxstylestrong{file} or \sphinxstylestrong{MySQL,} using the classes
provided in the module \sphinxcode{\sphinxupquote{input\_formats}}.


\section{Pathways}
\label{\detokenize{index:pathways}}
\sphinxstylestrong{pypath} includes data and predefined format descriptions for more than 25
high quality, literature curated databases. The inut formats are defined in
the \sphinxcode{\sphinxupquote{data\_formats}} module. For some resources data downloaded on the fly,
where it is not possible, data is redistributed with the module. Descriptions
and comprehensive information about the resources is available in the
\sphinxcode{\sphinxupquote{descriptions}} module.


\section{Structural features}
\label{\detokenize{index:structural-features}}
One of the modules called \sphinxcode{\sphinxupquote{intera}} provides many classes for representing
structures and mechanisms behind protein interactions. These are \sphinxcode{\sphinxupquote{Residue}}
(optionally mutated), \sphinxcode{\sphinxupquote{Motif}}, \sphinxcode{\sphinxupquote{Ptm}}, \sphinxcode{\sphinxupquote{Domain}}, \sphinxcode{\sphinxupquote{DomainMotif}},
\sphinxcode{\sphinxupquote{DomainDomain}} and \sphinxcode{\sphinxupquote{Interface}}. All these classes have \sphinxcode{\sphinxupquote{\_\_eq\_\_()}}
methods to test equality between instances, and also \sphinxcode{\sphinxupquote{\_\_contains\_\_()}}
methods to look up easily if a residue is within a short motif or protein
domain, or is the target residue of a PTM.


\section{Sequences}
\label{\detokenize{index:sequences}}
The module \sphinxcode{\sphinxupquote{seq}} contains a simple class for quick lookup any residue or
segment in \sphinxstylestrong{UniProt} protein sequences while being aware of isoforms.


\section{Tissue expression}
\label{\detokenize{index:tissue-expression}}
For 3 protein expression databases there are functions and modules for
downloading and combining the expression data with the network. These are the
Human Protein Atlas, the ProteomicsDB and GIANT. The \sphinxcode{\sphinxupquote{giant}} and
\sphinxcode{\sphinxupquote{proteomicsdb}} modules can be used also as stand alone Python clients for
these resources.


\section{Functional annotations}
\label{\detokenize{index:functional-annotations}}
\sphinxstylestrong{GSEA} and \sphinxstylestrong{Gene Ontology} are two approaches for annotating genes and
gene products, and enrichment analysis technics aims to use these annotations
to highlight the biological functions a given set of genes is related to. Here
the \sphinxcode{\sphinxupquote{enrich}} module gives abstract classes to calculate enrichment
statistics, while the \sphinxcode{\sphinxupquote{go}} and the \sphinxcode{\sphinxupquote{gsea}} modules give access to GO and
GSEA data, and make it easy to count enrichment statistics for sets of genes.


\section{Drug compounds}
\label{\detokenize{index:drug-compounds}}
\sphinxstylestrong{UniChem} submodule provides an interface to effectively query the UniChem
service, use connectivity search with custom settings, and translate SMILEs to
ChEMBL IDs with ChEMBL web service.

\sphinxstylestrong{ChEMBL} submodule queries directly your own ChEMBL MySQL instance, has the
features to search targets and compounds from custom assay types and
relationship types, to get activity values, binding domains, and action types.
You need to download the ChEMBL MySQL dump, and load into your own server.


\section{Technical}
\label{\detokenize{index:technical}}
\sphinxstylestrong{MySQL} submodule helps to manage MySQL connections and track queries. It is
able to run queries parallely to optimize CPU and memory usage on the server,
handling queues, and serve the result by server side or client side storage.
The \sphinxcode{\sphinxupquote{chembl}} and potentially the \sphinxcode{\sphinxupquote{mapping}} modules rely on this \sphinxcode{\sphinxupquote{mysql}}
module.

The most important function in module \sphinxcode{\sphinxupquote{dataio}} is a very flexible \sphinxstylestrong{download
manager} built around \sphinxcode{\sphinxupquote{curl}}. The function \sphinxcode{\sphinxupquote{dataio.curl()}} accepts
numerous arguments, tries to deal in a smart way with local \sphinxstylestrong{cache,}
authentication, redirects, uncompression, character encodings, FTP and HTTP
transactions, and many other stuff. Cache can grow to several GBs, and takes
place in \sphinxcode{\sphinxupquote{./cache}} by default. Please be aware of this, and use for example
symlinks in case of using multiple working directories.

A simple \sphinxstylestrong{webservice} comes with this module: the \sphinxcode{\sphinxupquote{server}} module based on
\sphinxcode{\sphinxupquote{twisted.web.server}} opens a custom port and serves plain text tables over
HTTP with REST style querying.


\chapter{OmniPath in R}
\label{\detokenize{index:omnipath-in-r}}
You can download the data from the webservice and load into R. Look
\sphinxhref{https://github.com/saezlab/pypath/tree/master/r\_import}{here} for an
example.


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{p}
\item\relax\sphinxstyleindexentry{pypath.annot}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.annot}}
\item\relax\sphinxstyleindexentry{pypath.complex}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.complex}}
\item\relax\sphinxstyleindexentry{pypath.intera}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.intera}}
\item\relax\sphinxstyleindexentry{pypath.intercell}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.intercell}}
\item\relax\sphinxstyleindexentry{pypath.main}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.main}}
\item\relax\sphinxstyleindexentry{pypath.ptm}\sphinxstyleindexpageref{reference:\detokenize{module-pypath.ptm}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}